<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 How bias affects DA results | Implications of taxonomic bias for microbial differential-abundance analysis</title>
  <meta name="description" content="3 How bias affects DA results | Implications of taxonomic bias for microbial differential-abundance analysis" />
  <meta name="generator" content="bookdown 0.27.3 and GitBook 2.6.7" />

  <meta property="og:title" content="3 How bias affects DA results | Implications of taxonomic bias for microbial differential-abundance analysis" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 How bias affects DA results | Implications of taxonomic bias for microbial differential-abundance analysis" />
  
  
  

<meta name="author" content="Michael R. McLaren" />
<meta name="author" content="Jacob T. Nearing" />
<meta name="author" content="Amy D. Willis" />
<meta name="author" content="Karen G. Lloyd" />
<meta name="author" content="Benjamin J. Callahan" />


<meta name="date" content="2022-10-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="abundance-measurement.html"/>
<link rel="next" href="case-studies.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
    TeX: {extensions: ["cancel.js"]},
});
MathJax.Hub.Config({
    TeX: {
        Macros : {
            md: '^{\\times}\\!/',
        }
    },
});
</script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="abundance-measurement.html"><a href="abundance-measurement.html"><i class="fa fa-check"></i><b>2</b> How bias affects abundance measurements</a>
<ul>
<li class="chapter" data-level="2.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#a-model-of-mgs-measurements"><i class="fa fa-check"></i><b>2.1</b> A model of MGS measurements</a></li>
<li class="chapter" data-level="2.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#relative-abundance"><i class="fa fa-check"></i><b>2.2</b> Relative abundance</a></li>
<li class="chapter" data-level="2.3" data-path="abundance-measurement.html"><a href="abundance-measurement.html#absolute-abundance"><i class="fa fa-check"></i><b>2.3</b> Absolute abundance</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#leveraging-information-about-total-community-abundance"><i class="fa fa-check"></i><b>2.3.1</b> Leveraging information about total-community abundance</a></li>
<li class="chapter" data-level="2.3.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#leveraging-information-about-a-reference-species"><i class="fa fa-check"></i><b>2.3.2</b> Leveraging information about a reference species</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>3</b> How bias affects DA results</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-abundance.html"><a href="differential-abundance.html#fold-change-between-a-pair-of-samples"><i class="fa fa-check"></i><b>3.1</b> Fold change between a pair of samples</a></li>
<li class="chapter" data-level="3.2" data-path="differential-abundance.html"><a href="differential-abundance.html#regression-analysis-of-many-samples"><i class="fa fa-check"></i><b>3.2</b> Regression analysis of many samples</a></li>
<li class="chapter" data-level="3.3" data-path="differential-abundance.html"><a href="differential-abundance.html#rank-based-analyses"><i class="fa fa-check"></i><b>3.3</b> Rank-based analyses</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="case-studies.html"><a href="case-studies.html"><i class="fa fa-check"></i><b>4</b> Case studies  </a>
<ul>
<li class="chapter" data-level="4.1" data-path="case-studies.html"><a href="case-studies.html#foliar-fungi-experiment"><i class="fa fa-check"></i><b>4.1</b> Foliar fungi experiment</a></li>
<li class="chapter" data-level="4.2" data-path="case-studies.html"><a href="case-studies.html#vaginal-microbiomes-of-pregnant-women"><i class="fa fa-check"></i><b>4.2</b> Vaginal microbiomes of pregnant women</a></li>
<li class="chapter" data-level="4.3" data-path="case-studies.html"><a href="case-studies.html#human-gut-microbiomes"><i class="fa fa-check"></i><b>4.3</b> Human gut microbiomes</a></li>
<li class="chapter" data-level="4.4" data-path="case-studies.html"><a href="case-studies.html#microbial-growth-in-marine-sediments"><i class="fa fa-check"></i><b>4.4</b> Microbial growth in marine sediments</a></li>
<li class="chapter" data-level="4.5" data-path="case-studies.html"><a href="case-studies.html#summary-and-discussion"><i class="fa fa-check"></i><b>4.5</b> Summary and discussion</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="solutions.html"><a href="solutions.html"><i class="fa fa-check"></i><b>5</b> Solutions</a>
<ul>
<li class="chapter" data-level="5.1" data-path="solutions.html"><a href="solutions.html#ratio-based-relative-da-analysis"><i class="fa fa-check"></i><b>5.1</b> Ratio-based relative DA analysis</a></li>
<li class="chapter" data-level="5.2" data-path="solutions.html"><a href="solutions.html#calibrate-compositions"><i class="fa fa-check"></i><b>5.2</b> Calibration using community controls</a></li>
<li class="chapter" data-level="5.3" data-path="solutions.html"><a href="solutions.html#absolute-abundance-methods-with-more-stable-fes"><i class="fa fa-check"></i><b>5.3</b> Absolute-abundance methods with more stable FEs</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="solutions.html"><a href="solutions.html#use-complementary-mgs-and-total-abundance-measurements"><i class="fa fa-check"></i><b>5.3.1</b> Use complementary MGS and total-abundance measurements</a></li>
<li class="chapter" data-level="5.3.2" data-path="solutions.html"><a href="solutions.html#normalize-to-a-reference-species"><i class="fa fa-check"></i><b>5.3.2</b> Normalize to a reference species</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="solutions.html"><a href="solutions.html#bias-sensitivity-analysis"><i class="fa fa-check"></i><b>5.4</b> Bias sensitivity analysis</a></li>
<li class="chapter" data-level="5.5" data-path="solutions.html"><a href="solutions.html#bias-aware-meta-analysis"><i class="fa fa-check"></i><b>5.5</b> Bias-aware meta-analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="appendix-regression.html"><a href="appendix-regression.html"><i class="fa fa-check"></i><b>A</b> Linear regression</a>
<ul>
<li class="chapter" data-level="A.1" data-path="appendix-regression.html"><a href="appendix-regression.html#simple-linear-regression"><i class="fa fa-check"></i><b>A.1</b> Simple linear regression</a>
<ul>
<li class="chapter" data-level="A.1.1" data-path="appendix-regression.html"><a href="appendix-regression.html#review-of-simple-linear-regression"><i class="fa fa-check"></i><b>A.1.1</b> Review of simple linear regression</a></li>
<li class="chapter" data-level="A.1.2" data-path="appendix-regression.html"><a href="appendix-regression.html#measurement-error-in-the-response"><i class="fa fa-check"></i><b>A.1.2</b> Measurement error in the response</a></li>
<li class="chapter" data-level="A.1.3" data-path="appendix-regression.html"><a href="appendix-regression.html#specific-application-to-taxonomic-bias"><i class="fa fa-check"></i><b>A.1.3</b> Specific application to taxonomic bias</a></li>
</ul></li>
<li class="chapter" data-level="A.2" data-path="appendix-regression.html"><a href="appendix-regression.html#gamma-poisson-regression"><i class="fa fa-check"></i><b>A.2</b> Gamma-Poisson regression</a>
<ul>
<li class="chapter" data-level="A.2.1" data-path="appendix-regression.html"><a href="appendix-regression.html#background"><i class="fa fa-check"></i><b>A.2.1</b> Background</a></li>
<li class="chapter" data-level="A.2.2" data-path="appendix-regression.html"><a href="appendix-regression.html#inferring-lfcs-in-proportions-with-and-without-bias-correction"><i class="fa fa-check"></i><b>A.2.2</b> Inferring LFCs in proportions with and without bias correction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="B" data-path="supplemental-figures.html"><a href="supplemental-figures.html"><i class="fa fa-check"></i><b>B</b> Supplemental figures</a></li>
<li class="chapter" data-level="" data-path="funding.html"><a href="funding.html"><i class="fa fa-check"></i>Funding</a></li>
<li class="chapter" data-level="" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Implications of taxonomic bias for microbial differential-abundance analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-abundance" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> How bias affects DA results<a href="differential-abundance.html#differential-abundance" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<!-- Previous title: Differential-abundance methods vary in their robustness to taxonomic bias  -->
<p>We now turn to how the errors in MGS-based abundance measurements affect DA results.
Our focus is on <em>multiplicative DA analyses</em> that estimate the fold differences (FD) in the relative or absolute abundance of species across time points or between experimental conditions.
Multiplicative DA has a direct ecological interpretation via the multiplicative processes of exponential growth and decay.
Many DA methods operate on additive differences in log-transformed abundances and are therefore multiplicative; these include popular methods such as DESeq2 (<span class="citation">Love, Huber, and Anders (<a href="#ref-love2014mode" role="doc-biblioref">2014</a>)</span>), ALDEx2 (<span class="citation">Fernandes et al. (<a href="#ref-fernandes2014unif" role="doc-biblioref">2014</a>)</span>), corncob (<span class="citation">Martin, Witten, and Willis (<a href="#ref-martin2020mode" role="doc-biblioref">2020</a>)</span>), and time series methods based on the Lotka-Volterra model (e.g., <span class="citation">Stein et al. (<a href="#ref-stein2013ecol" role="doc-biblioref">2013</a>)</span>).
We also consider non-parametric rank-based methods; these methods are popular in case-control studies for which ecological interpretability is often of secondary importance to discovering stable associations between species and the condition of interest.</p>
<div id="fold-change-between-a-pair-of-samples" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Fold change between a pair of samples<a href="differential-abundance.html#fold-change-between-a-pair-of-samples" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The building blocks of a multiplicative DA analysis are the measured FDs in a species’ abundance between pairs of individual samples.
<!-- The FE in the measured FD between two samples is given by the ratio of the FEs for the individual-sample measurements. -->
If the fold error (FE) in a species’ abundance measurement is constant across samples, then it will not affect the measured FDs between samples.
<!-- Ratios -->
In Section <a href="abundance-measurement.html#abundance-measurement">2</a>, we showed that consistent taxonomic bias creates a constant FE in the ratio between two species <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>, equal to the ratio in their efficiencies (Equation <a href="abundance-measurement.html#eq:ratio-error">(2.9)</a>).
This error completely cancels when calculating the FD of this ratio from sample <span class="math inline">\(a\)</span> to <span class="math inline">\(b\)</span>,
<span class="math display" id="eq:ratio-fc-error">\[\begin{align}
  \tag{3.1}
\underbrace{\frac{\tilde R_{i/j}^{(b)}}{\tilde R_{i/j}^{(a)}}}_\text{measured FD}
  = \frac
    {R_{i/j}^{(b)} \cdot \cancel{B_i / B_j}}
    {R_{i/j}^{(a)} \cdot \cancel{B_i / B_j}}
  =
  \underbrace{\frac{R_{i/j}^{(b)}}{R_{i/j}^{(a)}}}_\text{actual FD}
  .
\end{align}\]</span>
<!-- Proportions -->
In contrast, bias creates a FE in the proportion of a species (Equation <a href="abundance-measurement.html#eq:prop-error">(2.7)</a>) that varies inversely with the mean efficiency of the sample.
Therefore, this error does not cancel when calculating the FD in the proportion,
<span class="math display" id="eq:prop-fc-error">\[\begin{align}
  \tag{3.2}
\underbrace{\frac{\tilde P_i^{(b)}}{\tilde P_i^{(a)}}}_\text{measured FD}
  = \frac
    {P_i^{(b)} \cdot \cancel{B_i} / \bar B^{(b)}}
    {P_i^{(a)} \cdot \cancel{B_i} / \bar B^{(a)}}
  =
  \underbrace{\frac{P_i^{(b)}}{P_i^{(a)}}}_\text{actual FD}
  \cdot
  \underbrace{\left[\frac{\bar B^{(b)}}{\bar B^{(a)}}\right]^{-1}}_\text{FE}
  .
\end{align}\]</span>
The varying mean efficiency does not cancel, leaving an FE in the FD of the species’ proportion that is equal to the inverse FD in the mean efficiency.
Notably, this FE is the same for every species.</p>
<p>The different effects of taxonomic bias on the measured FDs in proportions versus ratios are illustrated in a hypothetical example in Figure <a href="abundance-measurement.html#fig:error-proportions">2.1</a> (bottom row).
In this example, the mean efficiency decreases 2.6-fold from Sample 1 to Sample 2 (FD of 0.4), causing the FD in the proportion of each species to appear 2.6-fold larger than its true value.
Though the FE in the FD is the same for each species, the biological implications of the error vary.
In particular, there are three distinct types of error: an increase in the magnitude, a decrease in the magnitude, and a reversal in the direction of the measured FD.
We can see each type of error in Figure <a href="abundance-measurement.html#fig:error-proportions">2.1</a>.
For Species 1, which increases in abundance and thus moves in the opposite direction of the mean efficiency, we see an increase in magnitude of the measured FD (actual FD: 2.3, measured FD: 6.5).
For Species 2, which decreases and thus moves in the same direction as the mean efficiency but by a larger factor, we see a decrease in magnitude of the measured FD (actual FD: 0.15, measured FD: 0.44).
For Species 3, which decreases but by a smaller factor than the mean efficiency, the species actually appears to increase—a reversal in direction of the measured FD (actual FD: 0.6, measured FD: 1.8).
In contrast, the FDs in the ratios among species are identical in the actual and measured taxonomic profiles.
For example, the ratio of Species 2 to Species 3 shows the same 4-fold decrease in the actual profiles (<span class="math inline">\(1\)</span> to <span class="math inline">\(1/4\)</span>) and in the measured profiles (<span class="math inline">\(3\)</span> to <span class="math inline">\(3/4\)</span>).</p>
<!-- The ratios in sample 1 are Actual: 1:1:1, Observed: 1:18:6 -->
<!-- The ratios in sample 2 are Actual: 15:1:4, Observed: 5:6:8 -->
<p>Taxonomic bias affects measured FDs in a species’ absolute abundance in a similar fashion.
First, consider a species’ abundance measurement <span class="math inline">\(\tilde A_i^{(a)}\)</span> derived from a non-MGS measurement of total-community abundance using Equation <a href="abundance-measurement.html#eq:density-prop-meas">(2.11)</a>.
Equation <a href="abundance-measurement.html#eq:density-prop-error">(2.13)</a> indicates that the FE in this measurement equals the (constant) species efficiency multiplied by <span class="math inline">\({\bar B^{{\text{[tot]}}(a)}}/{\bar B^{(a)}}\)</span>, the ratio of the mean efficiency of the total-abundance measurement to that of the MGS measurement.
This ratio can vary, creating error in the measured FD between samples.
A notable special case is if the FDs in the mean efficiency of the total-abundance measurement mirrors that of the MGS measurement; then, the two will offset each other and lead to more stable FEs (and hence more accurate FDs) in <span class="math inline">\(\tilde A_i^{(a)}\)</span>.
We discuss how this possibility might be exploited in real experimental workflows in Section <a href="solutions.html#solutions">5</a>.</p>
<p>Now consider a species’ abundance derived from a reference species using Equation <a href="abundance-measurement.html#eq:density-ratio-meas">(2.14)</a>.
Equation <a href="abundance-measurement.html#eq:density-ratio-error">(2.15)</a> indicates that the FE equals a constant ratio in species’ efficiencies multiplied by the FE in the measurement of the reference species’ abundance.
If the abundance of the reference species can be determined up to constant FE across samples, then the FE in <span class="math inline">\(\tilde A_i^{(a)}\)</span> will also be constant, leading to accurate FDs.</p>
</div>
<div id="regression-analysis-of-many-samples" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Regression analysis of many samples<a href="differential-abundance.html#regression-analysis-of-many-samples" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In most cases, a DA analysis can be understood as regression of microbial abundance variables against one or more covariates describing predictive properties of the sample.
The simplest such regression relates a microbial abundance variable <span class="math inline">\(y\)</span> to a covariate <span class="math inline">\(x\)</span> and a <em>residual (or unexplained) error</em> <span class="math inline">\(\varepsilon\)</span> via the simple linear regression formula
<span class="math display" id="eq:regression">\[\begin{align}
  \tag{3.3}
  y^{(a)} = \alpha + \beta x^{(a)} + \varepsilon^{(a)}.
\end{align}\]</span>
A multiplicative DA analysis can be conducted by setting <span class="math inline">\(y\)</span> equal to the logarithm of the untransformed abundance measurement (<span class="math inline">\(\tilde P_i\)</span>, <span class="math inline">\(\tilde R_{i/j}\)</span>, or <span class="math inline">\(\tilde A_i\)</span>).
For example, <span class="math inline">\(y\)</span> may be the log absolute abundance of species <span class="math inline">\(i\)</span> (<span class="math inline">\(\log A_i^{(a)}\)</span>) and <span class="math inline">\(x\)</span> may be a continuous variable (such as pH) or a binary variable (such as <span class="math inline">\(x=1\)</span> for treated patients and <span class="math inline">\(x=0\)</span> for untreated controls).
The regression coefficients <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span> are parameters to be estimated from the data.
A DA regression analysis is primarily interested in the slope coefficient <span class="math inline">\(\beta\)</span>, which determines how the average or expected value of <span class="math inline">\(y\)</span> increases with <span class="math inline">\(x\)</span>.
For a binary covariate, <span class="math inline">\(\beta\)</span> equals the average difference in <span class="math inline">\(y\)</span> between samples with <span class="math inline">\(x=1\)</span> versus those with <span class="math inline">\(x=0\)</span>.
When <span class="math inline">\(y\)</span> equals the logarithm of a species’ relative or absolute abundance, the coefficient <span class="math inline">\(\beta\)</span> equals the average log fold difference (LFD) in abundance between conditions corresponding to a unit increase in <span class="math inline">\(x\)</span>.
Although researchers often use more sophisticated models, the simple linear regression model in Equation <a href="differential-abundance.html#eq:regression">(3.3)</a> provides an intuitive basis for understanding the effect of taxonomic bias on regression-based DA analyses.</p>
<p>Appendix <a href="appendix-regression.html#appendix-regression">A</a> describes a general framework for finding the error in the estimated slope coefficient <span class="math inline">\(\hat \beta\)</span> caused by taxonomic bias; here, we summarize and illustrate these results for the various abundance measures described in Section <a href="abundance-measurement.html#abundance-measurement">2</a>.</p>
<p>The effect that taxonomic bias has on the estimate <span class="math inline">\(\hat \beta\)</span> mirrors its effect on the measured FDs between a pair of individual samples:
For abundance measures with a FE that is constant across samples, bias does not affect <span class="math inline">\(\hat \beta\)</span>.
For abundance measures where the FE varies, bias can affect <span class="math inline">\(\hat \beta\)</span>, with its exact effect depending on how the mean efficiency varies across samples.</p>
<p>We illustrate the constant-FE case by considering a regression analysis of the log ratio between species <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>, <span class="math inline">\(\log \tilde R_{i/j}\)</span>.
Taxonomic bias creates a constant FE in <span class="math inline">\(\tilde R_{i/j}\)</span> that translates to a constant additive error in <span class="math inline">\(\log \tilde R_{i/j}\)</span>.
A constant (additive) shift in a response variable only affects the estimated intercept <span class="math inline">\(\hat \alpha\)</span> and not the estimated slope <span class="math inline">\(\hat \beta\)</span>.
Thus, in this case, taxonomic bias does not affect <span class="math inline">\(\hat \beta\)</span>.</p>
<p>We illustrate the varying-FE case by considering the absolute abundance of a species <span class="math inline">\(i\)</span> obtained by multiplying the species’ MGS proportion by a non-MGS measure of total-abundance, using Equation <a href="abundance-measurement.html#eq:density-prop-meas">(2.11)</a>.
For simplicity, suppose that total-community abundance has been measured completely accurately.
In this case, the measured abundance <span class="math inline">\(\tilde A_i\)</span> has a FE that varies inversely with the mean efficiency <span class="math inline">\(\bar B\)</span>.
If <span class="math inline">\(\bar B\)</span> is stable across samples, then the additive error in <span class="math inline">\(\log \tilde A_i\)</span> will be approximately constant and <span class="math inline">\(\hat \beta\)</span> is again unaffected—bias has no effect on the DA result.
If, on the other hand, <span class="math inline">\(\log \bar B\)</span> tends to linearly vary with the covariate <span class="math inline">\(x\)</span>, then the additive error in <span class="math inline">\(\log \tilde A_i\)</span> will linearly vary in the opposite direction and cause a systematic error in <span class="math inline">\(\hat \beta\)</span> that is equal in magnitude but opposite in sign to the slope of <span class="math inline">\(\log \bar B\)</span> with <span class="math inline">\(x\)</span>.
A third possibility is that <span class="math inline">\(\bar B\)</span> varies in an essentially random manner that is independent of the covariate <span class="math inline">\(x\)</span>.
In this scenario, the error from taxonomic bias in the measurement <span class="math inline">\(\log \tilde A_{i}\)</span> acts to increase the noise in <span class="math inline">\(\hat \beta\)</span> (i.e., increase its standard error), but does not cause a systematic error.</p>
<p>Figure <a href="differential-abundance.html#fig:regression-example">3.1</a> shows a simulated example of a regression analysis of the absolute abundances of 10 species across two conditions in which the mean efficiency is systematically greater in the condition <span class="math inline">\(x=1\)</span> than in the condition <span class="math inline">\(x=0\)</span>.
The increase in the log mean efficiency from <span class="math inline">\(x=0\)</span> to <span class="math inline">\(x=1\)</span> (Figure <a href="differential-abundance.html#fig:regression-example">3.1</a> A and B) corresponds to an artificial decrease in the estimated LFD between conditions for each species (Figure <a href="differential-abundance.html#fig:regression-example">3.1</a> C and D).
The absolute error created by bias on the LFD estimate is the same for each species; however, its scientific impact varies.
For the three species with large positive LFDs (Species 9, 10, and 1), bias decreases the magnitude of the LFD estimate.
In contrast, for the three species with large negative LFDs (Species 8, 2, and 3), bias increases the magnitude of the estimate.
For one species with a small positive LFD (Species 5), the effect of bias results in a negative LFD estimate (direction error).
The three remaining species (Species 7, 6, and 4) have LFDs near zero, but bias causes them to have large negative LFDs.</p>
<!-- Figure: Regression example -->
<div class="figure"><span style="display:block;" id="fig:regression-example"></span>
<img src="notebook/_posts/2021-08-03-simulate-regression-example/simulate-regression-example_files/figure-html5/main-figure-2-1.svg" alt="Taxonomic bias distorts multi-sample differential abundance analysis when the mean efficiency of samples is associated with the covariate of interest. This figure shows the results of a regression analysis of simulated microbiomes consisting of 10 species and 50 samples from two environmental conditions, which are indexed by \(x=0\) and \(x=1\). Panel A shows the (log) measurement efficiencies of the 10 species against their differential-abundance parameter values (expected LFD in absolute abundance between a sample from condition \(x=1\) to a sample from condition \(x=0\)). The simulation was created so that the species with the largest efficiency (Sp. 9) also has the largest positive expected LFD. The increased relative abundance of Sp. 9 in condition \(x=1\) drives an increase in the (log) mean efficiency (Panel B), which induces a systematic and consistent negative shift in the estimates of the mean LFD for each species (Panels C and D). Panel D shows the estimated mean LFD with 95% confidence interval for each species, when either estimated the actual abundances or the inaccurate measured abundances. The error (estimate from measured abundances minus estimate from actual abundances; red arrow in Panel D) equals the negative mean LFD the mean efficiency (red arrow in Panel B)." width="90%" />
<p class="caption">
Figure 3.1: <strong>Taxonomic bias distorts multi-sample differential abundance analysis when the mean efficiency of samples is associated with the covariate of interest.</strong> This figure shows the results of a regression analysis of simulated microbiomes consisting of 10 species and 50 samples from two environmental conditions, which are indexed by <span class="math inline">\(x=0\)</span> and <span class="math inline">\(x=1\)</span>. Panel A shows the (log) measurement efficiencies of the 10 species against their differential-abundance parameter values (expected LFD in absolute abundance between a sample from condition <span class="math inline">\(x=1\)</span> to a sample from condition <span class="math inline">\(x=0\)</span>). The simulation was created so that the species with the largest efficiency (Sp. 9) also has the largest positive expected LFD. The increased relative abundance of Sp. 9 in condition <span class="math inline">\(x=1\)</span> drives an increase in the (log) mean efficiency (Panel B), which induces a systematic and consistent negative shift in the estimates of the mean LFD for each species (Panels C and D). Panel D shows the estimated mean LFD with 95% confidence interval for each species, when either estimated the actual abundances or the inaccurate measured abundances. The error (estimate from measured abundances minus estimate from actual abundances; red arrow in Panel D) equals the negative mean LFD the mean efficiency (red arrow in Panel B).
</p>
</div>

<!-- end Figure -->
</div>
<div id="rank-based-analyses" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Rank-based analyses<a href="differential-abundance.html#rank-based-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Rank-based DA methods work by first applying a cross-sample rank-transformation to the abundance variable <span class="math inline">\(y\)</span>: The sample in which <span class="math inline">\(y\)</span> has the smallest value receives a rank of 1, the sample with the second smallest value receives a rank of 2, etc.
These methods then analyze how the average rank of <span class="math inline">\(y\)</span> varies across sample conditions (<span class="citation">Conover (<a href="#ref-conover2012ther" role="doc-biblioref">2012</a>)</span>).
Rank-based methods commonly used in microbiome DA analysis include the Wilcoxon-Mann-Whitney and Kruskal-Wallis tests for a discrete covariate (e.g. case vs control) and Spearman’s rank correlation for a continuous covariate (e.g. pH); these tests have collectively been applied to species proportions (<span class="citation">Callahan et al. (<a href="#ref-callahan2017repl" role="doc-biblioref">2017</a>)</span>,<span class="citation">Fettweis et al. (<a href="#ref-fettweis2019thev" role="doc-biblioref">2019</a>)</span>), ratios (<span class="citation">Fernandes et al. (<a href="#ref-fernandes2014unif" role="doc-biblioref">2014</a>)</span>), and absolute abundances (<span class="citation">Vieira-Silva et al. (<a href="#ref-vieirasilva2019quan" role="doc-biblioref">2019</a>)</span>).</p>
<p>Multiplying a variable by a constant factor does not change its cross-sample ranks, making rank-based methods unaffected by a constant FE in the abundance variable <span class="math inline">\(y\)</span>.
Thus in our MGS measurement model, taxonomic bias does not affect rank-based DA analyses of species ratios or absolute abundances derived from a reference species.
But for species proportions and absolute abundances derived from a total-community abundance measurement, variation in the mean efficiency <span class="math inline">\(\bar B\)</span> can affect the cross-sample ranks and cause error in DA results if <span class="math inline">\(\bar B\)</span> is associated with the covariate.
Further work is needed to determine whether rank-based DA methods are significantly more robust to variation in <span class="math inline">\(\bar B\)</span> than multiplicative DA methods.</p>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-callahan2017repl" class="csl-entry">
Callahan, Benjamin J, Daniel B DiGiulio, Daniela S Aliaga Goltsman, Christine L Sun, Elizabeth K Costello, Pratheepa Jeganathan, Joseph R Biggio, et al. 2017. <span>“<span class="nocase">Replication and refinement of a vaginal microbial signature of preterm birth in two racially distinct cohorts of US women</span>.”</span> <em>Proc. Natl. Acad. Sci. U. S. A.</em> 114 (37): 9966–71. <a href="https://doi.org/10.1073/pnas.1705899114">https://doi.org/10.1073/pnas.1705899114</a>.
</div>
<div id="ref-conover2012ther" class="csl-entry">
Conover, W Jay. 2012. <span>“The Rank Transformation-an Easy and Intuitive Way to Connect Many Nonparametric Methods to Their Parametric Counterparts for Seamless Teaching Introductory Statistics Courses.”</span> <em>Wiley Interdiscip. Rev. Comput. Stat.</em> 4 (5): 432–38. <a href="https://doi.org/10.1002/wics.1216">https://doi.org/10.1002/wics.1216</a>.
</div>
<div id="ref-fernandes2014unif" class="csl-entry">
Fernandes, Andrew D, Jennifer NS Reid, Jean M Macklaim, Thomas A McMurrough, David R Edgell, and Gregory B Gloor. 2014. <span>“<span class="nocase">Unifying the analysis of high-throughput sequencing datasets: characterizing RNA-seq, 16S rRNA gene sequencing and selective growth experiments by compositional data analysis</span>.”</span> <em>Microbiome</em> 2 (1): 15. <a href="https://doi.org/10.1186/2049-2618-2-15">https://doi.org/10.1186/2049-2618-2-15</a>.
</div>
<div id="ref-fettweis2019thev" class="csl-entry">
Fettweis, Jennifer M., Myrna G. Serrano, Jamie Paul Brooks, David J. Edwards, Philippe H. Girerd, Hardik I. Parikh, Bernice Huang, et al. 2019. <span>“<span class="nocase">The vaginal microbiome and preterm birth</span>.”</span> <em>Nat. Med.</em> 25 (6): 1012–21. <a href="https://doi.org/10.1038/s41591-019-0450-2">https://doi.org/10.1038/s41591-019-0450-2</a>.
</div>
<div id="ref-love2014mode" class="csl-entry">
Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. <span>“<span class="nocase">Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</span>.”</span> <em>Genome Biol.</em> 15 (12): 550. <a href="https://doi.org/10.1186/s13059-014-0550-8">https://doi.org/10.1186/s13059-014-0550-8</a>.
</div>
<div id="ref-martin2020mode" class="csl-entry">
Martin, Bryan D., Daniela Witten, and Amy D. Willis. 2020. <span>“<span class="nocase">Modeling microbial abundances and dysbiosis with beta-binomial regression</span>.”</span> <em>Ann. Appl. Stat.</em> 14 (1): 94–115. <a href="https://doi.org/10.1214/19-AOAS1283">https://doi.org/10.1214/19-AOAS1283</a>.
</div>
<div id="ref-stein2013ecol" class="csl-entry">
Stein, Richard R., Vanni Bucci, Nora C. Toussaint, Charlie G. Buffie, Gunnar Rätsch, Eric G. Pamer, Chris Sander, and João B. Xavier. 2013. <span>“<span class="nocase">Ecological Modeling from Time-Series Inference: Insight into Dynamics and Stability of Intestinal Microbiota</span>.”</span> Edited by Christian von Mering. <em>PLoS Comput. Biol.</em> 9 (12): e1003388. <a href="https://doi.org/10.1371/journal.pcbi.1003388">https://doi.org/10.1371/journal.pcbi.1003388</a>.
</div>
<div id="ref-vieirasilva2019quan" class="csl-entry">
Vieira-Silva, Sara, João Sabino, Mireia Valles-Colomer, Gwen Falony, Gunter Kathagen, Clara Caenepeel, Isabelle Cleynen, Schalk van der Merwe, Séverine Vermeire, and Jeroen Raes. 2019. <span>“<span class="nocase">Quantitative microbiome profiling disentangles inflammation- and bile duct obstruction-associated microbiota alterations across PSC/IBD diagnoses</span>.”</span> <em>Nat. Microbiol.</em> 4 (11): 1826–31. <a href="https://doi.org/10.1038/s41564-019-0483-9">https://doi.org/10.1038/s41564-019-0483-9</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="abundance-measurement.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="case-studies.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/mikemc/differential-abundance-theory/edit/main/differential-abundance.Rmd",
"text": "Edit"
},
"history": {
"link": "https://github.com/mikemc/differential-abundance-theory/commits/main/differential-abundance.Rmd",
"text": null
},
"view": {
"link": "https://github.com/mikemc/differential-abundance-theory/blob/main/differential-abundance.Rmd",
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
