<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 How taxonomic bias affects abundance measurements | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements</title>
  <meta name="description" content="2 How taxonomic bias affects abundance measurements | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="2 How taxonomic bias affects abundance measurements | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 How taxonomic bias affects abundance measurements | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements" />
  
  
  

<meta name="author" content="Michael R. McLaren" />
<meta name="author" content="Jacob T. Nearing" />
<meta name="author" content="Amy D. Willis" />
<meta name="author" content="Karen G. Lloyd" />
<meta name="author" content="Benjamin J. Callahan" />


<meta name="date" content="2021-11-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introduction.html"/>
<link rel="next" href="differential-abundance.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
TeX: {extensions: ["cancel.js"]}
});
</script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="abundance-measurement.html"><a href="abundance-measurement.html"><i class="fa fa-check"></i><b>2</b> How taxonomic bias affects abundance measurements</a>
<ul>
<li class="chapter" data-level="2.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#model-of-mgs-measurement"><i class="fa fa-check"></i><b>2.1</b> Model of MGS measurement</a></li>
<li class="chapter" data-level="2.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#relative-abundances-proportions-and-ratios"><i class="fa fa-check"></i><b>2.2</b> Relative abundances (proportions and ratios)</a></li>
<li class="chapter" data-level="2.3" data-path="abundance-measurement.html"><a href="abundance-measurement.html#absolute-abundances-cell-densities"><i class="fa fa-check"></i><b>2.3</b> Absolute abundances (cell densities)</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>3</b> How taxonomic bias affects differential-abundance analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-abundance.html"><a href="differential-abundance.html#change-between-a-pair-of-samples"><i class="fa fa-check"></i><b>3.1</b> Change between a pair of samples</a></li>
<li class="chapter" data-level="3.2" data-path="differential-abundance.html"><a href="differential-abundance.html#regression-analysis-of-many-samples"><i class="fa fa-check"></i><b>3.2</b> Regression analysis of many samples</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="implications.html"><a href="implications.html"><i class="fa fa-check"></i><b>4</b> Implications for real-world inference </a>
<ul>
<li class="chapter" data-level="4.1" data-path="implications.html"><a href="implications.html#systematic-error-in-slope-or-lfc-estimates"><i class="fa fa-check"></i><b>4.1</b> Systematic error in slope or LFC estimates</a></li>
<li class="chapter" data-level="4.2" data-path="implications.html"><a href="implications.html#loss-of-precision-and-power"><i class="fa fa-check"></i><b>4.2</b> Loss of precision and power</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="solutions.html"><a href="solutions.html"><i class="fa fa-check"></i><b>5</b> Potential solutions</a>
<ul>
<li class="chapter" data-level="5.1" data-path="solutions.html"><a href="solutions.html#analyses-of-fold-changes-in-species-ratios"><i class="fa fa-check"></i><b>5.1</b> Analyses of fold changes in species ratios</a></li>
<li class="chapter" data-level="5.2" data-path="solutions.html"><a href="solutions.html#calibrate-compositions"><i class="fa fa-check"></i><b>5.2</b> Calibration using community controls</a></li>
<li class="chapter" data-level="5.3" data-path="solutions.html"><a href="solutions.html#calibration-using-reference-species"><i class="fa fa-check"></i><b>5.3</b> Calibration using reference species</a></li>
<li class="chapter" data-level="5.4" data-path="solutions.html"><a href="solutions.html#choosing-complementary-total-density-and-metagenomics-measurements"><i class="fa fa-check"></i><b>5.4</b> Choosing complementary total-density and metagenomics measurements</a></li>
<li class="chapter" data-level="5.5" data-path="solutions.html"><a href="solutions.html#bias-sensitivity-analysis"><i class="fa fa-check"></i><b>5.5</b> Bias-sensitivity analysis</a></li>
<li class="chapter" data-level="5.6" data-path="solutions.html"><a href="solutions.html#bias-aware-meta-analysis"><i class="fa fa-check"></i><b>5.6</b> Bias-aware meta-analysis</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="model-details.html"><a href="model-details.html"><i class="fa fa-check"></i><b>A</b> Model details</a>
<ul>
<li class="chapter" data-level="A.1" data-path="model-details.html"><a href="model-details.html#mgs-measurement"><i class="fa fa-check"></i><b>A.1</b> MGS measurement</a>
<ul>
<li class="chapter" data-level="A.1.1" data-path="model-details.html"><a href="model-details.html#revised-main-text-equations"><i class="fa fa-check"></i><b>A.1.1</b> Revised main text equations</a></li>
</ul></li>
<li class="chapter" data-level="A.2" data-path="model-details.html"><a href="model-details.html#total-density-bias"><i class="fa fa-check"></i><b>A.2</b> Taxonomic bias in total-density measurement</a></li>
<li class="chapter" data-level="A.3" data-path="model-details.html"><a href="model-details.html#total-density-ref"><i class="fa fa-check"></i><b>A.3</b> Using reference species for total-density normalization</a></li>
<li class="chapter" data-level="A.4" data-path="model-details.html"><a href="model-details.html#linearity-of-extraction"><i class="fa fa-check"></i><b>A.4</b> Linearity of extraction</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html"><i class="fa fa-check"></i><b>B</b> Review of experimental methods for obtaining absolute densities</a>
<ul>
<li class="chapter" data-level="B.1" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#measurement-of-total-cell-density"><i class="fa fa-check"></i><b>B.1</b> Measurement of total cell density</a></li>
<li class="chapter" data-level="B.2" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#measurement-of-total-dna-density-post-extraction"><i class="fa fa-check"></i><b>B.2</b> Measurement of total DNA density post extraction</a></li>
<li class="chapter" data-level="B.3" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#equivolumetric-protocol"><i class="fa fa-check"></i><b>B.3</b> Equivolumetric protocol</a></li>
<li class="chapter" data-level="B.4" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#housekeeping-species"><i class="fa fa-check"></i><b>B.4</b> Housekeeping species</a></li>
<li class="chapter" data-level="B.5" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#spike-ins"><i class="fa fa-check"></i><b>B.5</b> Spike-ins</a></li>
<li class="chapter" data-level="B.6" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#targeted-measurements"><i class="fa fa-check"></i><b>B.6</b> Targeted measurements</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="appendix-regression.html"><a href="appendix-regression.html"><i class="fa fa-check"></i><b>C</b> Error in estimated regression coefficients and standard errors</a></li>
<li class="chapter" data-level="D" data-path="supplemental-figures.html"><a href="supplemental-figures.html"><i class="fa fa-check"></i><b>D</b> Supplemental figures</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="abundance-measurement" class="section level1" number="2">
<h1><span class="header-section-number">2</span> How taxonomic bias affects abundance measurements</h1>
<p>To understand how bias affects the measured differential abundance between samples, we first consider how it affects measurements of individual samples.</p>
<div id="model-of-mgs-measurement" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Model of MGS measurement</h2>
<p>Our primary tool for understanding the impact of taxonomic bias on MGS measurement is the theoretical model of MGS measurement developed and empirically validated by <span class="citation">McLaren, Willis, and Callahan (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span>.
This model is the simplest model of MGS measurement that includes multiplicative taxonomic bias while respecting the <em>compositional</em> nature of sequencing measurements, in which the total read count for a sample is unrelated to its total cell number or density (<span class="citation">Gloor et al. (<a href="#ref-gloor2017micr" role="doc-biblioref">2017</a>)</span>).
We consider a set of microbial species that are measured by a specific MGS protocol that extracts, sequences, and bioinformatically analyzes samples to assign reads to various microbial species.
We assume that assigned reads have been correctly assigned to the given species and sample.
Thus we ignore the possibility of contamination or false-positive taxonomic assignment, though we allow that many reads may not receive a taxonomic assignment and so ultimately be discarded.
Unless otherwise stated, we treat the sequencing measurement as deterministic, ignoring the ‘noise’ or random error due to the random sampling of sequencing reads and other aspects of the MGS process.</p>
<p>Our model stipulates that the assigned read count of species <span class="math inline">\(i\)</span> in sample <span class="math inline">\(a\)</span> equals its cell density multiplied by a species-specific factor and a sample-specific factor,
<span class="math display" id="eq:measurement-model">\[\begin{align}
  \tag{2.1}
  \text{reads}_{i}(a)
  = \text{density}_{i}(a) \quad \cdot
    \underbrace{\text{efficiency}_{i}}_{\substack{\text{species specific,} \\  \text{sample independent}}}
    \cdot \quad
    \underbrace{\text{sequencing effort}(a)}_{\substack{\text{species independent,} \\  \text{sample specific}}}.
\end{align}\]</span>
The species-specific factor is the <em>relative measurement efficiency</em> (or <em>efficiency</em> for short) of the species, which represents how much more easily that species is measured (converted from cells to assigned reads) relative to an abitrarily chosen reference species (<span class="citation">McLaren, Willis, and Callahan (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span>).
The sample-specific factor, which we call the <em>sequencing effort</em> of the sample, reflects the fact that the number of reads per unit cell density varies among samples even in the absence of taxonomic bias due to variation in total density, library normalization, and total sequencing-run output.
Equation <a href="abundance-measurement.html#eq:measurement-model">(2.1)</a> implies that the total reads from all species in the sample equal
<span class="math display" id="eq:total-reads">\[\begin{align}
  \tag{2.2}
  \text{total reads}(a)
    = \text{total density}(a) \cdot \text{mean efficiency}(a) \cdot \text{sequencing effort}(a),
\end{align}\]</span>
where
<span class="math display" id="eq:mean-efficiency">\[\begin{align}
  \tag{2.3}
  \text{mean efficiency}(a) 
  \equiv \frac{\sum_{j}\text{density}_j(a)\cdot \text{efficiency}_j}{\text{total density}(a)}
\end{align}\]</span>
is the average or mean efficiency of cells in the sample.</p>
</div>
<div id="relative-abundances-proportions-and-ratios" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Relative abundances (proportions and ratios)</h2>
<p>We distinguish between two types of species-level <em>relative abundances</em>.
The first type consists of the <em>ratios</em> among the abundances of two or more species, such as a ratio of 5:4:2 among three species.
The second type consists of the <em>proportion</em> of an individual species, equal to its abundance divided by that of a particular root taxon (e.g. Prokaryotes).
DA analyses of proportions consider how the proportion of a species changes indepently of other species, whereas DA analyses of ratios consider the conserted change in relative abundance among two or more species.
Although proportion-based analysis is dominant, microbiome researchers increasingly use ratio-based analyses adapted from the statistical framework of Compositional Data Analysis (CoDA).
The different manner in which taxonomic bias affects ratios vs. proportions (<span class="citation">McLaren, Willis, and Callahan (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span>) forms the basis in subsequent sections for understanding the extent to which its effects cancel in DA analysis.</p>
<p>The standard measurement of the proportion of species <span class="math inline">\(i\)</span> in sample <span class="math inline">\(a\)</span> is given by the ratio of its read count to the total,
<span class="math display" id="eq:prop-meas">\[\begin{align}
  \tag{2.4}
  \widehat{\text{prop}}_{i}(a) = \frac{\text{reads}_i(a)}{\text{total reads}(a)}.
\end{align}\]</span>
From Equations <a href="abundance-measurement.html#eq:measurement-model">(2.1)</a>, <a href="abundance-measurement.html#eq:total-reads">(2.2)</a>, and <a href="abundance-measurement.html#eq:prop-meas">(2.4)</a>, it follows that
<span class="math display" id="eq:prop-error">\[\begin{align}
  \tag{2.5}
  \widehat{\text{prop}}_{i}(a)
  &amp;= \text{prop}_{i}(a) \cdot \frac{\text{efficiency}_{i}}{\text{mean efficiency}(a)},
\end{align}\]</span>
where <span class="math inline">\(\text{prop}_{i}(a) = \text{density}_{i}(a) / \text{total density}(a)\)</span> is the actual proportion.
Equation <a href="abundance-measurement.html#eq:prop-error">(2.5)</a> says that taxonomic bias creates a multiplicative error equal to the ratio of the species’ efficiency to the mean efficiency in the sample.
Consequently, the same species can be over- or under-measured depending on the composition of the given sample (Figure <a href="abundance-measurement.html#fig:error-proportions">2.1</a>).
For instance, in samples from two hypothetical communities in Figure <a href="abundance-measurement.html#fig:error-proportions">2.1</a>, Species 3 has an efficiency of 6 and is under-measured in Sample 1 (which has a mean efficiency of 8.33) but over-measured in Sample 2 (which has a mean efficiency of 3.15).</p>
<p>The measured ratio between species <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> is given by the ratio of their read counts,
<span class="math display" id="eq:ratio-meas">\[\begin{align}
  \tag{2.6}
  \widehat{\text{ratio}}_{i/j}(a) = \frac{\text{reads}_i(a)}{\text{reads}_j(a)}.
\end{align}\]</span>
From Equations <a href="abundance-measurement.html#eq:measurement-model">(2.1)</a> and <a href="abundance-measurement.html#eq:ratio-meas">(2.6)</a>, it follows that
<span class="math display" id="eq:ratio-error">\[\begin{align}
  \tag{2.7}
  \widehat{\text{ratio}}_{i/j}(a)
%  \frac{\text{reads}_{i}(a)}{\text{reads}_{j}(a)}
  &amp;= \frac{\text{density}_{i}(a)}{\text{density}_{j}(a)} \cdot \frac{\text{efficiency}_{i}}{\text{efficiency}_{j}}.
\end{align}\]</span>
The fold error in the ratio between two species simply equals the ratio in their efficiencies and is therefore constant across samples.
For instance, in the example of Figure <a href="abundance-measurement.html#fig:error-proportions">2.1</a>, the ratio of Species 3 (with an efficiency of 6) to Species 1 (with an efficiency of 1) is over-estimated by a factor of 6 in both communities despite their varying compositions.</p>
<!-- begin figure -->
<div class="figure"><span style="display:block;" id="fig:error-proportions"></span>
<img src="figures/illustrations/error-proportions.svg" alt="Taxonomic bias creates sample-dependent multiplicative errors in species proportions, which can lead to inaccurate fold changes between samples. Top row: Error in proportions measured by MGS in two microbiome samples that contain different relative abundances of three species. Bottom row: Error in the measured fold-change in the third species that is derived from these measurements." width="90%" />
<p class="caption">
Figure 2.1: <strong>Taxonomic bias creates sample-dependent multiplicative errors in species proportions, which can lead to inaccurate fold changes between samples.</strong> Top row: Error in proportions measured by MGS in two microbiome samples that contain different relative abundances of three species. Bottom row: Error in the measured fold-change in the third species that is derived from these measurements.
</p>
</div>

<!-- end figure -->
<p><strong>Difference between species proportions and species ratios:</strong>
Consistent taxonomic bias at the species level creates a consistent fold error in the ratios among species, but a varying fold error in the proportions of individual species.
But a proportion is, after all, just a ratio between two taxa in which the denominator taxon is the entire taxonomic domain under study.
The difference in behavior from that of ratios between species arises because the efficiency of higher-level taxa, which consist of the aggregate of many species, varies across samples with the changing relative abundances among its constituent species.</p>
</div>
<div id="absolute-abundances-cell-densities" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Absolute abundances (cell densities)</h2>
<p>A wide range of experimental methods exist for converting the relative-abundance information in MGS measurements into measurements of (absolute) cell densities;
however, these methods can be generally partitioned into two types, based on whether the absolute-abundance information derives from knowing the density of the total community or knowing the density of one or more individual species.</p>
<p><strong>Methods using the density of the total community:</strong>
The total microbial density of a community can be assayed by a variety of methods (reviewed in Appendix <a href="review-absolute-methods.html#review-absolute-methods">B</a>).
Once the total density is known, the density of individual species can be estimated by multiplying the total density by the species’ proportions (as measured by MGS),
<span class="math display" id="eq:density-prop-meas">\[\begin{align}
  \tag{2.8}
  \widehat{\text{density}}_{i}(a) 
  &amp;= \widehat{\text{prop}}_{i}(a) \cdot \widehat{\text{total density}}(a)
\\&amp;= \text{reads}_{i}(a) \cdot \frac{\widehat{\text{total density}}(a)}{\text{total reads}(a)}.
\end{align}\]</span>
We can rewrite the species’ density measurements to account for the effect of taxonomic bias on the MGS proportions (Equation <a href="abundance-measurement.html#eq:prop-error">(2.5)</a>) and any error in the total density measurement,
<span class="math display" id="eq:density-prop-error">\[\begin{align}
  \tag{2.9}
  \widehat{\text{density}}_{i}(a) 
  = \text{density}_{i}(a) \cdot \frac{\text{efficiency}_{i}}{\text{mean efficiency}(a)} 
%  \cdot \frac{\widehat{\text{total density}}(a)}{\text{total density}(a)}.
  \cdot \begin{array}{c} \text{fold error in} \\ \widehat{\text{total density}}(a) \end{array}.
\end{align}\]</span>
The variable fold error that bias creates in the measured proportions directly transfers to the density measurements.</p>
<p>Taxonomic bias can also create systematic error in the total-density measurements (Appendix <a href="model-details.html#total-density-bias">A.2</a>).
For now, we suppose that the total density can be measured accurately;
later, Section <a href="solutions.html#solutions">5</a> describes how experiments might be designed such that taxonomic bias in the total density measurement offsets that in the MGS proportions, leading to a (more) consistent fold error in the measured densities.</p>
<p><strong>Methods using the density of one or more reference species:</strong>
The density of one or more species can be directly measured using targeted measurement methods like species-specific qPCR; alternatively, the densities of one or more species may be thought to be constant across samples either from background knowledge or because they have been spiked in at a fixed density (these situations are reviewed in Appendix <a href="review-absolute-methods.html#review-absolute-methods">B</a>).
These <em>reference species</em> with known or constant density can be used to convert read counts to densities for all species.
For instance, given the density of a reference species <span class="math inline">\(r\)</span>, we can estimate the density of a focal species <span class="math inline">\(i\)</span> by
<span class="math display" id="eq:density-ratio-meas">\[\begin{align}
  \tag{2.10}
%  \widehat{\text{density}}_{i}(a) = \frac{\text{reads}_{i}(a)}{\text{reads}_{r}(a)} \cdot \widehat{\text{density}}_{r}(a).
  \widehat{\text{density}}_{i}(a) = \text{reads}_{i}(a) \cdot \frac{\widehat{\text{density}}_{r}(a)}{\text{reads}_{r}(a)}.
\end{align}\]</span>
Alternatively, if we only know that the reference species is constant, we can set <span class="math inline">\(\widehat{\text{density}}_{r}(a)\)</span> in Equation <a href="abundance-measurement.html#eq:density-ratio-meas">(2.10)</a> equal to 1 to estimate the density in unknown but fixed units, which is sufficient for multiplicative DA analysis.
(Appendix <a href="model-details.html#model-details">A</a> considers possible extensions to multiple reference species.)
It follows from Equation <a href="abundance-measurement.html#eq:ratio-error">(2.7)</a> that the error caused by taxonomic bias in this density measurement is
<span class="math display" id="eq:density-ratio-error">\[\begin{align}
  \tag{2.11}
  \widehat{\text{density}}_{i}(a) 
  = \text{density}_{i}(a) \cdot \frac{\text{efficiency}_{i}}{\text{efficiency}_{r}} 
  \cdot \begin{array}{c} \text{fold error in} \\ \widehat{\text{density}_r}(a) \end{array}.
\end{align}\]</span>
Here, it is the constant error in the measured ratio of species <span class="math inline">\(i\)</span> to <span class="math inline">\(r\)</span> that propagates into the density measurement.
There will generally be systematic error in the density of the reference species; however, if the systematic fold error is constant across samples, so will be that of the densities of other species.</p>
<!-- NOTE: the above doesn't quite make sense for DNA spike-ins -->
<p><strong>Differences between the two approaches:</strong>
Equations <a href="abundance-measurement.html#eq:density-prop-error">(2.9)</a> and <a href="abundance-measurement.html#eq:density-ratio-error">(2.11)</a> show that the manner in which taxonomic bias in the sequencing measurement impacts the two approaches mirrors that of proportions and ratios.
If the fold error in the total-density or reference species measurement is negligible (or at least constant) then the fold errors in species densities by the total-density approach vary across samples with the mean efficiency, whereas the fold errors by the reference-species approach are constant.
The fundamental determinant of whether constant error is obtained is not which of these two equations is ultimately used, but rather whether densities of individual species, which (we assume) have constant efficiency, or the total community, which has varying efficiency, provide the absolute-abundance information.
Studies using spike-ins as constant reference species will often, instead of using Equation <a href="abundance-measurement.html#eq:density-ratio-meas">(2.10)</a>, first use the spike-in species to estimate the total density, and then estimate the density of focal species with Equation <a href="abundance-measurement.html#eq:density-prop-meas">(2.8)</a>.
If the calculation is done carefully, however, this method still yields constant fold errors so long as the species efficiencies (including those of the spike-in) are constant across samples (Appendix <a href="model-details.html#total-density-ref">A.3</a>).</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-gloor2017micr" class="csl-entry">
Gloor, Gregory B., Jean M. Macklaim, Vera Pawlowsky-Glahn, and Juan J. Egozcue. 2017. <span>“<span>Microbiome Datasets Are Compositional: And This Is Not Optional</span>.”</span> <em>Front. Microbiol.</em> 8 (November): 2224. <a href="https://doi.org/10.3389/fmicb.2017.02224">https://doi.org/10.3389/fmicb.2017.02224</a>.
</div>
<div id="ref-mclaren2019cons" class="csl-entry">
McLaren, Michael R, Amy D Willis, and Benjamin J Callahan. 2019. <span>“<span class="nocase">Consistent and correctable bias in metagenomic sequencing experiments</span>.”</span> <em>Elife</em> 8 (September): 46923. <a href="https://doi.org/10.7554/eLife.46923">https://doi.org/10.7554/eLife.46923</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-abundance.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/mikemc/differential-abundance-theory/edit/main/abundance-measurement.Rmd",
"text": "Edit"
},
"history": {
"link": "https://github.com/mikemc/differential-abundance-theory/commits/main/abundance-measurement.Rmd",
"text": null
},
"view": {
"link": "https://github.com/mikemc/differential-abundance-theory/blob/main/abundance-measurement.Rmd",
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
