<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>A Model details | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements</title>
  <meta name="description" content="A Model details | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="A Model details | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A Model details | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements" />
  
  
  

<meta name="author" content="Michael R. McLaren" />
<meta name="author" content="Jacob T. Nearing" />
<meta name="author" content="Amy D. Willis" />
<meta name="author" content="Karen G. Lloyd" />
<meta name="author" content="Benjamin J. Callahan" />


<meta name="date" content="2021-11-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="solutions.html"/>
<link rel="next" href="review-absolute-methods.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
TeX: {extensions: ["cancel.js"]}
});
</script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="abundance-measurement.html"><a href="abundance-measurement.html"><i class="fa fa-check"></i><b>2</b> How taxonomic bias affects abundance measurements</a>
<ul>
<li class="chapter" data-level="2.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#model-of-mgs-measurement"><i class="fa fa-check"></i><b>2.1</b> Model of MGS measurement</a></li>
<li class="chapter" data-level="2.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#relative-abundances-proportions-and-ratios"><i class="fa fa-check"></i><b>2.2</b> Relative abundances (proportions and ratios)</a></li>
<li class="chapter" data-level="2.3" data-path="abundance-measurement.html"><a href="abundance-measurement.html#absolute-abundances-cell-densities"><i class="fa fa-check"></i><b>2.3</b> Absolute abundances (cell densities)</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>3</b> How taxonomic bias affects differential-abundance analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-abundance.html"><a href="differential-abundance.html#change-between-a-pair-of-samples"><i class="fa fa-check"></i><b>3.1</b> Change between a pair of samples</a></li>
<li class="chapter" data-level="3.2" data-path="differential-abundance.html"><a href="differential-abundance.html#regression-analysis-of-many-samples"><i class="fa fa-check"></i><b>3.2</b> Regression analysis of many samples</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="implications.html"><a href="implications.html"><i class="fa fa-check"></i><b>4</b> Implications for real-world inference </a>
<ul>
<li class="chapter" data-level="4.1" data-path="implications.html"><a href="implications.html#systematic-error-in-slope-or-lfc-estimates"><i class="fa fa-check"></i><b>4.1</b> Systematic error in slope or LFC estimates</a></li>
<li class="chapter" data-level="4.2" data-path="implications.html"><a href="implications.html#loss-of-precision-and-power"><i class="fa fa-check"></i><b>4.2</b> Loss of precision and power</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="solutions.html"><a href="solutions.html"><i class="fa fa-check"></i><b>5</b> Potential solutions</a>
<ul>
<li class="chapter" data-level="5.1" data-path="solutions.html"><a href="solutions.html#analyses-of-fold-changes-in-species-ratios"><i class="fa fa-check"></i><b>5.1</b> Analyses of fold changes in species ratios</a></li>
<li class="chapter" data-level="5.2" data-path="solutions.html"><a href="solutions.html#calibrate-compositions"><i class="fa fa-check"></i><b>5.2</b> Calibration using community controls</a></li>
<li class="chapter" data-level="5.3" data-path="solutions.html"><a href="solutions.html#calibration-using-reference-species"><i class="fa fa-check"></i><b>5.3</b> Calibration using reference species</a></li>
<li class="chapter" data-level="5.4" data-path="solutions.html"><a href="solutions.html#choosing-complementary-total-density-and-metagenomics-measurements"><i class="fa fa-check"></i><b>5.4</b> Choosing complementary total-density and metagenomics measurements</a></li>
<li class="chapter" data-level="5.5" data-path="solutions.html"><a href="solutions.html#bias-sensitivity-analysis"><i class="fa fa-check"></i><b>5.5</b> Bias-sensitivity analysis</a></li>
<li class="chapter" data-level="5.6" data-path="solutions.html"><a href="solutions.html#bias-aware-meta-analysis"><i class="fa fa-check"></i><b>5.6</b> Bias-aware meta-analysis</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="model-details.html"><a href="model-details.html"><i class="fa fa-check"></i><b>A</b> Model details</a>
<ul>
<li class="chapter" data-level="A.1" data-path="model-details.html"><a href="model-details.html#mgs-measurement"><i class="fa fa-check"></i><b>A.1</b> MGS measurement</a>
<ul>
<li class="chapter" data-level="A.1.1" data-path="model-details.html"><a href="model-details.html#revised-main-text-equations"><i class="fa fa-check"></i><b>A.1.1</b> Revised main text equations</a></li>
</ul></li>
<li class="chapter" data-level="A.2" data-path="model-details.html"><a href="model-details.html#total-density-bias"><i class="fa fa-check"></i><b>A.2</b> Taxonomic bias in total-density measurement</a></li>
<li class="chapter" data-level="A.3" data-path="model-details.html"><a href="model-details.html#total-density-ref"><i class="fa fa-check"></i><b>A.3</b> Using reference species for total-density normalization</a></li>
<li class="chapter" data-level="A.4" data-path="model-details.html"><a href="model-details.html#linearity-of-extraction"><i class="fa fa-check"></i><b>A.4</b> Linearity of extraction</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html"><i class="fa fa-check"></i><b>B</b> Review of experimental methods for obtaining absolute densities</a>
<ul>
<li class="chapter" data-level="B.1" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#measurement-of-total-cell-density"><i class="fa fa-check"></i><b>B.1</b> Measurement of total cell density</a></li>
<li class="chapter" data-level="B.2" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#measurement-of-total-dna-density-post-extraction"><i class="fa fa-check"></i><b>B.2</b> Measurement of total DNA density post extraction</a></li>
<li class="chapter" data-level="B.3" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#equivolumetric-protocol"><i class="fa fa-check"></i><b>B.3</b> Equivolumetric protocol</a></li>
<li class="chapter" data-level="B.4" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#housekeeping-species"><i class="fa fa-check"></i><b>B.4</b> Housekeeping species</a></li>
<li class="chapter" data-level="B.5" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#spike-ins"><i class="fa fa-check"></i><b>B.5</b> Spike-ins</a></li>
<li class="chapter" data-level="B.6" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#targeted-measurements"><i class="fa fa-check"></i><b>B.6</b> Targeted measurements</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="appendix-regression.html"><a href="appendix-regression.html"><i class="fa fa-check"></i><b>C</b> Error in estimated regression coefficients and standard errors</a></li>
<li class="chapter" data-level="D" data-path="supplemental-figures.html"><a href="supplemental-figures.html"><i class="fa fa-check"></i><b>D</b> Supplemental figures</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="model-details" class="section level1" number="6">
<h1><span class="header-section-number">A</span> Model details</h1>
<p>This appendix provides a more detailed description of our deterministic model of the microbiome measurement process, which it uses to derive some of the additional claims of the main text.</p>
<div id="mgs-measurement" class="section level2" number="6.1">
<h2><span class="header-section-number">A.1</span> MGS measurement</h2>
<p>This section expands the description of the model in the main text to more explicitly lay out our assumptions and to consider the effect of variation in measurement efficiencies within a taxonomic group on its MGS measurement.</p>
<p>A taxon can be any group of organisms.
For any taxon <span class="math inline">\(I\)</span> and sample <span class="math inline">\(a\)</span>, we define the absolute efficiency of that taxon and that sample as the number of reads assigned to that taxon divided by the number of cells of that taxon in the given sample.
We assume that these assigned reads really do come from that particular taxon and sample (though this assumption is often violated in practice).</p>
<p>The efficiency of a taxon <span class="math inline">\(I\)</span> is the average number of reads produced by each of its constituent cells.
This number varies due to</p>
<ol style="list-style-type: decimal">
<li>Genetic variation among cells</li>
<li>Non-genetic variation among cells</li>
<li>Randomness in the experimental process</li>
</ol>
<p>Variation encoded in the genome of cells determines properties that affect its efficiency, such as how easy it is to lyse, how easily PCR primers will bind to their target locus, how many copies of a marker gene are in the genome, and whether its sequencing reads can be taxonomically assigned.
This variation created by genetically encoded traits is what we refer to as ‘taxonomic bias’ and is the primary focus of this article.
In principle, even a single nucleotide change (e.g. in a primer-binding site) could affect the efficiency of a cell.
Nevertheless, cells will more similar genomes will tend to have more similar genetically-encoded efficiencies.
For the purposes of our analysis, we suppose that cells within the same species have approximately identical genetically-encoded efficiencies.
The impact of violations in this assumption can be understood using the approach we describe below for taxa above the species level.
More generally, our results can be read as applying to the finest taxonomic unit that is reported by the given MGS method.</p>
<p>Genetically identical cells can differ in cell state such that their efficiencies vary significantly.
An extreme example is the effect of sporulation: Spores are generally harder to lyse than vegetative cells, and <span class="citation">VanInsberghe et al. (<a href="#ref-vaninsberghe2020diar" role="doc-biblioref">2020</a>)</span> found that the efficiency of a common DNA extraction protocol was 800X on vegetative cells vs spores of a strain of <em>Clostridium difficile</em>.
Physiological differences might also arise simply due where they are in the cell cycle.
To the extent that the cells of a given genotype tend to have a similar distribution of cell states across samples, the efficiency of the genotype will remain stable.
We assume this stability throughout our analysis.
Violations in this assumption can again be understood using an approach similar to that we use to considering taxa above the species level.</p>
<p>The final number of reads also varies due to idiosyncratic events that we typically think of (and model) as ‘random’, such as the precise handling of a sample during pipetting or loading of a sequencing flow cell.
Such variation may or may not be dwarfed by the ‘random’ variation in the sample composition (i.e., the variation not explained by the covariates in a regression analysis).
For simplicity, we ignore this random variation when considering the impact of taxonomic bias on MGS measurement.
The effect of random variation in regression analysis is addressed in Appendix
<a href="appendix-regression.html#appendix-regression">C</a> and the effect of random counting error is discussed in Section <a href="implications.html#implications">4</a>.</p>
<p>The absolute efficiency of a taxon can also vary simply because more effort is put towards sequencing that sample, either intentionally or unintentionally.
The sequencing effort might depend on the taxonomic composition and bias of a sample.
For example, a sample with higher output DNA will often be diluted more prior to amplification or sequencing.
The critical assumption we make regarding sequencing effort is that it affects all species in the sample equally and hence doesn’t affect their relative abundances.</p>
<p>With these assumptions in place, we can partition the multiplicative difference between the reads assigned to a taxon and its density in the source sample into a taxon-specific factor and a sample-specific factor.
There is an extra degree of freedom, which we settle by equating the taxon-specific factors with relative measurement efficiencies and arbitrarily choose the relative efficiency of a particular species to equal 1.</p>
<p>Our model for the read counts of a species <span class="math inline">\(i\)</span> in sample <span class="math inline">\(a\)</span> can thus be expressed mathematically as
<span class="math display">\[\begin{align}
  \text{reads}_{i}(a)
  = \text{density}_{i}(a) \cdot \text{efficiency}_{i} \cdot \text{effort}(a).
\end{align}\]</span>
Note that the species’ efficiencies are properties of the species <span class="math inline">\(i\)</span> but not the sample <span class="math inline">\(a\)</span>.
Similarly, let <span class="math inline">\(\text{density}_{I}(a)\)</span>, <span class="math inline">\(\text{reads}_{I}(a)\)</span>, and <span class="math inline">\(\text{efficiency}_{I}(a)\)</span> denote the density, read count, and relative efficiency of an arbitrary group of species <span class="math inline">\(I\)</span> in sample <span class="math inline">\(a\)</span>.
The read count for taxon <span class="math inline">\(I\)</span> in sample <span class="math inline">\(a\)</span> can be written
<span class="math display" id="eq:measurement-model-general">\[\begin{align}
  \tag{A.1}
  \text{reads}_{I}(a)
  = \text{density}_{I}(a) \cdot \text{efficiency}_{I}(a) \cdot \text{effort}(a).
\end{align}\]</span>
The efficiency of taxon <span class="math inline">\(I\)</span> in sample <span class="math inline">\(a\)</span> equals the weighted average of its constituent species,
<span class="math display" id="eq:efficiency-general">\[\begin{align}
  \tag{A.2}
  \text{efficiency}_I(a) 
  \equiv \frac{\sum_{i\in I}\text{density}_i(a)\cdot \text{efficiency}_i}{\text{density}_I(a)}.
\end{align}\]</span>
The ratio between the read counts of two taxa <span class="math inline">\(I\)</span> and <span class="math inline">\(J\)</span> is
<span class="math display" id="eq:ratio-error-general">\[\begin{align}
  \tag{A.3}
  \widehat{\text{ratio}}_{I/J}(a)
  &amp;= \frac{\text{density}_{I}(a)}{\text{density}_{J}(a)} \cdot \frac{\text{efficiency}_{I}(a)}{\text{efficiency}_{J}(a)}.
\end{align}\]</span>
Equation <a href="abundance-measurement.html#eq:prop-error">(2.5)</a> for the error in a species’ proportion and Equation <a href="abundance-measurement.html#eq:ratio-error">(2.7)</a> for the error in the ratio between two species both follow directly from this more general expression.</p>
<div id="revised-main-text-equations" class="section level3" number="6.1.1">
<h3><span class="header-section-number">A.1.1</span> Revised main text equations</h3>
<p>Let <span class="math inline">\(S\)</span> be the set of all species.
Then
<span class="math display">\[\begin{align}
  \text{reads}_S(a)
    = \text{density}_S(a) \cdot \text{efficiency}_S(a) \cdot \text{effort}(a),
\end{align}\]</span>
where
<span class="math display">\[\begin{align}
  \text{efficiency}_S(a) 
  \equiv \frac{\sum_{j\in S}\text{density}_j(a)\cdot \text{efficiency}_j}{\text{density}_S(a)}
\end{align}\]</span>
Also,
<span class="math display">\[\begin{align}
  \widehat{\text{prop}}_{i}(a) = \frac{\text{reads}_i(a)}{\text{reads}_S(a)}.
\end{align}\]</span>
and
<span class="math display">\[\begin{align}
  \widehat{\text{prop}}_{i}(a)
  &amp;= \text{prop}_{i}(a) \cdot \frac{\text{efficiency}_{i}}{\text{efficiency}_S(a)}.
\end{align}\]</span></p>
</div>
</div>
<div id="total-density-bias" class="section level2" number="6.2">
<h2><span class="header-section-number">A.2</span> Taxonomic bias in total-density measurement</h2>
<p>To understand the impact of taxonomic bias in the total-density measurement on community normalization, we model error in total-density measurements similarly to that for MGS measurements.
Let <span class="math inline">\(\widehat{\text{density}}_{S}(a)\)</span> be the measurement of the density of all species <span class="math inline">\(S\)</span>.
We suppose that this number is the sum of the (unobserved) contributions from each species.
The contribution of a species <span class="math inline">\(i\)</span> is proportional to its actual density <span class="math inline">\(\text{density}_{i}(a)\)</span> and its <em>absolute efficiency for the total-density measurement</em> <span class="math inline">\(\text{efficiency}_{i}^{\text{tot}}(a)\)</span>.
The contributions may also be multiplied by a common sample-specific error factor to account for non-linearity of extraction and/or random error, which we ignore for now.
The measured total density therefore equals
<span class="math display">\[\begin{align}
  \widehat{\text{density}}_S(a) 
  &amp;= \sum_{i\in S} \text{density}_i(a) \cdot \text{efficiency}^{\text{tot}}_i
\\&amp;= \text{density}_S(a) \cdot \text{efficiency}^{\text{tot}}_S(a)
\end{align}\]</span>
where
<span class="math display" id="eq:app-total-mean-efficiency">\[\begin{align}
  \tag{A.4}
  \text{efficiency}^{\text{tot}}_S(a) 
  \equiv \frac{\sum_{i \in S}\text{density}_j(a)\cdot \text{efficiency}^{\text{tot}}_i}{\text{density}_S(a)}
\end{align}\]</span>
is the mean efficiency in the sample with respect to the total-density measurement.</p>
<p>The error in the total density depends on the species composition through the mean efficiency of the sample; hence spurious changes in measured density can occur simply due to shifts from high- to low-efficiency species (or vice versa).</p>
<p>How does this error affect the measured densities of individual species when the total is used for normalization?
Substituting the mean efficiency of the total measurement for the error term in Equation <a href="abundance-measurement.html#eq:density-prop-error">(2.9)</a> gives
<span class="math display" id="eq:app-density-prop-error">\[\begin{align}
  \tag{A.5}
  \widehat{\text{density}}_{i}(a) 
  = \text{density}_{i}(a) \cdot \frac{\text{efficiency}_{i} \cdot \text{efficiency}^{\text{tot}}_S(a)}{\text{efficiency}_S(a)}.
\end{align}\]</span>
The mean efficiency of the total measurement appears in the numerator, while the mean efficiency of the MGS measurement is in the denominator.
To the extent that these two are positively associated across samples (specifically, their logarithms are positively linearly correlated), then the variation in one will tend to be offset by the variation in the other.
In this case the (geometric) variation in the ratio will be reduced, and the fold error in the species density measurement will tend to be more consistent than if the total density measurement was taxonomically unbiased (i.e. <span class="math inline">\(\text{efficiency}^\text{tot}_S(a) = 1\)</span> for all samples).
The error in species densities for individual samples may not be reduced; but the error in fold changes and regression analysis across samples will be.</p>
<p>The extent to which the log mean efficiency of the MGS measurement and the total measurement are correlated depends on the species efficiencies as well as the changes in species composition across samples, making it difficult to make general statements about this effect.
We can get a sense for the effect by considering the correlation between log efficiencies across species between the two measurement types.
Suppose that distribution over species of log efficiencies of the MGS measurement has variance <span class="math inline">\(\sigma^{2}\)</span>, the log efficiencies for the total density measurement has variance <span class="math inline">\(\sigma_{\text{tot}}^{2}\)</span>, and the correlation between the two is <span class="math inline">\(\rho\)</span>.
The variance in the difference <span class="math inline">\(\log \text{efficiency}_{i}^{\text{tot}} - \log \text{efficiency}_{i}\)</span> is thus <span class="math inline">\(\sigma^{2} + \sigma_{\text{tot}}^{2} - \rho \sigma \sigma_{\text{tot}}\)</span>.
All else equal, the smaller this variance, the smaller we expect the variation in the error factor <span class="math inline">\(\text{efficiency}^{\text{tot}}_S(a)/ \text{efficiency}_S(a)\)</span>.</p>
<p><strong>Unclassified reads:</strong> It is common that a given MGS protocol generates sequencing reads from particular taxa that the bioinformatics portion entirely discards, due to an inability to taxonomically assign the reads to the chosen taxonomic units.
For instance, 16S data that is analyzed with closed-reference OTU assignment and shotgun data analyzed by mapping to a database of reference genomes will be unable to assign sequences that are less than a chosen similarity cutoff (e.g. 97%) from the reference sequences.
These taxa may form an appreciable fraction of the community and thus lead to a large fraction of the total sequenced reads for a given sample may remain unassigned.</p>
<p>So far, we have defined the efficiency of the MGS protocol in terms of taxonomically classified reads and so have treated such taxa as having an MGS efficiency of 0.
If such species are included in the total density measurement, then we have a mismatch between the two measurement types.
However, we may be able to remove this mismatch by altering the equation we use to form species density measurements.
In particular, we can compute the proportion of the focal species from the ratio of its assigned reads to the total sequenced reads, rather than to the total set of assigned reads.
This approach raises additional questions around how to treat reads from assignable species that are filtered during quality control steps.</p>
<p>Suppose that the set of species that can be classified by our bioinformatics pipeline, <span class="math inline">\(S\)</span>, is a proper subset of a larger set of species <span class="math inline">\(S&#39;\)</span> that are present and yield sequencing reads in our sample.
Suppose that there is no additional taxonomic bias in our MGS measurement; that is, all species in <span class="math inline">\(S\)</span> have an efficiency of 1, and the species in <span class="math inline">\(S&#39; \setminus S\)</span> have an efficiency of 0.
Moreover, suppose that all reads from the <span class="math inline">\(S\)</span> species are classified; that is, there is no loss of reads due to routine QC filtering.
Further suppose that we are capable of perfectly measuring the total density of all cells from the full set of species <span class="math inline">\(S&#39;\)</span>; that is, <span class="math inline">\(\text{efficiency}^{\text{tot}}_{i} = 1\)</span> for all species <span class="math inline">\(i \in S&#39;\)</span>.</p>
<p>In this case, we can overcome the mismatch between MGS and total measurement when we use them to measure species densities, by using the total reads rather than the assigned reads to compute the species proportions.
That is, instead of
<span class="math display">\[\begin{align}
  \widehat{\text{prop}}_{i}(a) = \frac{\text{reads}_i(a)}{\text{reads}_S(a)},
\end{align}\]</span>
we take
<span class="math display">\[\begin{align}
  \widehat{\text{prop}}_{i}(a) 
  &amp;= \frac{\text{reads}_i(a)}{\text{total reads}(a)}
\\&amp;= \frac{\text{reads}_i(a)}{\text{reads}_{S&#39;}(a)}.
\end{align}\]</span>
By accounting for the contribution of unknown species when computing proportions, we are able to resolve the mismatch and obtain accurate species densities.</p>
</div>
<div id="total-density-ref" class="section level2" number="6.3">
<h2><span class="header-section-number">A.3</span> Using reference species for total-density normalization</h2>
<p>Constant reference species are sometimes used to measure total density of <span class="math inline">\(S\)</span> by the ratio of <span class="math inline">\(S\)</span> reads to <span class="math inline">\(R\)</span> reads.
For example, a study of <em>Arabidopsis</em> microbiomes used the ratio of bacterial to host reads in shotgun sequencing as a proxy for total bacterial density, which they then used for total-community normalization of 16S amplicon sequencing measurements (<span class="citation">Karasov et al. (<a href="#ref-karasov2020ther" role="doc-biblioref">2020</a>)</span>, <span class="citation">Regalado et al. (<a href="#ref-regalado2019comb" role="doc-biblioref">2020</a>)</span>).
<span class="citation">Chng et al. (<a href="#ref-chng2020meta" role="doc-biblioref">2020</a>)</span> similarly used the ratio of bacterial to host or diet reads in shotgun sequencing of mouse fecal samples as a proxy for total bacterial density (though they did not use this measurement for community normalization).
<span class="citation">Smets et al. (<a href="#ref-smets2016amet" role="doc-biblioref">2016</a>)</span> similarly used the ratio of non-spike-in to spike-in reads to estimate total density.</p>
<p>What is the impact of taxonomic bias on these total density estimates and the species densities derived from them?
The measured density of <span class="math inline">\(S\)</span> is
<span class="math display">\[\begin{align}
  \widehat{\text{density}}_{S}(a) 
  &amp;= \frac{\text{reads}_S(a)}{\text{reads}_{R}(a)}
\\&amp;= \frac{\text{density}_S(a)}{\text{density}_{R}} \cdot \frac{\text{efficiency}_S(a)}{\text{efficiency}_{R}}
\\&amp;= \text{density}_S(a) \cdot \text{efficiency}_S(a) \cdot \text{constant}.
\end{align}\]</span>
Hence variation in the mean efficiency among the species <span class="math inline">\(S\)</span> across samples creates a variable fold error in the density measurement, similar to direct measurements of total density.</p>
<p>The impact of this variable error for species density measurement via Equation <a href="abundance-measurement.html#eq:density-prop-meas">(2.8)</a> depends on whether the species proportions are derived from the same or a different MGS measurement.</p>
<p>If proportions from the same MGS measurement are used, then the measured species densities are
<span class="math display">\[\begin{align}
  \widehat{\text{density}}_{i}(a) 
  &amp;= \frac{\text{reads}_i(a)}{\text{reads}_{S}(a)} \cdot \widehat{\text{density}}_{S}(a)
\\&amp;= \frac{\text{reads}_{i}(a)}{\text{reads}_{R}(a)}.
\end{align}\]</span>
where the second line follows by substitution of the previous equation.
The variable error in the total density cancels with that in the proportion, yielding a constant fold error.
In fact, the measurement we obtain is exactly the same as that from reference normalization (Equation <a href="abundance-measurement.html#eq:density-ratio-meas">(2.10)</a>).</p>
<p>The situation differs when the total density and community composition are estimated using different sequencing and/or bioinformatic methods, as in the plant microbiome example above.
In this case, the mean efficiency of the total density measurement will not equal that in the proportions, and the more general case of total-density normalization discussed in Appendix <a href="model-details.html#total-density-bias">A.2</a> applies.</p>
<p>Although we only consider constant reference species, similar behavior occurs if for varying reference species assayed by targeted measurements.</p>
</div>
<div id="linearity-of-extraction" class="section level2" number="6.4">
<h2><span class="header-section-number">A.4</span> Linearity of extraction</h2>
<p>Some popular absolute-abundance methods use post-extraction DNA (or RNA) density as a proxy for cell density in the original sample; in particular, using florescence-based total DNA quantification or using qPCR or ddPCR to quantify a marker-gene.
In addition, some have suggested normalization of species to spike-ins of extraneous DNA added after extraction.
Both of these approaches presuppose that DNA extraction is linear in the sense that the DNA concentration is proportional to the cell concentration in the original sample (possibly after correction by known dilution or concentration factors).
Deviations from linearity can occur due to random and systematic variation in DNA yields.
Here we consider the impact of these deviations on species density measurements.</p>
<p>One source of systematic variation in yield is taxonomic bias.
We already explored the impact of taxonomic bias in extraction.</p>
<p>NOTE: This might not be a correct way to talk about this; should perhaps make a more explicit model to verify this argument.</p>
<p>To understand the impact of non-linearity after bias has been accounted for, let <span class="math inline">\(C(a)\)</span> be a sample-specific factor that accounts for DNA extraction non-linearity after controlling for bias.
From Equation <a href="abundance-measurement.html#eq:density-prop-error">(2.9)</a>, we have the more general equation
<span class="math display" id="eq:app-density-prop-error">\[\begin{align}
  \tag{A.5}
  \widehat{\text{density}}_{i}(a) 
  = \text{density}_{i}(a) \cdot \frac{\text{efficiency}_{i} \cdot \text{efficiency}^{\text{tot}}_S(a)}{\text{efficiency}_S(a)} \cdot C(a).
\end{align}\]</span>
<span class="math inline">\(C(a)\)</span> might fluctuate randomly among samples, or might vary systematically — for example, if DNA yield saturates at high concentrations then <span class="math inline">\(C(a)\)</span> will be smaller in higher-concentration samples.
If DNA extraction is not linear, then measures of cell density might give more accurate fold changes even if they share less bias with the MGS measurement.</p>
<p>Post-extraction spike-ins and targeted measurements face a similar problem.
These perfectly control for bias in fold change estimation <em>if</em> DNA extraction is linear.
However, if not then the fold error in <span class="math inline">\(\widehat{\text{density}_r}(a)\)</span> for a species with a targeted measurement will vary with <span class="math inline">\(C(a)\)</span> and cause error in fold changes.
The same problem applies to DNA spike-ins added after DNA extraction (although different accounting is needed since <span class="math inline">\(\text{density}_r(a)\)</span> no longer has meaning).</p>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-chng2020meta" class="csl-entry">
Chng, Kern Rei, Tarini Shankar Ghosh, Yi Han Tan, Tannistha Nandi, Ivor Russel Lee, Amanda Hui Qi Ng, Chenhao Li, et al. 2020. <span>“<span class="nocase">Metagenome-wide association analysis identifies microbial determinants of post-antibiotic ecological recovery in the gut</span>.”</span> <em>Nat. Ecol. Evol.</em> 4 (9): 1256–67. <a href="https://doi.org/10.1038/s41559-020-1236-0">https://doi.org/10.1038/s41559-020-1236-0</a>.
</div>
<div id="ref-karasov2020ther" class="csl-entry">
Karasov, Talia L., Manuela Neumann, Alejandra Duque-Jaramillo, Sonja Kersten, Ilja Bezrukov, Birgit Schröppel, Efthymia Symeonidi, et al. 2020. <span>“<span class="nocase">The relationship between microbial population size and disease in the Arabidopsis thaliana phyllosphere</span>.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/828814">https://doi.org/10.1101/828814</a>.
</div>
<div id="ref-regalado2019comb" class="csl-entry">
Regalado, Julian, Derek S. Lundberg, Oliver Deusch, Sonja Kersten, Talia Karasov, Karin Poersch, Gautam Shirsekar, and Detlef Weigel. 2020. <span>“<span class="nocase">Combining whole-genome shotgun sequencing and rRNA gene amplicon analyses to improve detection of microbe–microbe interaction networks in plant leaves</span>.”</span> <em>ISME J.</em>, May, 823492. <a href="https://doi.org/10.1038/s41396-020-0665-8">https://doi.org/10.1038/s41396-020-0665-8</a>.
</div>
<div id="ref-smets2016amet" class="csl-entry">
Smets, Wenke, Jonathan W. Leff, Mark A. Bradford, Rebecca L. McCulley, Sarah Lebeer, and Noah Fierer. 2016. <span>“<span class="nocase">A method for simultaneous measurement of soil bacterial abundances and community composition via 16S rRNA gene sequencing</span>.”</span> <em>Soil Biol. Biochem.</em> 96: 145–51. <a href="https://doi.org/10.1016/j.soilbio.2016.02.003">https://doi.org/10.1016/j.soilbio.2016.02.003</a>.
</div>
<div id="ref-vaninsberghe2020diar" class="csl-entry">
VanInsberghe, David, Joseph A. Elsherbini, Bernard Varian, Theofilos Poutahidis, Susan Erdman, and Martin F. Polz. 2020. <span>“<span class="nocase">Diarrhoeal events can trigger long-term Clostridium difficile colonization with recurrent blooms</span>.”</span> <em>Nat. Microbiol.</em> 5 (4): 642–50. <a href="https://doi.org/10.1038/s41564-020-0668-2">https://doi.org/10.1038/s41564-020-0668-2</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="solutions.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="review-absolute-methods.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/mikemc/differential-abundance-theory/edit/main/appendix.Rmd",
"text": "Edit"
},
"history": {
"link": "https://github.com/mikemc/differential-abundance-theory/commits/main/appendix.Rmd",
"text": null
},
"view": {
"link": "https://github.com/mikemc/differential-abundance-theory/blob/main/appendix.Rmd",
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
