<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>B Review of experimental methods for obtaining absolute densities | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements</title>
  <meta name="description" content="B Review of experimental methods for obtaining absolute densities | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="B Review of experimental methods for obtaining absolute densities | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="B Review of experimental methods for obtaining absolute densities | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements" />
  
  
  

<meta name="author" content="Michael R. McLaren" />
<meta name="author" content="Jacob T. Nearing" />
<meta name="author" content="Amy D. Willis" />
<meta name="author" content="Karen G. Lloyd" />
<meta name="author" content="Benjamin J. Callahan" />


<meta name="date" content="2021-11-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="model-details.html"/>
<link rel="next" href="appendix-regression.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
TeX: {extensions: ["cancel.js"]}
});
</script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="abundance-measurement.html"><a href="abundance-measurement.html"><i class="fa fa-check"></i><b>2</b> How taxonomic bias affects abundance measurements</a>
<ul>
<li class="chapter" data-level="2.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#model-of-mgs-measurement"><i class="fa fa-check"></i><b>2.1</b> Model of MGS measurement</a></li>
<li class="chapter" data-level="2.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#relative-abundances-proportions-and-ratios"><i class="fa fa-check"></i><b>2.2</b> Relative abundances (proportions and ratios)</a></li>
<li class="chapter" data-level="2.3" data-path="abundance-measurement.html"><a href="abundance-measurement.html#absolute-abundances-cell-densities"><i class="fa fa-check"></i><b>2.3</b> Absolute abundances (cell densities)</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>3</b> How taxonomic bias affects differential-abundance analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-abundance.html"><a href="differential-abundance.html#change-between-a-pair-of-samples"><i class="fa fa-check"></i><b>3.1</b> Change between a pair of samples</a></li>
<li class="chapter" data-level="3.2" data-path="differential-abundance.html"><a href="differential-abundance.html#regression-analysis-of-many-samples"><i class="fa fa-check"></i><b>3.2</b> Regression analysis of many samples</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="implications.html"><a href="implications.html"><i class="fa fa-check"></i><b>4</b> Implications for real-world inference </a>
<ul>
<li class="chapter" data-level="4.1" data-path="implications.html"><a href="implications.html#systematic-error-in-slope-or-lfc-estimates"><i class="fa fa-check"></i><b>4.1</b> Systematic error in slope or LFC estimates</a></li>
<li class="chapter" data-level="4.2" data-path="implications.html"><a href="implications.html#loss-of-precision-and-power"><i class="fa fa-check"></i><b>4.2</b> Loss of precision and power</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="solutions.html"><a href="solutions.html"><i class="fa fa-check"></i><b>5</b> Potential solutions</a>
<ul>
<li class="chapter" data-level="5.1" data-path="solutions.html"><a href="solutions.html#analyses-of-fold-changes-in-species-ratios"><i class="fa fa-check"></i><b>5.1</b> Analyses of fold changes in species ratios</a></li>
<li class="chapter" data-level="5.2" data-path="solutions.html"><a href="solutions.html#calibrate-compositions"><i class="fa fa-check"></i><b>5.2</b> Calibration using community controls</a></li>
<li class="chapter" data-level="5.3" data-path="solutions.html"><a href="solutions.html#calibration-using-reference-species"><i class="fa fa-check"></i><b>5.3</b> Calibration using reference species</a></li>
<li class="chapter" data-level="5.4" data-path="solutions.html"><a href="solutions.html#choosing-complementary-total-density-and-metagenomics-measurements"><i class="fa fa-check"></i><b>5.4</b> Choosing complementary total-density and metagenomics measurements</a></li>
<li class="chapter" data-level="5.5" data-path="solutions.html"><a href="solutions.html#bias-sensitivity-analysis"><i class="fa fa-check"></i><b>5.5</b> Bias-sensitivity analysis</a></li>
<li class="chapter" data-level="5.6" data-path="solutions.html"><a href="solutions.html#bias-aware-meta-analysis"><i class="fa fa-check"></i><b>5.6</b> Bias-aware meta-analysis</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="model-details.html"><a href="model-details.html"><i class="fa fa-check"></i><b>A</b> Model details</a>
<ul>
<li class="chapter" data-level="A.1" data-path="model-details.html"><a href="model-details.html#mgs-measurement"><i class="fa fa-check"></i><b>A.1</b> MGS measurement</a>
<ul>
<li class="chapter" data-level="A.1.1" data-path="model-details.html"><a href="model-details.html#revised-main-text-equations"><i class="fa fa-check"></i><b>A.1.1</b> Revised main text equations</a></li>
</ul></li>
<li class="chapter" data-level="A.2" data-path="model-details.html"><a href="model-details.html#total-density-bias"><i class="fa fa-check"></i><b>A.2</b> Taxonomic bias in total-density measurement</a></li>
<li class="chapter" data-level="A.3" data-path="model-details.html"><a href="model-details.html#total-density-ref"><i class="fa fa-check"></i><b>A.3</b> Using reference species for total-density normalization</a></li>
<li class="chapter" data-level="A.4" data-path="model-details.html"><a href="model-details.html#linearity-of-extraction"><i class="fa fa-check"></i><b>A.4</b> Linearity of extraction</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html"><i class="fa fa-check"></i><b>B</b> Review of experimental methods for obtaining absolute densities</a>
<ul>
<li class="chapter" data-level="B.1" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#measurement-of-total-cell-density"><i class="fa fa-check"></i><b>B.1</b> Measurement of total cell density</a></li>
<li class="chapter" data-level="B.2" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#measurement-of-total-dna-density-post-extraction"><i class="fa fa-check"></i><b>B.2</b> Measurement of total DNA density post extraction</a></li>
<li class="chapter" data-level="B.3" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#equivolumetric-protocol"><i class="fa fa-check"></i><b>B.3</b> Equivolumetric protocol</a></li>
<li class="chapter" data-level="B.4" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#housekeeping-species"><i class="fa fa-check"></i><b>B.4</b> Housekeeping species</a></li>
<li class="chapter" data-level="B.5" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#spike-ins"><i class="fa fa-check"></i><b>B.5</b> Spike-ins</a></li>
<li class="chapter" data-level="B.6" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#targeted-measurements"><i class="fa fa-check"></i><b>B.6</b> Targeted measurements</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="appendix-regression.html"><a href="appendix-regression.html"><i class="fa fa-check"></i><b>C</b> Error in estimated regression coefficients and standard errors</a></li>
<li class="chapter" data-level="D" data-path="supplemental-figures.html"><a href="supplemental-figures.html"><i class="fa fa-check"></i><b>D</b> Supplemental figures</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="review-absolute-methods" class="section level1" number="7">
<h1><span class="header-section-number">B</span> Review of experimental methods for obtaining absolute densities</h1>
<p>There are many experimental techniques to be able to add absolute-density information to MGS measurements.
Here we review the experimental techniques; the next section considers the implications for systematic error.</p>
<p><em>NOTE: Right now I don’t consistently address why various targeted methods might be expected to produce constant fold errors. In revision, seek to connect each method with the relevant theory.</em></p>
<div id="measurement-of-total-cell-density" class="section level2" number="7.1">
<h2><span class="header-section-number">B.1</span> Measurement of total cell density</h2>
<p>Total cell density in the original sample can be directly measured by cell counting, either via microscopy (<span class="citation">Kevorkian et al. (<a href="#ref-kevorkian2018esti" role="doc-biblioref">2018</a>)</span>, <span class="citation">Lloyd et al. (<a href="#ref-lloyd2020evid" role="doc-biblioref">2020</a>)</span>) or flow cytometry (<span class="citation">Props et al. (<a href="#ref-props2017abso" role="doc-biblioref">2017</a>)</span>, <span class="citation">Vandeputte et al. (<a href="#ref-vandeputte2017quan" role="doc-biblioref">2017</a>)</span>).
Total cell density or biomass can also be measured via properties assumed to be proportional to cell density, such as fluorescence (as in fluorescence spectroscopy, <span class="citation">Wang et al. (<a href="#ref-wang2021curr" role="doc-biblioref">2021</a>)</span>), components of microbial cell membranes (as in PLFA analysis, <span class="citation">Smets et al. (<a href="#ref-smets2016amet" role="doc-biblioref">2016</a>)</span>), and the rate of microbial respiration (as in SIR method, <span class="citation">Smets et al. (<a href="#ref-smets2016amet" role="doc-biblioref">2016</a>)</span>).
So far, it is primarily the cell counting methods that have been used for species-density measurement (rather than simply for the total density), by multiplying the estimated total density by the MGS proportions.</p>
</div>
<div id="measurement-of-total-dna-density-post-extraction" class="section level2" number="7.2">
<h2><span class="header-section-number">B.2</span> Measurement of total DNA density post extraction</h2>
<p>It is also common to use density of bulk DNA or a marker gene as a proxy for total community density.
Marker-gene density is typically measured with qPCR or ddPCR using ‘universal primers’ that target the marker gene of interest (typically the 16S gene for bacterial microbiome experiments) (<span class="citation">Tettamanti Boshier et al. (<a href="#ref-tettamantiboshier2020comp" role="doc-biblioref">2020</a>)</span>, <span class="citation">Jian et al. (<a href="#ref-jian2020quan" role="doc-biblioref">2020</a>)</span>, <span class="citation">Galazzo et al. (<a href="#ref-galazzo2020howt" role="doc-biblioref">2020</a>)</span>).
Bulk DNA density can be measured using fluorescence-based DNA quantification assays (<span class="citation">Contijoch et al. (<a href="#ref-contijoch2019gutm" role="doc-biblioref">2019</a>)</span>; <span class="citation">Korpela et al. (<a href="#ref-korpela2018inte" role="doc-biblioref">2018</a>)</span>).
In either case, the DNA density is measured after DNA extraction and so is affected by taxonomic bias in the extraction process, such as variation in lysis efficiency among species.
Other sources of bias that affect the DNA density measurement include variation in marker-gene copy number (for marker-gene density) and variation in genome size (for bulk DNA density).</p>
<p>The measured DNA density is typically used as a direct proxy for cell density in the original sample.
In particular, it is assumed that a doubling of cell density in the original sample leads to a doubling of DNA density in the extraction (possibly after adjustment for known dilution factors).
This <em>linearity assumption</em> may be violated for several reasons.
First, because of taxonomic bias.
For example, samples dominated by easy-to-lyse species will give more DNA per cell than samples dominated by hard-to-lyse species.
Second, systematic non-linearity may occur in the DNA yield as a function of input, even if species composition is held fixed.
For example, DNA yield may saturate at high sample inputs.
Third, the DNA yield may vary apparently randomly due to subtle differences in sample chemistry or handling during the experiment.</p>
</div>
<div id="equivolumetric-protocol" class="section level2" number="7.3">
<h2><span class="header-section-number">B.3</span> Equivolumetric protocol</h2>
<p>A large part of the reason that there is not a direct correspondence between total density in the sample and total reads sequenced is that MGS experiments are typically intentionally designed to yield a similar number of sequencing reads from each sample, regardless of total density.
<span class="citation">Cruz, Christoff, and Oliveira (<a href="#ref-cruz2021equi" role="doc-biblioref">2021</a>)</span> propose instead designing the MGS experiment so as to make total reads proportional total density.
The ‘equivolumetric protocol’ they develop represents a first attempt in this direction.
In their protocol, total reads is a saturating function of total density; this function can be measured with a calibration experiment, and the calibration curve used to predict the total density in the source sample.
This total density estimate is then used to scale the read counts to estimate species densities in a manner equivalent to the total-community density method.</p>
</div>
<div id="housekeeping-species" class="section level2" number="7.4">
<h2><span class="header-section-number">B.4</span> Housekeeping species</h2>
<p>We use <em>housekeeping species</em> (by analogy with housekeeping genes used for normalization in RNAseq experiments) to refer to species whose density is assumed to be constant, either in the MGS sample or in the source ecosystem it is derived from.</p>
<p>Housekeeping species can sometimes be identified from prior scientific knowledge.
Several studies that have employed shotgun sequencing of host-associated microbiomes have use the plant or animal host for this purpose.
A study of <em>Arabidopsis</em> microbiomes used the ratio of bacterial to host reads in shotgun sequencing as a proxy for total bacterial density, which they then used for total-community normalization of 16S amplicon sequencing measurements (<span class="citation">Karasov et al. (<a href="#ref-karasov2020ther" role="doc-biblioref">2020</a>)</span>, <span class="citation">Regalado et al. (<a href="#ref-regalado2019comb" role="doc-biblioref">2020</a>)</span>).
<span class="citation">Chng et al. (<a href="#ref-chng2020meta" role="doc-biblioref">2020</a>)</span> similarly used the ratio of bacterial to host reads in shotgun sequencing of mouse fecal samples as a proxy for total bacterial density (though they did not use this measurement for community normalization).
They also use reads from dietary plants for the same purpose.
<span class="citation">Wallace et al. (<a href="#ref-wallace2021thed" role="doc-biblioref">2021</a>)</span> used shotgun sequencing to study the virome of <em>Drosophila</em>, and normalized virus reads to <em>Drosophila</em> reads to measure viral abundance per fly.
Organelle reads can also be used.
<span class="citation">Diener et al. (<a href="#ref-diener2021nonr" role="doc-biblioref">2021</a>)</span> use mitochondria reads in 16S sequencing of mouse fecal pellets to assess total microbial load, though in a qualitative fashion (as mitochondrial reads were only non-zero at very low bacterial densities induced by antibiotics).</p>
<p>In some cases, there may also be microbes or viruses thought to have stable densities.
A recent example is that the most abundant DNA virus in human feces, crAssphage, and the most abundant RNA virus in human feces, Pepper Mild Mottle Virus, have been treated as stable reference species in wastewater monitoring for SARS-CoV-2.
Although primarily used in the context of qPCR measurements, these viruses could also be used as references in RNA and DNA shotgun sequencing experiments.</p>
<p>Housekeeping species may not be known <em>a priori</em>; to address this case, several methods have been put forward to computationally identify unchanging microbes species directly from MGS measurements.
These studies are often focused on mammalian gut bacterial communities.
It is perhaps unreasonable to expect bacterial species to be unchanging across hosts, but weaker assumptions can be made to develop normalization methods for the MGS measurements with a similar spirit to reference-based normalization.
These studies have instead developed normalization methods based on assumptions such as that most species do not change between any pair of samples (<span class="citation">David et al. (<a href="#ref-david2014host" role="doc-biblioref">2014</a>)</span>) or that the mean (log) abundance between two sample conditions is unchanged for at least some species (<span class="citation">Mandal et al. (<a href="#ref-mandal2015anal" role="doc-biblioref">2015</a>)</span>, <span class="citation">Kumar et al. (<a href="#ref-kumar2018anal" role="doc-biblioref">2018</a>)</span>).</p>
<p>When housekeeping species are sequenced along with the primary MGS measurement, they can be used to obtain species densities via reference-normalization (Equation <a href="abundance-measurement.html#eq:density-ratio-meas">(2.10)</a>.
In this case, the only relevant taxonomic bias is that of the primary MGS measurement; if it is constant then it will cancel in fold-change calculations.
Because the density of the housekeeping species is unknown, we can consider either that the density of focal species has a constant error, or is in units of the housekeeping species.
Non-constant error might arise if the species is treated as constant when it is in fact not.</p>
<p>Housekeeping species have also been used to estimate total community density by <span class="math inline">\(\text{reads}_{S} / \text{reads}_{R}\)</span>.
This estimate has been used to study variation in total density across samples, or for total-density normalization.</p>
</div>
<div id="spike-ins" class="section level2" number="7.5">
<h2><span class="header-section-number">B.5</span> Spike-ins</h2>
<p>Spike-in methods differ by the biological spike-in material:
Cellular spike-ins can be added prior to DNA extraction, and DNA spike-ins can be added prior to or following DNA extraction.
In either case, a variety of methods can be used to actually leverage the spike-ins for absolute density analysis.</p>
<p><strong>Cellular spike-ins:</strong>
Cellular spike-ins are added to the sample prior to DNA extraction.
Some sample processing has typically occurred prior to spiking, for the purposes of storage (e.g. a freeze thaw cycle) and homogenization.
We should expect there to be taxonomic bias between the spike-in species and those naturally in the sample due to genetic differences and because of physiological differences induced by the sample processing prior to spiking and the experimental procedure used to grow and prepare the spike-in cells.
Our analysis acknowledges this bias, but assumes that it is consistent across samples.
The nominal density of the spike-in species added to each sample is subject to random and systematic fold error; but systematic fold error that is shared across samples will induce a constant fold error in species densities and so not impact DA analysis.
For instance, if source stock is actually 1.5X higher concentration than thought, the true spike-in concentration will be 1.5X greater than nominal in all samples and not pose a problem for accurate DA inference beyond leading to a greater than intended sequencing effort being expended on the spike-in.</p>
<p>Example studies include <span class="citation">Stämmler et al. (<a href="#ref-stammler2016adju" role="doc-biblioref">2016</a>)</span>, <span class="citation">Ji et al. (<a href="#ref-ji2019quan" role="doc-biblioref">2019</a>)</span>, and <span class="citation">Rao et al. (<a href="#ref-rao2021mult" role="doc-biblioref">2021</a>)</span>.</p>
<p><strong>DNA spike-ins:</strong>
Another possibility is to add DNA spike-ins, derived from natural or artificial sequence.
DNA spike-ins can be added the samples before DNA extraction (e.g. <span class="citation">Smets et al. (<a href="#ref-smets2016amet" role="doc-biblioref">2016</a>)</span>, <span class="citation">Tkacz, Hortala, and Poole (<a href="#ref-tkacz2018abso" role="doc-biblioref">2018</a>)</span>, <span class="citation">Zemb et al. (<a href="#ref-zemb2020abso" role="doc-biblioref">2020</a>)</span>) or after DNA extraction (e.g. <span class="citation">Hardwick et al. (<a href="#ref-hardwick2018synt" role="doc-biblioref">2018</a>)</span>, <span class="citation">Tkacz, Hortala, and Poole (<a href="#ref-tkacz2018abso" role="doc-biblioref">2018</a>)</span>).
Adding spike-ins prior to extraction is thought to be preferable as it makes it possible to detect and correct for variation in DNA extraction yield among samples
(<span class="citation">Tkacz, Hortala, and Poole (<a href="#ref-tkacz2018abso" role="doc-biblioref">2018</a>)</span>, <span class="citation">Zemb et al. (<a href="#ref-zemb2020abso" role="doc-biblioref">2020</a>)</span>, <span class="citation">Harrison et al. (<a href="#ref-harrison2021theq" role="doc-biblioref">2021</a>)</span>).
Below, we consider the distinction between pre- and post-extraction spike-ins in the light of taxonomic bias coupled with other sources of variation in extraction efficiency.</p>
<p><strong>How spike-ins are used:</strong>
Like housekeeping species, spike-ins have been used in a variety of ways to analyze absolute abundances.
Let <span class="math inline">\(R\)</span> (for reference) be the spike-in species and <span class="math inline">\(S\)</span> be the native species.
<span class="citation">Smets et al. (<a href="#ref-smets2016amet" role="doc-biblioref">2016</a>)</span> used the ratio of <span class="math inline">\(S\)</span> to <span class="math inline">\(R\)</span> reads as an estimate of total density, which they were interested in for its own sake, though one could imagine then also using this total density estimate in community normalization (Equation <a href="abundance-measurement.html#eq:density-prop-meas">(2.8)</a>).
<span class="citation">Zemb et al. (<a href="#ref-zemb2020abso" role="doc-biblioref">2020</a>)</span> used the spike-ins to measure total density from the ratio of <span class="math inline">\(S\)</span> to <span class="math inline">\(R\)</span> qPCR abundance estimates and then used Equation <a href="abundance-measurement.html#eq:density-prop-meas">(2.8)</a>.
<span class="citation">Stämmler et al. (<a href="#ref-stammler2016adju" role="doc-biblioref">2016</a>)</span> and others used the ratio-based method of Equation <a href="abundance-measurement.html#eq:density-ratio-meas">(2.10)</a>.</p>
</div>
<div id="targeted-measurements" class="section level2" number="7.6">
<h2><span class="header-section-number">B.6</span> Targeted measurements</h2>
<p>A variety of methods exist for targeted measurement of absolute density of a specific species (or higher-order taxon).
The most common approach is to use qPCR or ddPCR to measure the concentration of a marker gene in the extracted DNA, using primers scoped to the target taxon.
This approach is therefore subject to sources of taxonomic bias including extraction, marker-gene copy number, and primer-binding.
It is also subject to non-species-specific variation in extraction yields unless these are otherwise controlled for.
It is also possible to directly measure cell density using methods.
Some species can be measured by CFU counting after plating on selective media (REFs), and ddPCR has been used to direct measure cells (<span class="citation">Dreo et al. (<a href="#ref-dreo2014opti" role="doc-biblioref">2014</a>)</span>, <span class="citation">Morella et al. (<a href="#ref-morella2018rapi" role="doc-biblioref">2018</a>)</span>) and viruses (<span class="citation">Pavšič, Žel, and Milavec (<a href="#ref-pavsic2016digi" role="doc-biblioref">2016</a>)</span>, <span class="citation">Morella et al. (<a href="#ref-morella2018rapi" role="doc-biblioref">2018</a>)</span>) without first performing an extraction.
Species-specific florescent probes also make it possible to measure individual species via microscopy or flow cytometry (REFs).</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-chng2020meta" class="csl-entry">
Chng, Kern Rei, Tarini Shankar Ghosh, Yi Han Tan, Tannistha Nandi, Ivor Russel Lee, Amanda Hui Qi Ng, Chenhao Li, et al. 2020. <span>“<span class="nocase">Metagenome-wide association analysis identifies microbial determinants of post-antibiotic ecological recovery in the gut</span>.”</span> <em>Nat. Ecol. Evol.</em> 4 (9): 1256–67. <a href="https://doi.org/10.1038/s41559-020-1236-0">https://doi.org/10.1038/s41559-020-1236-0</a>.
</div>
<div id="ref-contijoch2019gutm" class="csl-entry">
Contijoch, Eduardo J, Graham J Britton, Chao Yang, Ilaria Mogno, Zhihua Li, Ruby Ng, Sean R Llewellyn, et al. 2019. <span>“<span class="nocase">Gut microbiota density influences host physiology and is shaped by host and microbial factors</span>.”</span> <em>Elife</em> 8 (January). <a href="https://doi.org/10.7554/eLife.40553">https://doi.org/10.7554/eLife.40553</a>.
</div>
<div id="ref-cruz2021equi" class="csl-entry">
Cruz, Giuliano Netto Flores, Ana Paula Christoff, and Luiz Felipe Valter de Oliveira. 2021. <span>“<span class="nocase">Equivolumetric Protocol Generates Library Sizes Proportional to Total Microbial Load in 16S Amplicon Sequencing</span>.”</span> <em>Front. Microbiol.</em> 12 (February): 1–16. <a href="https://doi.org/10.3389/fmicb.2021.638231">https://doi.org/10.3389/fmicb.2021.638231</a>.
</div>
<div id="ref-david2014host" class="csl-entry">
David, Lawrence A, Arne C Materna, Jonathan Friedman, Maria I Campos-Baptista, Matthew C Blackburn, Allison Perrotta, Susan E Erdman, and Eric J Alm. 2014. <span>“<span class="nocase">Host lifestyle affects human microbiota on daily timescales</span>.”</span> <em>Genome Biol.</em> 15 (7): R89. <a href="https://doi.org/10.1186/gb-2014-15-7-r89">https://doi.org/10.1186/gb-2014-15-7-r89</a>.
</div>
<div id="ref-diener2021nonr" class="csl-entry">
Diener, Christian, Anna C. H. Hoge, Sean M. Kearney, Ulrike Kusebauch, Sushmita Patwardhan, Robert L. Moritz, Susan E. Erdman, and Sean M. Gibbons. 2021. <span>“<span class="nocase">Non-responder phenotype reveals apparent microbiome-wide antibiotic tolerance in the murine gut</span>.”</span> <em>Commun. Biol.</em> 4 (1). <a href="https://doi.org/10.1038/s42003-021-01841-8">https://doi.org/10.1038/s42003-021-01841-8</a>.
</div>
<div id="ref-dreo2014opti" class="csl-entry">
Dreo, Tanja, Manca Pirc, Živa Ramšak, Jernej Pavšič, Mojca Milavec, Jana Žel, and Kristina Gruden. 2014. <span>“<span class="nocase">Optimising droplet digital PCR analysis approaches for detection and quantification of bacteria: a case study of fire blight and potato brown rot</span>.”</span> <em>Anal. Bioanal. Chem.</em> 406 (26): 6513–28. <a href="https://doi.org/10.1007/s00216-014-8084-1">https://doi.org/10.1007/s00216-014-8084-1</a>.
</div>
<div id="ref-galazzo2020howt" class="csl-entry">
Galazzo, Gianluca, Niels van Best, Birke J. Benedikter, Kevin Janssen, Liene Bervoets, Christel Driessen, Melissa Oomen, et al. 2020. <span>“<span class="nocase">How to Count Our Microbes? The Effect of Different Quantitative Microbiome Profiling Approaches</span>.”</span> <em>Front. Cell. Infect. Microbiol.</em> 10 (August). <a href="https://doi.org/10.3389/fcimb.2020.00403">https://doi.org/10.3389/fcimb.2020.00403</a>.
</div>
<div id="ref-hardwick2018synt" class="csl-entry">
Hardwick, Simon A., Wendy Y. Chen, Ted Wong, Bindu S. Kanakamedala, Ira W. Deveson, Sarah E. Ongley, Nadia S. Santini, et al. 2018. <span>“<span class="nocase">Synthetic microbe communities provide internal reference standards for metagenome sequencing and analysis</span>.”</span> <em>Nat. Commun.</em> 9 (1): 3096. <a href="https://doi.org/10.1038/s41467-018-05555-0">https://doi.org/10.1038/s41467-018-05555-0</a>.
</div>
<div id="ref-harrison2021theq" class="csl-entry">
Harrison, Joshua G., W. John Calder, Bryan Shuman, and C. Alex Buerkle. 2021. <span>“<span class="nocase">The quest for absolute abundance: The use of internal standards for DNA‐based community ecology</span>.”</span> <em>Mol. Ecol. Resour.</em> 21 (1): 30–43. <a href="https://doi.org/10.1111/1755-0998.13247">https://doi.org/10.1111/1755-0998.13247</a>.
</div>
<div id="ref-ji2019quan" class="csl-entry">
Ji, Brian W., Ravi U. Sheth, Purushottam D. Dixit, Yiming Huang, Andrew Kaufman, Harris H. Wang, and Dennis Vitkup. 2019. <span>“<span class="nocase">Quantifying spatiotemporal variability and noise in absolute microbiota abundances using replicate sampling</span>.”</span> <em>Nat. Methods</em> 16 (8): 731–36. <a href="https://doi.org/10.1038/s41592-019-0467-y">https://doi.org/10.1038/s41592-019-0467-y</a>.
</div>
<div id="ref-jian2020quan" class="csl-entry">
Jian, Ching, Panu Luukkonen, Hannele Yki-Järvinen, Anne Salonen, and Katri Korpela. 2020. <span>“<span class="nocase">Quantitative PCR provides a simple and accessible method for quantitative microbiota profiling</span>.”</span> Edited by Ivone Vaz-Moreira. <em>PLoS One</em> 15 (1): e0227285. <a href="https://doi.org/10.1371/journal.pone.0227285">https://doi.org/10.1371/journal.pone.0227285</a>.
</div>
<div id="ref-karasov2020ther" class="csl-entry">
Karasov, Talia L., Manuela Neumann, Alejandra Duque-Jaramillo, Sonja Kersten, Ilja Bezrukov, Birgit Schröppel, Efthymia Symeonidi, et al. 2020. <span>“<span class="nocase">The relationship between microbial population size and disease in the Arabidopsis thaliana phyllosphere</span>.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/828814">https://doi.org/10.1101/828814</a>.
</div>
<div id="ref-kevorkian2018esti" class="csl-entry">
Kevorkian, Richard, Jordan T Bird, Alexander Shumaker, and Karen G Lloyd. 2018. <span>“<span class="nocase">Estimating Population Turnover Rates by Relative Quantification Methods Reveals Microbial Dynamics in Marine Sediment</span>.”</span> <em>Appl. Environ. Microbiol.</em> 84 (1): e01443–17. <a href="https://doi.org/10.1128/AEM.01443-17">https://doi.org/10.1128/AEM.01443-17</a>.
</div>
<div id="ref-korpela2018inte" class="csl-entry">
Korpela, Katri, Elin W. Blakstad, Sissel J. Moltu, Kenneth Strømmen, Britt Nakstad, Arild E. Rønnestad, Kristin Brække, Per O. Iversen, Christian A. Drevon, and Willem de Vos. 2018. <span>“<span class="nocase">Intestinal microbiota development and gestational age in preterm neonates</span>.”</span> <em>Sci. Rep.</em> 8 (1): 1–9. <a href="https://doi.org/10.1038/s41598-018-20827-x">https://doi.org/10.1038/s41598-018-20827-x</a>.
</div>
<div id="ref-kumar2018anal" class="csl-entry">
Kumar, M. Senthil, Eric V. Slud, Kwame Okrah, Stephanie C. Hicks, Sridhar Hannenhalli, and Héctor Corrada Bravo. 2018. <span>“<span class="nocase">Analysis and correction of compositional bias in sparse sequencing count data</span>.”</span> <em>BMC Genomics</em> 19 (1): 799. <a href="https://doi.org/10.1186/s12864-018-5160-5">https://doi.org/10.1186/s12864-018-5160-5</a>.
</div>
<div id="ref-lloyd2020evid" class="csl-entry">
Lloyd, Karen G., Jordan T. Bird, Joy Buongiorno, Emily Deas, Richard Kevorkian, Talor Noordhoek, Jacob Rosalsky, and Taylor Roy. 2020. <span>“<span class="nocase">Evidence for a Growth Zone for Deep-Subsurface Microbial Clades in Near-Surface Anoxic Sediments</span>.”</span> <em>Appl. Environ. Microbiol.</em> 86 (19): 1–15. <a href="https://doi.org/10.1128/AEM.00877-20">https://doi.org/10.1128/AEM.00877-20</a>.
</div>
<div id="ref-mandal2015anal" class="csl-entry">
Mandal, Siddhartha, Will Van Treuren, Richard A. White, Merete Eggesbø, Rob Knight, and Shyamal D. Peddada. 2015. <span>“<span class="nocase">Analysis of composition of microbiomes: a novel method for studying microbial composition</span>.”</span> <em>Microb. Ecol. Heal. Dis.</em> 26 (1): 27663. <a href="https://doi.org/10.3402/mehd.v26.27663">https://doi.org/10.3402/mehd.v26.27663</a>.
</div>
<div id="ref-morella2018rapi" class="csl-entry">
Morella, Norma M., Shangyang Christopher Yang, Catherine A. Hernandez, and Britt Koskella. 2018. <span>“<span class="nocase">Rapid quantification of bacteriophages and their bacterial hosts in vitro and in vivo using droplet digital PCR</span>.”</span> <em>J. Virol. Methods</em> 259 (May): 18–24. <a href="https://doi.org/10.1016/j.jviromet.2018.05.007">https://doi.org/10.1016/j.jviromet.2018.05.007</a>.
</div>
<div id="ref-pavsic2016digi" class="csl-entry">
Pavšič, Jernej, Jana Žel, and Mojca Milavec. 2016. <span>“<span class="nocase">Digital PCR for direct quantification of viruses without DNA extraction</span>.”</span> <em>Anal. Bioanal. Chem.</em> 408 (1): 67–75. <a href="https://doi.org/10.1007/s00216-015-9109-0">https://doi.org/10.1007/s00216-015-9109-0</a>.
</div>
<div id="ref-props2017abso" class="csl-entry">
Props, Ruben, Frederiek-Maarten Kerckhof, Peter Rubbens, Jo De Vrieze, Emma Hernandez Sanabria, Willem Waegeman, Pieter Monsieurs, Frederik Hammes, and Nico Boon. 2017. <span>“<span class="nocase">Absolute quantification of microbial taxon abundances</span>.”</span> <em>ISME J.</em> 11 (2): 584–87. <a href="https://doi.org/10.1038/ismej.2016.117">https://doi.org/10.1038/ismej.2016.117</a>.
</div>
<div id="ref-rao2021mult" class="csl-entry">
Rao, Chitong, Katharine Z. Coyte, Wayne Bainter, Raif S. Geha, Camilia R. Martin, and Seth Rakoff-Nahoum. 2021. <span>“<span class="nocase">Multi-kingdom ecological drivers of microbiota assembly in preterm infants</span>.”</span> <em>Nature</em> 591 (7851): 633–38. <a href="https://doi.org/10.1038/s41586-021-03241-8">https://doi.org/10.1038/s41586-021-03241-8</a>.
</div>
<div id="ref-regalado2019comb" class="csl-entry">
Regalado, Julian, Derek S. Lundberg, Oliver Deusch, Sonja Kersten, Talia Karasov, Karin Poersch, Gautam Shirsekar, and Detlef Weigel. 2020. <span>“<span class="nocase">Combining whole-genome shotgun sequencing and rRNA gene amplicon analyses to improve detection of microbe–microbe interaction networks in plant leaves</span>.”</span> <em>ISME J.</em>, May, 823492. <a href="https://doi.org/10.1038/s41396-020-0665-8">https://doi.org/10.1038/s41396-020-0665-8</a>.
</div>
<div id="ref-smets2016amet" class="csl-entry">
Smets, Wenke, Jonathan W. Leff, Mark A. Bradford, Rebecca L. McCulley, Sarah Lebeer, and Noah Fierer. 2016. <span>“<span class="nocase">A method for simultaneous measurement of soil bacterial abundances and community composition via 16S rRNA gene sequencing</span>.”</span> <em>Soil Biol. Biochem.</em> 96: 145–51. <a href="https://doi.org/10.1016/j.soilbio.2016.02.003">https://doi.org/10.1016/j.soilbio.2016.02.003</a>.
</div>
<div id="ref-stammler2016adju" class="csl-entry">
Stämmler, Frank, Joachim Gläsner, Andreas Hiergeist, Ernst Holler, Daniela Weber, Peter J. Oefner, André Gessner, and Rainer Spang. 2016. <span>“<span class="nocase">Adjusting microbiome profiles for differences in microbial load by spike-in bacteria</span>.”</span> <em>Microbiome</em> 4 (1): 28. <a href="https://doi.org/10.1186/s40168-016-0175-0">https://doi.org/10.1186/s40168-016-0175-0</a>.
</div>
<div id="ref-tettamantiboshier2020comp" class="csl-entry">
Tettamanti Boshier, Florencia A., Sujatha Srinivasan, Anthony Lopez, Noah G. Hoffman, Sean Proll, David N. Fredricks, and Joshua T. Schiffer. 2020. <span>“<span class="nocase">Complementing 16S rRNA Gene Amplicon Sequencing with Total Bacterial Load To Infer Absolute Species Concentrations in the Vaginal Microbiome</span>.”</span> <em>mSystems</em> 5 (2): 1–14. <a href="https://doi.org/10.1128/mSystems.00777-19">https://doi.org/10.1128/mSystems.00777-19</a>.
</div>
<div id="ref-tkacz2018abso" class="csl-entry">
Tkacz, Andrzej, Marion Hortala, and Philip S. Poole. 2018. <span>“<span class="nocase">Absolute quantitation of microbiota abundance in environmental samples</span>.”</span> <em>Microbiome</em> 6 (1): 110. <a href="https://doi.org/10.1186/s40168-018-0491-7">https://doi.org/10.1186/s40168-018-0491-7</a>.
</div>
<div id="ref-vandeputte2017quan" class="csl-entry">
Vandeputte, Doris, Gunter Kathagen, Kevin D’hoe, Sara Vieira-Silva, Mireia Valles-Colomer, João Sabino, Jun Wang, et al. 2017. <span>“<span class="nocase">Quantitative microbiome profiling links gut community variation to microbial load</span>.”</span> <em>Nature</em> 551 (7681): 507. <a href="https://doi.org/10.1038/nature24460">https://doi.org/10.1038/nature24460</a>.
</div>
<div id="ref-wallace2021thed" class="csl-entry">
Wallace, Megan A., Kelsey A. Coffman, Clément Gilbert, Sanjana Ravindran, Gregory F. Albery, Jessica Abbott, Eliza Argyridou, et al. 2021. <span>“<span class="nocase">The discovery, distribution, and diversity of DNA viruses associated with Drosophila melanogaster in Europe</span>.”</span> <em>Virus Evol.</em> 7 (1): 1–23. <a href="https://doi.org/10.1093/ve/veab031">https://doi.org/10.1093/ve/veab031</a>.
</div>
<div id="ref-wang2021curr" class="csl-entry">
Wang, Xiaofan, Samantha Howe, Feilong Deng, and Jiangchao Zhao. 2021. <span>“<span class="nocase">Current Applications of Absolute Bacterial Quantification in Microbiome Studies and Decision-Making Regarding Different Biological Questions</span>.”</span> <em>Microorganisms</em> 9 (9): 1797. <a href="https://doi.org/10.3390/microorganisms9091797">https://doi.org/10.3390/microorganisms9091797</a>.
</div>
<div id="ref-zemb2020abso" class="csl-entry">
Zemb, Olivier, Caroline S Achard, Jerome Hamelin, Marie‐Léa De Almeida, Béatrice Gabinaud, Laurent Cauquil, Lisanne M. G. Verschuren, and Jean-jacques Godon. 2020. <span>“<span class="nocase">Absolute quantitation of microbes using 16S rRNA gene metabarcoding: A rapid normalization of relative abundances by quantitative PCR targeting a 16S rRNA gene spike‐in standard</span>.”</span> <em>Microbiologyopen</em> 9 (3): 1–21. <a href="https://doi.org/10.1002/mbo3.977">https://doi.org/10.1002/mbo3.977</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="model-details.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="appendix-regression.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/mikemc/differential-abundance-theory/edit/main/appendix.Rmd",
"text": "Edit"
},
"history": {
"link": "https://github.com/mikemc/differential-abundance-theory/commits/main/appendix.Rmd",
"text": null
},
"view": {
"link": "https://github.com/mikemc/differential-abundance-theory/blob/main/appendix.Rmd",
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
