# How taxonomic bias affects abundance measurements {#abundance-measurement}

## Model of MGS measurement

Our primary tool for understanding the impact of taxonomic bias on MGS measurement is the theoretical model of MGS measurement developed and empirically validated by @mclaren2019cons.
This model is the simplest that respects the multiplicative nature of taxonomic bias and the *compositional* nature of MGS measurements, in which the total read count for a sample is unrelated to its total cell number or density (@gloor2017micr).
The model as described by @mclaren2019cons considers only relative abundances; here we present a straightforward extension that allows us to also consider absolute abundances.

<!-- TODO: Is S the set of species that could be in the sample, or the set that can be mapped to? Also, what about the host H? -->
<!-- This issue ties in with our general notation issue, and the problem of unassigned reads. -->

We consider a set of microbiome samples measured by a specific MGS protocol that extracts, sequences, and taxonomically assigns reads to a set of microbial species $S$.
We make several simplifying assumptions to facilitate our analysis and presentation.
First, we consider only species-level assignment, and suppose that reads that cannot be uniquely assigned to a single species in $S$ are discarded.
Second, we ignore the possibility that reads are misassigned to either the wrong species or sample.
Third, we suppose that taxonomic bias acts consistently across samples at the level of species---that is, a given species is always measured more efficiently than another to the same degree, regardless of variation in changes in sub-species-level composition or the sample matrix.
Finally, unless otherwise stated, we treat the sequencing measurement as deterministic, ignoring the 'random' variation in read counts that arise from the sampling of sequencing reads and other aspects of the MGS process.
These assumptions, though unrealistic descriptions of most MGS experiments, serve our purpose of clearly demonstrating when and why consistent taxonomic bias creates errors in DA analysis.
The results we derive from them form the basis for understanding the impact of bias in more complex scenarios.

Our model describes the mathematical relationship between the read counts obtained by MGS and the true absolute abundances of each species in a sample.
For concreteness, we take _absolute abundance_, or simply _abundance_, to refer to cell concentration, or number of cells per unit volume, in a sample.
Thus we take 'absolute' simply to mean not relative to other species; however, our results also apply were 'absolute' taken to mean all cells in a sample or in a sampled ecosystem.
Similarly, our results apply to abundance units other than cells, such as biomass and genome copy number.
Our model stipulates that the assigned read count of a species $i$ in a sample $a$ equals its abundance multiplied by species-specific and sample-specific factors,
\begin{align}
  (\#eq:measurement-model)
  \text{reads}_{i}(a)
  = \text{abun}_{i}(a) \quad \cdot
    \underbrace{\text{efficiency}_{i}}_{\substack{\text{species specific,} \\  \text{sample independent}}}
    \cdot \quad
    \underbrace{\text{effort}(a)}_{\substack{\text{species independent,} \\  \text{sample specific}}}.
\end{align}
The species-specific factor, $\text{efficiency}_{i}$, equals the *relative measurement efficiency* (or simply *efficiency*) of the species---how much more easily that species is measured (converted from cells to assigned reads) relative to a arbitrary fixed reference species (@mclaren2019cons).
We assume the efficiencies of particular species are consistent across samples.
The variation in efficiency among species corresponds to the taxonomic bias of the MGS protocol.
The sample-specific factor, $\text{effort}(a)$, we call the *sequencing effort* for that sample; it captures the variation in the total number of assigned reads due to experimental features such as library normalization and total sequencing-run output.
Equation \@ref(eq:measurement-model) implies that the total number of assigned reads in sample $a$ equals
\begin{align}
  (\#eq:total-reads)
  \text{reads}_S(a)
    = \text{abun}_S(a) \cdot \text{efficiency}_S(a) \cdot \text{effort}(a),
\end{align}
where $\text{abun}_{S}(a) \equiv \sum_{j\in S}\text{abun}_j(a)$ is the total abundance of species in $S$ and 
\begin{align}
  (\#eq:mean-efficiency)
  \text{efficiency}_S(a) 
    \equiv \frac{\sum_{j\in S}\text{abun}_j(a) \cdot \text{efficiency}_j}{\text{abun}_S(a)}
\end{align}
is the _mean efficiency_ over all species in $S$.
<!-- The subscript $S$ serves as a reminder that many reads may not be assigned---for example, reads that fail to pass quality control or that come from the host or other taxonomic domains than what is being studied. -->

## Relative abundance (proportions and ratios)

We first review results from @mclaren2019cons on how bias affects measurements of relative abundance, which are essential to understanding both relative and absolute DA analysis.
We distinguish between two types of species-level *relative abundances* within a sample.
The *proportion* of species $i$ in sample $a$ equals its abundance relative to the total abundance of all species in $S$,
\begin{align}
  (\#eq:prop)
  \text{prop}_{i}(a) 
    &\equiv \frac{\text{abun}_i(a)}{\text{abun}_S(a)}.
%  \\&= \frac{\text{abun}_i(a)}{\sum_{i \in S}\text{abun}_i(a)}.
\end{align}
The *ratio* between two species $i$ and $j$ equals the abundance of $i$ relative to that of $j$,
\begin{align}
  (\#eq:ratio)
  \text{ratio}_{i/j}(a) = \frac{\text{abun}_i(a)}{\text{abun}_j(a)}.
\end{align}
Proportions and ratios each form the basis for popular relative DA methods.

The proportion of a species is typically measured by its proportion of assigned reads,
\begin{align}
  (\#eq:prop-meas)
  \widehat{\text{prop}}_{i}(a) = \frac{\text{reads}_i(a)}{\text{reads}_S(a)}.
\end{align}
We can rewrite the right-hand side (using Equations \@ref(eq:measurement-model), \@ref(eq:total-reads), and \@ref(eq:prop-meas)) to find
\begin{align}
  (\#eq:prop-error)
  \widehat{\text{prop}}_{i}(a)
  &= \text{prop}_{i}(a) \cdot \underbrace{\frac{\text{efficiency}_{i}}{\text{efficiency}_S(a)}}_{\substack{\text{variable} \\ \text{fold error}}}.
\end{align}
Taxonomic bias creates a multiplicative error in the species' proportion equal to its efficiency relative to the mean efficiency in the sample.
Consequently, the species's proportion is measured as too high in samples that are dominated by species with lower efficiencies, and measured as too low in samples that are dominated by species with higher efficiencies.
This phenomenon is illustrated in two hypothetical communities in Figure \@ref(fig:error-proportions). 
Species 3 has an efficiency of 6; it is under-measured in Sample 1, which has a mean efficiency of 8.33, but over-measured in Sample 2, which has a mean efficiency of 3.15.

The measured ratio between species $i$ and $j$ is given by the ratio of their read counts, 
\begin{align}
  (\#eq:ratio-meas)
  \widehat{\text{ratio}}_{i/j}(a) = \frac{\text{reads}_i(a)}{\text{reads}_j(a)}.
\end{align}
From Equations \@ref(eq:measurement-model) and \@ref(eq:ratio-meas), it follows that 
\begin{align}
  (\#eq:ratio-error)
  \widehat{\text{ratio}}_{i/j}(a)
%  &= \frac{\text{abun}_{i}(a)}{\text{abun}_{j}(a)} 
  &= \text{ratio}_{i/j}(a) \cdot \underbrace{\frac{\text{efficiency}_{i}}{\text{efficiency}_{j}}}_{\substack{\text{constant} \\ \text{fold error}}}.
\end{align}
Taxonomic bias creates a multiplicative measurement error in the ratio that is equal to the ratio in their efficiencies; the error is therefore constant across samples.
For instance, in Figure \@ref(fig:error-proportions), the ratio of Species 3 (with an efficiency of 6) to Species 1 (with an efficiency of 1) is over-estimated by a factor of 6 in both communities despite their varying compositions.

<!-- begin figure -->

```{r error-proportions, out.width = '90%', fig.cap = '(ref:cap-error-proportions)', cache = TRUE}
"figures/illustrations/error-proportions.svg" %>%
  include_svg
```

(ref:cap-error-proportions) **Taxonomic bias creates sample-dependent multiplicative errors in species proportions, which can lead to inaccurate fold changes between samples.** Top row: Error in proportions measured by MGS in two hypothetical microbiome samples that contain different relative abundances of three species. Bottom row: Error in the measured fold-change in the third species that is derived from these measurements. Species' proportions may be measured as too high or too low depending on sample composition. For instance, Species 3 has an efficiency of 6 and is under-measured in Sample 1 (which has a mean efficiency of 8.33) but over-measured in Sample 2 (which has a mean efficiency of 3.15). 

<!-- end figure -->

Thus taxonomic bias creates variable fold errors in species proportions, but constant fold errors in species ratios.
Consequently, a perfect cancellation of errors can only occur in ratio-based DA analysis.

**Higher-order taxa:** How does bias affect the proportions and ratios of higher-order taxa, such as a genera, phyla, or even all organisms?
More generally, we can identify an higher-order taxon $I$ with a set of species, $\{i \in I\}$.
The abundance and reads of a taxon $I$ in a sample $a$ equal the sums over its constituent species, $\text{abun}_{I}(a) = \sum_{i \in I}\text{abun}_{i}(a)$ and $\text{reads}_{I}(a) = \sum_{i \in I}\text{reads}_{i}(a)$.
The efficiency of taxon $I$ equals the weighted average of its constituent species,
\begin{align}
  (\#eq:efficiency-general)
  \text{efficiency}_I(a) 
    \equiv \frac{\sum_{i\in I}\text{abun}_i(a)\cdot \text{efficiency}_i}{\text{abun}_I(a)}.
\end{align}
Thus the efficiency of a higher-level taxon may vary across samples (@mclaren2019cons).
An exception is if the constituent species have the same efficiency or always appear in the same abundances relative to each other, in which case the efficiency of the higher-order taxon would stay constant.
As an example, suppose that Species 1 and Species 2 in Figure \@ref(fig:error-proportions) were in the same phylum.
The efficiency of the phylum would then be $\tfrac{1}{2} \cdot 1 + \tfrac{1}{2} \cdot 18 = 9.5$ in Sample 1 and $\tfrac{15}{16} \cdot 1 + \tfrac{1}{16} \cdot 18 \approx 2.1$ in Sample 2.
Equations \@ref(eq:prop-error) and \@ref(eq:ratio-error) continue to describe the measurement error in proportions and ratios involving higher-order taxa, so long as the sample-dependent taxon efficiency defined by Equation \@ref(eq:efficiency-general) is used.
In this way, we can see that both proportions and ratios among higher-order taxa may have inconsistent fold errors.

## Absolute abundance

Various methods can be used to convert the relative abundances from MGS into absolute abundances.
These methods are subject to taxonomic bias in the MGS measurements and, in some cases, in supplemental non-MGS measurements used for the conversion.
Here we describe how various methods for measuring absolute abundance are predicted to be affected by taxonomic bias, with an eye towards determining whether the measurement error is variable or constant.
We consider two general classes of methods: Those in which absolute-abundance information is derived from measurement of the aggregate abundance of the total community, and those in which it is derived from targeted measurement (or prior knowledge) of one or more particular species.

### Total-community normalization

Measurements of species absolute abundances are often obtained by making a (non-MGS) measurement of total community abundance, equating it with the aggregate abundance of the species $S$ measured by MGS, and multiplying this total by the proportions from MGS,
\begin{align}
  (\#eq:density-prop-meas)
  \widehat{\text{abun}}_{i}(a) 
  &= \widehat{\text{prop}}_{i}(a) \cdot \widehat{\text{abun}}_S(a).
\end{align}
This measurement is affected by taxonomic bias in the MGS measurement as well as in the total-abundance measurement.
For example, measurement of total community abundance by 16S qPCR is affected by variation among species in extraction efficiency, 16S copy number, and PCR binding and amplification bias.
If the species-level efficiencies of the total-abundance measurement are constant across samples and we neglect other sources, then we can express the total-abundance measurement as
\begin{align}
  (\#eq:total-density-error)
  \widehat{\text{abun}}_S(a) 
  &= \sum_{i\in S} \text{abun}_i(a) \cdot \text{efficiency}^{\text{tot}}_i
\\&= \text{abun}_S(a) \cdot \text{efficiency}^{\text{tot}}_S(a),
\end{align}
where $\text{efficiency}_{i}^{\text{tot}}(a)$ is the absolute measurement efficiency of species $i$ for the total-abundance measurement and $\text{efficiency}^{\text{tot}}_S(a)$ is the mean efficiency of the total-abundance measurement in the sample.
<!-- Note, we have assumed that the total-abundance and MGS measurements are sensitive to the same species -->
The equations \@ref(eq:total-density-error) and \@ref(eq:prop-error) for the error in total-abundance and proportion measurements imply that the species abundance measurement \@ref(eq:density-prop-meas) has error given by
\begin{align}
  (\#eq:density-prop-error)
  \widehat{\text{abun}}_{i}(a) 
  = \text{abun}_{i}(a) \cdot \frac{\text{efficiency}_{i} \cdot \text{efficiency}^{\text{tot}}_S(a)}{\text{efficiency}_S(a)}.
\end{align}

Equation \@ref(eq:density-prop-error) indicates that the multiplicative error in the measured absolute abundance of a species equals its MGS efficiency relative to the mean MGS efficiency in the sample, multiplied by the mean efficiency of the total measurement.
As in the case of proportions (Equation \@ref(eq:prop-error)), the error depends on sample composition through the two mean efficiency terms and so may vary across samples.
On the other hand, if the mean efficiency of the total-abundance measurement mirrors that of the MGS measurement, the two can offset and lead to a relatively stable error.
We discuss how this possibility might apply to real experimental workflows below.

### Reference-species normalization

Suppose we had a measurement of the (absolute) abundance of a _reference species_ $r$.
In the absence of taxonomic bias, all species are expected to have the same ratio of reads to abundance in a sample (Equation \@ref(eq:measurement-model)).
Thus the abundance-to-reads ratio for species $r$ can serve as a conversion factor allowing us to obtain the abundance of an arbitrary species $i$,
\begin{align}
  (\#eq:density-ratio-meas)
  \widehat{\text{abun}}_{i}(a) = \text{reads}_{i}(a) \cdot \frac{\widehat{\text{abun}}_{r}(a)}{\text{reads}_{r}(a)}.
\end{align}
The abundance of one or more reference species can be directly measured using targeted measurement methods like species-specific qPCR and used with Equation 
\@ref(eq:density-ratio-meas) to obtain absolute abundance for all species.
To our knowledge, this approach has not previously been suggested.
<!-- (Appendix \@ref(model-details) considers possible extensions to multiple reference species.) -->
Instead, reference-based measurement has been used in the context of spike-in experiments or normalization to a species (such as the host) which is treated as having a constant abundance.
In a spike-in experiment, the abundance of reference species are added in known (up to experimental error) abundances to each sample.
When normalizing to a species that is treated as having a constant but unknown abundance, we set $\widehat{\text{abun}}_{r}(a)$ in Equation \@ref(eq:density-ratio-meas) equal to 1 and interpret the abundances as having fixed but unknown units, which is sufficient for multiplicative DA analysis.

The error in the abundance measurement \@ref(eq:density-ratio-meas) due to taxonomic bias in the MGS measurement is given by
\begin{align}
  (\#eq:density-ratio-error)
  \widehat{\text{abun}}_{i}(a) 
  = \text{abun}_{i}(a) \cdot \frac{\text{efficiency}_{i}}{\text{efficiency}_{r}}
  \cdot \begin{array}{c} \text{fold error in} \\ \widehat{\text{abun}_r}(a) \end{array}.
\end{align}
The constant error in the measured ratio of species $i$ to $r$ (see Equation \@ref(eq:ratio-error)) propagates to the abundance measurement.
There will generally also be systematic error in the abundance of the reference species; however, if the systematic fold error is constant across samples, so will be that of the abundances of other species.

Spike-ins are instead sometimes used to measure absolute abundances using Equation \@ref(eq:density-prop-meas).
In this case, an intermediate step is taken in which the total community abundance is first measured by the ratio of non-spike-in to spike-in reads.
<!-- CHECK -->
If done correctly, this calculation yields results that are identical to directly applying Equation \@ref(eq:density-ratio-meas) (Appendix \@ref(total-density-ref)).

**Difference between the two approaches:**
Reference-species normalization yields constant fold errors because it is based on species-level read counts and abundances and we assume that efficiencies are constant at the species level.
In contrast, total-abundance normalization is based on aggregates of species (for the calculation of proportion and the total-abundance measurement) and so depends on mean efficiencies, which can vary across samples.
