# How taxonomic bias affects abundance measurements {#abundance-measurement}

To understand how bias affects the measured differential abundance between samples, we first consider how it affects measurements of individual samples.

## Model of MGS measurement

Our primary tool for understanding the impact of taxonomic bias MGS measurement is the theoretical model of MGS measurement developed and empirically validated by @mclaren2019cons.
This model is the simplest model of MGS measurement that includes multiplicative taxonomic bias while respecting the *compositional* nature of sequencing measurements, in which the total read count for a sample is unrelated to its total cell number or density (@gloor2017micr).
We consider a set of microbial species that are measured by a specific MGS protocol.
The protocol assigns each sequencing read to at most one sample and species and we ignore the possibility of contamination or false-positive taxonomic assignments---we assume that reads assigned to a given species and sample really do come from that species and sample.
Unless otherwise stated, we treat the sequencing measurement as deterministic, ignoring the random error due to the sampling of sequencing reads and aspects of the sequencing process.
Our model stipulates that the assigned read count of species $i$ in sample $a$ equals its cell density multiplied by a species-specific factor and a sample-specific factor,
\begin{align}
  (\#eq:measurement-model)
  \text{reads}_{i}(a)
  = \text{density}_{i}(a) \quad \cdot
    \underbrace{\text{efficiency}_{i}}_{\substack{\text{species specific,} \\  \text{sample independent}}}
    \cdot \quad
    \underbrace{\text{sequencing effort}(a)}_{\substack{\text{species independent,} \\  \text{sample specific}}}.
\end{align}
The species-specific factor is the *relative measurement efficiency* (or *efficiency* for short) of the species, which represents how much more easily that species is measured (converted from cells to assigned reads) relative to an abitrarily chosen reference species (@mclaren2019cons).
The sample-specific factor, which we call the *sequencing effort* of the sample, reflects the fact that the number of reads per unit cell density varies among samples even in the absence of taxonomic bias due to variation in total density, library normalization, and total sequencing-run output.
Equation \@ref(eq:measurement-model) implies that the total reads from all species in the sample equal
\begin{align}
  (\#eq:total-reads)
  \text{total reads}(a)
    = \text{total density}(a) \cdot \text{mean efficiency}(a) \cdot \text{sequencing effort}(a),
\end{align}
where 
\begin{align}
  (\#eq:mean-efficiency)
  \text{mean efficiency}(a) 
  \equiv \frac{\sum_{j}\text{density}_j(a)\cdot \text{efficiency}_j}{\text{total density}(a)}
\end{align}
is the average or mean efficiency of cells in the sample.

## Relative abundances (proportions and ratios)

The term *relative abundance* carries multiple meanings in microbiome science.
Often, relative abundance refers to the number of cells of a given species divided by the total number of cells from the taxonomic domain of interest (e.g. Prokaryotes).
We use the term *proportion* for this type of relative abundance.
A second view, drawn from the statistical framework of Compositional Data Analysis (CoDA), equates the relative abundances in a sample with the *ratios* among two or more species (e.g. a ratio of 5:4:2 among three species).
By this view, it is meaningless to refer to the relative abundance of a species independently of one or more others.
Confusingly, many microbiome papers use 'relative abundance' to mean proportion despite analysing these 'relative abundances' with CoDA-inspired ratio-based analyses.
Additionally, as we will see, DA analyses based on proportions and ratios are differently affected by bias.
For these reasons, we distinguish between proportions and ratios throughout.
Although other interpretations exist^[e.g. the rank order within a sample, or value associated with a species after a centered-log-ratio transform], we focus on the proportion and ratio views as they underly most relative and absolute DA methods and the differences between them clarify when and why the effects of bias cancel in a DA analysis.

The standard measurement of the proportion of species $i$ in sample $a$ is given by the ratio of its read count to the total,
\begin{align}
  (\#eq:prop-meas)
  \widehat{\text{prop}}_{i}(a) = \frac{\text{reads}_i(a)}{\text{total reads(a)}}.
\end{align}
From Equations \@ref(eq:measurement-model), \@ref(eq:total-reads), and \@ref(eq:prop-meas), it follows that
\begin{align}
  (\#eq:prop-error)
  \widehat{\text{prop}}_{i}(a)
  &= \text{prop}_{i}(a) \cdot \frac{\text{efficiency}_{i}}{\text{mean efficiency}(a)},
\end{align}
where $\text{prop}_{i}(a) = \text{density}_{i}(a) / \text{total density}(a)$ is the actual proportion.
Equation \@ref(eq:prop-error) says that taxonomic bias creates a multiplicative error equal to the ratio of the species' efficiency to the mean efficiency in the sample.
Consequently, the same species can be over- or under-measured depending on the composition of the given sample (Figure \@ref(fig:error-proportions)).
For instance, in samples from two hypothetical communities in Figure \@ref(fig:error-proportions), Species 3 has an efficiency of 6 and is under-measured in Sample 1 (which has a mean efficiency of 8.33) but over-measured in Sample 2 (which has a mean efficiency of 3.15).

The measured ratio between species $i$ and $j$ is given by the ratio of their read counts, 
\begin{align}
  (\#eq:ratio-meas)
  \widehat{\text{ratio}}_{i/j}(a) = \frac{\text{reads}_i(a)}{\text{reads}_j(a)}.
\end{align}
From Equations \@ref(eq:measurement-model) and \@ref(eq:ratio-meas), it follows that 
\begin{align}
  (\#eq:ratio-error)
  \widehat{\text{ratio}}_{i/j}(a)
%  \frac{\text{reads}_{i}(a)}{\text{reads}_{j}(a)}
  &= \frac{\text{density}_{i}(a)}{\text{density}_{j}(a)} \cdot \frac{\text{efficiency}_{i}}{\text{efficiency}_{j}}.
\end{align}
The fold error in the ratio between two species simply equals the ratio in their efficiencies and is therefore constant across samples.
For instance, in the example of Figure \@ref(fig:error-proportions), the ratio of Species 3 (with an efficiency of 6) to Species 1 (with an efficiency of 1) is over-estimated by a factor of 6 in both communities despite their varying compositions.

<!-- begin figure -->

```{r error-proportions, out.width = '90%', fig.cap = '(ref:cap-error-proportions)', cache = TRUE}
svg_path <- here("figures/illustrations/error-proportions.svg")
if (knitr::is_html_output()) {
  knitr::include_graphics(svg_path)
} else if (knitr::is_latex_output()) {
  # cowplot::ggdraw() + cowplot::draw_image(svg_path)
  pdf_path <- file.path("/tmp", "bookdown-differential-abundance-theory",
    "error-proportions.pdf")
  args <- c(
    "--export-area-page", 
    "--export-type=pdf", 
    str_c("--export-filename=", pdf_path),
    svg_path
  )
  system2("inkscape", args = args)
  knitr::include_graphics(pdf_path)
  rm(pdf_path)
}
rm(svg_path)
```

(ref:cap-error-proportions) **Taxonomic bias creates sample-dependent multiplicative errors in species proportions, which can lead to inaccurate fold changes between samples.** Top row: Error in proportions measured by MGS in two microbiome samples that contain different relative abundances of three species. Bottom row: Error in the measured fold-change in the third species that is derived from these measurements.

<!-- end figure -->

## Absolute abundances (cell densities)

A wide range of experimental methods exist for converting the relative-abundance information in MGS measurements into measurements of (absolute) cell densities;
however, these methods can be generally partitioned into two types, based on whether the absolute-abundance information derives from knowing the density of the total community or knowing the density of one or more individual species.

**Methods using the density of the total community:**
The total microbial density of a community can be assayed by a variety of methods (reviewed in Appendix \@ref(review-absolute-methods)).
Once the total density is known, the density of individual species can be estimated by multiplying the total density by the species' proportions (as measured by MGS),
\begin{align}
  (\#eq:density-prop-meas)
  \widehat{\text{density}}_{i}(a) 
  &= \widehat{\text{prop}}_{i}(a) \cdot \widehat{\text{total density}}(a)
\\&= \text{reads}_{i}(a) \cdot \frac{\widehat{\text{total density}}(a)}{\text{total reads}(a)}.
\end{align}
We can rewrite the species' density measurements to account for the effect of taxonomic bias on the MGS proportions (Equation \@ref(eq:prop-error)) and any error in the total density measurement, 
\begin{align}
  (\#eq:density-prop-error)
  \widehat{\text{density}}_{i}(a) 
  = \text{density}_{i}(a) \cdot \frac{\text{efficiency}_{i}}{\text{mean efficiency}(a)} 
%  \cdot \frac{\widehat{\text{total density}}(a)}{\text{total density}(a)}.
  \cdot \begin{array}{c} \text{fold error in} \\ \widehat{\text{total density}}(a) \end{array}.
\end{align}
The variable fold error that bias creates in the measured proportions directly transfers to the density measurements.

Taxonomic bias can also create systematic error in the total-density measurements (Appendix \@ref(appendix-total-density)).
For now, we suppose that the total density can be measured accurately; 
later, Section \@ref(solutions) describes how experiments might be designed such that taxonomic bias in the total density measurement offsets that in the MGS proportions, leading to a (more) consistent fold error in the measured densities.

**Methods using the density of one or more reference species:**
The density of one or more species can be directly measured using targeted measurement methods like species-specific qPCR; alternatively, the densities of one or more species may be thought to be constant across samples either from background knowledge or because they have been spiked in at a fixed density (these situations are reviewed in Appendix \@ref(review-absolute-methods)).
These _reference species_ with known or constant density can be used to convert read counts to densities for all species.
For instance, given the density of a reference species $r$, we can estimate the density of a focal species $i$ by 
\begin{align}
  (\#eq:density-ratio-meas)
%  \widehat{\text{density}}_{i}(a) = \frac{\text{reads}_{i}(a)}{\text{reads}_{r}(a)} \cdot \widehat{\text{density}}_{r}(a).
  \widehat{\text{density}}_{i}(a) = \text{reads}_{i}(a) \cdot \frac{\widehat{\text{density}}_{r}(a)}{\text{reads}_{r}(a)}.
\end{align}
Alternatively, if we only know that the reference species is constant, we can set $\widehat{\text{density}}_{r}(a)$ in Equation \@ref(eq:density-ratio-meas) equal to 1 to estimate the density in unknown but fixed units, which is sufficient for multiplicative DA analysis.
(Appendix \@ref(model-details) considers possible extensions to multiple reference species.)
It follows from Equation \@ref(eq:ratio-error) that the error caused by taxonomic bias in this density measurement is
\begin{align}
  (\#eq:density-ratio-error)
  \widehat{\text{density}}_{i}(a) 
  = \text{density}_{i}(a) \cdot \frac{\text{efficiency}_{i}}{\text{efficiency}_{r}} 
  \cdot \begin{array}{c} \text{fold error in} \\ \widehat{\text{density}_r}(a) \end{array}.
\end{align}
Here, it is the constant error in the measured ratio of species $i$ to $r$ that propagates into the density measurement.
There will generally be systematic error in the density of the reference species; however, if the systematic fold error is constant across samples, so will be that of the densities of other species.

**Differences between the two approaches:**
Equations \@ref(eq:density-prop-error) and \@ref(eq:density-ratio-error) show that the manner in which taxonomic bias in the sequencing measurement impacts the two approaches mirrors that of proportions and ratios.
If the fold error in the total-density or reference species measurement is negligible (or at least constant) then the fold errors in species densities by the total-density approach vary across samples with the mean efficiency, whereas the fold errors by the reference-species approach are constant. 
The fundamental determinant of whether constant error is obtained is not which of these two equations is ultimately used, but rather whether densities of individual species, which (we assume) have constant efficiency, or the total community, which has varying efficiency, provide the absolute-abundance information.
Studies using spike-ins as constant reference species will often, instead of using Equation \@ref(eq:density-ratio-meas), first use the spike-in species to estimate the total density, and then estimate the density of focal species with Equation \@ref(eq:density-prop-meas).
If the calculation is done carefully, however, this method still yields constant fold errors so long as the species efficiencies (including those of the spike-in) are constant across samples (Appendix \@ref(appendix-total-density)).
