# Case studies Â {#case-studies}

To better understand the potential impact of taxonomic bias on DA analysis in practice, we conducted several case studies spanning a range of biological scenarios and sequencing technologies.

## Foliar fungi experiment

Measurement of control communities along with the primary experimental samples can enable researchers to directly measure and remove the effect of bias prior to or concurrent with downstream DA analysis (@mclaren2019cons).
Gnotobiotic community experiments are well suited to this form of _calibration via community controls_ since, unlike most natural ecosystems, it is possible to assemble 'mock communities' containing all species in known relative abundances.

@leopold2020host are the first to use mock-based calibration in a gnotobiotic microbiome experiment.
In a study of host-commensal-pathogen interactions, @leopold2020host inoculated plants with 8 commensal fungal species and subsequently exposed plants to a fungal pathogen.
To investigate the joint effect of colonization order and host genotype, plants varied in source location and in which commensal colonized first.
The authors used ITS amplicon sequencing to measure communities before and after pathogen infection, along with mock communities with quantified genome concentrations of DNA from the 9 species.
These DNA mock communities allowed the authors to directly measure the bias due to the sequencing workflow following DNA extraction and correct for it when analyzing the primary experimental samples following the method of @mclaren2019cons.
Although extraction is not accounted for, considerable bias might still be expected due to preferential PCR binding and amplification and the substantial ribosomal copy-number-variation (CNV) in fungi (@lofgren2019geno).
<!-- TODO: add citations -->
The authors found a 13X difference between the most and least efficiently measured commensal, while the pathogen was measured 40X more efficiently than the least efficiently measured commensal.
We reanalyzed this experiment to understand the impact that bias might have had on the study's analyses were it not accounted for.

@leopold2020host used the ITS sequencing profiles of the pre-infection samples to understand the impact of host genotype and an experimental treatment---which species was allowed to colonize first---on the commensal community composition prior to infection.
Two DA analyses were conducted to address these questions.
First, the authors used linear regression of log species proportion (via negative-binomial regression to account for random read-sampling error) to ask whether host genotype, treatment, and their interaction significantly impacted overall community composition.
Second, the authors quantified the strength of priority effects---the advantage of being allowed to colonize first---in different host genotypes by the LFC in the proportion of a species when allowed to colonize first versus later with the majority of commensals.

Since these analysis are both based on proportions, bias could in principle impact their results.
We re-ran both analyses with and without bias correction, finding nearly identical results regardless of whether bias is accounted for.^[The two sets of analyses can be seen at the following links: [negative-binomial regression analysis](https://mikemc.github.io/differential-abundance-theory/notebook/posts/2022-01-06-leopold2020host-original-regression-analysis/index.html); [priority-effects analysis](https://mikemc.github.io/differential-abundance-theory/notebook/posts/2022-01-05-leopold2020host-priority-effects/index.html)]
To understand why, we compared the multiplicative variation taxa proportions with that of the mean efficiency.
The proportions of each taxon varied substantially across samples, much more so than the mean efficiency, which had a geometric range of 1.62X and a geometric standard deviation of just 1.05X (SI Figure \@ref(fig:leopold2020host-variation)).
<!-- Explanation -->
Why does the mean efficiency vary so little despite substantial multiplicative variation in the efficiency among taxa and in the taxa proportions across samples?
This answer relates to the fact that the mean efficiency is a weighted _arithmetic_ average of species proportions and so is insensitive to large multiplicative variation in low-proportion species.
The pre-infection samples in the @leopold2020host study are always dominated by the three species with the highest efficiencies, whose efficiencies vary only by 3X (and just 1.5X between the two most dominant).
The lowest efficiency species, despite having large variation in their proportion on the log scale, are always rare (proportions $\ll 1$, even after bias correction) and hence don't significantly affect the mean efficiency.
<!-- Conclusion -->
Because the log mean efficiency varies much less than the log proportions for each taxon, there is little room for it to distort the results of DA analyses based on variation in log proportions.

<!-- Notes: -->
<!-- - Consider whether to use log-transformed abundances and efficiencies throughout, and standard (non-geometric) ranges and standard deviations -->
<!-- - Consider connecting back to the previous section and reminding of the condition for the variation in mean efficiency to impact regression analyses -->
<!-- - Consider comparing the correlations among taxa with covariates; or at least looking at that of the mean efficiency, since correlation is also an important piece of the requirement -->
<!-- - The priority-effect analysis isn't done as a regression, though it could have been (but would have been a slightly different calculation). I might want to explain this more here and in the notebook post. -->
<!-- - I don't consider/mention whether there are actually 'significant' DA results; that would perhaps be a useful piece of the story. Or: I should perhaps say what the result was of each of the two analyses, so it is more concrete what results are not changing. -->
<!-- - Consider expanding on the impact of bias, using Penicillium as an example: It's proportion is generally ~6X lower without correction, but its changes across samples are not affected by correction -->

We additionally considered the potential for bias to impact analysis of how commensal taxa responded to infection.
In particular, it is interesting to consider whether any commensals are able to increase in absolute concentration in response to infection.
The experiment performed by @leopold2020host does not allow absolute-abundance measurement; nevertheless, the results of the previous sections allow us to consider the impact that bias would have on an analysis of absolute genome concentration measured using the total-abundance normalization method (Equation \@ref(eq:density-prop-meas)).
It is first useful to consider how bias impacts the estimated change in log proportion of each commensal following infection.
The pathogen is absent in pre-infection samples but tends to dominate the community post-infection, raising the mean efficiency of post-infection samples (SI Figure \@ref(fig:leopold2020host-infection-mean-efficiency-dist)).
Across different host genotypes, the average increase in mean efficiency ranges from 2.5X to 5.2X. 
We used simple linear regression to measure the average change in log proportion of each commensal for each host genotype, with and without bias correction.
Regardless of whether bias is accounted for, the log proportion of each commensal decreased, which is unsurprising given the pathogen's growth and the sum-to-one constraint of species proportions.
Without bias correction, however, the LFCs were lower by an amount corresponding to the inverse change in log mean efficiency.
The magnitude errors created were substantial for a large fraction of commensal-host pairs; in several cases, the LFC estimates without bias correction were clearly negative, whereas the estimates with bias correction were close to 0.

Now, consider the impact this shift in mean efficiency might have on a regression analysis with log absolute genome concentration as the response, if measured using the total-abundance normalization method (Equation \@ref(eq:density-prop-meas)).
We consider two scenarios.
_Scenario 1:_ Total genome concentration in each sample is perfectly known and used to measure the concentrations of individual taxa via Equation \@ref(eq:density-prop-meas).
In this case, bias in the MGS measurements would create absolute errors in the LFCs for genome concentration identical to those for proportions.
The scientific error, however, may be much worse.
Since the sum-to-one constraint of proportions no longer applies, the LFCs will necessarily be larger (less negative), such that the absolute error caused by the change in mean efficiency could plausibly create substantial biological errors where some taxa that persist or increase in abundance instead appear to decrease.
_Scenario 2:_ Total ITS concentration is measured by qPCR with the same primers and PCR protocol used for ITS sequencing, and used without any correction to measure individual taxa concentration with the total-abundance normalization method.
In this case, we should expect the increase in mean efficiency to have little to no impact on LFCs, since it would be offset by the increase in mean efficiency of the total-concentration measurement.
Moreover, since the total ITS concentration and ITS sequencing are both performed on the extracted DNA, we can also expect the effect of taxonomic bias caused by DNA sequencing to be offset.
Thus, the systematic shift in mean efficiency following infection may or may not significantly impact the LFC estimates depending on how the total-abundance measurement is made prior to normalization.

<!-- cut text -->
<!-- - Genome concentration was chosen as the measure of abundance, due to the tendency for fungi to grow in forms where cells are not easily demarcated. -->
<!-- - The original study focused on how host genotype and the pre-infection community affected disease severity after challenge with a pathogen, and did not analyze the impact of infection on commensals. -->
<!-- - The post-colonization samples were not analyzed by @leopold2020host in part because the reads from post-colonization samples tended to be dominated by the pathogen, giving little ability to infer the relative abundance of the rarer and lower efficiency species. -->
<!-- - This was presumably due to high pathogen growth and aided by the pathogen's post-extraction efficiency being substantially larger than the other species; the high efficiency of the pathogen typically leads to a roughly 6X increase in the mean efficiency of post-colonization samples (REF SI/FIG). -->

## Vaginal microbiomes of pregnant women

The vaginal microbiome during pregnancy has been a source of intensive study due to its apparent connection with the health of both mother and her developing child.
Many MGS studies have found associations of specific microbial species and community characteristics with rates of urinary tract and sexually-transmitted infections, bacterial vaginosis (BV), and preterm birth.
Yet these associations vary across studies of different populations and using different MGS methods.
DA analyses of the vaginal microbiome are commonly based on proportions, creating an opportunity for taxonomic bias to impact results.
A number of studies have experimentally demonstrated substantial taxonomic bias among MGS protocols and individual steps (such as extraction and PCR amplification) in vaginal samples or _in vitro_ samples of vaginally-associated species (@yuan2012eval,@brooks2015thet,@gill2016eval,@graspeuntner2018sele).
<!-- TODO: perhaps add ref on taxonomic resolution and also sampling methods -->
Bias has been proposed as a potential explanation for discrepancies across studies (@callahan2017repl, Others?), but there has so far been little quantitative analysis of this possibility.
Here we use empirical bias measurements from control samples to investigate the role of bias in proportion-based DA analyses of vaginal microbiomes from a recent large-cohort study of pregnant women.

As part of the Multi-Omic Microbiome Study: Pregnancy Initiative (MOMS-PI) study, @fettweis2019thev collected longitudinal samples from over 1500 pregnant women, including nearly 600 that were measured by amplicon sequencing of the 16S V1-V3 hypervariable region to yield species-level bacterial taxonomic profiles.
Taxonomic bias of this MGS protocol was previously investigated by @brooks2015thet and @mclaren2019cons, using measurements by @brooks2015thet of cellular mock communities of seven common, clinically-relevant vaginal bacterial species. 
<!-- TODO: check on clinical relevance -->
Of these, _Lactobacillus iners_ had the highest efficiency, which was nearly 30X larger than that of the species with the lowest efficiency, _Gardnerella vaginalis_.
A second _Lactobacillus_ species, _L. crispatus_, had an efficiency that was approximately 2X less than _L. iners_ and 15X greater than _G. vaginalis_.
These species, along with the unculturable _Lachnospiraceae BVAB1_, are the most common top (most-abundant) species (SI analysis) and can reach high proportions in individual samples, indicating that shifts between them might drive large changes in the mean efficiency which might in turn impact DA results.

We sought to assess this possibility using a joint analysis of the control measurements from @brooks2015thet and the microbiome measurements from the MOMS-PI study.
To obtain the taxonomic bias among all species identified in the MOMS-PI measurements, we used the taxonomic relationships with the seven control species to impute the efficiencies for the remaining species.
We used these imputed efficiencies to calibrate (correct the effect of bias in) the MOMS-PI measurements, examine variation in the mean efficiency varied across samples, and compare the results of a DA analysis with and without bias correction.

<!-- Such imputed efficiencies carry considerable uncertainty, which we ignore to simplify our presentation.  -->
<!-- Notably, while this uncertainty can majorly impact abundances in individual samples, it has a much weaker impact on LFCs across samples (SI analysis TODO). -->

The mean efficiency varies substantially across vaginal samples (Figure \@ref(fig:momspi-mean-efficiency-dist).
This variation appears to be primarily driven by variation in which species is most abundant, samples in which a _Lactobacillus_ species is most abundant (after calibration) typically have an efficiency that is 3-20X greater than samples in which _G. vaginalis_ is most abundant.
Shifts between _Lactobacillus_-dominance and _Gardnerella_-dominance are common in between-women comparisons and occasionally occur between consecutive visits in individual women (SI Figure \@ref(fig:momspi-mean-efficiency-fcs)).
These shifts typically result in substantial fold changes in mean efficiency (SI Figure \@ref(fig:momspi-mean-efficiency-fcs)) and can cause spurious fold changes in the trajectories of lower-abundance species (SI Figure \@ref(fig:momspi-trajectory)).
Decreases in mean efficiency during transitions from _Lactobacillus_ to _Gardnerella_ dominance would be more extreme for vaginal microbiome primer sets that fail to amplify _Gardnerella_ (@graspeuntner2018sele).

<!-- begin figure -->

```{r momspi-mean-efficiency-dist, fig.cap = '(ref:cap-momspi-mean-efficiency-dist)'}
fs::path(
  "notebook/_posts/2021-11-01-momspi-summary/momspi-summary_files", 
  "figure-html5/momspi-mean-efficiency-dist-1.svg"
  ) %>%
  include_svg
```

(ref:cap-momspi-mean-efficiency-dist) **The mean efficiency in vaginal samples from the MOMS-PI study varies with the most abundant species.**

<!-- end figure -->

We next considered whether the observed variation in mean efficiency could cause systematic error in a DA analysis.
In particular, we hypothesized that DA analysis of species proportions versus a covariate that is associated with _Lactobacillus_ and/or _Gardnerella_ would be particularly prone to spurious results.
Patient metadata was not available due to privacy restrictions; we therefore sought a clinically relevant covariate to use in a regression analysis that could be determined directly from the microbiome profiles and that we _a priori_ expected to be associated with the proportions of _Lactobacillus_ and _Gardnerella_.
Alpha diversity metrics such as species richness, the Shannon index, and the (Inverse) Simpson index have been repeatedly found to be strongly positively associated with bacterial vaginosis (BV; @srinivasan2012bact, @cartwright2018mult).^[@cartwright2018mult found that an observed Simpson Diversity Index of 0.82 (corresponding to an order-2 effective number of species of 5.6) classified high diversity samples as BV positive with a sensitivity of 100% and specificity of 85.1%.]
In addition, it is commonly observed that samples from women with and without BV that are dominated by _Lactobacillus spp._ tend to have higher diversity, whereas samples dominated by _Gardnerella_ tend to have lower diversity.
We therefore chose to perform a DA analysis of species proportion versus alpha diversity, hypothesizing that _Lactobacillus_ and _Gardnerella_ would drive a negative association of mean efficiency with diversity and thereby distort DA estimates for all species.
We split samples into low, medium, and high diversity groups based on Shannon diversity in observed (uncalibrated) microbiome profiles.
We then estimated the LFC in proportion from low- to high-diversity samples for 30 species passing a basic prevalence threshold, with and without bias correction.
A large fraction of entries in the species count matrix were small counts or zeros.
We used therefore used gamma-Poisson (or negative-binomial) regression of the observed read counts to fit a linear model to log species proportions while accounting for uncertainty from the random sampling of reads.
We accounted for bias by including an offset term in the linear model equal to the log ratio of the species' efficiency to the sample mean efficiency.

The mean efficiency was on average lower in high-diversity samples due to a reduction in _Lactobacillus_ dominance (LFC of -1.6, or a 4.8X reduction). 
Our results from the previous section therefore suggest that accounting for bias should reduce the estimated LFC for each species by about this same amount.
We found that, as expected, bias correction reduced the LFC of each species, though the effect was variable and typically somewhat smaller.
Bias correction led to substantial magnitude changes for several species and sign changes in 3 species; however, the relative difference between the corrected and uncorrected LFCs was modest for the 19/30 species with the largest LFCs of over 2 (corresponding to a FC of 7.4X) even after correction.
These large LFCs occur in species that rarely if ever reach high proportions and so are likely driven by the sum-to-one constraint: The high diversity communities are those not dominated by _Lactobacillus_ spp. (or to a lesser extent _G. vaginalis_ and BVAB1).
Therefore it is not clear how well it will generalize to regressions on other covariates.

## Human gut microbiome

The vaginal microbiome is characterised by low species diversity within and across samples, so that variation in efficiency among a few dominant species could drive large variation in the mean efficiency among samples.
The human gut microbiome has substantially greater diversity both within and across samples, which we should expect to result in lower variation in mean efficiency.
The reason is that the mean efficiency is a weighted average of efficiencies of individual species, and so will tend to vary less the more species that are averaged over.
Here we use publically available gut microbiome profiles to consider how the distinct dynamics of the gut microbiome impact the variation in the mean efficiency and its potential to distort DA results.

First we considered variation in healthy subjects and asked how the distinct structures of vaginal and gut communities impacted mean efficiency variation.
We considered vaginal and stool samples from the original Human Microbiome Project (@huttenhower2012stru); these and other results in this case study are based on species-level bacterial relative-abundances produced by Metaphlan3 and made available through the curatedMetagenomicData R package (CITE).
We simulated taxonomic bias with a geometric standard deviation of 3, with and without phylogenetic conservation, 
by sampling from a multivariate normal distribution, with efficiencies either IID or with a covariance equal to the phylogenetic covariance matrix to simulate bias as an evolving trait.
We then compared the distribution of mean efficiency across samples.
We also considered how species-level diversity patterns could help explain this variation.

As expected, the mean efficiency varied substantially less in the higher-diversity gut microbiome samples than in the vaginal microbiome samples (median geometric standard deviation of X versus Y); yet it remained non-negligible.
There are several potential reasons why.
First, although richness is typically around 100 bacterial species in the Metaphlan profiles, the order-2 effective number of species ($^2D$; equivalent to the Inverse Simpson diversity index) is typically only around 10 and often lower. 
This number is more relevant when we consider the reduction in variation when computing a weighted average over species (Appendix; standard deviation in mean efficiency scales as $1 / \sqrt(^2D)$  when efficiencies are IID).
A order-2 diversity that is much lower than species richness arises from the skewed distribution of abundances within a gut microbiome sample, with the most abundant 1-2 species often forming a majority of reads.
It is these high-proportion species, captured by the order-2 diversity, that ultimately drive the variation in mean efficiency, not the large number of low-proportion species that drive the richness.
A separate but related reason is that, although the total number of species across gut microbiomes is large, a small number of species form the vast majority of reads and tend to be the most abundant in individual samples.
The overall effect is similar, though less extreme, as that of the dominance of a few key species in the vaginal microbiome: Variation in the efficiencies among these species has an outsized impact on the mean efficiency, again leading to a weakening of the averaging affect when we consider variation across samples.

A notable aspect of gut microbiome variation, shared with with the vaginal microbiome, is that the most abundant species tend to come from a small set of groups.
In the HMP1 gut data, these are various Bacteroides and Prevotella species; in other gut datasets, Gram positives such as X and Y are also common.
In patients with diarrhea or infections, we may see dominance of Enterobactericeae like E. coli and Klebsiella spp.
More generally, there is phylogenetic conservation at the level of ecological traits.
There is also reports of phylogenetic conservation at the level of the species efficiency for a given protocol.
An implication is that we may have less 'effective diversity' from the perspective of variation in the mean effiency.

(Although this was not a big effect in the simulation)

Might we see correlations between mean efficency and covariates?
One way is for a small number of taxonomic groups to cause major shifts, and also have conserved efficiencies.
An example occurs in an dataset we examined of accurate diarrhaea in children from Dhakka.
Stool from children experiencing accute diarrhea were overwhelmingly dominated by Preoteobacteria, whereas stool from healthy controls were dominated by a roughly even mixture of Actinobacteria, Firmicutes, and Bacteroides.
The Proteobacteria species in the diarrheaic stool were dominated by E. coli and to a lesser extent Klebsiella spp, all in the Enterobactericeae family.
Hence a Gram+/- bias, a bias favoring Enterobactericeae, or a bias favoring E. coli would all be capable of driving systematic variation in the mean efficiency in this study.

On the other hand, in other datasets we examined, the variation in phylum proportions that occured was primarily not associated with the clinical covariate.
In these cases, a Gram or phylum-level bias might primary add noise rather than systematic error, depending on the sample size of the study and the magnitude of true effects.

(Perhaps put this in the conclusion.)
There is another reason to consider that the effect of bias should not be dismissed in the higher-diversity natural ecosystems like the gut.
So far we have only considered the variation in mean efficiency; however, it is the variation in mean efficiency relative to the true effect sizes that determines the signficance of the error.
In conditions where the mean efficiency varies less, we should also expect lower true effect sizes, so it isn't at all obvious that we should be less worried.

## Microbial growth in marine sediments

Our route for mean efficiency to become associated with the covariate is if the covariate quantifies a biological process that preferentially selects for a microbial trait that also tends to increase or decrease measurement efficiency.
We illustrate this potential mechanism with a study of microbial growth in marine sediments.

The surface layers of marine sediments harbor diverse and abundant microbiomes. Total cell density and species richness decrease with depth as resources are consumed in older, deeper sediment layers; however, some taxa are able to grow and increase in density as they are slowly buried.
@lloyd2020evid performed a systematic assessment of growth rates of bacterial and archaeal taxa over a depth of 10 cm (corresponding to ~40 years of burial time) in sediment of the White Oak River estuary.
To estimate growth rate, the authors first measured absolute cell density of microbial taxa using the total-abundance normalization method (Equation \@ref(eq:density-prop-meas)), with taxa proportions measured with 16S amplicon sequencing and total community density measured by directly counting cells using epifluorescence microscopy (CARD-FISH).
The authors refer to these absolute densities as FRAxC measurements, for 'fraction of 16S reads times total cell counts'.
The FRAxC measurements were used to infer growth rate from the slope of a simple linear regression of log cell density against burial time over the first 3 cm below the bioirrigation layer (corresponding to $\sim 8$ years of burial).
To validate the inference of positive growers, the authors compared the growth rates from FRAxC-based inference from two sediment cores, qPCR measurements in these cores for a few reference taxa, and FRAxC-based inference in two replicate laboratory incubation experiments.

Taxonomic bias could lead to systematic error in FRAxC-derived growth rates if sample mean efficiency tends to systematically vary with burial time (or equivalently, depth).
One possibility is that microbes with tougher cell walls tend to persist longer (alive or dead) in the sediment, while at the same time being more difficult to extract DNA from than microbes with weaker cell walls.
In this case, we would expect the relative abundance of tougher species to increase with depth and hence the mean extraction efficiency to decrease, which in turn would lead to inflated growth-rate estimates for all taxa; a possible end result would be that taxa that decay sufficiently slowly would be mistakenly inferred to have positive growth.

```{r}
x <- here(
  'notebook/_posts/2022-01-08-lloyd2020evi-case-study/_output/table1-tidy.rds'
) %>% readRDS
bathy_core30_fraxc <- x %>%
  filter(exp_name == 'Core 30', str_detect(clade, 'Bathyarchaeota'), 
    meas_type == 'FRAxC') %>%
  pull(doubling_rate)
bathy_core30_qpcr <- x %>%
  filter(exp_name == 'Core 30', str_detect(clade, 'Bathyarchaeota'), 
    meas_type == 'qPCR') %>%
  pull(doubling_rate)
bathy_core32_fraxc <- x %>%
  filter(exp_name == 'Core 32', str_detect(clade, 'Bathyarchaeota'), 
    meas_type == 'FRAxC') %>%
  pull(doubling_rate)
bathy_core32_qpcr <- x %>%
  filter(exp_name == 'Core 32', str_detect(clade, 'Bathyarchaeota'), 
    meas_type == 'qPCR') %>%
  pull(doubling_rate)
mbgd_core32_fraxc <- x %>%
  filter(exp_name == 'Core 32', str_detect(clade, 'MBG-D'), 
    meas_type == 'FRAxC') %>%
  pull(doubling_rate)
mbgd_core32_qpcr <- x %>%
  filter(exp_name == 'Core 32', str_detect(clade, 'MBG-D'), 
    meas_type == 'qPCR') %>%
  pull(doubling_rate)
```

We can test the hypothesis that systematic variation in log mean efficiency with depth distorts inferred growth rates using the qPCR measurements of two clades also collected by @lloyd2020evid.
In Section \@ref(review-absolute-methods), we argue that taxonomic bias in targeted qPCR measurements is expected to create a roughly constant fold error and yield fold changes in absolute density that are relatively unaffected by changes in mean efficiency.
Thus comparing qPCR to FRAxC growth rates on the same reference taxa allows us to estimate the systematic error in FRAxC growth rates.
Because the systematic error in regression slopes due to bias is the same for each taxon (Section \@ref(differential-abundance)), these comparisons allow us to draw conclusions about the accuracy of FRAxC growth rates for all taxa.
<!-- Our aim is to compare the growth rates for taxa with both FRAxC and qPCR values for the two sediment cores, as reported in Table 1. -->
The first soil core included qPCR measurements of a single archaeal clade, _Bathyarchaeota_, for which growth rates by qPCR and FRAxC were nearly identical (doubling rates of `r round(bathy_core30_fraxc, 3)`/yr by FRAxC and `r round(bathy_core30_qpcr, 3)`/yr by qPCR).
The second soil core included qPCR measurements of _Bathyarchaeota_ and a second clade, _Thermoprofundales_/MBG-D.
In this core, FRAxC and qPCR growth rates differed more substantially, with growth rates from FRAxC being larger by `r round(bathy_core32_fraxc - bathy_core32_qpcr, 3)`/yr for _Bathyarchaeota_ (`r round(bathy_core32_fraxc, 3)`/yr by FRAxC and `r round(bathy_core32_qpcr, 3)`/yr by qPCR) and by `r round(mbgd_core32_fraxc - mbgd_core32_qpcr, 3)`/yr for _Thermoprofundales_/MBG-D (`r round(mbgd_core32_fraxc, 3)`/yr by FRAxC and `r round(mbgd_core32_qpcr, 3)`/yr by qPCR).
A low number of experimental samples and noise in both the FRAxC and qPCR measurements place significant uncertainty in these measurements; however, the fact that FRAxC-derived growth rates are larger than qPCR-derived rates in all three cases is consistent with our hypothesis that mean efficiency decreases with depth in a manner that systematically biases FRAxC-derived rates to higher values.
The differences in growth rate are small in absolute terms; however, the maximum observed difference of 0.086/yr suggests an error large enough to impact results for some taxa classified as positive growers, whose FRAxC growth rates ranged between 0.04/yr and 0.5/yr.
Overall, the comparison between FRAxC and qPCR measurements gives support to the study conclusions, but suggest that species at the lower end of this range of positive FRAxC-derived rates may in fact be merely persisting or even slowly declining in abundance.

<!-- notes/questions -->

<!-- - also bring in the incubation experiments as providing further evidence? -->
<!-- - discuss that kl et al attempted to account for bias, incorrected concluded consistent bias couldn't mess up results? -->
<!-- - discuss other mech's that could impact mean eff, which might even push in other directions? -->

<!-- Caveats that might be instructive to mention -->

<!-- - qPCR and seq analysis are both at ~family level; this is prob not a big deal when it comes to inferring the systematic variation in mean efficiency though, since qPCR and 16S seq share a lot of bias -->
<!-- - relic DNA -->
<!-- - possibility of non-genetic variation -> change in efficiency with depth. This could still lead to a change in mean efficiency, but would offset the effect for these particular species. -->
<!-- - possible error/bias in CARD-FISH -->
    
## Summary and conclusions

The impact of bias can depend on protocol, biological system, and type of DA analysis being done.
Though these case studies span a highly limited range of possibilities, when combined with the theoretical results of Section \@ref(differential-abundance) suggest some general conclusions about how and when bias will impact DA analyses based on fold changes in proportions.

<!-- P: Canceling of consistent bias does occur to an important degree -->

First, the error caused by consistent taxonomic bias _can mostly cancel_ in cross-sample comparisons and so not impact DA analyses of fold changes in species proportions.
Our theoretical results from Section \@ref(differential-abundance) indicate that when the mean efficiency is roughly constant across samples, the error in proportions cancels in fold change calculations and is absorbed by the intercept term in regression models.
We observed this scenario in the analysis of pre-infection fungal microbiome samples of @leopold2020host and when analyzing the trajectories of the vaginal microbiomes of women when the dominant species remained constant^[This point could use more support in the MOMSPI case study], and a stable mean efficiency within marine soil cores is plausible and consistent with (though unable to be fully determined by) the different growth rate estimates in the @lloyd2020evid experiment.
Section \@ref(differential-abundance) showed that absolute DA analysis using total-abundance normalization are susceptible to errors as with DA analysis of proportions; yet our analysis of two (hypothetical) approaches to analyzing absolute changes in response to infection in the fungal microbiome experiment indicates that the increasingly common approach of pairing marker-gene sequencing with qPCR of the total marker density can largely mitigate this effect, since here what matters is the variation in the ratio of mean efficiencies of the MGS measurement to the total-density measurement and this ratio may remain roughly constant if bias is largely shared by the two measurements.

Yet in other cases, the mean efficiency can vary substantially and create substantial error in fold changes between pairs of samples.
We saw examples when comparing fungal microbiomes pre- and post-infection in the foliar-fungi experiment and comparing vaginal microbiomes with different dominant species in the MOMS-PI experiment.
The impact of these errors on results of DA regression analysis across many samples depends on whether the mean efficiency varies systematically with the covariate of interest.
<!-- P: but bias can still cause substantial systematic error -->
In these two examples, systematic variation of the mean efficiency arose as high-efficiency species tended to dominate samples in one of the sample conditions (post-infection foliar samples or low-diversity vaginal samples).
Thus one way for significant error in DA regression results to arise is when a small number of (or just one) species that have particularly high or low mean efficiencies, form a large proportion of the community in a substantial fraction of samples, and are associated with the covariate of interest.
<!-- This dynamic was responsible for the errors observed in the DA analysis of pre- and post-infection fungal microbiomes from @leopold2020host and in the vaginal microbiome profiles of the MOMS-PI study. -->

For many experimental systems, however, there may not be any one species that frequently forms a large fraction of the community.
In these systems, systematic variation of the mean efficiency can still arise through the collective change of many species that are associated with the covariate of interest.
We described a hypothesized scenario in the marine-sediment case study in which increased burial time (the covariate) selects for lysis-resistant (and so lower efficiency) species.
As another example, treatment with a certain antibiotic might selectively kill easier-to-lyze Gram negative species (or harder-to-lyse Gram positives) and thereby decreasing (or increasing) the mean efficiency in samples collected from a host post-treatment.
Such situations can arise whenever there is a microbial trait that is associated with both measurement efficiency and the biological processes of interest.
An example besides cell-wall structure is ribosomal copy number, which increases a species' efficiency in ribosomal amplicon measurements and is positively associated with metabolic rate.
<!-- Even when there is not a single trait that affects efficiency and the focal biological process,  -->
A more generic mechanism by which mean-efficiency associations might arise is from the evolutionary relationships among species.
Species with more recent common ancestry are expected to be more similar across a wide range of heritable traits, which include traits that affect measurement efficiency (such as cell wall structure, genome size, ribosomal copy number, PCR binding sequence) as well as traits that affect the biological processes under study.
For example, differences in the phylum-level composition of samples from two conditions might be driven by many species that show phylum-level conservation in a relevant biological trait.
If these species also show phylum-level conservation in (potentially different) traits that affect measurement efficiency, an association of mean efficiency with the condition can arise.

<!-- P: Bias can also reduce the precision and power of DA analyses -->

So long as the mean efficiency does not vary systematically with the regression covariates, the error in fold changes serve primarily to reduce the precision and power of regression analyses without systematically distorting estimates and so does not necessarily lead to invalid inferences.
But the reduction in precision could be substantial, particularly given the small sample sizes common in many microbiome studies, and thereby substantially limit the ability to draw meaningful conclusions from a study^[I suspect this non-systematic variation in the mean efficiency is the more typical situation, but we currently lack an example of it in the case studies. We can find a semi-real example by taking real microbiome data from a case-control-type design and simulating a strong degree of bias. I did this with the HMP2 IBD study, but thinnk it might be better to find a different example where there are some clear DA results.].

<!-- Conclusion: Need solutions -->

These observations provide reasons to both worry and hope.
It seems likely that in many experiments the mean efficiency is consistent or is at least not associated with the covariate, so that DA inferences remain valid.
Yet it is not obvious _a priori_ in which studies this condition holds, and there are plausible mechanisms that can create problematic associations of the mean efficiency even in ecosystems with high species diversity.
Thus while we should not discount the large set of existing DA results, we should seek ways to better assess the robustness of results from previous studies and to measure and correct the error caused by bias in future ones.

<!-- ## cut from conclusions -->

<!-- ### bias can push species below the detection limit -->

<!-- - Another mechanism, not considered here, by which bias can dramatically reduce precision, is by pushing speices below the sequencing detection limit. -->
<!-- - e.g. of primers that don't detect Gard; problems with estimates of rare and/or low eff taxa on leopold -->
<!-- - this mechanism can interact with variation in the mean efficiency - e.g. of fungal samples after Mel increases -->

<!-- ### Other cases where bias can matter that we didn't consider -->

<!-- Optional for this draft; may also want to leave for final 'conclusions' / discussion -->

<!-- - DA analyses that aren't based on fold changes - don't get canceling, and might have bigger impact -->
<!-- - taxonomic aggregation - e.g. of Lactobacillus in vaginal; (mabye: Firmicutes in gut) - can get a varying efficiency within the taxon -->
<!-- - ?? zero replacement, interpretation -->

