# Case studies Â {#case-studies}

To better understand the potential impact of taxonomic bias on DA analysis in practice, we conducted several case studies spanning a range of biological scenarios and sequencing technologies.

## Foliar fungi experiment

To study the interactions between a host plant and its fungal symbionts and pathogen, @leopold2020host inoculated plants with 8 fungal species and, after allowing these fungal microbiomes to establish, exposed plants to a fungal pathogen.
To investigate the role of colonization order and host genotype, plants varied in source location and in which symbiont was allowed to colonize first.
The authors used ITS amplicon sequencing to measure communities before and after pathogen infection, along with mock communities with quantified genome concentrations of DNA from the 9 species.
These DNA mock communities allowed the authors to directly measure the bias due to the sequencing workflow following DNA extraction and correct for it when analyzing the primary experimental samples following the method of @mclaren2019cons.
Although extraction is not accounted for, considerable bias might still be expected due to the substantial ribosomal copy-number-variation (CNV) in fungi (CITE) and preferential PCR binding and amplification (CITE).
The authors found a 13X difference between the most and least efficiently measured symbiont, while the pathogen was measured 40X more efficiently than the least efficiently measured commensal.
We reanalyzed this experiment to understand the impact that bias might have had on the study's analyses were it not accounted for (SI Analysis Fungi).

@leopold2020host used the ITS sequencing profiles of the pre-infection samples to understand the impact of host genotype and an experimental treatment---which species was allowed to colonize first---on the community composition after all species were later allowed to colonize.
The authors used linear regression of log species proportions, using negative-binomial regression to account for random read-sampling error, to ask whether host genotype, treatment, and their interaction significantly impacted overall log species proportions and to quantify the advantage of being allowed to colonize first (priority effects).
Bias was accounted for via the use of species- and sample-specific offset terms in the regression model (SI / appendix).
In principle, bias could significantly impact the results of such regressions through variation in the log mean efficiency among samples.
Yet bias has a negligible impact on the study's analyses, with nearly identical DA results being obtained regardless of whether bias correction is used (Ref SI, which will run the analyses from the paper with and without bias correction).
To understand why, we compared the variation in log species proportions across samples to that in log mean efficiency (Figure X).
Individual species show substantial variation in log proportion (give numbers); nevertheless, the log mean efficiency hardly varies (number; perhaps ref a figure).
This difference is due to the communities always being dominated by just three species, whose efficiencies happen to be the highest among the commensal species and only vary by 3X (and just 1.5X between the two most dominant).
The lowest efficiency species, despite having large variation in their proportion on the log scale, are always rare (proportions $\ll 1$, even after bias correction) and hence don't induce significant variation in the mean efficiency.
Due to the stable mean efficiency across samples, bias primarily impacts the intercept terms and has a negligible impact on the estimates and standard errors of slope (DA) terms in the regressions.

Although not analyzed in the study, the authors also sequenced the fungal communities of plants after pathogen colonization.
Comparison of the pre- and post-colonization samples suggests a simple, biologically relevant scenario where bias could confound attempts to infer microbial dynamics.
A natural question to ask in such an experiment is, how do various species in the existing community respond to pathogen invasion?
Relative abundances alone offer limited information, as they are unable to account for the potential sharp increase in the pathogen abundance post-infection, making such an experiment a natural target for absolute-abundance methods.
One possibility would be to measure total genome concentration using qPCR of the single-copy gene that was used to quantify concentrations for the mock communities.
Genome concentration could then be multiplied by ITS sequencing proportions to infer absolute abundances for DA analysis.
Suppose the genome-concentration measurement is 100% accurate.
In this case, it would be likely be important to account for bias in the ITS measurement.
The reason is that the high efficiency of the pathogen typically leads to a roughly 6X increase in the mean efficiency of post-colonization samples (REF SI/FIG), which would tend to cause a concomitant 6X decrease in the infered fold changes in abundance in response to colonization for each species.
Alternatively, one might measure total ITS concentration via qPCR, possibly using the same primers used for ITS sequencing.
In this case, bias correction using the measured efficiencies is unnecessary and would be harmful if not also done for the qPCR measurement.
The reason is that much (if not most) of the bias in the sequencing measurement is shared by the ITS qPCR measurement, including that due to extraction, CNV, and PCR amplification.
Appendix X shows that this shared bias in fact leads to more accurate inferences of fold changes in the genome concentrations of individual species, even while likely giving inaccurate changes in total genome concentration.
In summary, accurate inference of species abundances following pathogen colonization requires accounting for the increase in mean efficiency among samples; however, measuring total community abundance with a method (here, ITS qPCR) that is proportional to the mean efficiency naturally does so without needing to explicitly measure or correct for species efficiencies.
Similarly, the (ratio-based) reference-species approaches discussed in the previous section might instead be used.

<!-- The post-colonization samples were not analyzed by @leopold2020host in part because the reads from post-colonization samples tended to be dominated by the pathogen, giving little ability to infer the relative abundance of the rarer and lower efficiency species. -->
<!-- This was presumably due to high pathogen growth and aided by the pathogen's post-extraction efficiency being substantially larger than the other species; -->
<!-- the high efficiency of the pathogen typically leads to a roughly 6X increase in the mean efficiency of post-colonization samples (REF SI/FIG). -->

TODO:

- Give numerical e.g. for the lowest eff species, Penicillium, whose prop is typically 6X too low but the magnitude of error in FCs is generally smaller
- Check that strength of correlation with covariates isn't an important part of the story as well
- Analysis: re-run DL analysis with and without calibrated offsets

## Vaginal microbiomes of pregant women

### Outline

The vaginal microbiomes of pregnant women have been a source of intensive study due to its perceived connection with the health of both mother and her developing child.
Many MGS studies have found associations of specific microbial species and community characteristics with rates of urinary tract and sexually-transmitted infections, bacterial vaginosis (BV), and preterm births.
Yet these associations and the baseline, apparently healthy, community state vary across studies of different populations and using different MGS methods.
A number of studies have experimentally probed the taxonomic bias of different MGS protocols on vaginally-associated species (@yuan2012eval,@brooks2015thet,@gill2016eval,@graspeuntner2018sele,todo ref on taxonomic resolution and also sampling methods).
These studies have revealed substantial bias among protocols, including variation in extraction efficiency by 10-40X and an inability for certain primer sets to amplify highly-prevalent species.
Taxonomic bias has been proposed as a potential explanation for discrepancies across studies (@callahan2017repl, Others), there has so far been little quantitative analysis of this potential explanation.
Here we re-analyze data from published studies to examine the potential impact of taxonomic bias on DA analysis of vaginal communities of pregnant women.

- P on brooks, momspi studies + calibration
  - GC
  - Intro the studies
  - intro ability to use calibration to look at impact of bias - perhaps don't explain as such
  - how we did it given limited species composition of the mock

We first examine what the taxonomic bias observed in control experiments by @brooks2015thet might imply for DA analysis of vaginal microbiomes of pregnant women reported by @fettweis2019thev.
As part of the Multi-Omic Microbiome Study: Pregnancy Initiative (MOMS-PI) study, @fettweis2019thev collected longitudinal samples from over 1500 pregnant women, including nearly 600 that were measured by amplicon sequencing of the 16S V1-V3 hypervariable region to yield species-level taxonomic profiles.
The effect of taxonomic bias in the 16S protocol had previously been investigated by @brooks2015thet via the construction and measurement of cellular mock communities of seven vaginally-relevant bacterial species.
We reanalyzed these data in @mclaren2019cons using a new approach that makes it possible to measure the relative efficiencies among species from mock community measurements.
Once relative efficiencies of all species are estimated, it is possible to quantitatively examine the impact of bias and to _calibrate_, or correct the effect of bias, in samples with unknown composition (@mclaren2019cons).
To account for species not in the mock communities, we used a simple method to impute the relative efficiencies of remaining species using their taxonomic relationship to the mock species.

maybe say more about this
- Such imputation carries with it substantial uncertainty; however, 
- These species collectively constitute a substantial majority of taxomically assigned reads in the MOMSPI 16S profiles (X for species level, and Y when expand to genus).
- analysis of sensitivty to uncertainty can be done and shows not a very big impact.

Efficiency varied subsantially among control species in the @brooks2015thet experiment, with an approximately 30X advantage of the highest efficiency _L. iners_ over the lowest efficiency _G. vaginalis_.
To assess the impact of bias on the observed fold changes in species proportions, we used our imputed efficiencies to consider how the mean efficiency varied across women.
The mean efficiency varies substantially across samples (SI Figure \@ref(fig:momspi-mean-efficiency-fcs; SUMMARY STATS), with greater variation between samples from different women than samples from the same woman (SI Figure \@ref(fig:momspi-mean-efficiency-fcs)). 
Vaginal microbiomes are often dominated by a single species often forming a vast majority of calibrated or uncalibrated read counts, with _Lactobacillus crispatus_, _Lactobacillus iners_, _G. vaginalis_, and BVAB1 being the most common (CHECK).
Consequently, the mean efficiency tends to track that of most dominant species, with _Lactobacillus_-dominated samples typically having an efficiency that is 3-20X greater _Gardnerella_-dominated samples (Figure \@ref(fig:momspi-mean-efficiency-dist)).
Shifts between _Lactobacillus_ and _Gardnerella_ dominance are common in between-women comparisons, and also occasionally occur between subsequent sampling points in individual women (SI Figure \@ref(fig:momspi-mean-efficiency-fcs)), and typically result in substantial fold changes in mean efficiency.
Examination of the multiplicative trajectories in the proportions of individual species indicates that such transitions can cause spurious fold changes in lower-abundance species (SI Figure TODO).
In contrast, bias-correction only affects the baseline value of species ratios and not the fold changes.

TODO:
- consider whether the error is significant, which depends on the species FC and not just the mean efficiency FC.
    - would be good to somehow visualize or summarize this.
    - should give a better interpretation/assessment of the bottom line here.
    - one way: discuss the variation in log mean efficiency versus that in species proportions. See a contrast to the Leopold situation; here the variation is lower but similar enough to potentially matter
- to show that variation within women is lower, compute standard deviation of log mean efficiency within women and average this (perhaps weighted by n)
- create 1-2 trajectories to show
- also show ratios
- writing/main issue: Consider how often the error matters, in FCs, and make trajectories, - make sure can support the idea that it usually doesn't matter, if that is true.

<!-- begin figure -->

```{r momspi-mean-efficiency-dist, fig.cap = '(ref:cap-momspi-mean-efficiency-dist)'}
svg_path <- fs::path(
  "notebook/_posts/2021-11-01-momspi-summary/momspi-summary_files", 
  "figure-html5/momspi-mean-efficiency-dist-1.svg"
)
if (knitr::is_html_output()) {
  fig_path <- svg_path
} else if (knitr::is_latex_output()) {
  fig_path <- svg_to_pdf(svg_path)
}
knitr::include_graphics(fig_path)
rm(svg_path, fig_path)
```

(ref:cap-momspi-mean-efficiency-dist) **The mean efficiency in vaginal samples from the MOMS-PI study varies with the most abundant species.**

<!-- end figure -->

- P: Simulated DA analysis
    - GC: We next sought to see how ...
    - Methods
    - Results
    - Discussion/caveats
    - notes: could do with fido as well, to show that the results are unaffected.
    
We next considered whether the variation in mean efficiency could cause systematic error in a DA analysis of species proportions.
Due to privacy protections, we did not have access to subject variables that could be used as covariates in a DA analysis.
We therefore performed a DA analysis of species proportion versus community alpha diversity.
A low-diversity, _Lactobacillus_-dominated microbiome is frequently cited as the typical healthy microbiome and high alpha diversity is considered an indication of bacterial vaginosis and has been repeatedly found to be postively associated with risk of preterm birth.
(Though notably diverse and low-_Lactobacillus_ microbiomes are also common in healthy women and even typical in certain populations, including segments of the MOMSPI cohort.)
We split the samples into classes with low and high Shannon diversity in the observed (uncorrected) community composition, leaving out samples with medium diversity, and performed a DA analysis of species proportion versus diversity class.
We evaluated several types of DA analyses: The simple linear regression model applied to log proportion (Section \@ref()); negative-binomial (or gamma-poisson) regression, which models expected log proportion with a linear model while also accounting for the random, discrete nature of sequencing read counts; and a rank-regression method that approximates the commonly-used non-parametric Wilcoxin test (TODO).
We performed each DA method with and without correcting for bias on a set of X species passing a minimum abundance prevalence criterion and compared the effect of bias on estimated regression coefficients for individual species.
The mean efficiency tended to be larger in the high-diversity class (presumably due to a lower proportion of Lactobacillus, CHECK); unsurprisingly, the effect of bias was therefore always to lower the regression coefficient.
For each method, bias caused significant magnitude and sign errors for X-Y species, including creating sign errors for X, Y, and Z species respectively.
The absolute error due to bias is the same for all species in the simple linear regression (see previous section), but for the other two methods varied substantially among species (though was always positive, corresponding to the decrease in log mean efficiency in the high-diversity class).
The DA results also differed substantially across methods.
Together, these variations meant that whether bias had a significant impact on a particular species was often method dependent.

    
- ?? P: Real application: Callahan et al, Gard bias - maybe in future
    - what bias could explain discrepancies?

MOVE: Decreases in mean efficiency during transitions from _Lactobacillus_ to _Gardnerella_ dominance can be expected to be even more extreme for commonly-used vaginal microbiome primers that fail to amplify _Gardnerella_.

## Gut microbiome

(Has not been updated)

It is clear that an individual species with an unusually high or low efficiency can drive large changes in the mean efficiency when it shifts from a very large ($\sim 1$) to very small proportion ($\ll 1$) of the sample.
Consideration of dynamics in the human gut illustrates another a way in which large shifts in higher-level taxa might create similar shifts, due to the fact that efficiencies do not vary independently among closely related species.
Gut microbiomes are dominated by two phyla: The Bacteroidetes and the Firmicutes, the ratio of which varies substantially across individuals.
Within a sample, there are usually multiple species contributing to the abundance of each phylum, with the median Inverse Simpson diversity being around 4 for each phylum in the HMP2 IBD study.
Thus changes from Bacteroidetes to Firmicutes dominance are unlikely to be driven by a single species.
However, if Firmicutes species tend to have systematically different efficiencies than Bacteroidetes species, then changes in phylum dominance can be expected to drive substantial changes in the mean efficiency even if many species contribute.
Indeed, several studies have found DNA extraction protocols to more efficiently lyse Gram-negative Bacteroidetes species than Gram-positive Firmicutes species (though substantial variation within phyla also occurs, @mclaren2019cons).

Systematic error in DA regression results requires that the variation in log mean efficiency is correlated with the covariate.
The examples above each suggest plausible biologically-significant scenarios in which such correlations might arise.
The Bacteroidetes-to-Firmicutes ratio has also been linked to host conditions.
Yet in the HMP2 IBD dataset, the large variation in this ratio is largely independent of disease status, suggesting that the primary effect of any variation in mean efficiency driven by these phyla would be to increase noise rather than create systematic error in DA analysis.

## Marine sediment growers

As sediment is slowly buried, the concentration of oxygen and nutrients decreases, and cell density decreases; @lloyd2020evid asked whether some taxa are nevertheless able to continue to grow and increase in cell density.
To measure density with depth, they used community 16S sequencing coupled with measurements of total community density using CARD-FISH, which they combined using community-normalization (Equation X).
(The authors refer to these absolute density measurements as FRAxC measurements.)
The authors assessed whether density of bacterial and archaeal families increased with depth, using linear regression on log cell density to infer the exponential growth (or decay) rate over the first 3 cm below the bioirrigation layer (corresponding to $\sim 8$ years of burial time). (Per time after converting depth -> time)
The authors did this for two cores, and checked the inferences from these sequencing-based measurements against targeted measurements (qPCR) of 16S concentration for particular taxa.
They further checked w/ sequencing-based growth-rate inference from two laboratory incubations.

The authors attempted to explicitly account for bias by arguing that a consistent taxonomic bias would cancel in the observed log-fold-changes that determine the inferred exponential growth rates (see also @kevorkian2018esti); however, their calculations neglected the impact of changes in mean efficiency which do not cancel and may lead to spurious conclusions (previous section).
We were therefore interested in whether the unaccounted for error caused by bias might have led to spurious inferences about which species were increasing with depth.
Supposing the CARD-FISH measurements accurately measured total density, our results from the previous section indicate that a systematic error in growth rates would arise if the log mean efficiency systematically changed with depth.
Diversity is sufficiently high that this systematic error is unlikely to be driven by just a few species, as in the vaginal microbiome and fungal case studies.
But it is possible that the covariate (burial time) selects directly on a species efficiency as well as its ability to grow. 
In particular, we considered that microbes with tougher cell walls would tend to persist longer (alive or dead) in the sediment, while at the same time being more difficult to extract DNA from than microbes with weaker cell walls.
In this case, we would expect the relative abundance of tougher species to increase with depth and hence the mean extraction efficiency to decrease, which in turn would lead to inflated growth-rate estimates for all taxa; a possible end result would be that taxa that are just persisting or even slowly dying off would be mistakenly inferred to have positive growth.

We can test the hypothesis that systematic variation in log mean efficiency with depth distorts inferred growth rates using the qPCR measurements of two clades also collected by @lloyd2020evid.
qPCR measurements of specific clades are also subject to taxonomic bias, but not the compositional constraint of sequencing, or at least are expected to experience it to a much weaker extent (Section \@ref(review-absolute-methods)). 
Hence we expect the systematic fold error in qPCR measurements to be approximately constant across samples and for targeted qPCR measurements to yield relatively accurate estimates of growth rates.
Comparing qPCR to FRAxC growth rates on the same taxa allows us to estimate the systematic error in the FRAxC growth rates, equal to the rate of change of log mean efficiency with depth.
Because the same error applies to all taxa, these targeted comparisons inform us about the impact of bias on the growth rates of all taxa.

We compared the growth rates for taxa with both FRAxC and qPCR values for the two sediment cores, as reported in Table 1.
The first soil core included qPCR measurements of a single archaeal clade, _Bathyarchaeota_, for which growth rates by qPCR and FRAxC were nearly identical (doubling rates of 0.099/yr by FRAxC and 0.097/yr by qPCR).
The second soil core included qPCR measurements of _Bathyarchaeota_ and a second clade, _Thermoprofundales_/MBG-D.
In this core, FRAxC and qPCR growth rates differed more substantially, with growth rates from FRAxC being larger by 0.012/yr for _Bathyarchaeota_ (0.112/yr by FRAxC and 0.100/yr by qPCR) and by 0.086/yr for _Thermoprofundales_/MBG-D (0.294/yr by FRAxC and 0.208/yr by qPCR).
A low number of experimental samples and noise in both the FRAxC and qPCR measurements place signficant uncertainty in these measurements; however, the fact that FRAxC-derived growth rates are larger than qPCR-derived rates in all three cases is consistent with our hypothesis that mean efficiency decreases with depth in a manner that systematically biases FRAxC-derived rates to higher values.
The differences in growth rate are small in absolute terms; however, the maximum observed difference of 0.086/yr is significant for this low nutrient environment, where growth rates of the identified positive growers ranged between 0.04/yr and 0.5/yr.
Overall, the comparison between FRAxC and qPCR measurements gives support to the study conclusions, but suggest that species at the lower end of this range of positive FRAxC-derived rates may in fact be merely persisting or even slowly declining in abundance.
    
Caveats that might be instructive to mention

- qPCR and seq analysis are both at ~family level; this is prob not a big deal when it comes to inferring the systematic variation in mean efficiency though, since qPCR and 16S seq share a lot of bias
- relic dna
- possibility of non-genetic variation -> change in efficiency with depth. This could still lead to a change in mean efficiency, but would offset the effect for these particular species.
- possible error/bias in CARD-FISH
    
## Conclusions

Systematic error in DA regression results requires that the variation in log mean efficiency is correlated with the covariate.
The MOMSPI regression example illustrates a case where the mean efficiency is highly variable and also correlated with the covariate (diversity class).
The Leopold T1 example illustrates a case where the mean efficiency simply doesn't vary significantly across samples, and hence there is no room for bias to affect regression coefficients.
