# Case studies Â {#case-studies}

To better understand the potential impact of taxonomic bias on DA analysis in practice, we conducted several case studies spanning a range of biological scenarios and sequencing technologies.

## Foliar fungi experiment

Measurement of control communities along with the primary experimental samples can enable researchers to directly measure and remove the effect of bias prior to or concurrent with downstream DA analysis (@mclaren2019cons).
Gnotobiotic community experiments are well suited to this form of _calibration via community controls_ since, unlike most natural ecosystems, it is possible to assemble 'mock communities' containing all species in known relative abundances.

@leopold2020host are the first to use mock-based calibration in a gnotobiotic microbiome experiment.
In a study of host-commensal-pathogen interactions, @leopold2020host inoculated plants with 8 commensal fungal species and subsequently exposed plants to a fungal pathogen.
To investigate the joint effect of colonization order and host genotype, plants varied in source location and in which commensal colonized first.
The authors used ITS amplicon sequencing to measure communities before and after pathogen infection, along with mock communities with quantified genome concentrations of DNA from the 9 species.
These DNA mock communities allowed the authors to directly measure the bias due to the sequencing workflow following DNA extraction and correct for it when analyzing the primary experimental samples following the method of @mclaren2019cons.
Although extraction is not accounted for, considerable bias might still be expected due to preferential PCR binding and amplification (CITE) and the substantial ribosomal copy-number-variation (CNV) in fungi (@lofgren2019geno).
The authors found a 13X difference between the most and least efficiently measured commensal, while the pathogen was measured 40X more efficiently than the least efficiently measured commensal.
We reanalyzed this experiment to understand the impact that bias might have had on the study's analyses were it not accounted for.

@leopold2020host used the ITS sequencing profiles of the pre-infection samples to understand the impact of host genotype and an experimental treatment---which species was allowed to colonize first---on the commensal community composition prior to infection.
Two DA analyses were conducted to address these questions.
First, the authors used linear regression of log species proportion (via negative-binomial regression to account for random read-sampling error) to ask whether host genotype, treatment, and their interaction significantly impacted overall community composition.
Second, the authors quantified the strength of priority effects---the advantage of being allowed to colonize first---in different host genotypes by the LFC in the proportion of a species when allowed to colonize first versus later with the majority of commensals.

Since these analysis are both based on proportions, bias could in principle impact their results.
We re-ran both analyses with and without bias correction, finding nearly identical results regardless of whether bias is accounted for^[The two sets of analyses can be seen at the following links: [negative-binomial regression analysis](https://mikemc.github.io/differential-abundance-theory/notebook/posts/2022-01-06-leopold2020host-original-regression-analysis/index.html); [priority-effects analysis](https://mikemc.github.io/differential-abundance-theory/notebook/posts/2022-01-05-leopold2020host-priority-effects/index.html)].
To understand why, we compared the multiplicative variation taxa proportions with that of the mean efficiency.
The proportions of each taxon varied substantially across samples, much more so than the mean efficiency, which had a geometric range of 1.57X and a geometric standard deviation of just 1.06X (Figure \@ref(fig:leopold2020host-variation)).
<!-- Explanation -->
Why does the mean efficiency vary so little despite substantial multiplicative variation in the efficiency among taxa and in the taxa proportions across samples?
This answer relates to the fact that the mean efficiency is a weighted _arithmetic_ average of species proportions and so is insensitive to large multiplicative variation in low-proportion species.
The pre-infection samples in the @leopold2020host study are always dominated by the three species with the highest efficiencies, whose efficiencies vary only by 3X (and just 1.5X between the two most dominant).
The lowest efficiency species, despite having large variation in their proportion on the log scale, are always rare (proportions $\ll 1$, even after bias correction) and hence don't significantly affect the mean efficiency.
<!-- Conclusion -->
Because the log mean efficiency varies much less than the log proportions for each taxon, there is little room for it to distort the results of DA analyses based on variation in log proportions.

<!-- Notes: -->
<!-- - Consider whether to use log-transformed abundances and efficiencies throughout, and standard (non-geometric) ranges and standard deviations -->
<!-- - Consider connecting back to the previous section and reminding of the condition for the variation in mean efficiency to impact regression analyses -->
<!-- - Consider comparing the correlations among taxa with covariates; or at least looking at that of the mean efficiency, since correlation is also an important piece of the requirement -->
<!-- - The priority-effect analysis isn't done as a regression, though it could have been (but would have been a slightly different calculation). I might want to explain this more here and in the notebook post. -->
<!-- - I don't consider/mention whether there are actually 'significant' DA results; that would perhaps be a useful piece of the story. Or: I should perhaps say what the result was of each of the two analyses, so it is more concrete what results are not changing. -->
<!-- - Consider expanding on the impact of bias, using Penicillium as an example: It's proportion is generally ~6X lower without correction, but its changes across samples are not affected by correction -->

We additionally considered the potential for bias to impact analysis of how commensal taxa responded to infection.
In particular, it is interesting to consider whether any commensals are able to increase in absolute concentration in response to infection.
The experiment performed by @leopold2020host does not allow absolute-abundance measurement; nevertheless, the results of the previous sections allow us to consider the impact that bias would have on an analysis of absolute genome concentration 
measured using the total-community normalization method (Equation \@ref(eq:density-prop-meas)).
It is first useful to consider how bias impacts the estimated change in log proportion of each commensal following infection.
The pathogen is absent in pre-infection samples but tends to dominate the community post-infection (Figure \@ref(fig:leopold2020host-infection-mean-efficiency-dist)), leading to an average increase in mean efficiency ranging from 3.5X to 5X across host genotypes and an expectation that commensals will typically decrease in proportion due to the sum-to-one constraint.
Without bias correction, the log proportion of commensals decrease in each host genotype, often substantially (log-e fold changes less than -2); however, bias leads to overly negative LFCs due to the higher mean efficiency in post-infection samples.
Bias correction substantially lessens the magnitude of LFC estimates, in some cases leading to estimates that are approximately 0 or slightly positive (Figure \@ref(fig:leopold2020host-infection-lfc)).

Now, consider the impact this shift in mean efficiency might have on a regression analysis with log absolute genome concentration as the response, if measured using the total-community normalization method (Equation \@ref(eq:density-prop-meas)).
We consider two scenarios.
_Scenario 1:_ Total genome concentration in each sample is perfectly known and used to measure the concentrations of individual taxa via Equation \@ref(eq:density-prop-meas).
In this case, bias in the MGS measurements would create absolute errors in the LFCs for genome concentration identical to those for proportions.
The scientific error, however, may be much worse.
Since the sum-to-one constraint of proportions no longer applies, the LFCs will necessarily be larger (less negative), such that the absolute error caused by the change in mean efficiency could plausibly create substantial biological errors where taxa that persist or even increase appear to decrease.
_Scenario 2:_ Total ITS concentration is measured by qPCR with the same primers and PCR protocol used for ITS sequencing, and used without any correction to measure individual taxa concentration with the total-community normalization method.
In this case, we should expect the increase in mean efficiency to have little to no impact on LFCs, since it would be offset by the increase in mean efficiency of the total-concentration measurement.
Moreover, since the total ITS concentration and ITS sequencing are both performed on the extracted DNA, we can also expect the effect of taxonomic bias caused by DNA sequencing to be offset.
Thus, the systematic shift in mean efficiency following infection may or may not significantly impact the LFC estimates depending on how the total-community measurement is made prior to normalization.

<!-- cut text -->
<!-- - Genome concentration was chosen as the measure of abundance, due to the tendency for fungi to grow in forms where cells are not easily demarcated. -->
<!-- - The original study focused on how host genotype and the pre-infection community affected disease severity after challenge with a pathogen, and did not analyze the impact of infection on commensals. -->
<!-- - The post-colonization samples were not analyzed by @leopold2020host in part because the reads from post-colonization samples tended to be dominated by the pathogen, giving little ability to infer the relative abundance of the rarer and lower efficiency species. -->
<!-- - This was presumably due to high pathogen growth and aided by the pathogen's post-extraction efficiency being substantially larger than the other species; the high efficiency of the pathogen typically leads to a roughly 6X increase in the mean efficiency of post-colonization samples (REF SI/FIG). -->

## Vaginal microbiomes of pregant women

The vaginal microbiomes of pregnant women have been a source of intensive study due to its perceived connection with the health of both mother and her developing child.
Many MGS studies have found associations of specific microbial species and community characteristics with rates of urinary tract and sexually-transmitted infections, bacterial vaginosis (BV), and preterm births.
Yet these associations and the baseline, apparently healthy, community state vary across studies of different populations and using different MGS methods.
A number of studies have experimentally probed the taxonomic bias of different MGS protocols on vaginally-associated species (@yuan2012eval,@brooks2015thet,@gill2016eval,@graspeuntner2018sele,todo ref on taxonomic resolution and also sampling methods).
These studies have revealed substantial bias among protocols, including variation in extraction efficiency by 10-40X and an inability for certain primer sets to amplify highly-prevalent species.
Taxonomic bias has been proposed as a potential explanation for discrepancies across studies (@callahan2017repl, Others), there has so far been little quantitative analysis of this potential explanation.
Here we re-analyze data from published studies to examine the potential impact of taxonomic bias on DA analysis of vaginal communities of pregnant women.

- P on brooks, momspi studies + calibration
  - GC
  - Intro the studies
  - intro ability to use calibration to look at impact of bias - perhaps don't explain as such
  - how we did it given limited species composition of the mock

We first examine what the taxonomic bias observed in control experiments by @brooks2015thet might imply for DA analysis of vaginal microbiomes of pregnant women reported by @fettweis2019thev.
As part of the Multi-Omic Microbiome Study: Pregnancy Initiative (MOMS-PI) study, @fettweis2019thev collected longitudinal samples from over 1500 pregnant women, including nearly 600 that were measured by amplicon sequencing of the 16S V1-V3 hypervariable region to yield species-level bacterial taxonomic profiles.
DA analyses of the vaginal microbiome, including those of @fettweis2019thev, are most commonly based on species proportions, creating opportunity for bias to have an impact on their results.
Previously, the authors evaluated taxonomic bias of this MGS protocol via the construction and measurement of cellular mock communities of seven vaginal bacterial species (@brooks2015thet), but did not directly account for this bias in @fettweis2019thev.
We used the control measurements from @brooks2015thet to infer the bias of the MOMS-PI 16S protocol and investigate its potential impact on DA analyses of vaginal profiles from the MOMS-PI study.
To estimate the relative efficiencies among all species, we directly estimated the efficiencies in the control samples following the method of @mclaren2019cons and imputed the remaining efficiencies based on taxonomic relationships.
We then used these imputed efficiencies to calibrate the measured taxonomic profiles from the MOMS-PI study and compare DA analyses with and without calibration.
Such imputed efficiencies carry considerable uncertainty, which we ignore to simplify our presentation. 
Notably, while this uncertainty can majorly impact abundances in individual samples, it has a much weaker impact on LFCs across samples (SI analysis).

To assess the potential for bias to affect fold changes in species proportions, we first used the imputed efficiencies to consider how the mean efficiency varied across women.
In contrast to the fungal experiment, the mean efficiency varies substantially across vaginal samples, primarily driven by variation in which species is most abundant (Figure \@ref(fig:momspi-mean-efficiency-dist).
The four most common top species in the observed and calibrated profiles were _Gardnerella vaginalis_, _Lactobacillus iners_, _Lactobacillus crispatus_, and _Lachnospiraceae BVAB1_.
All of these except _Lachnospiraceae BVAB1_ where included in the mock-community experiment, from which we estimated that _L. iners_ has an efficiency of 2X relative to _L. crispatus_ and 30X relative to _G. vaginalis_, with _L. iners_ and _G. vaginalis_ having the maximum and minimum efficiencies among the 7 control species.

HERE
- finish explaining how the variation is driven by changes in these taxa, esp G and L 
- add that the importance of these taxa, and the tendency for one species to domiannt, is a gneral well-suported feature of vaginal mb studies, though the identities varies by population and mgs protocol.
- segue into further considerations - what is the impact of this variation?
  - trajectories
    - impact of big shifts 
  - DA analysis

(TODO: confirm true that often these species dominate. Also check the efficiency numbers)


Consequently, the mean efficiency tends to track that of most dominant species, with _Lactobacillus_-dominated samples typically having an efficiency that is 3-20X greater _Gardnerella_-dominated samples (Figure \@ref(fig:momspi-mean-efficiency-dist)).

Shifts between _Lactobacillus_ and _Gardnerella_ dominance are common in between-women comparisons, and also occasionally occur between subsequent sampling points in individual women (SI Figure \@ref(fig:momspi-mean-efficiency-fcs)), and typically result in substantial fold changes in mean efficiency.
Examination of the multiplicative trajectories in the proportions of individual species indicates that such transitions can cause spurious fold changes in lower-abundance species (SI Figure TODO).
In contrast, bias-correction only affects the baseline value of species ratios and not the fold changes.

TODO:
- consider whether the error is significant, which depends on the species FC and not just the mean efficiency FC.
    - would be good to somehow visualize or summarize this.
    - should give a better interpretation/assessment of the bottom line here.
    - one way: discuss the variation in log mean efficiency versus that in species proportions. See a contrast to the Leopold situation; here the variation is lower but similar enough to potentially matter
- to show that variation within women is lower, compute standard deviation of log mean efficiency within women and average this (perhaps weighted by n)
- create 1-2 trajectories to show
- also show ratios
- writing/main issue: Consider how often the error matters, in FCs, and make trajectories, - make sure can support the idea that it usually doesn't matter, if that is true.

<!-- begin figure -->

```{r momspi-mean-efficiency-dist, fig.cap = '(ref:cap-momspi-mean-efficiency-dist)'}
svg_path <- fs::path(
  "notebook/_posts/2021-11-01-momspi-summary/momspi-summary_files", 
  "figure-html5/momspi-mean-efficiency-dist-1.svg"
)
if (knitr::is_html_output()) {
  fig_path <- svg_path
} else if (knitr::is_latex_output()) {
  fig_path <- svg_to_pdf(svg_path)
}
knitr::include_graphics(fig_path)
rm(svg_path, fig_path)
```

(ref:cap-momspi-mean-efficiency-dist) **The mean efficiency in vaginal samples from the MOMS-PI study varies with the most abundant species.**

<!-- end figure -->

- P: Simulated DA analysis
    - GC: We next sought to see how ...
    - Methods
    - Results
    - Discussion/caveats
    - notes: could do with fido as well, to show that the results are unaffected.
    
We next considered whether the variation in mean efficiency could cause systematic error in a DA analysis of species proportions.
Due to privacy protections, we did not have access to subject variables that could be used as covariates in a DA analysis.
We therefore performed a DA analysis of species proportion versus community alpha diversity.
A low-diversity, _Lactobacillus_-dominated microbiome is frequently cited as the typical healthy microbiome and high alpha diversity is considered an indication of bacterial vaginosis and has been repeatedly found to be postively associated with risk of preterm birth.
(Though notably diverse and low-_Lactobacillus_ microbiomes are also common in healthy women and even typical in certain populations, including segments of the MOMSPI cohort.)
We split the samples into classes with low and high Shannon diversity in the observed (uncorrected) community composition, leaving out samples with medium diversity, and performed a DA analysis of species proportion versus diversity class.
We evaluated several types of DA analyses: The simple linear regression model applied to log proportion (Section \@ref()); negative-binomial (or gamma-poisson) regression, which models expected log proportion with a linear model while also accounting for the random, discrete nature of sequencing read counts; and a rank-regression method that approximates the commonly-used non-parametric Wilcoxin test (TODO).
We performed each DA method with and without correcting for bias on a set of X species passing a minimum abundance prevalence criterion and compared the effect of bias on estimated regression coefficients for individual species.
The mean efficiency tended to be larger in the high-diversity class (presumably due to a lower proportion of Lactobacillus, CHECK); unsurprisingly, the effect of bias was therefore always to lower the regression coefficient.
For each method, bias caused significant magnitude and sign errors for X-Y species, including creating sign errors for X, Y, and Z species respectively.
The absolute error due to bias is the same for all species in the simple linear regression (see previous section), but for the other two methods varied substantially among species (though was always positive, corresponding to the decrease in log mean efficiency in the high-diversity class).
The DA results also differed substantially across methods.
Together, these variations meant that whether bias had a significant impact on a particular species was often method dependent.

    
- ?? P: Real application: Callahan et al, Gard bias - maybe in future
    - what bias could explain discrepancies?

MOVE: Decreases in mean efficiency during transitions from _Lactobacillus_ to _Gardnerella_ dominance can be expected to be even more extreme for commonly-used vaginal microbiome primers that fail to amplify _Gardnerella_.

## Gut microbiome

(Has not been updated)

It is clear that an individual species with an unusually high or low efficiency can drive large changes in the mean efficiency when it shifts from a very large ($\sim 1$) to very small proportion ($\ll 1$) of the sample.
Consideration of dynamics in the human gut illustrates another a way in which large shifts in higher-level taxa might create similar shifts, due to the fact that efficiencies do not vary independently among closely related species.
Gut microbiomes are dominated by two phyla: The Bacteroidetes and the Firmicutes, the ratio of which varies substantially across individuals.
Within a sample, there are usually multiple species contributing to the abundance of each phylum, with the median Inverse Simpson diversity being around 4 for each phylum in the HMP2 IBD study.
Thus changes from Bacteroidetes to Firmicutes dominance are unlikely to be driven by a single species.
However, if Firmicutes species tend to have systematically different efficiencies than Bacteroidetes species, then changes in phylum dominance can be expected to drive substantial changes in the mean efficiency even if many species contribute.
Indeed, several studies have found DNA extraction protocols to more efficiently lyse Gram-negative Bacteroidetes species than Gram-positive Firmicutes species (though substantial variation within phyla also occurs, @mclaren2019cons).

Systematic error in DA regression results requires that the variation in log mean efficiency is correlated with the covariate.
The examples above each suggest plausible biologically-significant scenarios in which such correlations might arise.
The Bacteroidetes-to-Firmicutes ratio has also been linked to host conditions.
Yet in the HMP2 IBD dataset, the large variation in this ratio is largely independent of disease status, suggesting that the primary effect of any variation in mean efficiency driven by these phyla would be to increase noise rather than create systematic error in DA analysis.

## Microbial growth in marine sediments

Our route for mean efficiency to become associated with the covariate is if the covariate quantifies a biological process that preferentially selects for a microbial trait that also tends to increase or decrease measurement efficiency.
We illustrate this potential mechanism with a study of microbial growth in marine sediments.

The surface layers of marine sediments harbor diverse and abundant microbiomes. Total cell density and species richness decrease with depth as resources are consumed in older, deeper sediment layers; however, some taxa are able to grow and increase in density as they are slowly buried.
@lloyd2020evid performed a systematic assessment of growth rates of bacterial and archaeal taxa over a depth of 10 cm (corresponding to ~40 years of burial time) in sediment of the White Oak River estuary.
To estimate growth rate, the authors first measured absolute cell density of microbial taxa using the total-community normalization method (Equation \@ref(eq:density-prop-meas)), with taxa proportions measured with 16S amplicon sequencing and total community density measured by directly counting cells using epifluorescence microscopy (CARD-FISH).
The authors refer to these absolute densities as FRAxC measurements, for 'fraction of 16S reads times total cell counts'.
The FRAxC measurements were used to infer growth rate from the slope of a simple linear regression of log cell density against burial time over the first 3 cm below the bioirrigation layer (corresponding to $\sim 8$ years of burial).
To validate the inference of positive growers, the authors compared the growth rates from FRAxC-based inference from two sediment cores, qPCR measurements in these cores for a few reference taxa, and FRAxC-based inference in two replicate laboratory incubation experiments.

Taxonomic bias could lead to systematic error in FRAxC-derived growth rates if sample mean efficiency tends to systematically vary with burial time (or equivalently, depth).
One possibility is that microbes with tougher cell walls tend to persist longer (alive or dead) in the sediment, while at the same time being more difficult to extract DNA from than microbes with weaker cell walls.
In this case, we would expect the relative abundance of tougher species to increase with depth and hence the mean extraction efficiency to decrease, which in turn would lead to inflated growth-rate estimates for all taxa; a possible end result would be that taxa that decay sufficiently slowly would be mistakenly inferred to have positive growth.

```{r}
x <- here(
  'notebook/_posts/2022-01-08-lloyd2020evi-case-study/_output/table1-tidy.rds'
) %>% readRDS
bathy_core30_fraxc <- x %>%
  filter(exp_name == 'Core 30', str_detect(clade, 'Bathyarchaeota'), 
    meas_type == 'FRAxC') %>%
  pull(doubling_rate)
bathy_core30_qpcr <- x %>%
  filter(exp_name == 'Core 30', str_detect(clade, 'Bathyarchaeota'), 
    meas_type == 'qPCR') %>%
  pull(doubling_rate)
bathy_core32_fraxc <- x %>%
  filter(exp_name == 'Core 32', str_detect(clade, 'Bathyarchaeota'), 
    meas_type == 'FRAxC') %>%
  pull(doubling_rate)
bathy_core32_qpcr <- x %>%
  filter(exp_name == 'Core 32', str_detect(clade, 'Bathyarchaeota'), 
    meas_type == 'qPCR') %>%
  pull(doubling_rate)
mbgd_core32_fraxc <- x %>%
  filter(exp_name == 'Core 32', str_detect(clade, 'MBG-D'), 
    meas_type == 'FRAxC') %>%
  pull(doubling_rate)
mbgd_core32_qpcr <- x %>%
  filter(exp_name == 'Core 32', str_detect(clade, 'MBG-D'), 
    meas_type == 'qPCR') %>%
  pull(doubling_rate)
```

We can test the hypothesis that systematic variation in log mean efficiency with depth distorts inferred growth rates using the qPCR measurements of two clades also collected by @lloyd2020evid.
In Section \@ref(review-absolute-methods), we argue that taxonomic bias in targeted qPCR measurements is expected to create a roughly constant fold error and yield fold changes in absolute density that are relatively unaffected by changes in mean efficiency.
Thus comparing qPCR to FRAxC growth rates on the same reference taxa allows us to estimate the systematic error in FRAxC growth rates.
Because the systematic error in regression slopes due to bias is the same for each taxon (Equation \@ref(eq:regression-error)), these comparisons allow us to draw conclusions about the accuracy of FRAxC growth rates for all taxa.
<!-- Our aim is to compare the growth rates for taxa with both FRAxC and qPCR values for the two sediment cores, as reported in Table 1. -->
The first soil core included qPCR measurements of a single archaeal clade, _Bathyarchaeota_, for which growth rates by qPCR and FRAxC were nearly identical (doubling rates of `r round(bathy_core30_fraxc, 3)`/yr by FRAxC and `r round(bathy_core30_qpcr, 3)`/yr by qPCR).
The second soil core included qPCR measurements of _Bathyarchaeota_ and a second clade, _Thermoprofundales_/MBG-D.
In this core, FRAxC and qPCR growth rates differed more substantially, with growth rates from FRAxC being larger by `r round(bathy_core32_fraxc - bathy_core32_qpcr, 3)`/yr for _Bathyarchaeota_ (`r round(bathy_core32_fraxc, 3)`/yr by FRAxC and `r round(bathy_core32_qpcr, 3)`/yr by qPCR) and by `r round(mbgd_core32_fraxc - mbgd_core32_qpcr, 3)`/yr for _Thermoprofundales_/MBG-D (`r round(mbgd_core32_fraxc, 3)`/yr by FRAxC and `r round(mbgd_core32_qpcr, 3)`/yr by qPCR).
A low number of experimental samples and noise in both the FRAxC and qPCR measurements place significant uncertainty in these measurements; however, the fact that FRAxC-derived growth rates are larger than qPCR-derived rates in all three cases is consistent with our hypothesis that mean efficiency decreases with depth in a manner that systematically biases FRAxC-derived rates to higher values.
The differences in growth rate are small in absolute terms; however, the maximum observed difference of 0.086/yr suggests an error large enough to impact results for some taxa classified as positive growers, whose FRAxC growth rates ranged between 0.04/yr and 0.5/yr.
Overall, the comparison between FRAxC and qPCR measurements gives support to the study conclusions, but suggest that species at the lower end of this range of positive FRAxC-derived rates may in fact be merely persisting or even slowly declining in abundance.

<!-- notes/questions -->

<!-- - also bring in the incubation experiments as providing further evidence? -->
<!-- - discuss that kl et al attempted to account for bias, incorrected concluded consistent bias couldn't mess up results? -->
<!-- - discuss other mech's that could impact mean eff, which might even push in other directions? -->

<!-- Caveats that might be instructive to mention -->

<!-- - qPCR and seq analysis are both at ~family level; this is prob not a big deal when it comes to inferring the systematic variation in mean efficiency though, since qPCR and 16S seq share a lot of bias -->
<!-- - relic DNA -->
<!-- - possibility of non-genetic variation -> change in efficiency with depth. This could still lead to a change in mean efficiency, but would offset the effect for these particular species. -->
<!-- - possible error/bias in CARD-FISH -->
    
## Conclusions

Systematic error in DA regression results requires that the variation in log mean efficiency is correlated with the covariate.
The MOMSPI regression example illustrates a case where the mean efficiency is highly variable and also correlated with the covariate (diversity class).
The Leopold T1 example illustrates a case where the mean efficiency simply doesn't vary significantly across samples, and hence there is no room for bias to affect regression coefficients.
