# Case studies  {#case-studies}

To better understand the potential impact of taxonomic bias on DA analysis in practice, we conducted several case studies spanning a range of biological scenarios and sequencing technologies.

## Foliar fungi experiment

The taxonomic bias species present in 'mock communities' of known composition can be directly measured, and the measured bias can then be used to correct downstream DA analysis (@mclaren2019cons).
In practice it is difficult to construct control communities that span the many species present in complex natural communities.
However, gnotobiotic community experiments are well suited to this form of _calibration via community controls_ since, unlike most natural ecosystems, it is possible to assemble mock communities containing all species in known relative abundances.

In a study of host-commensal-pathogen interactions, @leopold2020host inoculated plants with 8 commensal fungal species and subsequently exposed plants to a fungal pathogen.
The authors used ITS amplicon sequencing to measure communities before and after pathogen infection.
Motivated by the substantial ribosomal copy-number variation (CNV) in fungi (@lofgren2019geno), the authors also performed control measurements of mock communities that they constructed from quantified genomic DNA of the 9 species in the experiment; these controls were used to measure taxonomic bias with the method of @mclaren2019cons.
The authors found a 13X difference between the most and least efficiently measured commensal, while the pathogen was measured 40X more efficiently than the least efficiently measured commensal.

@leopold2020host performed two related DA analyses on the pre-infection communities: the first characterized the relative importance of host genetics and species arrival order on species relative abundances in the fully-established community, and the second quantified the strength of 'priority effects'---the advantage gained by a species from being allowed to colonize first.
Both analyses were based on fold changes in species proportions and so in principle were sensitive to taxonomic bias.
To improve accuracy, the authors incorporated the bias measured from the control samples with analysis-specific calibration procedures.

We repeated the two DA analyses of @leopold2020host with and without calibration and found that the results did not meaningfully differ.
To understand why, we examined the variation in species proportions and the mean efficiency across the pre-infection communities (SI Figure \@ref(fig:leopold2020host-variation)).
Despite the 13X variation in the efficiencies among species, the mean efficiency hardly varied across samples (SI Figure \@ref(fig:leopold2020host-variation)C), having a geometric range of 1.62X and a geometric standard deviation of 1.05X.
This consistency in the mean efficiency was despite the fact that each species each showed substantial multiplicative variation (SI Figure \@ref(fig:leopold2020host-variation)A).
But the pre-infection samples were always dominated by the three species with the highest efficiencies, which varied by 3X and by just 1.5X between the two most dominant.
The mean efficiency, equal to the proportion-weighted arithmetic average of species efficiencies, is insensitive to species present at low relative abundance and so remained relatively constant across samples.
Because the multiplicative variation in the mean efficiency was much smaller than that in the proportions of individual species, it had a negligible impact on the inferred fold changes and the DA analyses based on them.

We performed an additional DA analysis on the data from @leopold2020host to investigate whether any commensals increased in absolute concentration in response to infection.
The pathogen is absent in pre-infection samples but tends to dominate the community post-infection, resulting in a substantially higher mean efficiency in post-infection samples (SI Figure \@ref(fig:leopold2020host-infection-mean-efficiency-dist)).
Across different host genotypes, the average post-infection increase in mean efficiency ranged from 2.5X to 5.2X.
Using gamma-Poisson (or negative-binomial) regression of the observed read counts, we estimated the average LFC in commensal species proportions following infection with and without calibration (SI Figure \@ref(fig:leopold2020host-infection-lfc)).
Commensals typically decreased post-infection, which is expected given the pathogen's growth to most abundant community member and the sum-to-one constraint faced by proportions.
However, failure to account for bias caused the decrease to be overestimated, by an amount corresponding to the inverse change in log mean efficiency.
For commensal-host pairs with relatively small observed decreases, bias correction greatly reduced the magnitude of the negative LFCs or, in several cases, resulted in LFCs that were near 0 or slightly positive.

Although @leopold2020host did not include absolute-abundance measurements, we can consider the impact taxonomic bias would have on an absolute DA analysis in a simple scenario in which total genome concentration of each pre- and post-infection sample is perfectly known.
The total genomic concentrations of each sample, combined with the species proportions measured by MGS, can be combined into a measurement of species concentrations using the total-abundance normalization method (Equation \@ref(eq:density-prop-meas)).
In this case, the bias in the MGS measurements will create absolute errors in the estimated LFCs of genome concentration of equal magnitude to the errors in the LFC estimates for proportions.
The scientific error, however, might become worse.
Suppose that total abundance increased by 2X due to pathogen growth; in this case, species whose proportions remained approximately constant would have increased in absolute abundance by around 2X.
Bias, when not accounted for, shifts LFCs estimates downwards by 2.5X to 5.2X depending on host genotype.
Hence, without bias correction, we would instead conclude that these species decreased.

## Vaginal microbiomes of pregnant women

A growing number of MGS studies have used DA analysis to describe associations between members of the vaginal microbiome and health conditions including urinary tract infections, sexually-transmitted infections, bacterial vaginosis, and preterm birth.
However, these associations often vary between studies.
DA analyses of the vaginal microbiome are commonly based on proportions, creating an opportunity for taxonomic bias to impact results.
Substantial taxonomic bias has been experimentally demonstrated in MGS protocols applied to vaginal samples or in vitro samples of vaginally-associated species (@yuan2012eval, @brooks2015thet, @gill2016eval, @graspeuntner2018sele).
The different biases of different MGS protocols have been proposed as one explanation for discrepancies in DA results across studies (@callahan2017repl).


As part of the Multi-Omic Microbiome Study: Pregnancy Initiative (MOMS-PI) study, @fettweis2019thev collected longitudinal samples from over 1500 pregnant women, including nearly 600 that were taxonomically profiled by amplicon sequencing of the 16S V1-V3 hypervariable region.
The taxonomic bias of the MOMS-PI MGS protocol was previously quantified by @mclaren2019cons using control measurements by @brooks2015thet of cellular mock communities of seven common, clinically-relevant vaginal bacterial species.
Of these, _Lactobacillus iners_ had the highest efficiency, which was nearly 30X larger than that of the species with the lowest efficiency, _Gardnerella vaginalis_.
A second _Lactobacillus_ species, _L. crispatus_, had an efficiency that was approximately 2X less than _L. iners_ and 15X greater than _G. vaginalis_.
These species, along with the unculturable _Lachnospiraceae BVAB1_, are most frequently the most abundant species in a sample and commonly reach high proportions of over 70% (TODO SI analysis).
Therefore, shifts between these dominant species could drive large changes in the sample mean efficiency, which might in turn distort DA analyses.

We sought to understand the potential impact of bias on DA analysis of vaginal microbiome profiles from the MOMS-PI study.
To do so, we first examined variation in the mean efficiency across samples under the assumption that bias in the MOMS-PI study was accurately represented by the @brooks2015thet control measurements.
Using taxonomic relatedness to impute the efficiencies of species not in the controls, we were able to calibrate the MOMS-PI profiles and estimate the mean efficiency of each sample (TODO SI Analysis).
The mean efficiency varies substantially across vaginal samples (Figure 4.1), with samples in which a _Lactobacillus_ species is most abundant typically having a mean efficiency that is 3-20X greater than samples in which _G. vaginalis_ is most abundant.
The vaginal microbiome sometimes shifted between _Lactobacillus_-dominance and _Gardnerella_-dominance between consecutive visits in individual women (SI Figure \@ref(fig:momspi-mean-efficiency-fcs)), which caused magnitude and sign errors in the trajectories of lower-abundance species (SI Figure \@ref(fig:momspi-trajectory)).

<!-- begin figure -->

```{r momspi-mean-efficiency-dist, fig.cap = '(ref:cap-momspi-mean-efficiency-dist)'}
fs::path(
  "notebook/_posts/2021-11-01-momspi-summary/momspi-summary_files", 
  "figure-html5/momspi-mean-efficiency-dist-1.svg"
  ) %>%
  include_svg
```

(ref:cap-momspi-mean-efficiency-dist) **The mean efficiency in vaginal samples from the MOMS-PI study varies with the most abundant species.**

<!-- end figure -->

We hypothesized that proportion-based DA analyses of health condition that is associated with the abundance of common taxa with particularly high or low efficiencies, such as _Lactobacillus_ and/or _Gardnerella_ in the MOMS-PI study, would be particularly prone to spurious results.
A potential example may exist in the vaginal microbiome: Bacterial vaginosis (BV) has been repeatedly found to be strongly positively associated with higher-diversity communities with increased abundance of _G. vaginalis_ and decreased abundance of _Lactobacillus spp._ (@srinivasan2012bact, @cartwright2018mult).
As a proxy for clinical BV, we split samples from the MOMS-PI study into low diversity (non-BV), intermediate diversity, and high diversity (BV-like) groups based on Shannon diversity in observed (uncalibrated) microbiome profiles.
We estimated the LFC in proportion from non-BV to BV-like samples for 30 prevalent bacterial species, with and without bias correction, using gamma-Poisson (or negative-binomial) regression of the observed read counts to fit a linear model to log species proportions.
We accounted for bias by including an offset term in the linear model equal to the log ratio of the species' efficiency to the sample mean efficiency.

As expected, bias correction reduced the estimated LFC from non-BV to BV-like groups for every species tested, though the importance of this effect varied among species (TODO: median and range of change).
Bias correction led to substantial magnitude changes for several species and sign changes in 3 species; however, the relative difference between the corrected and uncorrected LFCs was modest for the 19/30 species with the largest changes, which each had a corrected LFC greater than 2 (corresponding to a FC greater than 7.4X).
Our theoretical results from the previous section suggest that bias correction should reduce the estimated LFC for each species by roughly the amount that the mean efficiency decreases between groups.
Consistent with this prediction, the mean efficiency was lower in high-diversity BV-like samples (mean LFC of -1.6) due to the lower abundance of the _Lactobacillus spp._
that are so efficiently measured by the MOMS-PI MGS protocol.

## Human gut microbiomes

We wondered whether the greater alpha (within-community) diversity in the human gut microbiome samples relative to the vaginal microbiome might reduce the impact of bias on DA analyses.
In a community with higher species-level alpha diversity, the mean efficiency effectively averages over a larger number of species.
Consequently, if communities are assembled randomly, the mean efficiency varies less as the order-2 alpha diversity $^2D$ increases (Appendix REF).^[Order-2 alpha diversity, also known as the order-2 Hill number or effective number of species, is equivalent to the Inverse Simpson diversity index and is given by $^2D = 1 / \sum_{i}P_{i}^2$, where $P_{i}$ is the proportion of species $i$ in the sample (@jost2006entr).]
But real communities are not randomly assembled; instead, species have complex patterns of positive and negative associations (REF).
Thus there is unlikely to be a simple 1-1 relationship between alpha diversity and variation in the mean efficiency.
Moreover, even if the mean efficiency does vary less across gut microbiome samples, if species (relative or absolute) abundances also vary less then the importance of bias for DA analyses might stay the same or even increase.

To investigate this question, we analyzed bacterial species profiles of vaginal and stool samples derived from shotgun sequencing in the Human Microbiome Project (@huttenhower2012stru, SI Methods \@ref(methods-gut)).
Stool profiles had substantially greater order-2 alpha diversity than vaginal profiles (TODO SI FIGURE; geometric average of TODO in stool and TODO in vaginal profiles).
To assess the relative importance of bias for proportion-based DA analyses in the two ecosystems, we quantified the variation in the mean efficiency across samples for a large number of possible taxonomic biases under the assumption that the measured profiles reflected the truth (TODO SI Analysis).
Across simulation replicates, the GSD in the mean efficiency was typically lower in stool than in vaginal samples by a geometric average of 1.5X (SI FIGURE TODO; geometric average GSD of 1.5 for stool samples and 2.2 for vaginal samples).
At the level of individual species, the average GSD in the proportion of a species in gut samples was around 1.8X lower than that in vaginal samples.
The negative effects of bias in a DA analysis of a species' proportion scales with the ratio of the GSD in mean efficiency to that in the species' proportion (Section \@ref(differential-abundance)).
Thus our results suggest that although the mean efficiency is likely to vary less in the gut, it may remain just as or more problematic for proportion-based DA analyses.

## Microbial growth in marine sediments

The surface layers of marine sediments harbor diverse and abundant microbiomes.
Total cell density and species richness decrease with depth as resources are consumed in older, deeper sediment layers; however, some taxa are able to grow and increase in density as they are slowly buried.
@lloyd2020evid performed a systematic assessment of growth rates of bacterial and archaeal taxa over a depth of 10 cm (corresponding to ~40 years of burial time) in sediment of the White Oak River estuary.
Taxa proportions were measured with 16S amplicon sequencing and total community densities were measured by counting cells using epifluorescence microscopy.
The authors then calculated cell concentrations for each taxon by multiplying the taxon’s proportion by the total concentration, a method they referred to as FRAxC measurements for 'fraction of 16S reads times total cell counts'.
Taxon-specific growth rates were inferred from the slope of a simple linear regression of the log of the calculated concentration of each taxon against burial time over the first 3 cm below the bioirrigation layer (corresponding to ~8 years of burial).

The FRAxC concentration measurements made by @lloyd2020evid correspond to the total-abundance approach to inferring absolute abundance (Equation \@ref(eq:density-prop-meas)).
Accordingly, taxonomic bias could lead to systematic error in FRAxC-derived growth rates if sample mean efficiency tends to systematically vary with burial time.
Such systematic variation could occur if microbes with tougher cell walls both persist longer in the sediment and are more difficult to extract DNA from than microbes with weaker cell walls.
In this scenario, the relative abundance of tougher species will increase with burial time, as the cells of weaker species degrade.
This shifting composition will cause the sample mean efficiency to decrease with time, leading to inflated growth-rate estimates for all taxa.
In this scenario, taxa that decay with depth sufficiently slowly would mistakenly be inferred to have positive growth from the FRAxC-calculated abundances.

Importantly, @lloyd2020evid also used qPCR to measure the absolute concentration of two taxa from these same samples.
These qPCR measurements allow us to test the hypothesis that systematic variation in log mean efficiency with depth distorts inferred growth rates.
In Section 8, we argue that taxonomic bias in targeted qPCR measurements is expected to create a roughly constant fold error and yield fold changes in absolute density that are relatively unaffected by changes in mean efficiency.
Thus comparing qPCR to FRAxC growth rates on the same reference taxa allows us to estimate the systematic error in FRAxC growth rates.
Because the systematic error in regression slopes due to bias is the same for each taxon (Section 3), these comparisons allow us to draw conclusions about the accuracy of FRAxC growth rates for all taxa.

```{r}
x <- here(
  'notebook/_posts/2022-01-08-lloyd2020evi-case-study/_output/table1-tidy.rds'
) %>% readRDS
bathy_core30_fraxc <- x %>%
  filter(exp_name == 'Core 30', str_detect(clade, 'Bathyarchaeota'), 
    meas_type == 'FRAxC') %>%
  pull(doubling_rate)
bathy_core30_qpcr <- x %>%
  filter(exp_name == 'Core 30', str_detect(clade, 'Bathyarchaeota'), 
    meas_type == 'qPCR') %>%
  pull(doubling_rate)
bathy_core32_fraxc <- x %>%
  filter(exp_name == 'Core 32', str_detect(clade, 'Bathyarchaeota'), 
    meas_type == 'FRAxC') %>%
  pull(doubling_rate)
bathy_core32_qpcr <- x %>%
  filter(exp_name == 'Core 32', str_detect(clade, 'Bathyarchaeota'), 
    meas_type == 'qPCR') %>%
  pull(doubling_rate)
mbgd_core32_fraxc <- x %>%
  filter(exp_name == 'Core 32', str_detect(clade, 'MBG-D'), 
    meas_type == 'FRAxC') %>%
  pull(doubling_rate)
mbgd_core32_qpcr <- x %>%
  filter(exp_name == 'Core 32', str_detect(clade, 'MBG-D'), 
    meas_type == 'qPCR') %>%
  pull(doubling_rate)
```

The first soil core included qPCR measurements of a single archaeal taxon, _Bathyarchaeota_, for which growth rates by qPCR and FRAxC were nearly identical (doubling rates of `r round(bathy_core30_fraxc, 3)`/yr by FRAxC and `r round(bathy_core30_qpcr, 3)`/yr by qPCR).
The second soil core included qPCR measurements of _Bathyarchaeota_ and a second taxon, _Thermoprofundales_/MBG-D.
In this core, FRAxC and qPCR growth rates differed more substantially, with growth rates from FRAxC being larger by `r round(bathy_core32_fraxc - bathy_core32_qpcr, 3)`/yr for _Bathyarchaeota_ (`r round(bathy_core32_fraxc, 3)`/yr by FRAxC and `r round(bathy_core32_qpcr, 3)`/yr by qPCR) and by `r round(mbgd_core32_fraxc - mbgd_core32_qpcr, 3)`/yr for _Thermoprofundales_/MBG-D (`r round(mbgd_core32_fraxc, 3)`/yr by FRAxC and `r round(mbgd_core32_qpcr, 3)`/yr by qPCR).
Uncertainty in the FRAxC- and qPCR-derived growth rates is not reported and is likely substantial; however, the fact that the FRAxC-derived rates are larger than qPCR-derived rates in all three cases is consistent with our hypothesis that mean efficiency decreased with depth in a manner that systematically biased FRAxC-derived rates to higher values.
The differences in growth rate are small in absolute terms; however, the maximum observed difference of 0.086/yr suggests an error large enough to impact results for some taxa classified as positive growers, whose FRAxC growth rates ranged between 0.04/yr and 0.5/yr.
In summary, the comparison between FRAxC and qPCR measurements supports the overall study conclusions, but also suggests that species at the low end of positive FRAxC-derived rates may be merely persisting or even slowly declining in abundance.

## Summary

The impact of bias can depend on protocol, biological system, and type of DA analysis being done.
Though these case studies span a highly limited range of possibilities, when combined with the theoretical results of Section \@ref(differential-abundance) suggest some general conclusions about how and when bias will impact DA analyses based on fold changes in proportions.

<!-- P: Canceling of consistent bias does occur to an important degree -->

First, the error caused by consistent taxonomic bias _can mostly cancel_ in cross-sample comparisons and so not impact DA analyses of fold changes in species proportions.
Our theoretical results from Section \@ref(differential-abundance) indicate that when the mean efficiency is roughly constant across samples, the error in proportions cancels in fold change calculations and is absorbed by the intercept term in regression models.
We observed this scenario in the analysis of pre-infection fungal microbiome samples of @leopold2020host and when analyzing the trajectories of the vaginal microbiomes of women when the dominant species remained constant^[This point could use more support in the MOMSPI case study], and a stable mean efficiency within marine soil cores is plausible and consistent with (though unable to be fully determined by) the different growth rate estimates in the @lloyd2020evid experiment.
Section \@ref(differential-abundance) showed that absolute DA analysis using total-abundance normalization are susceptible to errors as with DA analysis of proportions; yet our analysis of two (hypothetical) approaches to analyzing absolute changes in response to infection in the fungal microbiome experiment indicates that the increasingly common approach of pairing marker-gene sequencing with qPCR of the total marker density can largely mitigate this effect, since here what matters is the variation in the ratio of mean efficiencies of the MGS measurement to the total-density measurement and this ratio may remain roughly constant if bias is largely shared by the two measurements.

Yet in other cases, the mean efficiency can vary substantially and create substantial error in fold changes between pairs of samples.
We saw examples when comparing fungal microbiomes pre- and post-infection in the foliar-fungi experiment and comparing vaginal microbiomes with different dominant species in the MOMS-PI experiment.
The impact of these errors on results of DA regression analysis across many samples depends on whether the mean efficiency varies systematically with the covariate of interest.
<!-- P: but bias can still cause substantial systematic error -->
In these two examples, systematic variation of the mean efficiency arose as high-efficiency species tended to dominate samples in one of the sample conditions (post-infection foliar samples or low-diversity vaginal samples).
Thus one way for significant error in DA regression results to arise is when a small number of (or just one) species that have particularly high or low mean efficiencies, form a large proportion of the community in a substantial fraction of samples, and are associated with the covariate of interest.
<!-- This dynamic was responsible for the errors observed in the DA analysis of pre- and post-infection fungal microbiomes from @leopold2020host and in the vaginal microbiome profiles of the MOMS-PI study. -->

For many experimental systems, however, there may not be any one species that frequently forms a large fraction of the community.
In these systems, systematic variation of the mean efficiency can still arise through the collective change of many species that are associated with the covariate of interest.
We described a hypothesized scenario in the marine-sediment case study in which increased burial time (the covariate) selects for lysis-resistant (and so lower efficiency) species.
As another example, treatment with a certain antibiotic might selectively kill easier-to-lyse Gram negative species (or harder-to-lyse Gram positives) and thereby decrease (or increase) the mean efficiency in samples collected from a host post-treatment.
Such situations can arise whenever there is a microbial trait that is associated with both measurement efficiency and the biological processes of interest.
An example besides cell-wall structure is ribosomal copy number, which increases a species' efficiency in ribosomal amplicon measurements and is positively associated with metabolic rate.
<!-- Even when there is not a single trait that affects efficiency and the focal biological process,  -->
A more generic mechanism by which mean-efficiency associations might arise is from the evolutionary relationships among species.
Species with more recent common ancestry are expected to be more similar across a wide range of heritable traits, which include traits that affect measurement efficiency (such as cell wall structure, genome size, ribosomal copy number, PCR binding sequence) as well as traits that affect the biological processes under study.
For example, differences in the phylum-level composition of samples from two conditions might be driven by many species that show phylum-level conservation in a relevant biological trait.
If these species also show phylum-level conservation in (potentially different) traits that affect measurement efficiency, an association of mean efficiency with the condition can arise.

<!-- P: Bias can also reduce the precision and power of DA analyses -->

So long as the mean efficiency does not vary systematically with the regression covariates, the measurement error from bias primarily reduces the precision and power of regression analyses without systematically distorting estimates and so does not necessarily lead to invalid inferences.
But the reduction in precision could be substantial, particularly given the small sample sizes common in many microbiome studies, and thereby substantially limit the ability to draw meaningful conclusions from a study^[I suspect this non-systematic variation in the mean efficiency is the more typical situation, but we currently lack an example of it in the case studies. We can find a semi-real example by taking real microbiome data from a case-control-type design and simulating a strong degree of bias. I did this with the HMP2 IBD study, but thinnk it might be better to find a different example where there are some clear DA results.].

<!-- Conclusion: Need solutions -->

These observations provide reasons to both worry and hope.
It seems likely that in many experiments the mean efficiency is consistent or is at least not associated with the covariate, so that DA inferences remain valid.
Yet it is not obvious _a priori_ in which studies this condition holds, and there are plausible mechanisms that can create problematic associations of the mean efficiency even in ecosystems with high species diversity.
Thus while we should not discount the large set of existing DA results, we should seek ways to better assess the robustness of results from previous studies and to measure and correct the error caused by bias in future ones.

<!-- ## cut from conclusions -->

<!-- ### bias can push species below the detection limit -->

<!-- - Another mechanism, not considered here, by which bias can dramatically reduce precision, is by pushing speices below the sequencing detection limit. -->
<!-- - e.g. of primers that don't detect Gard; problems with estimates of rare and/or low eff taxa on leopold -->
<!-- - this mechanism can interact with variation in the mean efficiency - e.g. of fungal samples after Mel increases -->

<!-- ### Other cases where bias can matter that we didn't consider -->

<!-- Optional for this draft; may also want to leave for final 'conclusions' / discussion -->

<!-- - DA analyses that aren't based on fold changes - don't get canceling, and might have bigger impact -->
<!-- - taxonomic aggregation - e.g. of Lactobacillus in vaginal; (mabye: Firmicutes in gut) - can get a varying efficiency within the taxon -->
<!-- - ?? zero replacement, interpretation -->


