---
title: "Implications of taxonomic bias for microbial differential-abundance analysis"
author:
  - Michael R. McLaren^[North Carolina State University; now at Massachusetts Institute of Technology. Send correspondence to mmclaren@mit.edu]
  - Jacob T. Nearing^[Dalhousie University]
  - Amy D. Willis^[University of Washington]
  - Karen G. Lloyd^[University of Tennessee]
  - Benjamin J. Callahan^[North Carolina State University]
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
bibliography: [main.bib, packages.bib]
biblio-style: apalike
link-citations: yes
# description:
# For pandoc latex processing
documentclass: article
geometry:
  - top=1in
  - bottom=1in
  - left=1.5in
  - right=1.5in
output:
  bookdown::gitbook:
    css: style.css
    number_sections: true
    includes:
      in_header: header.html
    config:
      sharing:
        facebook: no
        github: yes
        twitter: no
        linkedin: no
        weibo: no
        instapaper: no
        vk: no
        all: ['facebook', 'twitter', 'linkedin', 'weibo', 'instapaper']
  bookdown::pdf_book:
    includes:
      in_header: latex/preamble.tex
    keep_tex: true
    dev: "pdf"
    papersize: letter
    toc_depth: 2
    toc_unnumbered: yes
    toc_appendix: yes
header-includes:
  - \newcommand{\tot}{{\text{tot}}}
  - \newcommand{\mtot}{{\text{[tot]}}}
  - \DeclareMathOperator{\FE}{FE}
---

# Abstract {-}

### Background {.unlisted .unnumbered}

Differential-abundance (DA) analyses assess how the species in a microbiome vary in their relative or absolute abundance across host or environmental conditions.
These analyses typically use sequencing-based community measurements, which are known to measure some species more efficiently than others.
Understanding how this \emph{taxonomic bias} affects DA analyses is essential for reliable and translatable findings.

### Results {.unlisted .unnumbered}

We characterized the effect of taxonomic bias on multiplicative DA analyses of relative and absolute abundances, using mathematical theory and real and simulated case studies.
We found that taxonomic bias can cause significant errors in a popular class of DA methods---those based on species proportions---if the average measurement efficiency of the community is associated with the condition of interest.
Our case-study analysis finds examples in which the average efficiency is stable or varies randomly across samples (low error due to bias) and in which it varies with the condition of interest (high error due to bias).
We show that the error due to bias can be avoided by using alternative DA methods (based on species ratios) or can be quantified and corrected using appropriate controls.

### Conclusions {.unlisted .unnumbered}

Several common methods of DA analysis are susceptible to errors driven by taxonomic bias; however, alternative methods exist which are theoretically robust to bias, provided that it acts consistently across samples.
Wide adoption of our recommendations can improve the reproducibility, interpretability, and translatability of microbiome DA studies.
