---
title: "Implications of taxonomic bias for microbial differential-abundance analysis"
author:
  - Michael R. McLaren^[Massachusetts Institute of Technology; send correspondence to m.mclaren42@gmail.com]
  - Jacob T. Nearing^[Dalhousie University]
  - Amy D. Willis^[University of Washington]
  - Karen G. Lloyd^[University of Tennessee]
  - Benjamin J. Callahan^[North Carolina State University]
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
bibliography: [main.bib, packages.bib]
biblio-style: apalike
link-citations: yes
# description:
# For pandoc latex processing
documentclass: article
geometry:
  - top=1in
  - bottom=1in
  - left=1.5in
  - right=1.5in
output:
  bookdown::gitbook:
    css: style.css
    number_sections: true
    includes:
      in_header: header.html
    config:
      sharing:
        facebook: no
        github: yes
        twitter: no
        linkedin: no
        weibo: no
        instapaper: no
        vk: no
        all: ['facebook', 'twitter', 'linkedin', 'weibo', 'instapaper']
  bookdown::pdf_book:
    includes:
      in_header: latex/preamble.tex
    keep_tex: true
    dev: "pdf"
    papersize: letter
    toc_depth: 2
    toc_unnumbered: yes
    toc_appendix: yes
header-includes:
  - \newcommand{\tot}{{\text{tot}}}
  - \newcommand{\mtot}{{\text{[tot]}}}
  - \DeclareMathOperator{\FE}{FE}
---

# Abstract {-}

<!-- **Abstract:** -->
Differential-abundance (DA) analyses enable microbiome researchers to assess how microbial species vary in relative or absolute abundance with specific host or environmental conditions, such as health status or pH.
These analyses typically use sequencing-based community measurements that are taxonomically biased to measure some species more efficiently than others.
Understanding the effects that taxonomic bias has on the results of a DA analysis is essential for achieving reliable and translatable findings; yet currently, these effects are unknown.
Here, we characterized these effects for DA analyses of both relative and absolute abundances, using a combination of mathematical theory and data analysis of real and simulated case studies.
We found that, for analyses based on species proportions, taxonomic bias can cause significant errors in DA results if the average measurement efficiency of the community is associated with the condition of interest.
These errors can be avoided by using more robust DA methods (based on species ratios) or quantified and corrected using appropriate controls.
Wide adoption of our recommendations can improve the reproducibility, interpretability, and translatability of microbiome DA studies.


```{r}
repo <- git2r::repository(".")
sha <- repo %>% git2r::commits(n = 1) %>% .[[1]] %>% git2r::sha()
sha_short <- str_sub(sha, 1, 7)
status <- repo %>% git2r::status()
```

::: {#preface-note .rmdnote}

This manuscript was rendered from commit `r sha_short` of https://github.com/mikemc/differential-abundance-theory.
Supporting data analyses can be found in the accompanying [computational research notebook](https://mikemc.github.io/differential-abundance-theory/notebook/).
Please post comments or questions on [GitHub](https://github.com/mikemc/differential-abundance-theory).
The manuscript is licensed under a [CC BY 4.0 License](https://creativecommons.org/licenses/by/4.0/).
See the [GitHub Releases](https://github.com/mikemc/differential-abundance-theory/releases) or [Zenodo record](https://doi.org/10.5281/zenodo.4552717) for earlier versions.

```{r}
if (!identical(status %>% unlist %>% length, 0L))
  print(status)
```

:::

