# Introduction {#introduction}

<!-- First paragraph: DA analysis -->
One of the most basic questions we can ask about microbial communities is: How do different microbial taxa vary in abundance—across space, time, and host or environmental conditions?
Advances in sequencing technology allow us to simultaneously measure the abundances of 100s to 1000s of species using marker-gene and shotgun metagenomic sequencing (jointly, MGS).
Although standard MGS measurements lose information about total microbial density—and so are typically used to analyze the abundances of taxa relative to each other or their total—new studies are increasingly employing strategies to enable the analysis of cell density or other measures of "absolute abundance."
These relative and absolute abundances serve as the basis for a _differential-abundance (DA) analysis_, in which the change in abundance of a microbial taxon across samples or conditions is used to learn about the biology of the taxon and its impact on the host and other microbes as well as to detect predictive biomarkers of host and environmental health and disease.

<!-- P: Taxonomic bias -->
Although MGS-based DA analysis has been widely deployed and achieved many notable successes, it faces serious concerns over accuracy and reproducibility due to the inherent technical limitations of MGS measurements.
In particular, MGS measurements are _taxonomically biased_: Taxa vary dramatically (e.g. 10-1000X) in how efficiently they are measured—that is, converted from cells into taxonomically classified sequencing reads—by a given MGS protocol (@mclaren2019cons).
As a result, the abundance measurements obtained by MGS are inaccurate representations of the actual abundances and also tend to differ across protocols, studies, and even experimental batches (@yeh2018taxo,@mclaren2019cons).
This bias arises from variation in how taxa respond to each step in an MGS protocol, from sample collection to bioinformatic classification.
Although often associated with variation in primer binding and amplification rates and marker-gene copy-number, large variation in DNA extraction efficiency and in the ability to correctly classify reads make taxonomic bias a feature of both shotgun and marker-gene measurements.
The error it causes have been found to in some cases to supersede sizable biological differences (e.g. @lozupone2013meta) and has plausibly caused replication failures for prominent findings such as the association of decreased Bacteroides and increased Firmicutes in stool with obesity (@finucane2014atax) and the association of certain taxa in the vagina of pregnant women with preterm birth (@callahan2017repl).

<!-- Revise below here to outline -->

<!-- P: Taxonomic bias and DA -->

The typical approach to countering taxonomic bias in DA analysis is to standardize the measurement protocol used within a given study.
In broad strokes, the thinking is that the measurements of samples measured by the same protocol will be affected by bias in the same way and so the inferred differences between samples (the focus of DA analysis) will be unaffected.
For example, if taxonomic bias consistently causes the measured proportion of a given species to be 10X too high, we can still accurately infer its fold changes across samples (@kevorkian2018esti,@lloyd2020evid).
However, @mclaren2019cons used theoretical arguments and sequencing measurements of defined bacterial (or 'mock') communities to show that consistent taxonomic bias can lead to variable fold errors in measured proportions (Figure \@ref(fig:basic-problem)).
These varying errors can lead to spurious conclusions for how the proportion of a taxon varies across samples, even in the direction of change (for example, causing a taxon that decreases appear to increase).
Yet @mclaren2019cons also found that the fold error in the ratios among species was constant, making fold changes in ratios robust to bias.
The implications of these findings for changes in absolute abundance---which remain subject to taxonomic bias in the underlying MGS measurement---and for DA analysis across many species and many samples---as commonly done in microbiome association testing---have yet to be investigated.

<!-- P: Objective and summary of the present article -->

Here we use a combination of theoretical analysis, simulation, and re-analysis of published experiments to consider when and why taxonomic bias in MGS measurements leads to spurious results in DA analysis of relative and absolute abundance.
Our analysis clarifies how the folk wisdom that taxonomic bias does not affect the analysis of change across samples is only partially correct and can give a false sense of security in the accuracy of DA results.
Yet we also present several potential solutions---methods for quantifying, correcting, or otherwise accounting for the effect of taxonomic bias in DA analyses that can be deployed today with only modest changes to existing experimental and analytical workflows.
Over time, application of these methods to past and future experiments will provide crucial quantitative information about the conditions under which taxonomic bias creates spurious results for various DA methodologies.
Collectively, these methods and insights may provide practical solutions to taxonomic bias in DA analysis and the confidence that is necessary to codify the statistical findings of microbiome studies into readily-translatable scientific knowledge.

<!-- begin figure -->

```{r basic-problem, fig.cap = '(ref:cap-basic-problem)'}
svg_path <- here::here(
  "notebook/_posts/2021-10-27-illustrate-the-problem/illustrate-the-problem_files", "figure-html5/brooks_lcrisp_prop_and_fc-1.svg"
)
if (knitr::is_html_output()) {
  knitr::include_graphics(svg_path)
} else if (knitr::is_latex_output()) {
  # cowplot::ggdraw() + cowplot::draw_image(svg_path)
  pdf_path <- file.path("/tmp", "bookdown-differential-abundance-theory",
    "error-proportions.pdf")
  args <- c(
    "--export-area-page", 
    "--export-type=pdf", 
    str_c("--export-filename=", pdf_path),
    svg_path
  )
  system2("inkscape", args = args)
  knitr::include_graphics(pdf_path)
  rm(pdf_path)
}
rm(svg_path)
```

(ref:cap-basic-problem) **Mock community experiments show that taxonomic bias can distort the measured fold change in an individual species' proportion across samples.** The figure shows the measured vs. actual proportions for a single bacterial species, _Lactobacillus crispatus_, in a set of bacterial cellular mock communities, and the resulting fold changes between community samples. The inconsistent error in the measured proportions of individual samples (Panel A) leads to inaccurate measurements of fold changes (Panel B). Mock communities were constructed and measured with 16S sequencing by @brooks2015thet. The data was re-analyzed by @mclaren2019cons, who showed that despite the inconsistency of the errors in Panel A, taxonomic bias acted consistently across samples.

<!-- end figure -->
