<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007ba5; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #007ba5; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #007ba5; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #5e5e5e; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>

<style>
  div.csl-bib-body { }
  div.csl-entry {
    clear: both;
    }
  .hanging div.csl-entry {
    margin-left:2em;
    text-indent:-2em;
  }
  div.csl-left-margin {
    min-width:2em;
    float:left;
  }
  div.csl-right-inline {
    margin-left:2em;
    padding-left:1em;
  }
  div.csl-indent {
    margin-left: 2em;
  }
</style>

  <!--radix_placeholder_meta_tags-->
  <title>MOMS-PI: Key dynamics</title>



  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2021-11-01"/>
  <meta property="article:created" itemprop="dateCreated" content="2021-11-01"/>
  <meta name="article:author" content="Michael R. McLaren"/>

  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="MOMS-PI: Key dynamics"/>
  <meta property="og:type" content="article"/>
  <meta property="og:locale" content="en_US"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="MOMS-PI: Key dynamics"/>

  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=The truth about metagenomics: quantifying and counteracting bias in 16S rRNA studies;citation_publication_date=2015;citation_publisher=BioMed Central;citation_volume=15;citation_doi=10.1186/s12866-015-0351-6;citation_author=J Paul Brooks;citation_author=David J Edwards;citation_author=Michael D Harwich;citation_author=Maria C Rivera;citation_author=Jennifer M Fettweis;citation_author=Myrna G Serrano;citation_author=Robert A Reris;citation_author=Nihar U Sheth;citation_author=Bernice Huang;citation_author=Philippe Girerd;citation_author=Jerome F Strauss;citation_author=Kimberly K Jefferson;citation_author=Gregory A Buck"/>
  <meta name="citation_reference" content="citation_title=The vaginal microbiome and preterm birth;citation_publication_date=2019;citation_publisher=Nature Publishing Group;citation_volume=25;citation_doi=10.1038/s41591-019-0450-2;citation_author=Jennifer M. Fettweis;citation_author=Myrna G. Serrano;citation_author=Jamie Paul Brooks;citation_author=David J. Edwards;citation_author=Philippe H. Girerd;citation_author=Hardik I. Parikh;citation_author=Bernice Huang;citation_author=Tom J. Arodz;citation_author=Laahirie Edupuganti;citation_author=Abigail L. Glascock;citation_author=Jie Xu;citation_author=Nicole R. Jimenez;citation_author=Stephany C. Vivadelli;citation_author=Stephen S. Fong;citation_author=Nihar U. Sheth;citation_author=Sophonie Jean;citation_author=Vladimir Lee;citation_author=Yahya A. Bokhari;citation_author=Ana M. Lara;citation_author=Shreni D. Mistry;citation_author=Robert A. Duckworth;citation_author=Steven P. Bradley;citation_author=Vishal N. Koparde;citation_author=X. Valentine Orenda;citation_author=Sarah H. Milton;citation_author=Sarah K. Rozycki;citation_author=Andrey V. Matveyev;citation_author=Michelle L. Wright;citation_author=Snehalata V. Huzurbazar;citation_author=Eugenie M. Jackson;citation_author=Ekaterina Smirnova;citation_author=Jonas Korlach;citation_author=Yu-Chih Tsai;citation_author=Molly R. Dickinson;citation_author=Jamie L. Brooks;citation_author=Jennifer I. Drake;citation_author=Donald O. Chaffin;citation_author=Amber L. Sexton;citation_author=Michael G. Gravett;citation_author=Craig E. Rubens;citation_author=N. Romesh Wijesooriya;citation_author=Karen D. Hendricks-MuÃ±oz;citation_author=Kimberly K. Jefferson;citation_author=Jerome F. Strauss;citation_author=Gregory A. Buck"/>
  <meta name="citation_reference" content="citation_title=Consistent and correctable bias in metagenomic sequencing experiments;citation_publication_date=2019;citation_volume=8;citation_doi=10.7554/eLife.46923;citation_author=Michael R McLaren;citation_author=Amy D Willis;citation_author=Benjamin J Callahan"/>
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","categories","date","draft","bibliography","output"]}},"value":[{"type":"character","attributes":{},"value":["MOMS-PI: Key dynamics"]},{"type":"NULL"},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url"]}},"value":[{"type":"character","attributes":{},"value":["Michael R. McLaren"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":[]}},"value":[]}]}]},{"type":"character","attributes":{},"value":["ref:brooks2015thet","ref:fettweis2019thev"]},{"type":"character","attributes":{},"value":["2021-11-01"]},{"type":"logical","attributes":{},"value":[false]},{"type":"character","attributes":{},"value":["../../../main.bib"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["self_contained","dev","toc"]}},"value":[{"type":"logical","attributes":{},"value":[false]},{"type":"character","attributes":{},"value":["svg"]},{"type":"logical","attributes":{},"value":[true]}]}]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-resource-manifest">
  {"type":"character","attributes":{},"value":["momspi-summary_files/anchor-4.2.2/anchor.min.js","momspi-summary_files/bowser-1.9.3/bowser.min.js","momspi-summary_files/distill-2.2.21/template.v2.js","momspi-summary_files/figure-html5/momspi-mean-efficiency-dist-1.svg","momspi-summary_files/figure-html5/momspi-mean-efficiency-dist-and-lacto-1.svg","momspi-summary_files/figure-html5/momspi-mean-efficiency-fcs-1.svg","momspi-summary_files/figure-html5/unnamed-chunk-4-1.svg","momspi-summary_files/header-attrs-2.11/header-attrs.js","momspi-summary_files/jquery-3.6.0/jquery-3.6.0.js","momspi-summary_files/jquery-3.6.0/jquery-3.6.0.min.js","momspi-summary_files/jquery-3.6.0/jquery-3.6.0.min.map","momspi-summary_files/popper-2.6.0/popper.min.js","momspi-summary_files/tippy-6.2.7/tippy-bundle.umd.min.js","momspi-summary_files/tippy-6.2.7/tippy-light-border.css","momspi-summary_files/tippy-6.2.7/tippy.css","momspi-summary_files/tippy-6.2.7/tippy.umd.min.js","momspi-summary_files/webcomponents-2.0.0/webcomponents.js"]}
  </script>
  <!--radix_placeholder_navigation_in_header-->
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          return "<p>" + $('#ref-' + ref).html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="momspi-summary_files/header-attrs-2.11/header-attrs.js"></script>
  <script src="momspi-summary_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <script src="momspi-summary_files/popper-2.6.0/popper.min.js"></script>
  <link href="momspi-summary_files/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="momspi-summary_files/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="momspi-summary_files/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="momspi-summary_files/anchor-4.2.2/anchor.min.js"></script>
  <script src="momspi-summary_files/bowser-1.9.3/bowser.min.js"></script>
  <script src="momspi-summary_files/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="momspi-summary_files/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"MOMS-PI: Key dynamics","authors":[{"author":"Michael R. McLaren","authorURL":{},"affiliation":"&nbsp;","affiliationURL":"#","orcidID":""}],"publishedDate":"2021-11-01T00:00:00.000-04:00","citationText":"McLaren, 2021"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>MOMS-PI: Key dynamics</h1>
<!--radix_placeholder_categories-->
<div class="dt-tags">
<div class="dt=tag">ref:brooks2015thet</div>
<div class="dt=tag">ref:fettweis2019thev</div>
</div>
<!--/radix_placeholder_categories-->

</div>

<div class="d-byline">
  Michael R. McLaren true 
  
<br/>2021-11-01
</div>

<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#setup">Setup</a>
<ul>
<li><a href="#r-setup">R setup</a></li>
<li><a href="#estimate-bias-from-brooks2015thet">Estimate bias from <span class="citation" data-cites="brooks2015thet">Brooks et al. (<span>2015</span>)</span></a></li>
<li><a href="#load-the-momspi-data">Load the MOMSPI data</a></li>
</ul></li>
<li><a href="#calibrate-the-momspi-measurements">Calibrate the MOMSPI measurements</a>
<ul>
<li><a href="#observed-and-calibrated-proportions">Observed and calibrated proportions</a></li>
<li><a href="#sample-summary-statistics">Sample summary statistics</a></li>
</ul></li>
<li><a href="#visual-analysis-of-mean-efficiency-distribution-and-role-of-key-species">Visual analysis of mean efficiency distribution and role of key species</a></li>
<li><a href="#fold-changes-in-mean-efficiency">Fold changes in mean efficiency</a></li>
<li><a href="#todo-example-community-trajectory">(TODO) Example community trajectory</a></li>
<li><a href="#session-info">Session info</a></li>
</ul>
</nav>
</div>
<p>This document is part of a series of analyses of the impact of taxonomic bias on vaginal community dynamics in MOMSPI dataset (<span class="citation" data-cites="fettweis2019thev">Fettweis et al. (<a href="#ref-fettweis2019thev" role="doc-biblioref">2019</a>)</span>), using the bias estimated from the <span class="citation" data-cites="brooks2015thet">Brooks et al. (<a href="#ref-brooks2015thet" role="doc-biblioref">2015</a>)</span> mock community data. The purpose of this document is to demonstrate the key dynamics we want to highlight in the manuscript.</p>
<h1 id="setup">Setup</h1>
<h2 id="r-setup">R setup</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># Tools for microbiome data</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://mikemc.github.io/speedyseq'>speedyseq</a></span><span class='op'>)</span>
<span class='co'># Tools for general purpose data manipulation and plotting</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://tidyverse.tidyverse.org'>tidyverse</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='http://fs.r-lib.org'>fs</a></span><span class='op'>)</span>
<span class='co'># ggplot helpers</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/eclarke/ggbeeswarm'>ggbeeswarm</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://wilkelab.org/cowplot/'>cowplot</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://patchwork.data-imaginist.com'>patchwork</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://scales.r-lib.org'>scales</a></span><span class='op'>)</span>
<span class='co'># stats helpers</span>
<span class='co'># library(broom)</span>

<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://mikemc.github.io/metacal'>metacal</a></span><span class='op'>)</span>; <span class='fu'><a href='https://rdrr.io/r/utils/packageDescription.html'>packageVersion</a></span><span class='op'>(</span><span class='st'>"metacal"</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code>[1] &#39;0.2.0.9008&#39;</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>theme_set</span><span class='op'>(</span><span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>colors_brooks</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
  <span class='st'>"Atopobium_vaginae"</span> <span class='op'>=</span> <span class='st'>"#009E73"</span>,
  <span class='st'>"Gardnerella_vaginalis"</span> <span class='op'>=</span> <span class='st'>"#56B4E9"</span>,
  <span class='st'>"Lactobacillus_crispatus"</span> <span class='op'>=</span> <span class='st'>"#D55E00"</span>,
  <span class='st'>"Lactobacillus_iners"</span> <span class='op'>=</span> <span class='st'>"#505050"</span>,
  <span class='st'>"Prevotella_bivia"</span> <span class='op'>=</span> <span class='st'>"#0072B2"</span>,
  <span class='st'>"Sneathia_amnii"</span> <span class='op'>=</span> <span class='st'>"#CC79A7"</span>,
  <span class='st'>"Streptococcus_agalactiae"</span> <span class='op'>=</span> <span class='st'>"#E69F00"</span><span class='op'>)</span>

<span class='va'>scale_y_custom</span> <span class='op'>&lt;-</span> <span class='fu'>scale_y_continuous</span><span class='op'>(</span>
    trans <span class='op'>=</span> <span class='st'>'log10'</span>,
    breaks <span class='op'>=</span> <span class='fu'><a href='https://scales.r-lib.org/reference/trans_breaks.html'>trans_breaks</a></span><span class='op'>(</span><span class='st'>'log10'</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fl'>10</span><span class='op'>^</span><span class='va'>x</span><span class='op'>)</span>,
    labels <span class='op'>=</span> <span class='fu'><a href='https://scales.r-lib.org/reference/trans_format.html'>trans_format</a></span><span class='op'>(</span><span class='st'>'log10'</span>, <span class='fu'><a href='https://scales.r-lib.org/reference/label_parse.html'>math_format</a></span><span class='op'>(</span><span class='fl'>10</span><span class='op'>^</span><span class='va'>.x</span><span class='op'>)</span><span class='op'>)</span>
  <span class='op'>)</span>
</code></pre>
</div>
</div>
<h2 id="estimate-bias-from-brooks2015thet">Estimate bias from <span class="citation" data-cites="brooks2015thet">Brooks et al. (<a href="#ref-brooks2015thet" role="doc-biblioref">2015</a>)</span></h2>
<p>We can estimate the efficiencies for the control species used in the cellular mocks of <span class="citation" data-cites="brooks2015thet">Brooks et al. (<a href="#ref-brooks2015thet" role="doc-biblioref">2015</a>)</span> by running the example code in the docs for <code>metacal::estimate_bias()</code>. This code uses the observed and actual species-level abundance tables which were produced by <span class="citation" data-cites="mclaren2019cons">McLaren, Willis, and Callahan (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span> from the SI files of <span class="citation" data-cites="brooks2015thet">Brooks et al. (<a href="#ref-brooks2015thet" role="doc-biblioref">2015</a>)</span> and are included in the metacal package.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>dr</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, package <span class='op'>=</span> <span class='st'>"metacal"</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/list.files.html'>list.files</a></span><span class='op'>(</span><span class='va'>dr</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code>[1] &quot;brooks2015-actual.csv&quot;      &quot;brooks2015-observed.csv&quot;   
[3] &quot;brooks2015-sample-data.csv&quot;</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>actual</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>dr</span>, <span class='st'>"brooks2015-actual.csv"</span><span class='op'>)</span> |&gt;
  <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.csv</a></span><span class='op'>(</span>row.names <span class='op'>=</span> <span class='st'>"Sample"</span><span class='op'>)</span> |&gt;
  <span class='fu'>as</span><span class='op'>(</span><span class='st'>"matrix"</span><span class='op'>)</span>
<span class='va'>observed</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>dr</span>, <span class='st'>"brooks2015-observed.csv"</span><span class='op'>)</span> |&gt;
  <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.csv</a></span><span class='op'>(</span>row.names <span class='op'>=</span> <span class='st'>"Sample"</span><span class='op'>)</span> |&gt;
  <span class='fu'><a href='https://rdrr.io/r/base/subset.html'>subset</a></span><span class='op'>(</span>select <span class='op'>=</span> <span class='op'>-</span> <span class='va'>Other</span><span class='op'>)</span> |&gt;
  <span class='fu'>as</span><span class='op'>(</span><span class='st'>"matrix"</span><span class='op'>)</span>

<span class='co'># Estimate bias with bootstrapping for error estimation</span>
<span class='va'>mc_fit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://mikemc.github.io/metacal/reference/estimate_bias.html'>estimate_bias</a></span><span class='op'>(</span><span class='va'>observed</span>, <span class='va'>actual</span>, margin <span class='op'>=</span> <span class='fl'>1</span>, boot <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>mc_fit</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code>Summary of a metacal bias fit.

Estimated relative efficiencies:
# A tibble: 7 Ã 4
  taxon                    estimate gm_mean gm_se
  &lt;chr&gt;                       &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;
1 Atopobium_vaginae           0.285   0.285  1.04
2 Gardnerella_vaginalis       0.160   0.160  1.05
3 Lactobacillus_crispatus     2.29    2.28   1.03
4 Lactobacillus_iners         4.68    4.69   1.02
5 Prevotella_bivia            1.79    1.79   1.04
6 Sneathia_amnii              4.59    4.60   1.04
7 Streptococcus_agalactiae    0.250   0.250  1.03

Geometric standard error estimated from 1000 bootstrap replicates.</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/base/rm.html'>rm</a></span><span class='op'>(</span><span class='va'>actual</span>, <span class='va'>observed</span>, <span class='va'>dr</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>control_species</span> <span class='op'>&lt;-</span> <span class='va'>mc_fit</span> <span class='op'>%&gt;%</span> <span class='va'>coef</span> <span class='op'>%&gt;%</span> <span class='va'>names</span>
<span class='va'>control_genera</span> <span class='op'>&lt;-</span> <span class='va'>control_species</span> <span class='op'>%&gt;%</span> <span class='fu'>str_extract</span><span class='op'>(</span><span class='st'>'^[^_]+'</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<h2 id="load-the-momspi-data">Load the MOMSPI data</h2>
<p>TODO: replace this code chunk with one that pulls data from the github repo into <code>_data/</code></p>
<p>Load the MOMSPI Stirrups profiles into phyloseq</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>path_momspi</span> <span class='op'>&lt;-</span> <span class='st'>'~/research/momspi'</span>

<span class='va'>otu</span> <span class='op'>&lt;-</span> <span class='fu'><a href='http://fs.r-lib.org/reference/path.html'>path</a></span><span class='op'>(</span><span class='va'>path_momspi</span>, <span class='st'>"output"</span>, <span class='st'>"stirrups-profiles"</span>, <span class='st'>"abundance-matrix.csv.bz2"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>read_csv</span><span class='op'>(</span>
    col_types <span class='op'>=</span> <span class='fu'>cols</span><span class='op'>(</span>.default <span class='op'>=</span> <span class='fu'>col_double</span><span class='op'>(</span><span class='op'>)</span>, sample_name <span class='op'>=</span> <span class='fu'>col_character</span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>
  <span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>otu_table</span><span class='op'>(</span>taxa_are_rows <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='va'>sam</span> <span class='op'>&lt;-</span> <span class='fu'><a href='http://fs.r-lib.org/reference/path.html'>path</a></span><span class='op'>(</span><span class='va'>path_momspi</span>, <span class='st'>"output"</span>, <span class='st'>"stirrups-profiles"</span>, <span class='st'>"sample-data.csv.bz2"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>read_csv</span><span class='op'>(</span>col_types <span class='op'>=</span> <span class='st'>"ccccccic"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span><span class='fu'>across</span><span class='op'>(</span><span class='va'>host_visit_number</span>, <span class='va'>factor</span>, ordered <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='va'>sample_data</span>
<span class='va'>tax</span> <span class='op'>&lt;-</span> <span class='fu'><a href='http://fs.r-lib.org/reference/path.html'>path</a></span><span class='op'>(</span><span class='va'>path_momspi</span>, <span class='st'>"output"</span>, <span class='st'>"stirrups-profiles"</span>, <span class='st'>"taxonomy.csv.bz2"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>read_csv</span><span class='op'>(</span>col_types <span class='op'>=</span> <span class='fu'>cols</span><span class='op'>(</span>.default <span class='op'>=</span> <span class='fu'>col_character</span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='va'>tax_table</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/mutate-phyloseq.html'>mutate_tax_table</a></span><span class='op'>(</span>
    species <span class='op'>=</span> <span class='fu'>case_when</span><span class='op'>(</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/NA.html'>is.na</a></span><span class='op'>(</span><span class='va'>genus</span><span class='op'>)</span> <span class='op'>~</span> <span class='va'>.otu</span><span class='op'>)</span>
  <span class='op'>)</span>
<span class='va'>momspi_raw</span> <span class='op'>&lt;-</span> <span class='fu'>phyloseq</span><span class='op'>(</span><span class='va'>otu</span>, <span class='va'>sam</span>, <span class='va'>tax</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/mutate-phyloseq.html'>mutate_tax_table</a></span><span class='op'>(</span><span class='fu'>across</span><span class='op'>(</span><span class='va'>.otu</span>, <span class='va'>str_replace</span>, 
      <span class='st'>"(?&lt;=Lactobacillus_crispatus)_cluster"</span>, <span class='st'>""</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/mutate-phyloseq.html'>mutate_sample_data</a></span><span class='op'>(</span><span class='va'>.</span>, sample_sum <span class='op'>=</span> <span class='fu'>sample_sums</span><span class='op'>(</span><span class='va'>.</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'>taxa_names</span><span class='op'>(</span><span class='va'>momspi_raw</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='fu'>str_subset</span><span class='op'>(</span><span class='st'>"crispatus"</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code>[1] &quot;Lactobacillus_crispatus&quot;       &quot;Lactobacillus_crispatus_type1&quot;</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/all.html'>all</a></span><span class='op'>(</span><span class='va'>control_species</span> <span class='op'>%in%</span> <span class='fu'>taxa_names</span><span class='op'>(</span><span class='va'>momspi_raw</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>L crisp type 1 is just a few reads so we can ignore it (check)</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='va'>momspi_raw</span> <span class='op'>%&gt;%</span> <span class='va'>sample_sums</span> <span class='op'>%&gt;%</span> <span class='va'>qplot</span> <span class='op'>+</span> <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>labs</span><span class='op'>(</span>x <span class='op'>=</span> <span class='st'>'sample sum'</span><span class='op'>)</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='va'>momspi_raw</span> <span class='op'>%&gt;%</span> <span class='va'>taxa_sums</span> <span class='op'>%&gt;%</span> <span class='va'>qplot</span> <span class='op'>+</span> <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>labs</span><span class='op'>(</span>x <span class='op'>=</span> <span class='st'>'otu sum'</span><span class='op'>)</span>
<span class='va'>p1</span> <span class='op'>+</span> <span class='va'>p2</span>
</code></pre>
</div>
<p><img src="momspi-summary_files/figure-html5/unnamed-chunk-4-1.svg" width="1950" /></p>
</div>
<p>Letâs filter samples with fewer than 1000 reads (also requirement used by the authors)</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>momspi</span> <span class='op'>&lt;-</span> <span class='va'>momspi_raw</span> <span class='op'>%&gt;%</span> 
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/filter-phyloseq.html'>filter_sample_data</a></span><span class='op'>(</span><span class='va'>sample_sum</span> <span class='op'>&gt;=</span> <span class='fl'>1e3</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>Note, we might want to be conservative in terms of composition shifts by making the min prop of species 1e-3. However this might add up to a large effect if we apply it to all species. Could instead filter to 1e4 reads and threshold at 1e-4, or 3e-4</p>
<p>Letâs see if we can filter the super-low-count species without affecting any remaining samples too much:</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='op'>(</span><span class='va'>momspi</span> <span class='op'>%&gt;%</span> <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/filter_taxa.html'>filter_taxa2</a></span><span class='op'>(</span><span class='op'>~</span><span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>.</span><span class='op'>)</span> <span class='op'>&gt;=</span> <span class='fl'>2e2</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='va'>sample_sums</span><span class='op'>)</span> <span class='op'>/</span> <span class='op'>(</span><span class='va'>momspi</span> <span class='op'>%&gt;%</span> <span class='va'>sample_sums</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   0.99    1.00    1.00    1.00    1.00    1.00 </code></pre>
</div>
<p>Dropping OTUs with less than 200 reads has a negligible impact on the samples, so letâs do that.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>momspi</span> <span class='op'>&lt;-</span> <span class='va'>momspi</span> <span class='op'>%&gt;%</span> <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/filter_taxa.html'>filter_taxa2</a></span><span class='op'>(</span><span class='op'>~</span><span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>.</span><span class='op'>)</span> <span class='op'>&gt;=</span> <span class='fl'>2e2</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<h1 id="calibrate-the-momspi-measurements">Calibrate the MOMSPI measurements</h1>
<p>Owing to the large sample size of the mock community experiment, the standard errors in the estimated bias for the control species are quite small, and our uncertainty in the true bias of the MOMSPI study is much larger than what these standard errors imply. Therefore I will use the point estimate of the bias for the control species as if it were known exactly, but we should consider this to just be a hypothesis of what the bias truly is in the MOMSPI study. This hypothetical nature applies even more strongly when we extrapolate to other species.</p>
<p>Since donât have a way to directly estimate the efficiencies of the other OTUs, weâll impute them as follows.</p>
<ol type="1">
<li>Set the efficiencies of the 7 control species to the directly estimated values</li>
<li>Compute efficiencies for the 6 control genera to the efficiency of the control species (if only one) or the geometric mean efficiency (if more than one; here this is just <em>Lactobacillus</em>)</li>
<li>Use these genera-level efficiencies for the efficiencies of OTUs that are not control species but are in the same genus as one</li>
<li>For other OTUs, use the geometric mean of the genera efficiencies; this is similar to using the mean of the 7 control species but gives <em>Lactobacillus</em> equal weight to the other genera.</li>
</ol>
<!--  -->
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>bias_species</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/coef.html'>coef</a></span><span class='op'>(</span><span class='va'>mc_fit</span><span class='op'>)</span> <span class='op'>%&gt;%</span> 
  <span class='fu'>enframe</span><span class='op'>(</span><span class='st'>"species"</span>, <span class='st'>"efficiency"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='va'>print</span>
</code></pre>
</div>
<pre><code># A tibble: 7 Ã 2
  species                  efficiency
  &lt;chr&gt;                         &lt;dbl&gt;
1 Atopobium_vaginae             0.285
2 Gardnerella_vaginalis         0.160
3 Lactobacillus_crispatus       2.29 
4 Lactobacillus_iners           4.68 
5 Prevotella_bivia              1.79 
6 Sneathia_amnii                4.59 
7 Streptococcus_agalactiae      0.250</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>bias_genus</span> <span class='op'>&lt;-</span> <span class='va'>bias_species</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span>genus <span class='op'>=</span> <span class='fu'>str_extract</span><span class='op'>(</span><span class='va'>species</span>, <span class='st'>"^[^_]+"</span><span class='op'>)</span>, .before <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>with_groups</span><span class='op'>(</span><span class='va'>genus</span>, <span class='va'>summarize</span>, <span class='fu'>across</span><span class='op'>(</span><span class='va'>efficiency</span>, <span class='va'>gm_mean</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='va'>print</span>
</code></pre>
</div>
<pre><code># A tibble: 6 Ã 2
  genus         efficiency
  &lt;chr&gt;              &lt;dbl&gt;
1 Atopobium          0.285
2 Gardnerella        0.160
3 Lactobacillus      3.27 
4 Prevotella         1.79 
5 Sneathia           4.59 
6 Streptococcus      0.250</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># Match on genus or species, depending on which is available; then set others</span>
<span class='co'># to average genus efficiency</span>
<span class='va'>bias_all</span> <span class='op'>&lt;-</span> <span class='fu'>tax_table</span><span class='op'>(</span><span class='va'>momspi</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='va'>as_tibble</span> <span class='op'>%&gt;%</span>
  <span class='fu'>left_join</span><span class='op'>(</span><span class='va'>bias_species</span>, by <span class='op'>=</span> <span class='st'>"species"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>left_join</span><span class='op'>(</span><span class='va'>bias_genus</span>, by <span class='op'>=</span> <span class='st'>"genus"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span>
    efficiency <span class='op'>=</span> <span class='fu'>case_when</span><span class='op'>(</span>
      <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/NA.html'>is.na</a></span><span class='op'>(</span><span class='va'>efficiency.x</span><span class='op'>)</span> <span class='op'>~</span> <span class='va'>efficiency.x</span>,
      <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/NA.html'>is.na</a></span><span class='op'>(</span><span class='va'>efficiency.y</span><span class='op'>)</span> <span class='op'>~</span> <span class='va'>efficiency.y</span>,
      <span class='cn'>TRUE</span> <span class='op'>~</span> <span class='fu'><a href='https://mikemc.github.io/metacal/reference/gm_mean.html'>gm_mean</a></span><span class='op'>(</span><span class='va'>bias_genus</span><span class='op'>$</span><span class='va'>efficiency</span><span class='op'>)</span>
    <span class='op'>)</span>
  <span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>select</span><span class='op'>(</span><span class='op'>-</span><span class='va'>efficiency.x</span>, <span class='op'>-</span><span class='va'>efficiency.y</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='co'># standardize to L. iners, the most efficiently measured</span>
  <span class='fu'>mutate</span><span class='op'>(</span>
    <span class='fu'>across</span><span class='op'>(</span><span class='va'>efficiency</span>, <span class='op'>~</span> <span class='va'>.</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='va'>.</span><span class='op'>)</span><span class='op'>)</span>
  <span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='va'>glimpse</span>
</code></pre>
</div>
<pre><code>Rows: 282
Columns: 10
$ .otu       &lt;chr&gt; &quot;Clostridiaceae_1_OTU17&quot;, &quot;Lachnospiraceae_BVAB1&quot;â¦
$ domain     &lt;chr&gt; NA, NA, NA, NA, NA, NA, &quot;Bacteria&quot;, &quot;Bacteria&quot;, &quot;â¦
$ phylum     &lt;chr&gt; NA, NA, NA, NA, NA, NA, &quot;Firmicutes&quot;, &quot;Firmicutesâ¦
$ class      &lt;chr&gt; NA, NA, NA, NA, NA, NA, &quot;Bacilli&quot;, &quot;Clostridia&quot;, â¦
$ order      &lt;chr&gt; NA, NA, NA, NA, NA, NA, &quot;Lactobacillales&quot;, &quot;Clostâ¦
$ family     &lt;chr&gt; NA, NA, NA, NA, NA, NA, &quot;Aerococcaceae&quot;, &quot;Veillonâ¦
$ genus      &lt;chr&gt; NA, NA, NA, NA, NA, NA, &quot;Abiotrophia&quot;, &quot;Acidaminoâ¦
$ control    &lt;chr&gt; &quot;FALSE&quot;, &quot;FALSE&quot;, &quot;FALSE&quot;, &quot;FALSE&quot;, &quot;FALSE&quot;, &quot;FALâ¦
$ species    &lt;chr&gt; NA, NA, NA, NA, NA, NA, &quot;Abiotrophia_defectiva&quot;, â¦
$ efficiency &lt;dbl&gt; 0.175212, 0.175212, 0.175212, 0.175212, 0.175212,â¦</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>bias_all_vec</span> <span class='op'>&lt;-</span> <span class='va'>bias_all</span> <span class='op'>%&gt;%</span> <span class='fu'>select</span><span class='op'>(</span><span class='va'>.otu</span>, <span class='va'>efficiency</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='va'>deframe</span> 
</code></pre>
</div>
</div>
<h2 id="observed-and-calibrated-proportions">Observed and calibrated proportions</h2>
<p>Compute observed and calibrated proportions with all taxa.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>momspi_obs</span> <span class='op'>&lt;-</span> <span class='va'>momspi</span> <span class='op'>%&gt;%</span> 
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/transform_sample_counts.html'>transform_sample_counts</a></span><span class='op'>(</span><span class='va'>close_elts</span><span class='op'>)</span>
<span class='co'># note, calibrate() automatically subsets to just the focal taxa if we haven't</span>
<span class='co'># already</span>
<span class='va'>momspi_cal</span> <span class='op'>&lt;-</span> <span class='va'>momspi_obs</span> <span class='op'>%&gt;%</span> <span class='fu'><a href='https://mikemc.github.io/metacal/reference/calibrate.html'>calibrate</a></span><span class='op'>(</span><span class='va'>bias_all_vec</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>Note, Iâm not using pseudo-props so the zeros are staying zero. TODO: Reconsider this choice.</p>
<h2 id="sample-summary-statistics">Sample summary statistics</h2>
<ul>
<li>Mean efficiencies</li>
<li>Most abundant species</li>
<li>Proportion of Lactobacillus <!--  --></li>
</ul>
<p>To compute the mean efficiencies, we can use <code>metacal::perturb()</code> to multiply efficiencies by the actual proportions for each sample (using <code>norm = "none"</code> to avoid renormalizing the results), then summing up the values from each sample with <code>phyloseq::sample_sums()</code>. Itâs also useful to get the mean efficiency among just the control species.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>tmp1</span> <span class='op'>&lt;-</span> <span class='va'>momspi_cal</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://mikemc.github.io/metacal/reference/perturb.html'>perturb</a></span><span class='op'>(</span><span class='va'>bias_all_vec</span>, norm <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>%&gt;%</span> 
  <span class='va'>sample_sums</span> <span class='op'>%&gt;%</span>
  <span class='fu'>enframe</span><span class='op'>(</span><span class='st'>'.sample'</span>, <span class='st'>'mean_efficiency'</span><span class='op'>)</span>
<span class='va'>tmp2</span> <span class='op'>&lt;-</span> <span class='va'>momspi_cal</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/filter-phyloseq.html'>filter_tax_table</a></span><span class='op'>(</span><span class='va'>.otu</span> <span class='op'>%in%</span> <span class='va'>control_species</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/transform_sample_counts.html'>transform_sample_counts</a></span><span class='op'>(</span><span class='va'>close_elts</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://mikemc.github.io/metacal/reference/perturb.html'>perturb</a></span><span class='op'>(</span><span class='va'>bias_species</span> <span class='op'>%&gt;%</span> <span class='va'>deframe</span>, norm <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>%&gt;%</span> 
  <span class='va'>sample_sums</span> <span class='op'>%&gt;%</span>
  <span class='fu'>enframe</span><span class='op'>(</span><span class='st'>'.sample'</span>, <span class='st'>'mean_efficiency_controls'</span><span class='op'>)</span>
<span class='va'>mean_eff</span> <span class='op'>&lt;-</span> <span class='fu'>left_join</span><span class='op'>(</span><span class='va'>tmp1</span>, <span class='va'>tmp2</span>, by <span class='op'>=</span> <span class='st'>'.sample'</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>left_join</span><span class='op'>(</span><span class='fu'>sample_data</span><span class='op'>(</span><span class='va'>momspi_cal</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='va'>as_tibble</span>, by <span class='op'>=</span> <span class='st'>'.sample'</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>For each sample, letâs get the most abundant species in the calibrated and uncalibrated profiles.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>get_top_species</span> <span class='op'>&lt;-</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>ps</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>x</span> <span class='op'>&lt;-</span> <span class='va'>ps</span> <span class='op'>%&gt;%</span>
    <span class='va'>otu_table</span> <span class='op'>%&gt;%</span>
    <span class='fu'>as_tibble</span><span class='op'>(</span>pivot <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
    <span class='fu'>with_groups</span><span class='op'>(</span><span class='va'>.sample</span>, <span class='va'>slice_max</span>, <span class='va'>.abundance</span>, n <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
    <span class='co'># add count to check for ties</span>
    <span class='fu'>add_count</span><span class='op'>(</span><span class='va'>.sample</span><span class='op'>)</span>
  <span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>anyDuplicated</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>$</span><span class='va'>.sample</span><span class='op'>)</span><span class='op'>)</span>
    <span class='kw'><a href='https://rdrr.io/r/base/stop.html'>stop</a></span><span class='op'>(</span><span class='st'>'Ties present'</span><span class='op'>)</span>
  <span class='va'>x</span> <span class='op'>%&gt;%</span> <span class='fu'>select</span><span class='op'>(</span><span class='va'>.sample</span>, <span class='va'>.otu</span>, <span class='va'>.abundance</span><span class='op'>)</span>
<span class='op'>}</span>
<span class='va'>top_species</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>
  Uncalibrated <span class='op'>=</span> <span class='va'>momspi_obs</span>,
  Calibrated <span class='op'>=</span> <span class='va'>momspi_cal</span>
  <span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>map_dfr</span><span class='op'>(</span><span class='va'>get_top_species</span>, .id <span class='op'>=</span> <span class='st'>'type'</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>Which species tend to be dominant?</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>top_species</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>type</span> <span class='op'>==</span> <span class='st'>'Uncalibrated'</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>count</span><span class='op'>(</span><span class='va'>.otu</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>arrange</span><span class='op'>(</span><span class='fu'>desc</span><span class='op'>(</span><span class='va'>n</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span>n<span class='op'>=</span><span class='fl'>20</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code># A tibble: 54 Ã 2
   .otu                                     n
   &lt;chr&gt;                                &lt;int&gt;
 1 Lactobacillus_iners                    777
 2 Lactobacillus_crispatus                424
 3 Lachnospiraceae_BVAB1                  238
 4 Gardnerella_vaginalis                  138
 5 Lactobacillus_gasseri_cluster           68
 6 Prevotella_cluster2                     53
 7 Atopobium_vaginae                       52
 8 Lactobacillus_jensenii                  50
 9 Sneathia_amnii                          23
10 Prevotella_bivia                        16
11 Streptococcus_cluster29                 16
12 Mycoplasma_girerdii                     11
13 Mycoplasma_hominis                       6
14 Corynebacterium_cluster58                5
15 Lactobacillus_delbrueckii                5
16 Bifidobacterium_longum_infantis_suis     4
17 Finegoldia_magna                         4
18 Prevotella_disiens                       4
19 Staphylococcus_cluster47                 4
20 Streptococcus_agalactiae                 4
# â¦ with 34 more rows</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>top_species</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>type</span> <span class='op'>==</span> <span class='st'>'Calibrated'</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>count</span><span class='op'>(</span><span class='va'>.otu</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>arrange</span><span class='op'>(</span><span class='fu'>desc</span><span class='op'>(</span><span class='va'>n</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span>n<span class='op'>=</span><span class='fl'>20</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code># A tibble: 60 Ã 2
   .otu                              n
   &lt;chr&gt;                         &lt;int&gt;
 1 Gardnerella_vaginalis           521
 2 Lactobacillus_iners             490
 3 Lactobacillus_crispatus         386
 4 Lachnospiraceae_BVAB1           185
 5 Atopobium_vaginae                63
 6 Lactobacillus_jensenii           52
 7 Lactobacillus_gasseri_cluster    46
 8 Streptococcus_cluster29          28
 9 Streptococcus_anginosus          20
10 Prevotella_cluster2              17
11 Mycoplasma_girerdii              15
12 Streptococcus_agalactiae         11
13 Finegoldia_magna                  9
14 Ureaplasma_cluster23              9
15 Corynebacterium_cluster58         7
16 Mycoplasma_hominis                7
17 Clostridiales_OTU22               5
18 Lactobacillus_delbrueckii         5
19 Prevotella_bivia                  5
20 Anaerococcus_OTU147               4
# â¦ with 40 more rows</code></pre>
</div>
<p>The most important species/OTU that we do not have direct bias information on at the species or genus level is Lachnospiraceae BVAB1.</p>
<p>The proportion of Lactobacillus in each sample - will simply do all control genera.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>genus_prop</span> <span class='op'>&lt;-</span> <span class='va'>momspi_cal</span> <span class='op'>%&gt;%</span>
  <span class='fu'>subset_taxa</span><span class='op'>(</span><span class='va'>genus</span> <span class='op'>%in%</span> <span class='va'>control_genera</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/tax_glom.html'>tax_glom</a></span><span class='op'>(</span><span class='st'>'genus'</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='va'>as_tibble</span> <span class='op'>%&gt;%</span>
  <span class='fu'>select</span><span class='op'>(</span><span class='va'>.sample</span>, <span class='va'>genus</span>, <span class='va'>.abundance</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>pivot_wider</span><span class='op'>(</span>names_from <span class='op'>=</span> <span class='va'>genus</span>, values_from <span class='op'>=</span> <span class='va'>.abundance</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>Letâs join the most abundant species and key genera proportions to the mean efficiecies for further analysis. For plotting, weâll create a version of the top taxa that only colors the taxa that are most often dominant.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>tmp</span> <span class='op'>&lt;-</span> <span class='va'>top_species</span> <span class='op'>%&gt;%</span> 
  <span class='fu'>select</span><span class='op'>(</span><span class='op'>-</span><span class='va'>.abundance</span><span class='op'>)</span> <span class='op'>%&gt;%</span> 
  <span class='fu'>pivot_wider</span><span class='op'>(</span>names_from <span class='op'>=</span> <span class='va'>type</span>, values_from <span class='op'>=</span> <span class='va'>.otu</span>, 
    names_glue <span class='op'>=</span> <span class='st'>'top_{str_to_lower(type)}'</span><span class='op'>)</span>
<span class='va'>mean_eff1</span> <span class='op'>&lt;-</span> <span class='va'>mean_eff</span> <span class='op'>%&gt;%</span>
  <span class='fu'>left_join</span><span class='op'>(</span><span class='va'>genus_prop</span>, by <span class='op'>=</span> <span class='st'>'.sample'</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>left_join</span><span class='op'>(</span><span class='va'>tmp</span>, by <span class='op'>=</span> <span class='st'>'.sample'</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span>
    top_plot <span class='op'>=</span> <span class='fu'>fct_lump_n</span><span class='op'>(</span><span class='va'>top_calibrated</span>, <span class='fl'>7</span><span class='op'>)</span>
  <span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/rm.html'>rm</a></span><span class='op'>(</span><span class='va'>tmp</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<h1 id="visual-analysis-of-mean-efficiency-distribution-and-role-of-key-species">Visual analysis of mean efficiency distribution and role of key species</h1>
<p>We can make a plot similar to that used for the Leopold et al analysis.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># First let's look at the distribution of mean efficiencies</span>
<span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='va'>mean_eff1</span> <span class='op'>%&gt;%</span>
  <span class='fu'>ggplot</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span><span class='va'>mean_efficiency</span>, fill <span class='op'>=</span> <span class='va'>top_plot</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_histogram</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='co'># annotate("point", y = 200, x = bias, color = "black", shape = "|", size = 7) +</span>
  <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>expand_limits</span><span class='op'>(</span>x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>bias_all_vec</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_fill_brewer</span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"qual"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>labs</span><span class='op'>(</span>
    <span class='co'># x = "Sample mean relative efficiency (SMRE)",</span>
    x <span class='op'>=</span> <span class='st'>"Mean efficiency"</span>, 
    y <span class='op'>=</span> <span class='st'>"Number of samples"</span>,
    fill <span class='op'>=</span> <span class='st'>'Most abundant species'</span>
  <span class='op'>)</span>
<span class='co'># Next let's plot Lactobacillus proportion against mean efficiency, as another</span>
  <span class='co'># look at the relationship between Lacto and mean efficiency.</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='va'>mean_eff1</span> <span class='op'>%&gt;%</span>
  <span class='fu'>ggplot</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span><span class='va'>mean_efficiency</span>, <span class='va'>Lactobacillus</span>, color <span class='op'>=</span> <span class='va'>top_plot</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_point</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>expand_limits</span><span class='op'>(</span>x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>bias_all_vec</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_y_continuous</span><span class='op'>(</span>
    trans <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/log_trans.html'>pseudo_log_trans</a></span><span class='op'>(</span>sigma <span class='op'>=</span> <span class='fl'>1e-2</span><span class='op'>)</span>,
    breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>0.01</span>, <span class='fl'>0.03</span>, <span class='fl'>0.1</span>, <span class='fl'>0.3</span>, <span class='fl'>1</span><span class='op'>)</span>
    <span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_color_brewer</span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"qual"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>labs</span><span class='op'>(</span>
    y <span class='op'>=</span> <span class='st'>"Lactobacillus proportion"</span>,
    x <span class='op'>=</span> <span class='st'>"Mean efficiency"</span>,
    color <span class='op'>=</span> <span class='st'>'Most abundant species'</span>
  <span class='op'>)</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># The yellow makes the L crispatus points a bit hard to see, so let's swap it's</span>
<span class='co'># color with the rarer L. gasseri in the final figure.</span>
<span class='va'>colors_top</span> <span class='op'>&lt;-</span> <span class='fu'>RColorBrewer</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html'>brewer.pal</a></span><span class='op'>(</span><span class='fl'>8</span>, <span class='st'>'Accent'</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>2</span>,<span class='fl'>3</span>,<span class='fl'>5</span>,<span class='fl'>4</span>,<span class='fl'>6</span>,<span class='fl'>7</span>,<span class='fl'>8</span><span class='op'>)</span><span class='op'>]</span> 
<span class='op'>(</span><span class='va'>p1</span> <span class='op'>+</span> <span class='fu'>ggtitle</span><span class='op'>(</span><span class='st'>"Distribution of sample mean efficiencies"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>/</span> 
  <span class='op'>(</span><span class='va'>p2</span> <span class='op'>+</span> 
    <span class='fu'>ggtitle</span><span class='op'>(</span><span class='st'>"Lactobacillus proportion vs. mean efficiency"</span><span class='op'>)</span> <span class='op'>+</span> 
    <span class='fu'>theme</span><span class='op'>(</span>legend.position <span class='op'>=</span> <span class='st'>'none'</span><span class='op'>)</span>
  <span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_annotation.html'>plot_annotation</a></span><span class='op'>(</span>tag_levels <span class='op'>=</span> <span class='st'>"A"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>guides <span class='op'>=</span> <span class='st'>'collect'</span><span class='op'>)</span> <span class='op'>&amp;</span>
  <span class='fu'>scale_fill_manual</span><span class='op'>(</span>
    values <span class='op'>=</span> <span class='va'>colors_top</span>,
    labels <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'>str_replace_all</span><span class='op'>(</span><span class='va'>x</span>, <span class='st'>'_'</span>, <span class='st'>' '</span><span class='op'>)</span>
  <span class='op'>)</span> <span class='op'>&amp;</span>
  <span class='fu'>scale_color_manual</span><span class='op'>(</span>
    values <span class='op'>=</span> <span class='va'>colors_top</span>,
    labels <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'>str_replace_all</span><span class='op'>(</span><span class='va'>x</span>, <span class='st'>'_'</span>, <span class='st'>' '</span><span class='op'>)</span>
  <span class='op'>)</span>
</code></pre>
</div>
<img src="momspi-summary_files/figure-html5/momspi-mean-efficiency-dist-and-lacto-1.svg" width="2880" />
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># The x-axis is very close but not completely aligned; I think I need to</span>
<span class='co'># manually set the x limits to get perfect alignment, but that leads to some</span>
<span class='co'># samples being dropped in the histogram plot; some investigation is needed</span>
<span class='co'># to go that route.</span>
</code></pre>
</div>
</div>
<p>This plot shows that Lactobacillus dominance drives a bimodal distribution in the mean efficiencies, and that Gardnerella dominance drives the low end. Note that the âotherâ and âBVAB1â dominated samples are lumped around the middle because of our assumption in imputing the efficiencies.</p>
<p>For the caption/methods: Note that the most abundant species is that inferred from the calibrated data; could just say that the calibrated data is treated as the actual composition.</p>
<p>The Lacto proportion doesnât seem that much more illuminating than the histogram colored by dominant species, so letâs make a version with just that for the manuscript.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>colors_top</span> <span class='op'>&lt;-</span> <span class='fu'>RColorBrewer</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html'>brewer.pal</a></span><span class='op'>(</span><span class='fl'>8</span>, <span class='st'>'Accent'</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>2</span>,<span class='fl'>3</span>,<span class='fl'>5</span>,<span class='fl'>4</span>,<span class='fl'>6</span>,<span class='fl'>7</span>,<span class='fl'>8</span><span class='op'>)</span><span class='op'>]</span> 
<span class='va'>p1</span> <span class='op'>+</span> 
<span class='co'>#  ggtitle("Distribution of sample mean efficiencies") +</span>
  <span class='fu'>scale_fill_manual</span><span class='op'>(</span>
    values <span class='op'>=</span> <span class='va'>colors_top</span>,
    labels <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'>str_replace_all</span><span class='op'>(</span><span class='va'>x</span>, <span class='st'>'_'</span>, <span class='st'>' '</span><span class='op'>)</span>
  <span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_color_manual</span><span class='op'>(</span>
    values <span class='op'>=</span> <span class='va'>colors_top</span>,
    labels <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'>str_replace_all</span><span class='op'>(</span><span class='va'>x</span>, <span class='st'>'_'</span>, <span class='st'>' '</span><span class='op'>)</span>
  <span class='op'>)</span>
</code></pre>
</div>
<p><img src="momspi-summary_files/figure-html5/momspi-mean-efficiency-dist-1.svg" width="2100" /></p>
</div>
<h1 id="fold-changes-in-mean-efficiency">Fold changes in mean efficiency</h1>
<p>How often do shifts in the mean efficiency occur within a single subject? How often do shifts from Lacto to non-Lacto dominance occur within a single subject? How often do shifts from Lacto to Gardnerella dominance occur within a single subject?</p>
<p>Letâs start by computing the pairwise changes in the sample mean efficiency between consecutive samples within women, and between the first samples of different women (first samples only to limit the number of sample pairs).</p>
<p>Since we may be missing some host visits, letâs create a new host visit number that has no gaps:</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>mean_eff2</span> <span class='op'>&lt;-</span> <span class='va'>mean_eff1</span> <span class='op'>%&gt;%</span>
  <span class='fu'>with_groups</span><span class='op'>(</span><span class='va'>host_subject_id</span>, <span class='va'>mutate</span>, 
    host_visit_rank <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/rank.html'>rank</a></span><span class='op'>(</span><span class='va'>host_visit_number</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>tb</span> <span class='op'>&lt;-</span> <span class='fu'>crossing</span><span class='op'>(</span>
  sample.1 <span class='op'>=</span> <span class='va'>mean_eff2</span><span class='op'>$</span><span class='va'>.sample</span>, 
  sample.2 <span class='op'>=</span> <span class='va'>mean_eff2</span><span class='op'>$</span><span class='va'>.sample</span>
  <span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>left_join</span><span class='op'>(</span><span class='va'>mean_eff2</span>, by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>sample.1 <span class='op'>=</span> <span class='st'>'.sample'</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>left_join</span><span class='op'>(</span><span class='va'>mean_eff2</span>, by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>sample.2 <span class='op'>=</span> <span class='st'>'.sample'</span><span class='op'>)</span>, suffix <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'.1'</span>, <span class='st'>'.2'</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span>
    comparison_type <span class='op'>=</span> <span class='fu'>case_when</span><span class='op'>(</span>
      <span class='va'>host_subject_id.1</span> <span class='op'>==</span> <span class='va'>host_subject_id.2</span> <span class='op'>&amp;</span> <span class='va'>host_visit_rank.2</span> <span class='op'>==</span> <span class='va'>host_visit_rank.1</span> <span class='op'>+</span> <span class='fl'>1</span> <span class='op'>~</span> 
        <span class='st'>'Between consecutive visits'</span>,
      <span class='va'>host_subject_id.1</span> <span class='op'>!=</span> <span class='va'>host_subject_id.2</span> <span class='op'>&amp;</span> <span class='va'>host_visit_rank.2</span> <span class='op'>==</span> <span class='fl'>1</span> <span class='op'>&amp;</span> <span class='va'>host_visit_rank.1</span> <span class='op'>==</span> <span class='fl'>1</span> <span class='op'>~</span>
        <span class='st'>"Between subjects' first visits"</span>
    <span class='op'>)</span>
  <span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/NA.html'>is.na</a></span><span class='op'>(</span><span class='va'>comparison_type</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span>
    switch_class <span class='op'>=</span> <span class='fu'>case_when</span><span class='op'>(</span>
      <span class='va'>top_plot.1</span> <span class='op'>==</span> <span class='va'>top_plot.2</span> <span class='op'>~</span> <span class='st'>'No change'</span>,
      <span class='fu'>str_detect</span><span class='op'>(</span><span class='va'>top_plot.1</span>, <span class='st'>'Gardnerella'</span><span class='op'>)</span> <span class='op'>&amp;</span> <span class='fu'>str_detect</span><span class='op'>(</span><span class='va'>top_plot.2</span>, <span class='st'>'Lactobacillus'</span><span class='op'>)</span> <span class='op'>~</span> <span class='st'>'Gard. to Lacto.'</span>,
      <span class='fu'>str_detect</span><span class='op'>(</span><span class='va'>top_plot.1</span>, <span class='st'>'Lactobacillus'</span><span class='op'>)</span> <span class='op'>&amp;</span> <span class='fu'>str_detect</span><span class='op'>(</span><span class='va'>top_plot.2</span>, <span class='st'>'Gardnerella'</span><span class='op'>)</span> <span class='op'>~</span> <span class='st'>'Lacto. to Gard.'</span>,
      <span class='fu'>str_detect</span><span class='op'>(</span><span class='va'>top_plot.1</span>, <span class='st'>'Lactobacillus'</span><span class='op'>)</span> <span class='op'>&amp;</span> <span class='fu'>str_detect</span><span class='op'>(</span><span class='va'>top_plot.2</span>, <span class='st'>'Lactobacillus'</span><span class='op'>)</span> <span class='op'>~</span> <span class='st'>'Lacto. to Lacto.'</span>,
      <span class='cn'>TRUE</span> <span class='op'>~</span> <span class='st'>'Other'</span>,
    <span class='op'>)</span>
  <span class='op'>)</span>

<span class='co'># tb_within &lt;- tb %&gt;%</span>
<span class='co'>#   filter(host_subject_id.1 == host_subject_id.2, host_visit_rank.2 == host_visit_rank.1 + 1)</span>
<span class='co'># tb_between &lt;- tb %&gt;%</span>
<span class='co'>#   filter(host_subject_id.1 != host_subject_id.2, host_visit_rank.2 == 1, host_visit_rank.1 == 1)</span>
</code></pre>
</div>
</div>
<p>Letâs look at the distribution of FCs in mean efficiency, coloring by whether there is a shift in dominant species.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>tb</span> <span class='op'>%&gt;%</span>
  <span class='fu'>ggplot</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span><span class='va'>mean_efficiency.2</span> <span class='op'>/</span> <span class='va'>mean_efficiency.1</span>, fill <span class='op'>=</span> <span class='va'>switch_class</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>facet_wrap</span><span class='op'>(</span><span class='op'>~</span><span class='va'>comparison_type</span>, ncol <span class='op'>=</span> <span class='fl'>1</span>, scales <span class='op'>=</span> <span class='st'>'free_y'</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_histogram</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='co'># expand_limits(x = range(bias_all_vec)) +</span>
  <span class='fu'>labs</span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='st'>"Fold change in mean efficiency"</span>, 
    y <span class='op'>=</span> <span class='st'>"Number of sample pairs"</span>,
    fill <span class='op'>=</span> <span class='st'>'Change in most\nabundant species'</span>
  <span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_fill_brewer</span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>'qual'</span>, palette <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="momspi-summary_files/figure-html5/momspi-mean-efficiency-fcs-1.svg" width="1800" /></p>
</div>
<p>Unsurprisingly, shifts in the mean efficiency tend to be larger between different subjects. Note, the between-subject panel is perfectly symmetric, but the consecutive-visits panel is not.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>tb</span> <span class='op'>%&gt;%</span>
  <span class='fu'>count</span><span class='op'>(</span><span class='va'>comparison_type</span>, <span class='va'>switch_class</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>with_groups</span><span class='op'>(</span><span class='va'>comparison_type</span>, <span class='va'>mutate</span>, prop <span class='op'>=</span> <span class='fu'><a href='https://mikemc.github.io/metacal/reference/close_elts.html'>close_elts</a></span><span class='op'>(</span><span class='va'>n</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span>digits <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
</code></pre>
</div>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">comparison_type</th>
<th style="text-align: left;">switch_class</th>
<th style="text-align: right;">n</th>
<th style="text-align: right;">prop</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Between consecutive visits</td>
<td style="text-align: left;">Gard. to Lacto.</td>
<td style="text-align: right;">60</td>
<td style="text-align: right;">0.044</td>
</tr>
<tr class="even">
<td style="text-align: left;">Between consecutive visits</td>
<td style="text-align: left;">Lacto. to Gard.</td>
<td style="text-align: right;">61</td>
<td style="text-align: right;">0.044</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Between consecutive visits</td>
<td style="text-align: left;">Lacto. to Lacto.</td>
<td style="text-align: right;">51</td>
<td style="text-align: right;">0.037</td>
</tr>
<tr class="even">
<td style="text-align: left;">Between consecutive visits</td>
<td style="text-align: left;">No change</td>
<td style="text-align: right;">829</td>
<td style="text-align: right;">0.601</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Between consecutive visits</td>
<td style="text-align: left;">Other</td>
<td style="text-align: right;">378</td>
<td style="text-align: right;">0.274</td>
</tr>
<tr class="even">
<td style="text-align: left;">Between subjectsâ first visits</td>
<td style="text-align: left;">Gard. to Lacto.</td>
<td style="text-align: right;">50460</td>
<td style="text-align: right;">0.152</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Between subjectsâ first visits</td>
<td style="text-align: left;">Lacto. to Gard.</td>
<td style="text-align: right;">50460</td>
<td style="text-align: right;">0.152</td>
</tr>
<tr class="even">
<td style="text-align: left;">Between subjectsâ first visits</td>
<td style="text-align: left;">Lacto. to Lacto.</td>
<td style="text-align: right;">50296</td>
<td style="text-align: right;">0.151</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Between subjectsâ first visits</td>
<td style="text-align: left;">No change</td>
<td style="text-align: right;">68072</td>
<td style="text-align: right;">0.205</td>
</tr>
<tr class="even">
<td style="text-align: left;">Between subjectsâ first visits</td>
<td style="text-align: left;">Other</td>
<td style="text-align: right;">113064</td>
<td style="text-align: right;">0.340</td>
</tr>
</tbody>
</table>
</div>
<h1 id="todo-example-community-trajectory">(TODO) Example community trajectory</h1>
<p>Pick a woman with multiple visits and a large change in mean efficiency</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>tb</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>host_subject_id.1</span> <span class='op'>==</span> <span class='va'>host_subject_id.2</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span>fc <span class='op'>=</span> <span class='va'>mean_efficiency.2</span> <span class='op'>/</span> <span class='va'>mean_efficiency.1</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>arrange</span><span class='op'>(</span><span class='fu'>desc</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>abs</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log</a></span><span class='op'>(</span><span class='va'>fc</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>select</span><span class='op'>(</span><span class='va'>fc</span>, <span class='va'>host_subject_id.1</span>, <span class='va'>top_plot.1</span>, <span class='va'>top_plot.2</span>, <span class='va'>host_visit_rank.1</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code># A tibble: 1,379 Ã 5
        fc host_subject_id.1 top_plot.1   top_plot.2  host_visit_rankâ¦
     &lt;dbl&gt; &lt;chr&gt;             &lt;fct&gt;        &lt;fct&gt;                  &lt;dbl&gt;
 1 17.1    EP229454          Gardnerellaâ¦ Lactobacilâ¦                1
 2 15.9    EP418446          Other        Lactobacilâ¦                2
 3  0.0655 EP663711          Lactobacillâ¦ Gardnerellâ¦                1
 4  0.0660 EP694022          Lactobacillâ¦ Gardnerellâ¦                1
 5 14.5    EP591995          Other        Lactobacilâ¦                2
 6 14.4    EP291588          Atopobium_vâ¦ Lactobacilâ¦                1
 7  0.0703 EP239957          Lactobacillâ¦ Other                      1
 8  0.0716 EP439098          Lactobacillâ¦ Gardnerellâ¦                3
 9 13.6    EP389265          Gardnerellaâ¦ Lactobacilâ¦                1
10 13.2    EP522897          Other        Lactobacilâ¦                1
# â¦ with 1,369 more rows</code></pre>
</div>
<h1 class="appendix" id="session-info">Session info</h1>
<details>
<summary>
Click for session info
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>sessioninfo</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/sessioninfo/man/session_info.html'>session_info</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code>â Session info ââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââ
 setting  value                       
 version  R version 4.1.1 (2021-08-10)
 os       Arch Linux                  
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       America/New_York            
 date     2021-11-10                  

â Packages ââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââââ
 package          * version    date       lib source                           
 ade4               1.7-18     2021-09-16 [1] CRAN (R 4.1.1)                   
 ape                5.5        2021-04-25 [1] CRAN (R 4.1.0)                   
 assertthat         0.2.1      2019-03-21 [1] CRAN (R 4.0.0)                   
 backports          1.2.1      2020-12-09 [1] CRAN (R 4.0.3)                   
 beeswarm           0.4.0      2021-06-01 [1] CRAN (R 4.1.0)                   
 Biobase            2.52.0     2021-05-19 [1] Bioconductor                     
 BiocGenerics       0.38.0     2021-05-19 [1] Bioconductor                     
 biomformat         1.20.0     2021-05-19 [1] Bioconductor                     
 Biostrings         2.60.1     2021-06-06 [1] Bioconductor                     
 bit                4.0.4      2020-08-04 [1] CRAN (R 4.0.2)                   
 bit64              4.0.5      2020-08-30 [1] CRAN (R 4.0.2)                   
 bitops             1.0-7      2021-04-24 [1] CRAN (R 4.1.0)                   
 broom              0.7.9      2021-07-27 [1] CRAN (R 4.1.0)                   
 bslib              0.3.1      2021-10-06 [1] CRAN (R 4.1.1)                   
 cellranger         1.1.0      2016-07-27 [1] CRAN (R 4.0.0)                   
 cli                3.1.0      2021-10-27 [1] CRAN (R 4.1.1)                   
 cluster            2.1.2      2021-04-17 [2] CRAN (R 4.1.1)                   
 codetools          0.2-18     2020-11-04 [2] CRAN (R 4.1.1)                   
 colorspace         2.0-2      2021-08-11 [1] R-Forge (R 4.1.1)                
 cowplot          * 1.1.1      2021-08-27 [1] Github (wilkelab/cowplot@555c9ae)
 crayon             1.4.2      2021-10-29 [1] CRAN (R 4.1.1)                   
 data.table         1.14.2     2021-09-27 [1] CRAN (R 4.1.1)                   
 DBI                1.1.1      2021-01-15 [1] CRAN (R 4.0.4)                   
 dbplyr             2.1.1      2021-04-06 [1] CRAN (R 4.0.5)                   
 digest             0.6.28     2021-09-23 [1] CRAN (R 4.1.1)                   
 distill            1.3        2021-10-13 [1] CRAN (R 4.1.1)                   
 downlit            0.2.1      2020-11-04 [1] CRAN (R 4.0.3)                   
 dplyr            * 1.0.7      2021-06-18 [1] CRAN (R 4.1.0)                   
 ellipsis           0.3.2      2021-04-29 [1] CRAN (R 4.1.0)                   
 evaluate           0.14       2019-05-28 [1] CRAN (R 4.0.0)                   
 fansi              0.5.0      2021-05-25 [1] CRAN (R 4.1.0)                   
 farver             2.1.0      2021-02-28 [1] CRAN (R 4.0.4)                   
 fastmap            1.1.0      2021-01-25 [1] CRAN (R 4.0.4)                   
 forcats          * 0.5.1      2021-01-27 [1] CRAN (R 4.0.4)                   
 foreach            1.5.1      2020-10-15 [1] CRAN (R 4.0.3)                   
 fs               * 1.5.0      2020-07-31 [1] CRAN (R 4.0.2)                   
 generics           0.1.1      2021-10-25 [1] CRAN (R 4.1.1)                   
 GenomeInfoDb       1.28.1     2021-07-01 [1] Bioconductor                     
 GenomeInfoDbData   1.2.6      2021-05-31 [1] Bioconductor                     
 ggbeeswarm       * 0.6.0      2017-08-07 [1] CRAN (R 4.0.0)                   
 ggplot2          * 3.3.5      2021-06-25 [1] CRAN (R 4.1.0)                   
 glue               1.4.2      2020-08-27 [1] CRAN (R 4.0.2)                   
 gtable             0.3.0      2019-03-25 [1] CRAN (R 4.0.0)                   
 haven              2.4.3      2021-08-04 [1] CRAN (R 4.1.1)                   
 here               1.0.1      2020-12-13 [1] CRAN (R 4.0.5)                   
 highr              0.9        2021-04-16 [1] CRAN (R 4.1.0)                   
 hms                1.1.1      2021-09-26 [1] CRAN (R 4.1.1)                   
 htmltools          0.5.2      2021-08-25 [1] CRAN (R 4.1.1)                   
 httr               1.4.2      2020-07-20 [1] CRAN (R 4.0.2)                   
 igraph             1.2.7      2021-10-15 [1] CRAN (R 4.1.1)                   
 IRanges            2.26.0     2021-05-19 [1] Bioconductor                     
 iterators          1.0.13     2020-10-15 [1] CRAN (R 4.0.3)                   
 jquerylib          0.1.4      2021-04-26 [1] CRAN (R 4.1.0)                   
 jsonlite           1.7.2      2020-12-09 [1] CRAN (R 4.0.3)                   
 knitr              1.36       2021-09-29 [1] CRAN (R 4.1.1)                   
 labeling           0.4.2      2020-10-20 [1] CRAN (R 4.0.3)                   
 lattice            0.20-44    2021-05-02 [2] CRAN (R 4.1.1)                   
 lifecycle          1.0.1      2021-09-24 [1] CRAN (R 4.1.1)                   
 lubridate          1.8.0      2021-10-07 [1] CRAN (R 4.1.1)                   
 magrittr           2.0.1      2020-11-17 [1] CRAN (R 4.0.3)                   
 MASS               7.3-54     2021-05-03 [2] CRAN (R 4.1.1)                   
 Matrix             1.3-4      2021-06-01 [2] CRAN (R 4.1.1)                   
 metacal          * 0.2.0.9008 2021-10-30 [1] Github (mikemc/metacal@c1f21db)  
 mgcv               1.8-36     2021-06-01 [2] CRAN (R 4.1.1)                   
 modelr             0.1.8      2020-05-19 [1] CRAN (R 4.0.0)                   
 multtest           2.48.0     2021-05-19 [1] Bioconductor                     
 munsell            0.5.0      2018-06-12 [1] CRAN (R 4.0.0)                   
 nlme               3.1-152    2021-02-04 [2] CRAN (R 4.1.1)                   
 nvimcom          * 0.9-102    2021-10-25 [1] local                            
 patchwork        * 1.1.1      2020-12-17 [1] CRAN (R 4.0.3)                   
 permute            0.9-5      2019-03-12 [1] CRAN (R 4.0.0)                   
 phyloseq         * 1.36.0     2021-05-19 [1] Bioconductor                     
 pillar             1.6.4      2021-10-18 [1] CRAN (R 4.1.1)                   
 pkgconfig          2.0.3      2019-09-22 [1] CRAN (R 4.0.0)                   
 plyr               1.8.6      2020-03-03 [1] CRAN (R 4.0.0)                   
 purrr            * 0.3.4      2020-04-17 [1] CRAN (R 4.0.0)                   
 R6                 2.5.1      2021-08-19 [1] CRAN (R 4.1.1)                   
 RColorBrewer       1.1-2      2014-12-07 [1] CRAN (R 4.0.0)                   
 Rcpp               1.0.7      2021-07-07 [1] CRAN (R 4.1.0)                   
 RCurl              1.98-1.5   2021-09-17 [1] CRAN (R 4.1.1)                   
 readr            * 2.0.2      2021-09-27 [1] CRAN (R 4.1.1)                   
 readxl             1.3.1      2019-03-13 [1] CRAN (R 4.0.0)                   
 reprex             2.0.1      2021-08-05 [1] CRAN (R 4.1.1)                   
 reshape2           1.4.4      2020-04-09 [1] CRAN (R 4.0.0)                   
 rhdf5              2.36.0     2021-05-19 [1] Bioconductor                     
 rhdf5filters       1.4.0      2021-05-19 [1] Bioconductor                     
 Rhdf5lib           1.14.2     2021-07-06 [1] Bioconductor                     
 rlang              0.4.12     2021-10-18 [1] CRAN (R 4.1.1)                   
 rmarkdown        * 2.11       2021-09-14 [1] CRAN (R 4.1.1)                   
 rprojroot          2.0.2      2020-11-15 [1] CRAN (R 4.0.3)                   
 rstudioapi         0.13       2020-11-12 [1] CRAN (R 4.0.3)                   
 rvest              1.0.2      2021-10-16 [1] CRAN (R 4.1.1)                   
 S4Vectors          0.30.0     2021-05-19 [1] Bioconductor                     
 sass               0.4.0      2021-05-12 [1] CRAN (R 4.1.0)                   
 scales           * 1.1.1      2020-05-11 [1] CRAN (R 4.0.0)                   
 sessioninfo        1.1.1      2018-11-05 [1] CRAN (R 4.0.0)                   
 speedyseq        * 0.5.3.9018 2021-06-29 [1] Github (mikemc/speedyseq@ceb941f)
 stringi            1.7.5      2021-10-04 [1] CRAN (R 4.1.1)                   
 stringr          * 1.4.0      2019-02-10 [1] CRAN (R 4.0.0)                   
 survival           3.2-11     2021-04-26 [2] CRAN (R 4.1.1)                   
 tibble           * 3.1.5      2021-09-30 [1] CRAN (R 4.1.1)                   
 tidyr            * 1.1.4      2021-09-27 [1] CRAN (R 4.1.1)                   
 tidyselect         1.1.1      2021-04-30 [1] CRAN (R 4.1.0)                   
 tidyverse        * 1.3.1      2021-04-15 [1] CRAN (R 4.1.0)                   
 tzdb               0.1.2      2021-07-20 [1] CRAN (R 4.1.0)                   
 useful             1.2.6      2018-10-08 [1] CRAN (R 4.0.0)                   
 utf8               1.2.2      2021-07-24 [1] CRAN (R 4.1.0)                   
 vctrs              0.3.8      2021-04-29 [1] CRAN (R 4.1.0)                   
 vegan              2.5-7      2020-11-28 [1] CRAN (R 4.0.3)                   
 vipor              0.4.5      2017-03-22 [1] CRAN (R 4.0.0)                   
 vroom              1.5.5      2021-09-14 [1] CRAN (R 4.1.1)                   
 withr              2.4.2      2021-04-18 [1] CRAN (R 4.0.5)                   
 xfun               0.27       2021-10-18 [1] CRAN (R 4.1.1)                   
 xml2               1.3.2      2020-04-23 [1] CRAN (R 4.0.0)                   
 XVector            0.32.0     2021-05-19 [1] Bioconductor                     
 yaml               2.2.1      2020-02-01 [1] CRAN (R 4.0.0)                   
 zlibbioc           1.38.0     2021-05-19 [1] Bioconductor                     

[1] /home/michael/.local/lib/R/library
[2] /usr/lib/R/library</code></pre>
</div>
</details>
<div class="sourceCode" id="cb13"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-brooks2015thet" class="csl-entry" role="doc-biblioentry">
Brooks, J Paul, David J Edwards, Michael D Harwich, Maria C Rivera, Jennifer M Fettweis, Myrna G Serrano, Robert A Reris, et al. 2015. <span>â<span class="nocase">The truth about metagenomics: quantifying and counteracting bias in 16S rRNA studies</span>.â</span> <em>BMC Microbiol.</em> 15 (1): 66. <a href="https://doi.org/10.1186/s12866-015-0351-6">https://doi.org/10.1186/s12866-015-0351-6</a>.
</div>
<div id="ref-fettweis2019thev" class="csl-entry" role="doc-biblioentry">
Fettweis, Jennifer M., Myrna G. Serrano, Jamie Paul Brooks, David J. Edwards, Philippe H. Girerd, Hardik I. Parikh, Bernice Huang, et al. 2019. <span>â<span class="nocase">The vaginal microbiome and preterm birth</span>.â</span> <em>Nat. Med.</em> 25 (6): 1012â21. <a href="https://doi.org/10.1038/s41591-019-0450-2">https://doi.org/10.1038/s41591-019-0450-2</a>.
</div>
<div id="ref-mclaren2019cons" class="csl-entry" role="doc-biblioentry">
McLaren, Michael R, Amy D Willis, and Benjamin J Callahan. 2019. <span>â<span class="nocase">Consistent and correctable bias in metagenomic sequencing experiments</span>.â</span> <em>Elife</em> 8 (September): 46923. <a href="https://doi.org/10.7554/eLife.46923">https://doi.org/10.7554/eLife.46923</a>.
</div>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
