<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>B Supplemental figures | Implications of taxonomic bias for microbial differential-abundance analysis</title>
  <meta name="description" content="B Supplemental figures | Implications of taxonomic bias for microbial differential-abundance analysis" />
  <meta name="generator" content="bookdown 0.27.3 and GitBook 2.6.7" />

  <meta property="og:title" content="B Supplemental figures | Implications of taxonomic bias for microbial differential-abundance analysis" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="B Supplemental figures | Implications of taxonomic bias for microbial differential-abundance analysis" />
  
  
  

<meta name="author" content="Michael R. McLaren" />
<meta name="author" content="Jacob T. Nearing" />
<meta name="author" content="Amy D. Willis" />
<meta name="author" content="Karen G. Lloyd" />
<meta name="author" content="Benjamin J. Callahan" />


<meta name="date" content="2022-08-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="appendix-regression.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
    TeX: {extensions: ["cancel.js"]},
});
MathJax.Hub.Config({
    TeX: {
        Macros : {
            md: '^{\\times}\\!/',
        }
    },
});
</script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="abundance-measurement.html"><a href="abundance-measurement.html"><i class="fa fa-check"></i><b>2</b> How bias affects abundance measurements</a>
<ul>
<li class="chapter" data-level="2.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#a-model-of-mgs-measurements"><i class="fa fa-check"></i><b>2.1</b> A model of MGS measurements</a></li>
<li class="chapter" data-level="2.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#relative-abundance"><i class="fa fa-check"></i><b>2.2</b> Relative abundance</a></li>
<li class="chapter" data-level="2.3" data-path="abundance-measurement.html"><a href="abundance-measurement.html#absolute-abundance"><i class="fa fa-check"></i><b>2.3</b> Absolute abundance</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#leveraging-information-about-total-community-abundance"><i class="fa fa-check"></i><b>2.3.1</b> Leveraging information about total-community abundance</a></li>
<li class="chapter" data-level="2.3.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#leveraging-information-about-a-reference-species"><i class="fa fa-check"></i><b>2.3.2</b> Leveraging information about a reference species</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>3</b> How bias affects DA results</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-abundance.html"><a href="differential-abundance.html#fold-change-between-a-pair-of-samples"><i class="fa fa-check"></i><b>3.1</b> Fold change between a pair of samples</a></li>
<li class="chapter" data-level="3.2" data-path="differential-abundance.html"><a href="differential-abundance.html#regression-analysis-of-many-samples"><i class="fa fa-check"></i><b>3.2</b> Regression analysis of many samples</a></li>
<li class="chapter" data-level="3.3" data-path="differential-abundance.html"><a href="differential-abundance.html#rank-based-analyses"><i class="fa fa-check"></i><b>3.3</b> Rank-based analyses</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="case-studies.html"><a href="case-studies.html"><i class="fa fa-check"></i><b>4</b> Case studies Â </a>
<ul>
<li class="chapter" data-level="4.1" data-path="case-studies.html"><a href="case-studies.html#foliar-fungi-experiment"><i class="fa fa-check"></i><b>4.1</b> Foliar fungi experiment</a></li>
<li class="chapter" data-level="4.2" data-path="case-studies.html"><a href="case-studies.html#vaginal-microbiomes-of-pregnant-women"><i class="fa fa-check"></i><b>4.2</b> Vaginal microbiomes of pregnant women</a></li>
<li class="chapter" data-level="4.3" data-path="case-studies.html"><a href="case-studies.html#human-gut-microbiomes"><i class="fa fa-check"></i><b>4.3</b> Human gut microbiomes</a></li>
<li class="chapter" data-level="4.4" data-path="case-studies.html"><a href="case-studies.html#microbial-growth-in-marine-sediments"><i class="fa fa-check"></i><b>4.4</b> Microbial growth in marine sediments</a></li>
<li class="chapter" data-level="4.5" data-path="case-studies.html"><a href="case-studies.html#summary-and-discussion"><i class="fa fa-check"></i><b>4.5</b> Summary and discussion</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="solutions.html"><a href="solutions.html"><i class="fa fa-check"></i><b>5</b> Solutions</a>
<ul>
<li class="chapter" data-level="5.1" data-path="solutions.html"><a href="solutions.html#ratio-based-relative-da-analysis"><i class="fa fa-check"></i><b>5.1</b> Ratio-based relative DA analysis</a></li>
<li class="chapter" data-level="5.2" data-path="solutions.html"><a href="solutions.html#calibrate-compositions"><i class="fa fa-check"></i><b>5.2</b> Calibration using community controls</a></li>
<li class="chapter" data-level="5.3" data-path="solutions.html"><a href="solutions.html#absolute-abundance-methods-with-more-stable-fes"><i class="fa fa-check"></i><b>5.3</b> Absolute-abundance methods with more stable FEs</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="solutions.html"><a href="solutions.html#use-complementary-mgs-and-total-abundance-measurements"><i class="fa fa-check"></i><b>5.3.1</b> Use complementary MGS and total-abundance measurements</a></li>
<li class="chapter" data-level="5.3.2" data-path="solutions.html"><a href="solutions.html#normalize-to-a-reference-species"><i class="fa fa-check"></i><b>5.3.2</b> Normalize to a reference species</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="solutions.html"><a href="solutions.html#bias-sensitivity-analysis"><i class="fa fa-check"></i><b>5.4</b> Bias sensitivity analysis</a></li>
<li class="chapter" data-level="5.5" data-path="solutions.html"><a href="solutions.html#bias-aware-meta-analysis"><i class="fa fa-check"></i><b>5.5</b> Bias-aware meta-analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="appendix-regression.html"><a href="appendix-regression.html"><i class="fa fa-check"></i><b>A</b> Linear regression</a>
<ul>
<li class="chapter" data-level="A.1" data-path="appendix-regression.html"><a href="appendix-regression.html#simple-linear-regression"><i class="fa fa-check"></i><b>A.1</b> Simple linear regression</a>
<ul>
<li class="chapter" data-level="A.1.1" data-path="appendix-regression.html"><a href="appendix-regression.html#review-of-simple-linear-regression"><i class="fa fa-check"></i><b>A.1.1</b> Review of simple linear regression</a></li>
<li class="chapter" data-level="A.1.2" data-path="appendix-regression.html"><a href="appendix-regression.html#measurement-error-in-the-response"><i class="fa fa-check"></i><b>A.1.2</b> Measurement error in the response</a></li>
<li class="chapter" data-level="A.1.3" data-path="appendix-regression.html"><a href="appendix-regression.html#specific-application-to-taxonomic-bias"><i class="fa fa-check"></i><b>A.1.3</b> Specific application to taxonomic bias</a></li>
</ul></li>
<li class="chapter" data-level="A.2" data-path="appendix-regression.html"><a href="appendix-regression.html#gamma-poisson-regression"><i class="fa fa-check"></i><b>A.2</b> Gamma-Poisson regression</a>
<ul>
<li class="chapter" data-level="A.2.1" data-path="appendix-regression.html"><a href="appendix-regression.html#background"><i class="fa fa-check"></i><b>A.2.1</b> Background</a></li>
<li class="chapter" data-level="A.2.2" data-path="appendix-regression.html"><a href="appendix-regression.html#inferring-lfcs-in-proportions-with-and-without-bias-correction"><i class="fa fa-check"></i><b>A.2.2</b> Inferring LFCs in proportions with and without bias correction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="B" data-path="supplemental-figures.html"><a href="supplemental-figures.html"><i class="fa fa-check"></i><b>B</b> Supplemental figures</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Implications of taxonomic bias for microbial differential-abundance analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="supplemental-figures" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">B</span> Supplemental figures<a href="supplemental-figures.html#supplemental-figures" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<!-- Case studies: leopold2020host -->
<div class="figure"><span style="display:block;" id="fig:leopold2020host-variation"></span>
<img src="notebook/_posts/2022-01-08-leopold2020host-case-study/leopold2020host-case-study_files/figure-html5/variation-in-proportions-and-mean-efficiency-1.svg" alt="In the pre-infection samples from Leopold and Busby (2020), multiplicative variation in taxa proportions is much larger than that in the mean efficiency. Panel A shows the distribution of the proportions of each commensal isolate (denoted by its genus) across all samples collected prior to pathogen inoculation; Panel C shows the distribution of the (estimated) sample mean efficiency across these same samples on the same scale; and Panel B shows the efficiency of each taxon estimated from DNA mock communities as point estimates and 90% bootstrap percentile confidence intervals. Efficiencies are shown relative to the most efficiently measured taxon (Fusarium)."  />
<p class="caption">
Figure B.1: <strong>In the pre-infection samples from <span class="citation">Leopold and Busby (<a href="#ref-leopold2020host" role="doc-biblioref">2020</a>)</span>, multiplicative variation in taxa proportions is much larger than that in the mean efficiency.</strong> Panel A shows the distribution of the proportions of each commensal isolate (denoted by its genus) across all samples collected prior to pathogen inoculation; Panel C shows the distribution of the (estimated) sample mean efficiency across these same samples on the same scale; and Panel B shows the efficiency of each taxon estimated from DNA mock communities as point estimates and 90% bootstrap percentile confidence intervals. Efficiencies are shown relative to the most efficiently measured taxon (<em>Fusarium</em>).
</p>
</div>

<p><br><br></p>
<div class="figure"><span style="display:block;" id="fig:leopold2020host-infection-mean-efficiency-dist"></span>
<img src="notebook/_posts/2022-01-08-leopold2020host-case-study/leopold2020host-case-study_files/figure-html5/infection-mean-efficiency-dist-1.svg" alt="The mean efficiency tends to increase after infection due to the high proportion of the pathogen."  />
<p class="caption">
Figure B.2: <strong>The mean efficiency tends to increase after infection due to the high proportion of the pathogen.</strong>
</p>
</div>

<p><br><br></p>
<div class="figure"><span style="display:block;" id="fig:leopold2020host-infection-lfc"></span>
<img src="notebook/_posts/2022-01-08-leopold2020host-case-study/leopold2020host-case-study_files/figure-html5/infection-gp-lfc-estimates-1.svg" alt="Bias correction increases the estimated log fold changes (LFCs) in commensal species proportions in response to pathogen infection. Shown are the estimated LFCs with 95% Bayesian credible intervals (CIs) derived from a Bayesian gamma Poisson regression. Negative values indicate that the speciesâ proportion decreased on average in response to infection, which is the default expectation given growth of the pathogen and the sum-to-one constraint of proportions. Bias leads to artificially low estimates, as the increase in the pathogen proportion increases the mean efficiency. In Western genotypes, several species whose proportion appears to decrease are instead found to remain stable or even to increase when bias is corrected."  />
<p class="caption">
Figure B.3: <strong>Bias correction increases the estimated log fold changes (LFCs) in commensal species proportions in response to pathogen infection.</strong> Shown are the estimated LFCs with 95% Bayesian credible intervals (CIs) derived from a Bayesian gamma Poisson regression. Negative values indicate that the speciesâ proportion decreased on average in response to infection, which is the default expectation given growth of the pathogen and the sum-to-one constraint of proportions. Bias leads to artificially low estimates, as the increase in the pathogen proportion increases the mean efficiency. In Western genotypes, several species whose proportion appears to decrease are instead found to remain stable or even to increase when bias is corrected.
</p>
</div>

<p><br><br></p>
<!-- Case study: MOMSPI -->
<div class="figure"><span style="display:block;" id="fig:momspi-mean-efficiency-fcs"></span>
<img src="notebook/_posts/2021-11-01-momspi-summary/momspi-summary_files/figure-html5/momspi-mean-efficiency-fcs-1.svg" alt="Fold changes in the mean efficiency within and between women in the MOMS-PI study."  />
<p class="caption">
Figure B.4: <strong>Fold changes in the mean efficiency within and between women in the MOMS-PI study.</strong>
</p>
</div>

<p><br><br></p>
<div class="figure"><span style="display:block;" id="fig:momspi-trajectory"></span>
<img src="notebook/_posts/2021-11-01-momspi-summary/momspi-summary_files/figure-html5/trajectories-within-subject-high-var-1.svg" alt="In vaginal microbiome measurements, shifts between Lactobacillus and Gardnerella dominance can drive spurious fold changes in other, lower-abundance species. The figure shows species proportions and mean efficiency trajectories over consecutive clinical visits for a subject in the MOMS-PI study whose microbiome samples showed substantial variation in mean efficiency. The subjectâs samples are dominated by Gardnerella vaginalis and Lachnospiraceae BVAB1 during the first three visits before transitioning to being dominated by Lactobacillus iners between visits 3 and 4. This transition drives a sharp increase in the mean efficiency, which significantly distorts the fold changes in the observed (uncalibrated) microbiome measurements for species with less dramatic fold changes. Two exemplar species are shown to illustrate the magnitude (Ureaplasma cluster 23) and direction (Megasphaera OTU70 type1) errors that can arise in this situation." width="75%" />
<p class="caption">
Figure B.5: <strong>In vaginal microbiome measurements, shifts between <em>Lactobacillus</em> and <em>Gardnerella</em> dominance can drive spurious fold changes in other, lower-abundance species.</strong> The figure shows species proportions and mean efficiency trajectories over consecutive clinical visits for a subject in the MOMS-PI study whose microbiome samples showed substantial variation in mean efficiency. The subjectâs samples are dominated by <em>Gardnerella vaginalis</em> and <em>Lachnospiraceae BVAB1</em> during the first three visits before transitioning to being dominated by <em>Lactobacillus iners</em> between visits 3 and 4. This transition drives a sharp increase in the mean efficiency, which significantly distorts the fold changes in the observed (uncalibrated) microbiome measurements for species with less dramatic fold changes. Two exemplar species are shown to illustrate the magnitude (<em>Ureaplasma cluster 23</em>) and direction (<em>Megasphaera OTU70 type1</em>) errors that can arise in this situation.
</p>
</div>

<p><br><br></p>
<div class="figure"><span style="display:block;" id="fig:momspi-regression"></span>
<img src="notebook/_posts/2021-12-15-momspi-regression-diversity/momspi-regression-diversity_files/figure-html5/regression-gp-1.svg" alt="Bias inflates estimated log fold changes (LFCs) in species proportions with diversity in vaginal microbiome samples from the MOMS-PI study. Samples were split into low, medium, and high diversity groups based on Shannon diversity in observed (uncalibrated) microbiome profiles. The LFC in proportion from low- to high-diversity samples was estimated for 30 common species using gamma-Poisson regression with and without bias correction. Panel A shows the distribution of point estimates; Panel B shows the point estimates and 95% Bayesian credible intervals for each species."  />
<p class="caption">
Figure B.6: <strong>Bias inflates estimated log fold changes (LFCs) in species proportions with diversity in vaginal microbiome samples from the MOMS-PI study.</strong> Samples were split into low, medium, and high diversity groups based on Shannon diversity in observed (uncalibrated) microbiome profiles. The LFC in proportion from low- to high-diversity samples was estimated for 30 common species using gamma-Poisson regression with and without bias correction. Panel A shows the distribution of point estimates; Panel B shows the point estimates and 95% Bayesian credible intervals for each species.
</p>
</div>

<p><br><br></p>
<!-- Case studies: Gut -->
<div class="figure"><span style="display:block;" id="fig:gut"></span>
<img src="notebook/_posts/2022-01-30-hmp-stool-vagina-comparison/hmp-stool-vagina-comparison_files/figure-html5/main-1.svg" alt="Compared to vaginal samples, stool samples have higher diversity, lower variation in mean efficiency, and lower variation in species proportions. This figure summarizes results from our case study of shotgun-sequenced stool and vaginal samples in the Human Microbiome Project. Panel A shows the distribution of order-2 alpha diversity (equal to the Inverse Simpson Index) across samples. Panel B shows the variation in the mean efficiency, as quantified by the GSD, across samples of a given type for each of 1000 simulated bias vectors. Panel C shows the variation in the proportion of each species commonly found in a given sample type. For Panel C, variation is measured by the GSD when a pseudo-value of \(10^{-4}\) is added to the raw proportions. Different choices of the pseudo-value substantially change the measured GSDs, but not the difference (ratio) between gut and vaginal species."  />
<p class="caption">
Figure B.7: <strong>Compared to vaginal samples, stool samples have higher diversity, lower variation in mean efficiency, and lower variation in species proportions.</strong> This figure summarizes results from our case study of shotgun-sequenced stool and vaginal samples in the Human Microbiome Project. Panel A shows the distribution of order-2 alpha diversity (equal to the Inverse Simpson Index) across samples. Panel B shows the variation in the mean efficiency, as quantified by the GSD, across samples of a given type for each of 1000 simulated bias vectors. Panel C shows the variation in the proportion of each species commonly found in a given sample type. For Panel C, variation is measured by the GSD when a pseudo-value of <span class="math inline">\(10^{-4}\)</span> is added to the raw proportions. Different choices of the pseudo-value substantially change the measured GSDs, but not the difference (ratio) between gut and vaginal species.
</p>
</div>

<p><br><br></p>
<!-- Solutions -->

</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-leopold2020host" class="csl-entry">
Leopold, Devin R, and Posy E Busby. 2020. <span>â<span class="nocase">Host Genotype and Colonist Arrival Order Jointly Govern Plant Microbiome Composition and Function</span>.â</span> <em>Curr. Biol.</em> 30 (16): 3260â3266.e5. <a href="https://doi.org/10.1016/j.cub.2020.06.011">https://doi.org/10.1016/j.cub.2020.06.011</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="appendix-regression.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/mikemc/differential-abundance-theory/edit/main/supplemental-figures.Rmd",
"text": "Edit"
},
"history": {
"link": "https://github.com/mikemc/differential-abundance-theory/commits/main/supplemental-figures.Rmd",
"text": null
},
"view": {
"link": "https://github.com/mikemc/differential-abundance-theory/blob/main/supplemental-figures.Rmd",
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
