<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 How bias affects abundance measurements | Characterizing and correcting the effect of taxonomic bias in microbial differential-abundance analysis</title>
  <meta name="description" content="2 How bias affects abundance measurements | Characterizing and correcting the effect of taxonomic bias in microbial differential-abundance analysis" />
  <meta name="generator" content="bookdown 0.27.3 and GitBook 2.6.7" />

  <meta property="og:title" content="2 How bias affects abundance measurements | Characterizing and correcting the effect of taxonomic bias in microbial differential-abundance analysis" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 How bias affects abundance measurements | Characterizing and correcting the effect of taxonomic bias in microbial differential-abundance analysis" />
  
  
  

<meta name="author" content="Michael R. McLaren" />
<meta name="author" content="Jacob T. Nearing" />
<meta name="author" content="Amy D. Willis" />
<meta name="author" content="Karen G. Lloyd" />
<meta name="author" content="Benjamin J. Callahan" />


<meta name="date" content="2022-07-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introduction.html"/>
<link rel="next" href="differential-abundance.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
    TeX: {extensions: ["cancel.js"]},
});
MathJax.Hub.Config({
    TeX: {
        Macros : {
            md: '^{\\times}\\!/',
        }
    },
});
</script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="abundance-measurement.html"><a href="abundance-measurement.html"><i class="fa fa-check"></i><b>2</b> How bias affects abundance measurements</a>
<ul>
<li class="chapter" data-level="2.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#model-of-mgs-measurement"><i class="fa fa-check"></i><b>2.1</b> Model of MGS measurement</a></li>
<li class="chapter" data-level="2.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#relative-abundance"><i class="fa fa-check"></i><b>2.2</b> Relative abundance</a></li>
<li class="chapter" data-level="2.3" data-path="abundance-measurement.html"><a href="abundance-measurement.html#absolute-abundance"><i class="fa fa-check"></i><b>2.3</b> Absolute abundance</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#leveraging-information-about-total-community-abundance"><i class="fa fa-check"></i><b>2.3.1</b> Leveraging information about total-community abundance</a></li>
<li class="chapter" data-level="2.3.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#leveraging-information-about-a-reference-species"><i class="fa fa-check"></i><b>2.3.2</b> Leveraging information about a reference species</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>3</b> How bias affects DA results</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-abundance.html"><a href="differential-abundance.html#fold-change-between-a-pair-of-samples"><i class="fa fa-check"></i><b>3.1</b> Fold change between a pair of samples</a></li>
<li class="chapter" data-level="3.2" data-path="differential-abundance.html"><a href="differential-abundance.html#regression-analysis-of-many-samples"><i class="fa fa-check"></i><b>3.2</b> Regression analysis of many samples</a></li>
<li class="chapter" data-level="3.3" data-path="differential-abundance.html"><a href="differential-abundance.html#rank-based-analyses"><i class="fa fa-check"></i><b>3.3</b> Rank-based analyses</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="case-studies.html"><a href="case-studies.html"><i class="fa fa-check"></i><b>4</b> Case studies  </a>
<ul>
<li class="chapter" data-level="4.1" data-path="case-studies.html"><a href="case-studies.html#foliar-fungi-experiment"><i class="fa fa-check"></i><b>4.1</b> Foliar fungi experiment</a></li>
<li class="chapter" data-level="4.2" data-path="case-studies.html"><a href="case-studies.html#vaginal-microbiomes-of-pregnant-women"><i class="fa fa-check"></i><b>4.2</b> Vaginal microbiomes of pregnant women</a></li>
<li class="chapter" data-level="4.3" data-path="case-studies.html"><a href="case-studies.html#human-gut-microbiomes"><i class="fa fa-check"></i><b>4.3</b> Human gut microbiomes</a></li>
<li class="chapter" data-level="4.4" data-path="case-studies.html"><a href="case-studies.html#microbial-growth-in-marine-sediments"><i class="fa fa-check"></i><b>4.4</b> Microbial growth in marine sediments</a></li>
<li class="chapter" data-level="4.5" data-path="case-studies.html"><a href="case-studies.html#summary-and-discussion"><i class="fa fa-check"></i><b>4.5</b> Summary and discussion</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="solutions.html"><a href="solutions.html"><i class="fa fa-check"></i><b>5</b> Potential solutions</a>
<ul>
<li class="chapter" data-level="5.1" data-path="solutions.html"><a href="solutions.html#ratio-based-relative-da-analysis"><i class="fa fa-check"></i><b>5.1</b> Ratio-based relative DA analysis</a></li>
<li class="chapter" data-level="5.2" data-path="solutions.html"><a href="solutions.html#calibrate-compositions"><i class="fa fa-check"></i><b>5.2</b> Calibration using community controls</a></li>
<li class="chapter" data-level="5.3" data-path="solutions.html"><a href="solutions.html#choose-absolute-abundance-methods-with-more-stable-fes"><i class="fa fa-check"></i><b>5.3</b> Choose absolute-abundance methods with more stable FEs</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="solutions.html"><a href="solutions.html#use-complementary-mgs-and-total-abundance-measurements"><i class="fa fa-check"></i><b>5.3.1</b> Use complementary MGS and total-abundance measurements</a></li>
<li class="chapter" data-level="5.3.2" data-path="solutions.html"><a href="solutions.html#normalize-to-a-reference-species"><i class="fa fa-check"></i><b>5.3.2</b> Normalize to a reference species</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="solutions.html"><a href="solutions.html#bias-sensitivity-analysis"><i class="fa fa-check"></i><b>5.4</b> Bias sensitivity analysis</a></li>
<li class="chapter" data-level="5.5" data-path="solutions.html"><a href="solutions.html#bias-aware-meta-analysis"><i class="fa fa-check"></i><b>5.5</b> Bias-aware meta-analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="mgs-measurement-details.html"><a href="mgs-measurement-details.html"><i class="fa fa-check"></i><b>A</b> MGS measurement details</a>
<ul>
<li class="chapter" data-level="A.1" data-path="mgs-measurement-details.html"><a href="mgs-measurement-details.html#spike-in-alt"><i class="fa fa-check"></i><b>A.1</b> Alternative method for measuring species absolute abundances from a spike-in</a></li>
<li class="chapter" data-level="A.2" data-path="mgs-measurement-details.html"><a href="mgs-measurement-details.html#multiple-references"><i class="fa fa-check"></i><b>A.2</b> Missing and multiple reference species</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="appendix-regression.html"><a href="appendix-regression.html"><i class="fa fa-check"></i><b>B</b> Linear regression</a>
<ul>
<li class="chapter" data-level="B.1" data-path="appendix-regression.html"><a href="appendix-regression.html#simple-linear-regression"><i class="fa fa-check"></i><b>B.1</b> Simple linear regression</a>
<ul>
<li class="chapter" data-level="B.1.1" data-path="appendix-regression.html"><a href="appendix-regression.html#review-of-simple-linear-regression"><i class="fa fa-check"></i><b>B.1.1</b> Review of simple linear regression</a></li>
<li class="chapter" data-level="B.1.2" data-path="appendix-regression.html"><a href="appendix-regression.html#measurement-error-in-the-response"><i class="fa fa-check"></i><b>B.1.2</b> Measurement error in the response</a></li>
<li class="chapter" data-level="B.1.3" data-path="appendix-regression.html"><a href="appendix-regression.html#specific-application-to-taxonomic-bias"><i class="fa fa-check"></i><b>B.1.3</b> Specific application to taxonomic bias</a></li>
</ul></li>
<li class="chapter" data-level="B.2" data-path="appendix-regression.html"><a href="appendix-regression.html#gamma-poisson-regression"><i class="fa fa-check"></i><b>B.2</b> Gamma-Poisson regression</a>
<ul>
<li class="chapter" data-level="B.2.1" data-path="appendix-regression.html"><a href="appendix-regression.html#background"><i class="fa fa-check"></i><b>B.2.1</b> Background</a></li>
<li class="chapter" data-level="B.2.2" data-path="appendix-regression.html"><a href="appendix-regression.html#inferring-lfcs-in-proportions-with-and-without-bias-correction"><i class="fa fa-check"></i><b>B.2.2</b> Inferring LFCs in proportions with and without bias correction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="C" data-path="diversity-and-mean-efficiency.html"><a href="diversity-and-mean-efficiency.html"><i class="fa fa-check"></i><b>C</b> Alpha diversity and variation in the mean efficiency</a></li>
<li class="chapter" data-level="D" data-path="supplemental-figures.html"><a href="supplemental-figures.html"><i class="fa fa-check"></i><b>D</b> Supplemental figures</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Characterizing and correcting the effect of taxonomic bias in microbial differential-abundance analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="abundance-measurement" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> How bias affects abundance measurements<a href="abundance-measurement.html#abundance-measurement" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This section extends the theoretical results of <span class="citation">McLaren, Willis, and Callahan (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span> to describe the effect that consistent taxonomic bias within an MGS experiment affects the relative and absolute abundances measured for various microbial species.
We show that some approaches to quantifying species abundance yield constant fold errors (FEs), while others yield FEs that depend on overall community composition and thus can vary across samples.</p>
<div id="model-of-mgs-measurement" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Model of MGS measurement<a href="abundance-measurement.html#model-of-mgs-measurement" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Our primary tool for understanding the impact of taxonomic bias on MGS measurement is the theoretical model of MGS measurement developed and empirically validated by <span class="citation">McLaren, Willis, and Callahan (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span>.
This model describes the mathematical relationship between the read counts obtained by MGS and the (actual) abundances of the various species in a sample.
Here we extend the model as first described <span class="citation">McLaren, Willis, and Callahan (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span>, which considers only relative abundances, to also consider absolute abundances.
For concreteness, we will consider <em>absolute abundance</em>, or simply <em>abundance</em>, to refer to the number of cells per unit volume in a sample (cell concentration).
That said, our results equally apply to other definitions of absolute abundance, such as the total number of cells in a sample or ecosystem and other abundance units such as biomass or genome copy number.</p>
<p>This model is the simplest that respects the multiplicative nature of taxonomic bias and the <em>compositional</em> nature of MGS measurements.
The actual abundance of a species in a given sample, multiplied by its <em>measurement efficiency</em>—its rate of conversion from cells to taxonomically assigned sequencing reads—determines the species’ read count in that sample.
Taxonomic bias presents as variation in the measurement efficiencies among species within an MGS experiment.
The read counts further depend on sample-specific experimental factors that are typically unknown, such that they are best interpreted as only providing relative abundances (such data is said to be <em>compositional</em>; <span class="citation">Gloor et al. (<a href="#ref-gloor2017micr" role="doc-biblioref">2017</a>)</span>).</p>
<p>We consider a set of microbiome samples measured by a specific MGS protocol that extracts, sequences, and taxonomically assigns reads to a set of microbial species <span class="math inline">\(S\)</span>.
We make several simplifying assumptions to facilitate our analysis and presentation.
First, we consider only species-level assignment, and suppose that reads that cannot be uniquely assigned to a single species in <span class="math inline">\(S\)</span> are discarded.
Second, we ignore the possibility that reads are misassigned to the wrong species or the wrong sample.
Third, we suppose that taxonomic bias acts consistently across samples at the species level—that is, a given species is always measured more efficiently than another to the same degree.
Finally, unless otherwise stated, we treat sequencing measurements as deterministic, ignoring the ‘random’ variation in read counts that arise from the sampling of sequencing reads and other aspects of the MGS process.
These assumptions, though unrealistic descriptions of most MGS experiments, serve the purpose of clearly demonstrating when and why consistent taxonomic bias creates errors in DA analysis.</p>
<p>Our model stipulates that the taxonomically-assigned read count for a species <span class="math inline">\(i\)</span> in a sample <span class="math inline">\(a\)</span> equals its abundance <span class="math inline">\(A_{i}^{(a)}\)</span> multiplied by a species-specific factor <span class="math inline">\(B_{i}\)</span> and a sample-specific factor <span class="math inline">\(F^{(a)}\)</span>,
<span class="math display" id="eq:mgs-model">\[\begin{align}
  \tag{2.1}
  M_i^{(a)} = A_i^{(a)} B_i F^{(a)}.
\end{align}\]</span>
The species-specific factor, <span class="math inline">\(B_{i}\)</span>, is the <em>relative measurement efficiency</em> (or simply <em>efficiency</em>) of the species relative to an arbitrary baseline species (<span class="citation">McLaren, Willis, and Callahan (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span>).
The variation in efficiency among species corresponds to the taxonomic bias of the MGS protocol.
The sample-specific factor, <span class="math inline">\(F^{(a)}\)</span>, describes the effective sequencing effort for that sample; it equals the number of reads per unit abundance that would be obtained for a species with an efficiency of 1.
We can write the total number of assigned reads for the sample as
<span class="math display" id="eq:total-reads">\[\begin{align}
  \tag{2.2}
  M_{\text{tot}}^{(a)}
    = A_{\text{tot}}^{(a)} \bar B^{(a)} F^{(a)},
\end{align}\]</span>
where
<span class="math inline">\(M_{{\text{tot}}}^{(a)} \equiv \sum_{j\in S} M_j^{(a)}\)</span> is the total read count and <span class="math inline">\(A_{{\text{tot}}}^{(a)} \equiv \sum_{j\in S}A_j^{(a)}\)</span> is the total abundance for all species <span class="math inline">\(S\)</span>, and
<span class="math display" id="eq:mean-efficiency">\[\begin{align}
  \tag{2.3}
  \bar B^{(a)} \equiv \frac{\sum_{i\in S} A_i^{(a)} B_i}{\sum_{i\in S} A_i^{(a)}}
\end{align}\]</span>
is the <em>sample mean efficiency</em>, defined as the mean efficiency of all species weighted by their abundance.</p>
</div>
<div id="relative-abundance" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Relative abundance<a href="abundance-measurement.html#relative-abundance" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We distinguish between two types of species-level <em>relative abundances</em> within a sample.
The <em>proportion</em> <span class="math inline">\(P_{i}^{(a)}\)</span> of species <span class="math inline">\(i\)</span> in sample <span class="math inline">\(a\)</span> equals its abundance relative to the total abundance of all species in <span class="math inline">\(S\)</span>,
<span class="math display" id="eq:prop">\[\begin{align}
  \tag{2.4}
  P_{i}^{(a)} &amp;\equiv \frac{A_i^{(a)}}{A_{\text{tot}}^{(a)}}.
\end{align}\]</span>
The <em>ratio</em> <span class="math inline">\(R_{i/j}^{(a)}\)</span> between two species <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> equals the abundance of <span class="math inline">\(i\)</span> divided by that of <span class="math inline">\(j\)</span>,
<span class="math display" id="eq:ratio">\[\begin{align}
  \tag{2.5}
  R_{i/j}^{(a)} \equiv \frac{A_i^{(a)}}{A_j^{(a)}}.
\end{align}\]</span></p>
<p>The measured proportion of a species is given by its proportion of all the assigned reads in a sample,
<span class="math display" id="eq:prop-meas">\[\begin{align}
  \tag{2.6}
  \tilde P_{i}^{(a)} &amp;\equiv \frac{M_i^{(a)}}{M_{\text{tot}}^{(a)}}.
\end{align}\]</span>
We use the tilde to distinguish the measurement from the actual quantity being measured.
From Equations <a href="abundance-measurement.html#eq:mgs-model">(2.1)</a>, <a href="abundance-measurement.html#eq:total-reads">(2.2)</a>, and <a href="abundance-measurement.html#eq:prop-meas">(2.6)</a>, it follows that the measured and actual proportion are related by
<span class="math display" id="eq:prop-error">\[\begin{align}
  \tag{2.7}
  \tilde P_{i}^{(a)} &amp;= P_{i}^{(a)} \cdot \frac{B_i}{\bar B^{(a)}}.
\end{align}\]</span>
<!-- Taxonomic bias thus creates a fold-error (FE) in the measured proportion $\tilde P_{i}^{(a)}$ of species $i$ equal to the efficiency $B_i$ of species $i$ $divided by the mean efficiency $\bar B^{(a)}$ in the sample. -->
Taxonomic bias thus creates a fold-error (FE) in the measured proportion of a species that is equal to its efficiency divided by the mean efficiency in the sample.
Since the mean efficiency varies across samples, so does the FE.
This phenomenon can be seen for Species 3 in the two hypothetical communities in Figure <a href="abundance-measurement.html#fig:error-proportions">2.1</a>.
Species 3, which has an efficiency of 6, is under-measured in Sample 1 (FE &lt; 1) but over-measured (FE &gt; 1) in Sample 2.
This difference occurs because the even distribution of species Sample 1 yields a mean efficiency of 8.33; in contrast, the lopsided distribution in Sample 2, which is dominated by the low-efficiency Species 1, has a mean efficiency of just 3.15.
A demonstration in bacterial mock communities is shown in <a href="https://doi.org/10.7554/eLife.46923.004">Figure 3C</a> of <span class="citation">McLaren, Willis, and Callahan (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span>.</p>
<p>The measured ratio <span class="math inline">\(\tilde R_{i/j}^{(a)}\)</span> between species <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> is given by the ratio of their read counts,
<span class="math display" id="eq:ratio-meas">\[\begin{align}
  \tag{2.8}
  \tilde R_{i/j}^{(a)} \equiv \frac{M_{i}^{(a)}}{M_{j}^{(a)}}.
\end{align}\]</span>
From Equations <a href="abundance-measurement.html#eq:mgs-model">(2.1)</a> and <a href="abundance-measurement.html#eq:ratio-meas">(2.8)</a>, it follows that the measured and actual ratio are related by
<span class="math display" id="eq:ratio-error">\[\begin{align}
  \tag{2.9}
  \tilde R_{i/j}^{(a)} = R_{i/j}^{(a)} \cdot \frac{B_i}{B_j}.
\end{align}\]</span>
Taxonomic bias thus creates a FE in the measured ratio that is equal to the ratio in the species’ efficiencies; the FE is therefore constant across samples.
For instance, in Figure <a href="abundance-measurement.html#fig:error-proportions">2.1</a>, the ratio of Species 3 (with an efficiency of 6) to Species 1 (with an efficiency of 1) is over-estimated by a factor of 6 in both communities despite their varying compositions.
A demonstration in bacterial mock communities is shown in <a href="https://doi.org/10.7554/eLife.46923.004">Figure 3D</a> of <span class="citation">McLaren, Willis, and Callahan (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span>.</p>
<!-- begin figure -->
<div class="figure"><span style="display:block;" id="fig:error-proportions"></span>
<img src="figures/illustrations/error-proportions.svg" alt="Taxonomic bias creates fold errors in species proportions that vary across samples and lead to inaccurate fold differences between samples. Top row: Error in proportions measured by MGS in two hypothetical microbiome samples that contain different relative abundances of three species. Bottom row: Error in the measured fold difference in the third species that is derived from these measurements. Species’ proportions may be measured as too high or too low depending on sample composition. For instance, Species 3 has an efficiency of 6 and is under-measured in Sample 1 (which has a mean efficiency of 8.33) but over-measured in Sample 2 (which has a mean efficiency of 3.15)." width="90%" />
<p class="caption">
Figure 2.1: <strong>Taxonomic bias creates fold errors in species proportions that vary across samples and lead to inaccurate fold differences between samples.</strong> Top row: Error in proportions measured by MGS in two hypothetical microbiome samples that contain different relative abundances of three species. Bottom row: Error in the measured fold difference in the third species that is derived from these measurements. Species’ proportions may be measured as too high or too low depending on sample composition. For instance, Species 3 has an efficiency of 6 and is under-measured in Sample 1 (which has a mean efficiency of 8.33) but over-measured in Sample 2 (which has a mean efficiency of 3.15).
</p>
</div>

<!-- end figure -->
<p><strong>Higher-order taxa:</strong>
We can consider a higher-order taxon <span class="math inline">\(I\)</span>, such as a genus or phylum, as a set of species, <span class="math inline">\(\{i \in I\}\)</span>.
The abundance of taxon <span class="math inline">\(I\)</span> in sample <span class="math inline">\(a\)</span> is the sum of the abundances of its constituent species, <span class="math inline">\(A_{I}^{(a)} \equiv \sum_{i \in I} A_{i}^{(a)}\)</span>.
Similarly, the read count of taxon <span class="math inline">\(I\)</span> is the sum <span class="math inline">\(M_{I}^{(a)} \equiv \sum_{i \in I} M_{i}^{(a)}\)</span>.
We further define the efficiency of taxon <span class="math inline">\(I\)</span> as the abundance-weighted average of the efficiencies of its constituent species,
<span class="math display" id="eq:efficiency-general">\[\begin{align}
  \tag{2.10}
  B_I^{(a)} \equiv \frac{\sum_{i\in I} A_{i}^{(a)} B_{i}}{\sum_{i\in I} A_{i}^{(a)}}.
\end{align}\]</span>
With these definitions, the read count for taxon <span class="math inline">\(I\)</span> can be expressed as
<span class="math inline">\(M_{I}^{(a)} = A_{I}^{(a)} B_I^{(a)} F^{(a)}\)</span>.
Thus <span class="math inline">\(B_I^{(a)}\)</span> plays a role analogous to the efficiency of an individual species, but differs in that it need not be constant across samples:
If the constituent species have different efficiencies, then the efficiency of the higher-order taxon <span class="math inline">\(I\)</span> depends on the relative abundances of its constituents and so will tend to vary across samples (<span class="citation">McLaren, Willis, and Callahan (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span>).
As an example, suppose that Species 1 and Species 2 in Figure <a href="abundance-measurement.html#fig:error-proportions">2.1</a> were in the same phylum.
The efficiency of the phylum would then be <span class="math inline">\(\tfrac{1}{2} \cdot 1 + \tfrac{1}{2} \cdot 18 = 9.5\)</span> in Sample 1 and <span class="math inline">\(\tfrac{15}{16} \cdot 1 + \tfrac{1}{16} \cdot 18 \approx 2.1\)</span> in Sample 2.
Equations <a href="abundance-measurement.html#eq:prop-error">(2.7)</a> and <a href="abundance-measurement.html#eq:ratio-error">(2.9)</a> continue to describe the measurement error in proportions and ratios involving higher-order taxa, so long as the sample-dependent, higher-order taxa efficiencies <span class="math inline">\(B_I^{(a)}\)</span> and <span class="math inline">\(B_J^{(a)}\)</span> are used.
In this way, we see that both proportions and ratios among higher-order taxa may have inconsistent FEs.</p>
</div>
<div id="absolute-abundance" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Absolute abundance<a href="abundance-measurement.html#absolute-abundance" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Several extensions of the standard MGS experiment make it possible to measure absolute species abundances.
These extensions fall into two general approaches.
The first approach leverages information about the abundance of the total community; for example, <span class="citation">Vandeputte et al. (<a href="#ref-vandeputte2017quan" role="doc-biblioref">2017</a>)</span> measured total-community abundance using flow cytometry and multiplied this number by MGS genus proportions to obtain the absolute abundances of individual genera (<span class="citation">Vandeputte et al. (<a href="#ref-vandeputte2017quan" role="doc-biblioref">2017</a>)</span>).
A second approach leverages information about the abundance of one or more individual species; for example, a researcher might ‘spike in’ a known, fixed amount of an extraneous species to all samples prior to MGS, and normalize the read counts of all species to the spike-in species (<span class="citation">Harrison et al. (<a href="#ref-harrison2021theq" role="doc-biblioref">2021</a>)</span>).
We consider each approach in detail to determine how taxonomic bias affects the resulting absolute-abundance measurements.</p>
<div id="leveraging-information-about-total-community-abundance" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Leveraging information about total-community abundance<a href="abundance-measurement.html#leveraging-information-about-total-community-abundance" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Suppose that the total abundance of all species in the sample, <span class="math inline">\(A_{{\text{tot}}}^{(a)}\)</span>, has been measured by a non-MGS method, yielding a measurement <span class="math inline">\(\tilde A_{\text{tot}}^{(a)}\)</span>.
The absolute abundance of an individual species can be measured by multiplying the species’ proportion from MGS by this total-abundance measurement,
<span class="math display" id="eq:density-prop-meas">\[\begin{align}
  \tag{2.11}
  \tilde A_i^{(a)} &amp;= \tilde P_i^{(a)} \tilde A_{\text{tot}}^{(a)}.
\end{align}\]</span>
Total-abundance measurements recently used for this purpose include counting cells with microscopy (<span class="citation">Lloyd et al. (<a href="#ref-lloyd2020evid" role="doc-biblioref">2020</a>)</span>) or flow cytometry (<span class="citation">Props et al. (<a href="#ref-props2017abso" role="doc-biblioref">2017</a>)</span>, <span class="citation">Vandeputte et al. (<a href="#ref-vandeputte2017quan" role="doc-biblioref">2017</a>)</span>, <span class="citation">Galazzo et al. (<a href="#ref-galazzo2020howt" role="doc-biblioref">2020</a>)</span>), measuring the concentration of a marker-gene with qPCR or ddPCR (<span class="citation">Zhang et al. (<a href="#ref-zhang2017soil" role="doc-biblioref">2017</a>)</span>, <span class="citation">Barlow, Bogatyrev, and Ismagilov (<a href="#ref-barlow2020aqau" role="doc-biblioref">2020</a>)</span>, <span class="citation">Galazzo et al. (<a href="#ref-galazzo2020howt" role="doc-biblioref">2020</a>)</span>, <span class="citation">Tettamanti Boshier et al. (<a href="#ref-tettamantiboshier2020comp" role="doc-biblioref">2020</a>)</span>), and measuring bulk DNA concentration with a florescence-based DNA quantification method (<span class="citation">Contijoch et al. (<a href="#ref-contijoch2019gutm" role="doc-biblioref">2019</a>)</span>).</p>
<p>Importantly, these methods of measuring total abundance are themselves subject to taxonomic bias.
Flow cytometry may, for example, yield lower cell counts for species whose cells tend to clump together or are prone to lysis during steps involved in sample collection, storage, and preparation.
Marker-gene concentrations measured by qPCR are affected by variation among species in extraction efficiency, marker-gene copy number, and PCR binding and amplification efficiency (<span class="citation">Lloyd et al. (<a href="#ref-lloyd2013meta" role="doc-biblioref">2013</a>)</span>).
We can easily understand the impact of taxonomic bias on total-abundance measurement under simplifying assumptions analogous to those in our MGS model.
Suppose that each species <span class="math inline">\(i\)</span> has an <em>absolute efficiency</em> <span class="math inline">\(B_{i}^{{\text{[tot]}}}\)</span> for the total-abundance measurement that is constant across samples.
Further, let <span class="math inline">\(\bar B^{{\text{[tot]}}(a)}\)</span> be the abundance-weighted average of these efficiencies in sample <span class="math inline">\(a\)</span>—that is, the mean efficiency of the total-abundance measurement.
Neglecting other error sources, the total-abundance measurement equals
<span class="math display" id="eq:total-density-error">\[\begin{align}
  \tag{2.12}
  \tilde A_{\text{tot}}^{(a)}
  &amp;= \sum_{i\in S} A_i^{(a)} B_{i}^{{\text{[tot]}}}
  \\&amp;= A_{\text{tot}}^{(a)} \bar B^{{\text{[tot]}}(a)}.
\end{align}\]</span>
<!-- Note: We have assumed that only species in S contribute to the total abundance measurement. --></p>
<p>Species abundance measurements derived by this method are affected by taxonomic bias in both the MGS and total-abundance measurement.
We can determine the resulting fold error (FE) by substituting Equations <a href="abundance-measurement.html#eq:prop-error">(2.7)</a> and <a href="abundance-measurement.html#eq:total-density-error">(2.12)</a> into Equation <a href="abundance-measurement.html#eq:density-prop-meas">(2.11)</a>, yielding
<span class="math display" id="eq:density-prop-error">\[\begin{align}
  \tag{2.13}
  \tilde A_{\text{tot}}^{(a)}
  = A_{\text{tot}}^{(a)} \cdot \frac{B_i \bar B^{{\text{[tot]}}(a)}}{\bar B^{(a)}}.
\end{align}\]</span>
Equation <a href="abundance-measurement.html#eq:density-prop-error">(2.13)</a> indicates that the FE in the measured absolute abundance of a species equals its MGS efficiency relative to the mean MGS efficiency in the sample, multiplied by the mean efficiency of the total measurement.
As in the case of proportions (Equation <a href="abundance-measurement.html#eq:prop-error">(2.7)</a>), the FE depends on sample composition through the two mean efficiency terms and so will vary across samples unless the two perfectly covary.</p>
</div>
<div id="leveraging-information-about-a-reference-species" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Leveraging information about a reference species<a href="abundance-measurement.html#leveraging-information-about-a-reference-species" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Suppose that the absolute abundance of a <em>reference species</em> <span class="math inline">\(r\)</span> has been fixed by the experimenter or been measured by independent means.
This known or measured abundance <span class="math inline">\(\tilde A_{r}^{(a)}\)</span> can be used in conjunction with the MGS read counts to obtain absolute abundances for all species.
In the absence of taxonomic bias, the ratio of a species’ absolute abundance to its MGS read count is the same for all species in a given sample (Equation <a href="abundance-measurement.html#eq:mgs-model">(2.1)</a>).
Hence the known ratio for the reference species can serve as conversion factor for obtaining the absolute abundance of a species <span class="math inline">\(i\)</span> from its read count,
<span class="math display" id="eq:density-ratio-meas">\[\begin{align}
  \tag{2.14}
  \tilde A_i^{(a)} &amp;= M_i^{(a)} \cdot \frac{\tilde A_r^{(a)}}{M_r^{(a)}}.
\end{align}\]</span>
Let <span class="math inline">\(\mathop{\mathrm{FE}}[\tilde A_r^{(a)}] \equiv {\tilde A_r^{(a)}}/{A_r^{(a)}}\)</span> be the FE in the reference measurement.
The effect on <span class="math inline">\(\tilde A_i^{(a)}\)</span> of taxonomic bias in the MGS measurement can be determined by substituting Equation <a href="abundance-measurement.html#eq:ratio-error">(2.9)</a> into Equation <a href="abundance-measurement.html#eq:density-ratio-meas">(2.14)</a>, yielding
<span class="math display" id="eq:density-ratio-error">\[\begin{align}
  \tag{2.15}
  \tilde A_i^{(a)} = A_i^{(a)} \cdot \frac{B_i}{B_r} \cdot
%    \frac{\tilde A_r^{(a)}}{A_r^{(a)}}.
    \mathop{\mathrm{FE}}\left[\tilde A_r^{(a)}\right].
\end{align}\]</span>
The FE in <span class="math inline">\(\tilde A_i^{(a)}\)</span> consists of two terms: the relative efficiency of species <span class="math inline">\(i\)</span> to species <span class="math inline">\(r\)</span> in the MGS measurement (<span class="math inline">\({B_i}/{B_r}\)</span>) and the FE in the reference species’ abundance (<span class="math inline">\({\tilde A_r^{(a)}}/{A_r^{(a)}}\)</span>).</p>
<p>A common application of this approach involves adding a ‘spike-in’ (as described above) in a known (and typically constant) abundance across samples (<span class="citation">Stämmler et al. (<a href="#ref-stammler2016adju" role="doc-biblioref">2016</a>)</span>, <span class="citation">Ji et al. (<a href="#ref-ji2019quan" role="doc-biblioref">2019</a>)</span>, <span class="citation">Tkacz, Hortala, and Poole (<a href="#ref-tkacz2018abso" role="doc-biblioref">2018</a>)</span>, <span class="citation">Harrison et al. (<a href="#ref-harrison2021theq" role="doc-biblioref">2021</a>)</span>, <span class="citation">Rao et al. (<a href="#ref-rao2021mult" role="doc-biblioref">2021</a>)</span>).
In this case, the reference abundance <span class="math inline">\(\tilde A_r^{(a)}\)</span> is determined from the concentration of the spike-in stock multiplied by the ratio of the spike-in to sample volumes.
(In practice, researchers in spike-in experiments have typically used a more indirect calculation to determine species abundances, but which yields identical results to <a href="abundance-measurement.html#eq:density-ratio-meas">(2.14)</a>; Appendix TODO).</p>
<p>Others have instead sought to determine naturally-occurring species that are thought to be constant across samples; we refer to such species as <em>housekeeping species</em> by analogy with the housekeeping genes used for absolute-abundance conversion in gene-expression studies (<span class="citation">Silver et al. (<a href="#ref-silver2006sele" role="doc-biblioref">2006</a>)</span>).
Housekeeping species can sometimes be identified using prior scientific knowledge; for example, in shotgun sequencing experiments, researchers have used sequencing reads from the plant or animal host as a reference (<span class="citation">Karasov et al. (<a href="#ref-karasov2020ther" role="doc-biblioref">2020</a>)</span>, <span class="citation">Regalado et al. (<a href="#ref-regalado2020comb" role="doc-biblioref">2020</a>)</span>, <span class="citation">Wallace et al. (<a href="#ref-wallace2021thed" role="doc-biblioref">2021</a>)</span>).
A related approach involves computationally identifying species that are constant between pairs of samples (<span class="citation">David et al. (<a href="#ref-david2014host" role="doc-biblioref">2014</a>)</span>) or between sample conditions (<span class="citation">Mandal et al. (<a href="#ref-mandal2015anal" role="doc-biblioref">2015</a>)</span>, <span class="citation">Kumar et al. (<a href="#ref-kumar2018anal" role="doc-biblioref">2018</a>)</span>).
The abundance of a housekeeping species is typically unknown; therefore, to estimate the abundances of other species, we simply set <span class="math inline">\(\tilde A_r^{(a)}\)</span> to 1 in Equation <a href="abundance-measurement.html#eq:density-ratio-meas">(2.14)</a>.
The resulting abundance measurements have unknown but fixed units, which is sufficient for measuring fold changes across samples.</p>
<p>We suggest an additional way of using the reference-species strategy even in the absence of a spike-in or constant species:
Performing targeted measurements of the absolute abundance of one or more naturally occurring species.
These species can then be used as reference species in Equation <a href="abundance-measurement.html#eq:density-ratio-meas">(2.14)</a> to measure the absolute abundances of all species.
The most common form of targeted measurement involves using qPCR or ddPCR to measure the concentration of a marker-gene in the extracted DNA.
It is also possible to directly measure cell concentration by performing ddPCR prior to DNA extraction (<span class="citation">Morella et al. (<a href="#ref-morella2018rapi" role="doc-biblioref">2018</a>)</span>), flow cytometry with species-specific florescent probes, or CFU counting on selective media.
Appendix TODO describes how using multiple reference species and/or statistical modeling can address the fact that any one native reference species is unlikely to be found in all samples.</p>

</div>
</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-barlow2020aqau" class="csl-entry">
Barlow, Jacob T., Said R. Bogatyrev, and Rustem F. Ismagilov. 2020. <span>“<span class="nocase">A quantitative sequencing framework for absolute abundance measurements of mucosal and lumenal microbial communities</span>.”</span> <em>Nat. Commun.</em> 11 (1): 1–13. <a href="https://doi.org/10.1038/s41467-020-16224-6">https://doi.org/10.1038/s41467-020-16224-6</a>.
</div>
<div id="ref-contijoch2019gutm" class="csl-entry">
Contijoch, Eduardo J, Graham J Britton, Chao Yang, Ilaria Mogno, Zhihua Li, Ruby Ng, Sean R Llewellyn, et al. 2019. <span>“<span class="nocase">Gut microbiota density influences host physiology and is shaped by host and microbial factors</span>.”</span> <em>Elife</em> 8 (January). <a href="https://doi.org/10.7554/eLife.40553">https://doi.org/10.7554/eLife.40553</a>.
</div>
<div id="ref-david2014host" class="csl-entry">
David, Lawrence A, Arne C Materna, Jonathan Friedman, Maria I Campos-Baptista, Matthew C Blackburn, Allison Perrotta, Susan E Erdman, and Eric J Alm. 2014. <span>“<span class="nocase">Host lifestyle affects human microbiota on daily timescales</span>.”</span> <em>Genome Biol.</em> 15 (7): R89. <a href="https://doi.org/10.1186/gb-2014-15-7-r89">https://doi.org/10.1186/gb-2014-15-7-r89</a>.
</div>
<div id="ref-galazzo2020howt" class="csl-entry">
Galazzo, Gianluca, Niels van Best, Birke J. Benedikter, Kevin Janssen, Liene Bervoets, Christel Driessen, Melissa Oomen, et al. 2020. <span>“<span class="nocase">How to Count Our Microbes? The Effect of Different Quantitative Microbiome Profiling Approaches</span>.”</span> <em>Front. Cell. Infect. Microbiol.</em> 10 (August). <a href="https://doi.org/10.3389/fcimb.2020.00403">https://doi.org/10.3389/fcimb.2020.00403</a>.
</div>
<div id="ref-gloor2017micr" class="csl-entry">
Gloor, Gregory B., Jean M. Macklaim, Vera Pawlowsky-Glahn, and Juan J. Egozcue. 2017. <span>“<span>Microbiome Datasets Are Compositional: And This Is Not Optional</span>.”</span> <em>Front. Microbiol.</em> 8 (November): 2224. <a href="https://doi.org/10.3389/fmicb.2017.02224">https://doi.org/10.3389/fmicb.2017.02224</a>.
</div>
<div id="ref-harrison2021theq" class="csl-entry">
Harrison, Joshua G., W. John Calder, Bryan Shuman, and C. Alex Buerkle. 2021. <span>“<span class="nocase">The quest for absolute abundance: The use of internal standards for DNA‐based community ecology</span>.”</span> <em>Mol. Ecol. Resour.</em> 21 (1): 30–43. <a href="https://doi.org/10.1111/1755-0998.13247">https://doi.org/10.1111/1755-0998.13247</a>.
</div>
<div id="ref-ji2019quan" class="csl-entry">
Ji, Brian W., Ravi U. Sheth, Purushottam D. Dixit, Yiming Huang, Andrew Kaufman, Harris H. Wang, and Dennis Vitkup. 2019. <span>“<span class="nocase">Quantifying spatiotemporal variability and noise in absolute microbiota abundances using replicate sampling</span>.”</span> <em>Nat. Methods</em> 16 (8): 731–36. <a href="https://doi.org/10.1038/s41592-019-0467-y">https://doi.org/10.1038/s41592-019-0467-y</a>.
</div>
<div id="ref-karasov2020ther" class="csl-entry">
Karasov, Talia L., Manuela Neumann, Alejandra Duque-Jaramillo, Sonja Kersten, Ilja Bezrukov, Birgit Schröppel, Efthymia Symeonidi, et al. 2020. <span>“<span class="nocase">The relationship between microbial population size and disease in the Arabidopsis thaliana phyllosphere</span>.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/828814">https://doi.org/10.1101/828814</a>.
</div>
<div id="ref-kumar2018anal" class="csl-entry">
Kumar, M. Senthil, Eric V. Slud, Kwame Okrah, Stephanie C. Hicks, Sridhar Hannenhalli, and Héctor Corrada Bravo. 2018. <span>“<span class="nocase">Analysis and correction of compositional bias in sparse sequencing count data</span>.”</span> <em>BMC Genomics</em> 19 (1): 799. <a href="https://doi.org/10.1186/s12864-018-5160-5">https://doi.org/10.1186/s12864-018-5160-5</a>.
</div>
<div id="ref-lloyd2020evid" class="csl-entry">
Lloyd, Karen G., Jordan T. Bird, Joy Buongiorno, Emily Deas, Richard Kevorkian, Talor Noordhoek, Jacob Rosalsky, and Taylor Roy. 2020. <span>“<span class="nocase">Evidence for a Growth Zone for Deep-Subsurface Microbial Clades in Near-Surface Anoxic Sediments</span>.”</span> <em>Appl. Environ. Microbiol.</em> 86 (19): 1–15. <a href="https://doi.org/10.1128/AEM.00877-20">https://doi.org/10.1128/AEM.00877-20</a>.
</div>
<div id="ref-lloyd2013meta" class="csl-entry">
Lloyd, Karen G., Megan K. May, Richard T. Kevorkian, and Andrew D. Steen. 2013. <span>“<span class="nocase">Meta-analysis of quantification methods shows that archaea and bacteria have similar abundances in the subseafloor</span>.”</span> <em>Appl. Environ. Microbiol.</em> 79 (24): 7790–99. <a href="https://doi.org/10.1128/AEM.02090-13">https://doi.org/10.1128/AEM.02090-13</a>.
</div>
<div id="ref-mandal2015anal" class="csl-entry">
Mandal, Siddhartha, Will Van Treuren, Richard A. White, Merete Eggesbø, Rob Knight, and Shyamal D. Peddada. 2015. <span>“<span class="nocase">Analysis of composition of microbiomes: a novel method for studying microbial composition</span>.”</span> <em>Microb. Ecol. Heal. Dis.</em> 26 (1): 27663. <a href="https://doi.org/10.3402/mehd.v26.27663">https://doi.org/10.3402/mehd.v26.27663</a>.
</div>
<div id="ref-mclaren2019cons" class="csl-entry">
McLaren, Michael R, Amy D Willis, and Benjamin J Callahan. 2019. <span>“<span class="nocase">Consistent and correctable bias in metagenomic sequencing experiments</span>.”</span> <em>Elife</em> 8 (September): 46923. <a href="https://doi.org/10.7554/eLife.46923">https://doi.org/10.7554/eLife.46923</a>.
</div>
<div id="ref-morella2018rapi" class="csl-entry">
Morella, Norma M., Shangyang Christopher Yang, Catherine A. Hernandez, and Britt Koskella. 2018. <span>“<span class="nocase">Rapid quantification of bacteriophages and their bacterial hosts in vitro and in vivo using droplet digital PCR</span>.”</span> <em>J. Virol. Methods</em> 259 (May): 18–24. <a href="https://doi.org/10.1016/j.jviromet.2018.05.007">https://doi.org/10.1016/j.jviromet.2018.05.007</a>.
</div>
<div id="ref-props2017abso" class="csl-entry">
Props, Ruben, Frederiek-Maarten Kerckhof, Peter Rubbens, Jo De Vrieze, Emma Hernandez Sanabria, Willem Waegeman, Pieter Monsieurs, Frederik Hammes, and Nico Boon. 2017. <span>“<span class="nocase">Absolute quantification of microbial taxon abundances</span>.”</span> <em>ISME J.</em> 11 (2): 584–87. <a href="https://doi.org/10.1038/ismej.2016.117">https://doi.org/10.1038/ismej.2016.117</a>.
</div>
<div id="ref-rao2021mult" class="csl-entry">
Rao, Chitong, Katharine Z. Coyte, Wayne Bainter, Raif S. Geha, Camilia R. Martin, and Seth Rakoff-Nahoum. 2021. <span>“<span class="nocase">Multi-kingdom ecological drivers of microbiota assembly in preterm infants</span>.”</span> <em>Nature</em> 591 (7851): 633–38. <a href="https://doi.org/10.1038/s41586-021-03241-8">https://doi.org/10.1038/s41586-021-03241-8</a>.
</div>
<div id="ref-regalado2020comb" class="csl-entry">
Regalado, Julian, Derek S. Lundberg, Oliver Deusch, Sonja Kersten, Talia Karasov, Karin Poersch, Gautam Shirsekar, and Detlef Weigel. 2020. <span>“<span class="nocase">Combining whole-genome shotgun sequencing and rRNA gene amplicon analyses to improve detection of microbe–microbe interaction networks in plant leaves</span>.”</span> <em>ISME J.</em>, May, 823492. <a href="https://doi.org/10.1038/s41396-020-0665-8">https://doi.org/10.1038/s41396-020-0665-8</a>.
</div>
<div id="ref-silver2006sele" class="csl-entry">
Silver, Nicholas, Steve Best, Jie Jiang, and Swee Lay Thein. 2006. <span>“Selection of Housekeeping Genes for Gene Expression Studies in Human Reticulocytes Using Real-Time <span>PCR</span>.”</span> <em>BMC Mol. Biol.</em> 7 (October): 33. <a href="https://doi.org/10.1186/1471-2199-7-33">https://doi.org/10.1186/1471-2199-7-33</a>.
</div>
<div id="ref-stammler2016adju" class="csl-entry">
Stämmler, Frank, Joachim Gläsner, Andreas Hiergeist, Ernst Holler, Daniela Weber, Peter J. Oefner, André Gessner, and Rainer Spang. 2016. <span>“<span class="nocase">Adjusting microbiome profiles for differences in microbial load by spike-in bacteria</span>.”</span> <em>Microbiome</em> 4 (1): 28. <a href="https://doi.org/10.1186/s40168-016-0175-0">https://doi.org/10.1186/s40168-016-0175-0</a>.
</div>
<div id="ref-tettamantiboshier2020comp" class="csl-entry">
Tettamanti Boshier, Florencia A., Sujatha Srinivasan, Anthony Lopez, Noah G. Hoffman, Sean Proll, David N. Fredricks, and Joshua T. Schiffer. 2020. <span>“<span class="nocase">Complementing 16S rRNA Gene Amplicon Sequencing with Total Bacterial Load To Infer Absolute Species Concentrations in the Vaginal Microbiome</span>.”</span> <em>mSystems</em> 5 (2): 1–14. <a href="https://doi.org/10.1128/mSystems.00777-19">https://doi.org/10.1128/mSystems.00777-19</a>.
</div>
<div id="ref-tkacz2018abso" class="csl-entry">
Tkacz, Andrzej, Marion Hortala, and Philip S. Poole. 2018. <span>“<span class="nocase">Absolute quantitation of microbiota abundance in environmental samples</span>.”</span> <em>Microbiome</em> 6 (1): 110. <a href="https://doi.org/10.1186/s40168-018-0491-7">https://doi.org/10.1186/s40168-018-0491-7</a>.
</div>
<div id="ref-vandeputte2017quan" class="csl-entry">
Vandeputte, Doris, Gunter Kathagen, Kevin D’hoe, Sara Vieira-Silva, Mireia Valles-Colomer, João Sabino, Jun Wang, et al. 2017. <span>“<span class="nocase">Quantitative microbiome profiling links gut community variation to microbial load</span>.”</span> <em>Nature</em> 551 (7681): 507. <a href="https://doi.org/10.1038/nature24460">https://doi.org/10.1038/nature24460</a>.
</div>
<div id="ref-wallace2021thed" class="csl-entry">
Wallace, Megan A., Kelsey A. Coffman, Clément Gilbert, Sanjana Ravindran, Gregory F. Albery, Jessica Abbott, Eliza Argyridou, et al. 2021. <span>“<span class="nocase">The discovery, distribution, and diversity of DNA viruses associated with Drosophila melanogaster in Europe</span>.”</span> <em>Virus Evol.</em> 7 (1): 1–23. <a href="https://doi.org/10.1093/ve/veab031">https://doi.org/10.1093/ve/veab031</a>.
</div>
<div id="ref-zhang2017soil" class="csl-entry">
Zhang, Zhaojing, Yuanyuan Qu, Shuzhen Li, Kai Feng, Shang Wang, Weiwei Cai, Yuting Liang, et al. 2017. <span>“<span class="nocase">Soil bacterial quantification approaches coupling with relative abundances reflecting the changes of taxa</span>.”</span> <em>Sci. Rep.</em> 7 (1): 1–11. <a href="https://doi.org/10.1038/s41598-017-05260-w">https://doi.org/10.1038/s41598-017-05260-w</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-abundance.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/mikemc/differential-abundance-theory/edit/main/abundance-measurement.Rmd",
"text": "Edit"
},
"history": {
"link": "https://github.com/mikemc/differential-abundance-theory/commits/main/abundance-measurement.Rmd",
"text": null
},
"view": {
"link": "https://github.com/mikemc/differential-abundance-theory/blob/main/abundance-measurement.Rmd",
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
