<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | Characterizing and correcting the effect of taxonomic bias in microbial differential-abundance analysis</title>
  <meta name="description" content="References | Characterizing and correcting the effect of taxonomic bias in microbial differential-abundance analysis" />
  <meta name="generator" content="bookdown 0.27.3 and GitBook 2.6.7" />

  <meta property="og:title" content="References | Characterizing and correcting the effect of taxonomic bias in microbial differential-abundance analysis" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | Characterizing and correcting the effect of taxonomic bias in microbial differential-abundance analysis" />
  
  
  

<meta name="author" content="Michael R. McLaren" />
<meta name="author" content="Jacob T. Nearing" />
<meta name="author" content="Amy D. Willis" />
<meta name="author" content="Karen G. Lloyd" />
<meta name="author" content="Benjamin J. Callahan" />


<meta name="date" content="2022-07-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="supplemental-figures.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
    TeX: {extensions: ["cancel.js"]},
});
MathJax.Hub.Config({
    TeX: {
        Macros : {
            md: '^{\\times}\\!/',
        }
    },
});
</script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="abundance-measurement.html"><a href="abundance-measurement.html"><i class="fa fa-check"></i><b>2</b> How bias affects abundance measurements</a>
<ul>
<li class="chapter" data-level="2.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#model-of-mgs-measurement"><i class="fa fa-check"></i><b>2.1</b> Model of MGS measurement</a></li>
<li class="chapter" data-level="2.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#relative-abundance"><i class="fa fa-check"></i><b>2.2</b> Relative abundance</a></li>
<li class="chapter" data-level="2.3" data-path="abundance-measurement.html"><a href="abundance-measurement.html#absolute-abundance"><i class="fa fa-check"></i><b>2.3</b> Absolute abundance</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#leveraging-information-about-total-community-abundance"><i class="fa fa-check"></i><b>2.3.1</b> Leveraging information about total-community abundance</a></li>
<li class="chapter" data-level="2.3.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#leveraging-information-about-a-reference-species"><i class="fa fa-check"></i><b>2.3.2</b> Leveraging information about a reference species</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>3</b> How bias affects DA results</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-abundance.html"><a href="differential-abundance.html#fold-change-between-a-pair-of-samples"><i class="fa fa-check"></i><b>3.1</b> Fold change between a pair of samples</a></li>
<li class="chapter" data-level="3.2" data-path="differential-abundance.html"><a href="differential-abundance.html#regression-analysis-of-many-samples"><i class="fa fa-check"></i><b>3.2</b> Regression analysis of many samples</a></li>
<li class="chapter" data-level="3.3" data-path="differential-abundance.html"><a href="differential-abundance.html#rank-based-analyses"><i class="fa fa-check"></i><b>3.3</b> Rank-based analyses</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="case-studies.html"><a href="case-studies.html"><i class="fa fa-check"></i><b>4</b> Case studies  </a>
<ul>
<li class="chapter" data-level="4.1" data-path="case-studies.html"><a href="case-studies.html#foliar-fungi-experiment"><i class="fa fa-check"></i><b>4.1</b> Foliar fungi experiment</a></li>
<li class="chapter" data-level="4.2" data-path="case-studies.html"><a href="case-studies.html#vaginal-microbiomes-of-pregnant-women"><i class="fa fa-check"></i><b>4.2</b> Vaginal microbiomes of pregnant women</a></li>
<li class="chapter" data-level="4.3" data-path="case-studies.html"><a href="case-studies.html#human-gut-microbiomes"><i class="fa fa-check"></i><b>4.3</b> Human gut microbiomes</a></li>
<li class="chapter" data-level="4.4" data-path="case-studies.html"><a href="case-studies.html#microbial-growth-in-marine-sediments"><i class="fa fa-check"></i><b>4.4</b> Microbial growth in marine sediments</a></li>
<li class="chapter" data-level="4.5" data-path="case-studies.html"><a href="case-studies.html#summary-and-discussion"><i class="fa fa-check"></i><b>4.5</b> Summary and discussion</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="solutions.html"><a href="solutions.html"><i class="fa fa-check"></i><b>5</b> Potential solutions</a>
<ul>
<li class="chapter" data-level="5.1" data-path="solutions.html"><a href="solutions.html#ratio-based-relative-da-analysis"><i class="fa fa-check"></i><b>5.1</b> Ratio-based relative DA analysis</a></li>
<li class="chapter" data-level="5.2" data-path="solutions.html"><a href="solutions.html#calibrate-compositions"><i class="fa fa-check"></i><b>5.2</b> Calibration using community controls</a></li>
<li class="chapter" data-level="5.3" data-path="solutions.html"><a href="solutions.html#choose-absolute-abundance-methods-with-more-stable-fes"><i class="fa fa-check"></i><b>5.3</b> Choose absolute-abundance methods with more stable FEs</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="solutions.html"><a href="solutions.html#use-complementary-mgs-and-total-abundance-measurements"><i class="fa fa-check"></i><b>5.3.1</b> Use complementary MGS and total-abundance measurements</a></li>
<li class="chapter" data-level="5.3.2" data-path="solutions.html"><a href="solutions.html#normalize-to-a-reference-species"><i class="fa fa-check"></i><b>5.3.2</b> Normalize to a reference species</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="solutions.html"><a href="solutions.html#bias-sensitivity-analysis"><i class="fa fa-check"></i><b>5.4</b> Bias sensitivity analysis</a></li>
<li class="chapter" data-level="5.5" data-path="solutions.html"><a href="solutions.html#bias-aware-meta-analysis"><i class="fa fa-check"></i><b>5.5</b> Bias-aware meta-analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="mgs-measurement-details.html"><a href="mgs-measurement-details.html"><i class="fa fa-check"></i><b>A</b> MGS measurement details</a>
<ul>
<li class="chapter" data-level="A.1" data-path="mgs-measurement-details.html"><a href="mgs-measurement-details.html#spike-in-alt"><i class="fa fa-check"></i><b>A.1</b> Alternative method for measuring species absolute abundances from a spike-in</a></li>
<li class="chapter" data-level="A.2" data-path="mgs-measurement-details.html"><a href="mgs-measurement-details.html#multiple-references"><i class="fa fa-check"></i><b>A.2</b> Missing and multiple reference species</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="appendix-regression.html"><a href="appendix-regression.html"><i class="fa fa-check"></i><b>B</b> Linear regression</a>
<ul>
<li class="chapter" data-level="B.1" data-path="appendix-regression.html"><a href="appendix-regression.html#simple-linear-regression"><i class="fa fa-check"></i><b>B.1</b> Simple linear regression</a>
<ul>
<li class="chapter" data-level="B.1.1" data-path="appendix-regression.html"><a href="appendix-regression.html#review-of-simple-linear-regression"><i class="fa fa-check"></i><b>B.1.1</b> Review of simple linear regression</a></li>
<li class="chapter" data-level="B.1.2" data-path="appendix-regression.html"><a href="appendix-regression.html#measurement-error-in-the-response"><i class="fa fa-check"></i><b>B.1.2</b> Measurement error in the response</a></li>
<li class="chapter" data-level="B.1.3" data-path="appendix-regression.html"><a href="appendix-regression.html#specific-application-to-taxonomic-bias"><i class="fa fa-check"></i><b>B.1.3</b> Specific application to taxonomic bias</a></li>
</ul></li>
<li class="chapter" data-level="B.2" data-path="appendix-regression.html"><a href="appendix-regression.html#gamma-poisson-regression"><i class="fa fa-check"></i><b>B.2</b> Gamma-Poisson regression</a>
<ul>
<li class="chapter" data-level="B.2.1" data-path="appendix-regression.html"><a href="appendix-regression.html#background"><i class="fa fa-check"></i><b>B.2.1</b> Background</a></li>
<li class="chapter" data-level="B.2.2" data-path="appendix-regression.html"><a href="appendix-regression.html#inferring-lfcs-in-proportions-with-and-without-bias-correction"><i class="fa fa-check"></i><b>B.2.2</b> Inferring LFCs in proportions with and without bias correction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="C" data-path="diversity-and-mean-efficiency.html"><a href="diversity-and-mean-efficiency.html"><i class="fa fa-check"></i><b>C</b> Alpha diversity and variation in the mean efficiency</a></li>
<li class="chapter" data-level="D" data-path="supplemental-figures.html"><a href="supplemental-figures.html"><i class="fa fa-check"></i><b>D</b> Supplemental figures</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Characterizing and correcting the effect of taxonomic bias in microbial differential-abundance analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered hasAnchor">
<h1>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h1>

<div id="refs" class="references csl-bib-body hanging-indent">
<div class="csl-entry">
Barlow, Jacob T., Said R. Bogatyrev, and Rustem F. Ismagilov. 2020. <span>“<span class="nocase">A quantitative sequencing framework for absolute abundance measurements of mucosal and lumenal microbial communities</span>.”</span> <em>Nat. Commun.</em> 11 (1): 1–13. <a href="https://doi.org/10.1038/s41467-020-16224-6">https://doi.org/10.1038/s41467-020-16224-6</a>.
</div>
<div class="csl-entry">
Brooks, J Paul, David J Edwards, Michael D Harwich, Maria C Rivera, Jennifer M Fettweis, Myrna G Serrano, Robert A Reris, et al. 2015. <span>“<span class="nocase">The truth about metagenomics: quantifying and counteracting bias in 16S rRNA studies</span>.”</span> <em>BMC Microbiol.</em> 15 (1): 66. <a href="https://doi.org/10.1186/s12866-015-0351-6">https://doi.org/10.1186/s12866-015-0351-6</a>.
</div>
<div class="csl-entry">
Callahan, Benjamin J, Daniel B DiGiulio, Daniela S Aliaga Goltsman, Christine L Sun, Elizabeth K Costello, Pratheepa Jeganathan, Joseph R Biggio, et al. 2017. <span>“<span class="nocase">Replication and refinement of a vaginal microbial signature of preterm birth in two racially distinct cohorts of US women</span>.”</span> <em>Proc. Natl. Acad. Sci. U. S. A.</em> 114 (37): 9966–71. <a href="https://doi.org/10.1073/pnas.1705899114">https://doi.org/10.1073/pnas.1705899114</a>.
</div>
<div class="csl-entry">
Cartwright, Charles P., Amanda J. Pherson, Ayla B. Harris, Matthew S. Clancey, and Melinda B. Nye. 2018. <span>“<span class="nocase">Multicenter study establishing the clinical validity of a nucleic-acid amplification–based assay for the diagnosis of bacterial vaginosis</span>.”</span> <em>Diagn. Microbiol. Infect. Dis.</em> 92 (3): 173–78. <a href="https://doi.org/10.1016/j.diagmicrobio.2018.05.022">https://doi.org/10.1016/j.diagmicrobio.2018.05.022</a>.
</div>
<div class="csl-entry">
Conover, W Jay. 2012. <span>“The Rank Transformation-an Easy and Intuitive Way to Connect Many Nonparametric Methods to Their Parametric Counterparts for Seamless Teaching Introductory Statistics Courses.”</span> <em>Wiley Interdiscip. Rev. Comput. Stat.</em> 4 (5): 432–38. <a href="https://doi.org/10.1002/wics.1216">https://doi.org/10.1002/wics.1216</a>.
</div>
<div class="csl-entry">
Contijoch, Eduardo J, Graham J Britton, Chao Yang, Ilaria Mogno, Zhihua Li, Ruby Ng, Sean R Llewellyn, et al. 2019. <span>“<span class="nocase">Gut microbiota density influences host physiology and is shaped by host and microbial factors</span>.”</span> <em>Elife</em> 8 (January). <a href="https://doi.org/10.7554/eLife.40553">https://doi.org/10.7554/eLife.40553</a>.
</div>
<div class="csl-entry">
David, Lawrence A, Arne C Materna, Jonathan Friedman, Maria I Campos-Baptista, Matthew C Blackburn, Allison Perrotta, Susan E Erdman, and Eric J Alm. 2014. <span>“<span class="nocase">Host lifestyle affects human microbiota on daily timescales</span>.”</span> <em>Genome Biol.</em> 15 (7): R89. <a href="https://doi.org/10.1186/gb-2014-15-7-r89">https://doi.org/10.1186/gb-2014-15-7-r89</a>.
</div>
<div class="csl-entry">
Fernandes, Andrew D, Jennifer NS Reid, Jean M Macklaim, Thomas A McMurrough, David R Edgell, and Gregory B Gloor. 2014. <span>“<span class="nocase">Unifying the analysis of high-throughput sequencing datasets: characterizing RNA-seq, 16S rRNA gene sequencing and selective growth experiments by compositional data analysis</span>.”</span> <em>Microbiome</em> 2 (1): 15. <a href="https://doi.org/10.1186/2049-2618-2-15">https://doi.org/10.1186/2049-2618-2-15</a>.
</div>
<div class="csl-entry">
Fettweis, Jennifer M., Myrna G. Serrano, Jamie Paul Brooks, David J. Edwards, Philippe H. Girerd, Hardik I. Parikh, Bernice Huang, et al. 2019. <span>“<span class="nocase">The vaginal microbiome and preterm birth</span>.”</span> <em>Nat. Med.</em> 25 (6): 1012–21. <a href="https://doi.org/10.1038/s41591-019-0450-2">https://doi.org/10.1038/s41591-019-0450-2</a>.
</div>
<div class="csl-entry">
Finucane, Mariel M., Thomas J. Sharpton, Timothy J. Laurent, and Katherine S. Pollard. 2014. <span>“<span class="nocase">A Taxonomic Signature of Obesity in the Microbiome? Getting to the Guts of the Matter</span>.”</span> Edited by Markus M. Heimesaat. <em>PLoS One</em> 9 (1): e84689. <a href="https://doi.org/10.1371/journal.pone.0084689">https://doi.org/10.1371/journal.pone.0084689</a>.
</div>
<div class="csl-entry">
Galazzo, Gianluca, Niels van Best, Birke J. Benedikter, Kevin Janssen, Liene Bervoets, Christel Driessen, Melissa Oomen, et al. 2020. <span>“<span class="nocase">How to Count Our Microbes? The Effect of Different Quantitative Microbiome Profiling Approaches</span>.”</span> <em>Front. Cell. Infect. Microbiol.</em> 10 (August). <a href="https://doi.org/10.3389/fcimb.2020.00403">https://doi.org/10.3389/fcimb.2020.00403</a>.
</div>
<div class="csl-entry">
Gelman, Andrew, Jennifer Hill, and Aki Vehtari. 2020. <em>Regression and Other Stories</em>. Cambridge University Press.
</div>
<div class="csl-entry">
Gill, Christina, Janneke H. H. M. van de Wijgert, Frances Blow, and Alistair C. Darby. 2016. <span>“<span class="nocase">Evaluation of Lysis Methods for the Extraction of Bacterial DNA for Analysis of the Vaginal Microbiota</span>.”</span> Edited by Peter E. Larsen. <em>PLoS One</em> 11 (9): e0163148. <a href="https://doi.org/10.1371/journal.pone.0163148">https://doi.org/10.1371/journal.pone.0163148</a>.
</div>
<div class="csl-entry">
Gloor, Gregory B., Jean M. Macklaim, Vera Pawlowsky-Glahn, and Juan J. Egozcue. 2017. <span>“<span>Microbiome Datasets Are Compositional: And This Is Not Optional</span>.”</span> <em>Front. Microbiol.</em> 8 (November): 2224. <a href="https://doi.org/10.3389/fmicb.2017.02224">https://doi.org/10.3389/fmicb.2017.02224</a>.
</div>
<div class="csl-entry">
Goodrich, Ben, Jonah Gabry, Imad Ali, and Sam Brilleman. 2020. <span>“Rstanarm: <span>Bayesian</span> Applied Regression Modeling via <span>Stan</span>.”</span> <a href="https://mc-stan.org/rstanarm">https://mc-stan.org/rstanarm</a>.
</div>
<div class="csl-entry">
Graspeuntner, Simon, Nathalie Loeper, Sven Künzel, John F. Baines, and Jan Rupp. 2018. <span>“<span class="nocase">Selection of validated hypervariable regions is crucial in 16S-based microbiota studies of the female genital tract</span>.”</span> <em>Sci. Rep.</em> 8 (1): 9678. <a href="https://doi.org/10.1038/s41598-018-27757-8">https://doi.org/10.1038/s41598-018-27757-8</a>.
</div>
<div class="csl-entry">
Greenland, Sander, John Copas, David R. Jones, David Spiegelhalter, Kenneth Rice, Ben Armstrong, Stephen Senn, et al. 2005. <span>“<span class="nocase">Multiple-bias modelling for analysis of observational data</span>.”</span> <em>J. R. Stat. Soc. Ser. A Stat. Soc.</em> 168 (2): 267–306. <a href="https://doi.org/10.1111/j.1467-985X.2004.00349.x">https://doi.org/10.1111/j.1467-985X.2004.00349.x</a>.
</div>
<div class="csl-entry">
Harrison, Joshua G., W. John Calder, Bryan Shuman, and C. Alex Buerkle. 2021. <span>“<span class="nocase">The quest for absolute abundance: The use of internal standards for DNA‐based community ecology</span>.”</span> <em>Mol. Ecol. Resour.</em> 21 (1): 30–43. <a href="https://doi.org/10.1111/1755-0998.13247">https://doi.org/10.1111/1755-0998.13247</a>.
</div>
<div class="csl-entry">
Hevroni, Gur, José Flores-Uribe, Oded Béjà, and Alon Philosof. 2020. <span>“<span class="nocase">Seasonal and diel patterns of abundance and activity of viruses in the Red Sea</span>.”</span> <em>Proc. Natl. Acad. Sci.</em>, November, 202010783. <a href="https://doi.org/10.1073/pnas.2010783117">https://doi.org/10.1073/pnas.2010783117</a>.
</div>
<div class="csl-entry">
Holmes, Susan, and Wolfgang Huber. 2018. <em>Modern Statistics for Modern Biology</em>. Cambridge University Press.
</div>
<div class="csl-entry">
Huttenhower, Curtis, Dirk Gevers, Rob Knight, Sahar Abubucker, Jonathan H. Badger, Asif T. Chinwalla, Heather H. Creasy, et al. 2012. <span>“<span class="nocase">Structure, function and diversity of the healthy human microbiome</span>.”</span> <em>Nature</em> 486 (7402): 207–14. <a href="https://doi.org/10.1038/nature11234">https://doi.org/10.1038/nature11234</a>.
</div>
<div class="csl-entry">
Ji, Brian W., Ravi U. Sheth, Purushottam D. Dixit, Yiming Huang, Andrew Kaufman, Harris H. Wang, and Dennis Vitkup. 2019. <span>“<span class="nocase">Quantifying spatiotemporal variability and noise in absolute microbiota abundances using replicate sampling</span>.”</span> <em>Nat. Methods</em> 16 (8): 731–36. <a href="https://doi.org/10.1038/s41592-019-0467-y">https://doi.org/10.1038/s41592-019-0467-y</a>.
</div>
<div class="csl-entry">
Jian, Ching, Anne Salonen, and Katri Korpela. 2021. <span>“<span class="nocase">Commentary: How to Count Our Microbes? The Effect of Different Quantitative Microbiome Profiling Approaches</span>.”</span> <em>Front. Cell. Infect. Microbiol.</em> 11 (March): 10–12. <a href="https://doi.org/10.3389/fcimb.2021.627910">https://doi.org/10.3389/fcimb.2021.627910</a>.
</div>
<div class="csl-entry">
Karasov, Talia L., Manuela Neumann, Alejandra Duque-Jaramillo, Sonja Kersten, Ilja Bezrukov, Birgit Schröppel, Efthymia Symeonidi, et al. 2020. <span>“<span class="nocase">The relationship between microbial population size and disease in the Arabidopsis thaliana phyllosphere</span>.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/828814">https://doi.org/10.1101/828814</a>.
</div>
<div class="csl-entry">
Kumar, M. Senthil, Eric V. Slud, Kwame Okrah, Stephanie C. Hicks, Sridhar Hannenhalli, and Héctor Corrada Bravo. 2018. <span>“<span class="nocase">Analysis and correction of compositional bias in sparse sequencing count data</span>.”</span> <em>BMC Genomics</em> 19 (1): 799. <a href="https://doi.org/10.1186/s12864-018-5160-5">https://doi.org/10.1186/s12864-018-5160-5</a>.
</div>
<div class="csl-entry">
Leek, Jeffrey T., Robert B. Scharpf, Héctor Corrada Bravo, David Simcha, Benjamin Langmead, W. Evan Johnson, Donald Geman, Keith Baggerly, and Rafael A. Irizarry. 2010. <span>“<span class="nocase">Tackling the widespread and critical impact of batch effects in high-throughput data</span>.”</span> <em>Nat. Rev. Genet.</em> 11 (10): 733–39. <a href="https://doi.org/10.1038/nrg2825">https://doi.org/10.1038/nrg2825</a>.
</div>
<div class="csl-entry">
Leopold, Devin R, and Posy E Busby. 2020. <span>“<span class="nocase">Host Genotype and Colonist Arrival Order Jointly Govern Plant Microbiome Composition and Function</span>.”</span> <em>Curr. Biol.</em> 30 (16): 3260–3266.e5. <a href="https://doi.org/10.1016/j.cub.2020.06.011">https://doi.org/10.1016/j.cub.2020.06.011</a>.
</div>
<div class="csl-entry">
Lloyd, Karen G., Jordan T. Bird, Joy Buongiorno, Emily Deas, Richard Kevorkian, Talor Noordhoek, Jacob Rosalsky, and Taylor Roy. 2020. <span>“<span class="nocase">Evidence for a Growth Zone for Deep-Subsurface Microbial Clades in Near-Surface Anoxic Sediments</span>.”</span> <em>Appl. Environ. Microbiol.</em> 86 (19): 1–15. <a href="https://doi.org/10.1128/AEM.00877-20">https://doi.org/10.1128/AEM.00877-20</a>.
</div>
<div class="csl-entry">
Lloyd, Karen G., Megan K. May, Richard T. Kevorkian, and Andrew D. Steen. 2013. <span>“<span class="nocase">Meta-analysis of quantification methods shows that archaea and bacteria have similar abundances in the subseafloor</span>.”</span> <em>Appl. Environ. Microbiol.</em> 79 (24): 7790–99. <a href="https://doi.org/10.1128/AEM.02090-13">https://doi.org/10.1128/AEM.02090-13</a>.
</div>
<div class="csl-entry">
Lofgren, Lotus A., Jessie K. Uehling, Sara Branco, Thomas D. Bruns, Francis Martin, and Peter G. Kennedy. 2019. <span>“<span class="nocase">Genome‐based estimates of fungal rDNA copy number variation across phylogenetic scales and ecological lifestyles</span>.”</span> <em>Mol. Ecol.</em> 28 (4): 721–30. <a href="https://doi.org/10.1111/mec.14995">https://doi.org/10.1111/mec.14995</a>.
</div>
<div class="csl-entry">
Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. <span>“<span class="nocase">Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</span>.”</span> <em>Genome Biol.</em> 15 (12): 550. <a href="https://doi.org/10.1186/s13059-014-0550-8">https://doi.org/10.1186/s13059-014-0550-8</a>.
</div>
<div class="csl-entry">
Lozupone, Catherine A, Jesse Stombaugh, Antonio Gonzalez, Gail Ackermann, Janet K Jansson, Jeffrey I Gordon, Doug Wendel, Yoshiki Va, and Rob Knight. 2013. <span>“<span class="nocase">Meta-analyses of studies of the human microbiota</span>.”</span> <em>Genome Res.</em>, 1704–14. <a href="https://doi.org/10.1101/gr.151803.112">https://doi.org/10.1101/gr.151803.112</a>.
</div>
<div class="csl-entry">
Mandal, Siddhartha, Will Van Treuren, Richard A. White, Merete Eggesbø, Rob Knight, and Shyamal D. Peddada. 2015. <span>“<span class="nocase">Analysis of composition of microbiomes: a novel method for studying microbial composition</span>.”</span> <em>Microb. Ecol. Heal. Dis.</em> 26 (1): 27663. <a href="https://doi.org/10.3402/mehd.v26.27663">https://doi.org/10.3402/mehd.v26.27663</a>.
</div>
<div class="csl-entry">
Martin, Bryan D., Daniela Witten, and Amy D. Willis. 2020. <span>“<span class="nocase">Modeling microbial abundances and dysbiosis with beta-binomial regression</span>.”</span> <em>Ann. Appl. Stat.</em> 14 (1): 94–115. <a href="https://doi.org/10.1214/19-AOAS1283">https://doi.org/10.1214/19-AOAS1283</a>.
</div>
<div class="csl-entry">
McLaren, Michael R, Amy D Willis, and Benjamin J Callahan. 2019. <span>“<span class="nocase">Consistent and correctable bias in metagenomic sequencing experiments</span>.”</span> <em>Elife</em> 8 (September): 46923. <a href="https://doi.org/10.7554/eLife.46923">https://doi.org/10.7554/eLife.46923</a>.
</div>
<div class="csl-entry">
Morella, Norma M., Shangyang Christopher Yang, Catherine A. Hernandez, and Britt Koskella. 2018. <span>“<span class="nocase">Rapid quantification of bacteriophages and their bacterial hosts in vitro and in vivo using droplet digital PCR</span>.”</span> <em>J. Virol. Methods</em> 259 (May): 18–24. <a href="https://doi.org/10.1016/j.jviromet.2018.05.007">https://doi.org/10.1016/j.jviromet.2018.05.007</a>.
</div>
<div class="csl-entry">
Nixon, Michelle Pistner, Jeffrey Letourneau, Lawrence David, Sayan Mukherjee, and Justin D. Silverman. 2022. <span>“<span class="nocase">A Statistical Analysis of Compositional Surveys</span>.”</span> <em>arXiv</em>, January. <a href="http://arxiv.org/abs/2201.03616">http://arxiv.org/abs/2201.03616</a>.
</div>
<div class="csl-entry">
Props, Ruben, Frederiek-Maarten Kerckhof, Peter Rubbens, Jo De Vrieze, Emma Hernandez Sanabria, Willem Waegeman, Pieter Monsieurs, Frederik Hammes, and Nico Boon. 2017. <span>“<span class="nocase">Absolute quantification of microbial taxon abundances</span>.”</span> <em>ISME J.</em> 11 (2): 584–87. <a href="https://doi.org/10.1038/ismej.2016.117">https://doi.org/10.1038/ismej.2016.117</a>.
</div>
<div class="csl-entry">
Quinn, Thomas P., and Ionas Erb. 2020. <span>“<span class="nocase">Interpretable Log Contrasts for the Classification of Health Biomarkers: a New Approach to Balance Selection</span>.”</span> Edited by Robert G. Beiko. <em>mSystems</em> 5 (2): 1–11. <a href="https://doi.org/10.1128/mSystems.00230-19">https://doi.org/10.1128/mSystems.00230-19</a>.
</div>
<div class="csl-entry">
Rao, Chitong, Katharine Z. Coyte, Wayne Bainter, Raif S. Geha, Camilia R. Martin, and Seth Rakoff-Nahoum. 2021. <span>“<span class="nocase">Multi-kingdom ecological drivers of microbiota assembly in preterm infants</span>.”</span> <em>Nature</em> 591 (7851): 633–38. <a href="https://doi.org/10.1038/s41586-021-03241-8">https://doi.org/10.1038/s41586-021-03241-8</a>.
</div>
<div class="csl-entry">
Regalado, Julian, Derek S. Lundberg, Oliver Deusch, Sonja Kersten, Talia Karasov, Karin Poersch, Gautam Shirsekar, and Detlef Weigel. 2020. <span>“<span class="nocase">Combining whole-genome shotgun sequencing and rRNA gene amplicon analyses to improve detection of microbe–microbe interaction networks in plant leaves</span>.”</span> <em>ISME J.</em>, May, 823492. <a href="https://doi.org/10.1038/s41396-020-0665-8">https://doi.org/10.1038/s41396-020-0665-8</a>.
</div>
<div class="csl-entry">
Rivera-Pinto, J., J. J. Egozcue, V. Pawlowsky-Glahn, R. Paredes, M. Noguera-Julian, and M. L. Calle. 2018. <span>“<span class="nocase">Balances: a New Perspective for Microbiome Analysis</span>.”</span> Edited by Catherine Lozupone. <em>mSystems</em> 3 (4): e00053–18. <a href="https://doi.org/10.1128/mSystems.00053-18">https://doi.org/10.1128/mSystems.00053-18</a>.
</div>
<div class="csl-entry">
Silver, Nicholas, Steve Best, Jie Jiang, and Swee Lay Thein. 2006. <span>“Selection of Housekeeping Genes for Gene Expression Studies in Human Reticulocytes Using Real-Time <span>PCR</span>.”</span> <em>BMC Mol. Biol.</em> 7 (October): 33. <a href="https://doi.org/10.1186/1471-2199-7-33">https://doi.org/10.1186/1471-2199-7-33</a>.
</div>
<div class="csl-entry">
Srinivasan, Sujatha, Noah G. Hoffman, Martin T. Morgan, Frederick A. Matsen, Tina L. Fiedler, Robert W. Hall, Frederick J. Ross, et al. 2012. <span>“<span class="nocase">Bacterial Communities in Women with Bacterial Vaginosis: High Resolution Phylogenetic Analyses Reveal Relationships of Microbiota to Clinical Criteria</span>.”</span> Edited by Adam J. Ratner. <em>PLoS One</em> 7 (6): e37818. <a href="https://doi.org/10.1371/journal.pone.0037818">https://doi.org/10.1371/journal.pone.0037818</a>.
</div>
<div class="csl-entry">
Stämmler, Frank, Joachim Gläsner, Andreas Hiergeist, Ernst Holler, Daniela Weber, Peter J. Oefner, André Gessner, and Rainer Spang. 2016. <span>“<span class="nocase">Adjusting microbiome profiles for differences in microbial load by spike-in bacteria</span>.”</span> <em>Microbiome</em> 4 (1): 28. <a href="https://doi.org/10.1186/s40168-016-0175-0">https://doi.org/10.1186/s40168-016-0175-0</a>.
</div>
<div class="csl-entry">
Stein, Richard R., Vanni Bucci, Nora C. Toussaint, Charlie G. Buffie, Gunnar Rätsch, Eric G. Pamer, Chris Sander, and João B. Xavier. 2013. <span>“<span class="nocase">Ecological Modeling from Time-Series Inference: Insight into Dynamics and Stability of Intestinal Microbiota</span>.”</span> Edited by Christian von Mering. <em>PLoS Comput. Biol.</em> 9 (12): e1003388. <a href="https://doi.org/10.1371/journal.pcbi.1003388">https://doi.org/10.1371/journal.pcbi.1003388</a>.
</div>
<div class="csl-entry">
Tettamanti Boshier, Florencia A., Sujatha Srinivasan, Anthony Lopez, Noah G. Hoffman, Sean Proll, David N. Fredricks, and Joshua T. Schiffer. 2020. <span>“<span class="nocase">Complementing 16S rRNA Gene Amplicon Sequencing with Total Bacterial Load To Infer Absolute Species Concentrations in the Vaginal Microbiome</span>.”</span> <em>mSystems</em> 5 (2): 1–14. <a href="https://doi.org/10.1128/mSystems.00777-19">https://doi.org/10.1128/mSystems.00777-19</a>.
</div>
<div class="csl-entry">
Tkacz, Andrzej, Marion Hortala, and Philip S. Poole. 2018. <span>“<span class="nocase">Absolute quantitation of microbiota abundance in environmental samples</span>.”</span> <em>Microbiome</em> 6 (1): 110. <a href="https://doi.org/10.1186/s40168-018-0491-7">https://doi.org/10.1186/s40168-018-0491-7</a>.
</div>
<div class="csl-entry">
Vandeputte, Doris, Gunter Kathagen, Kevin D’hoe, Sara Vieira-Silva, Mireia Valles-Colomer, João Sabino, Jun Wang, et al. 2017. <span>“<span class="nocase">Quantitative microbiome profiling links gut community variation to microbial load</span>.”</span> <em>Nature</em> 551 (7681): 507. <a href="https://doi.org/10.1038/nature24460">https://doi.org/10.1038/nature24460</a>.
</div>
<div class="csl-entry">
Vieira-Silva, Sara, João Sabino, Mireia Valles-Colomer, Gwen Falony, Gunter Kathagen, Clara Caenepeel, Isabelle Cleynen, Schalk van der Merwe, Séverine Vermeire, and Jeroen Raes. 2019. <span>“<span class="nocase">Quantitative microbiome profiling disentangles inflammation- and bile duct obstruction-associated microbiota alterations across PSC/IBD diagnoses</span>.”</span> <em>Nat. Microbiol.</em> 4 (11): 1826–31. <a href="https://doi.org/10.1038/s41564-019-0483-9">https://doi.org/10.1038/s41564-019-0483-9</a>.
</div>
<div class="csl-entry">
Wallace, Megan A., Kelsey A. Coffman, Clément Gilbert, Sanjana Ravindran, Gregory F. Albery, Jessica Abbott, Eliza Argyridou, et al. 2021. <span>“<span class="nocase">The discovery, distribution, and diversity of DNA viruses associated with Drosophila melanogaster in Europe</span>.”</span> <em>Virus Evol.</em> 7 (1): 1–23. <a href="https://doi.org/10.1093/ve/veab031">https://doi.org/10.1093/ve/veab031</a>.
</div>
<div class="csl-entry">
Wang, Yiwen, and Kim-Anh LêCao. 2020. <span>“<span class="nocase">Managing batch effects in microbiome data</span>.”</span> <em>Brief. Bioinform.</em> 21 (6): 1954–70. <a href="https://doi.org/10.1093/bib/bbz105">https://doi.org/10.1093/bib/bbz105</a>.
</div>
<div class="csl-entry">
Wasserman, Larry. 2004. <em><span class="nocase">All of Statistics</span></em>. Springer Texts in Statistics. New York, NY: Springer New York. <a href="https://doi.org/10.1007/978-0-387-21736-9">https://doi.org/10.1007/978-0-387-21736-9</a>.
</div>
<div class="csl-entry">
Williamson, Brian D, James P Hughes, and Amy D Willis. 2021. <span>“<span class="nocase">A multiview model for relative and absolute microbial abundances</span>.”</span> <em>Biometrics</em>, no. August 2019 (June): biom.13503. <a href="https://doi.org/10.1111/biom.13503">https://doi.org/10.1111/biom.13503</a>.
</div>
<div class="csl-entry">
Yeh, Yi-Chun, David M. Needham, Ella T. Sieradzki, and Jed A. Fuhrman. 2018. <span>“<span class="nocase">Taxon Disappearance from Microbiome Analysis Reinforces the Value of Mock Communities as a Standard in Every Sequencing Run</span>.”</span> <em>mSystems</em> 3 (3): e00023–18. <a href="https://doi.org/10.1128/mSystems.00023-18">https://doi.org/10.1128/mSystems.00023-18</a>.
</div>
<div class="csl-entry">
Yuan, Sanqing, Dora B. Cohen, Jacques Ravel, Zaid Abdo, and Larry J. Forney. 2012. <span>“<span class="nocase">Evaluation of Methods for the Extraction and Purification of DNA from the Human Microbiome</span>.”</span> Edited by Jack Anthony Gilbert. <em>PLoS One</em> 7 (3): e33865. <a href="https://doi.org/10.1371/journal.pone.0033865">https://doi.org/10.1371/journal.pone.0033865</a>.
</div>
<div class="csl-entry">
Zhang, Zhaojing, Yuanyuan Qu, Shuzhen Li, Kai Feng, Shang Wang, Weiwei Cai, Yuting Liang, et al. 2017. <span>“<span class="nocase">Soil bacterial quantification approaches coupling with relative abundances reflecting the changes of taxa</span>.”</span> <em>Sci. Rep.</em> 7 (1): 1–11. <a href="https://doi.org/10.1038/s41598-017-05260-w">https://doi.org/10.1038/s41598-017-05260-w</a>.
</div>
</div>
</div>










            </section>

          </div>
        </div>
      </div>
<a href="supplemental-figures.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/mikemc/differential-abundance-theory/edit/main/references.Rmd",
"text": "Edit"
},
"history": {
"link": "https://github.com/mikemc/differential-abundance-theory/commits/main/references.Rmd",
"text": null
},
"view": {
"link": "https://github.com/mikemc/differential-abundance-theory/blob/main/references.Rmd",
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
