<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>D Theory notes | Differential abundance analysis of taxonomically-biased microbiome measurements</title>
  <meta name="description" content="D Theory notes | Differential abundance analysis of taxonomically-biased microbiome measurements" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="D Theory notes | Differential abundance analysis of taxonomically-biased microbiome measurements" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="D Theory notes | Differential abundance analysis of taxonomically-biased microbiome measurements" />
  
  
  

<meta name="author" content="Michael R. McLaren" />
<meta name="author" content="Jacob T. Nearing" />
<meta name="author" content="Amy D. Willis" />
<meta name="author" content="Karen G. Lloyd" />
<meta name="author" content="Benjamin J. Callahan" />


<meta name="date" content="2022-02-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="appendix-regression.html"/>
<link rel="next" href="supplemental-figures.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
TeX: {extensions: ["cancel.js"]}
});
</script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="abundance-measurement.html"><a href="abundance-measurement.html"><i class="fa fa-check"></i><b>2</b> How taxonomic bias affects abundance measurements</a>
<ul>
<li class="chapter" data-level="2.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#model-of-mgs-measurement"><i class="fa fa-check"></i><b>2.1</b> Model of MGS measurement</a></li>
<li class="chapter" data-level="2.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#relative-abundance"><i class="fa fa-check"></i><b>2.2</b> Relative abundance (proportions and ratios)</a></li>
<li class="chapter" data-level="2.3" data-path="abundance-measurement.html"><a href="abundance-measurement.html#absolute-abundance"><i class="fa fa-check"></i><b>2.3</b> Absolute abundance</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#methods-that-measure-total-community-abundance"><i class="fa fa-check"></i><b>2.3.1</b> Methods that measure total-community abundance</a></li>
<li class="chapter" data-level="2.3.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#methods-that-determine-or-measure-a-reference-species"><i class="fa fa-check"></i><b>2.3.2</b> Methods that determine or measure a reference species</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="abundance-measurement.html"><a href="abundance-measurement.html#summary"><i class="fa fa-check"></i><b>2.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>3</b> DA methods vary in their robustness to bias</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-abundance.html"><a href="differential-abundance.html#fold-changes-between-a-pair-of-samples"><i class="fa fa-check"></i><b>3.1</b> Fold changes between a pair of samples</a></li>
<li class="chapter" data-level="3.2" data-path="differential-abundance.html"><a href="differential-abundance.html#regression-analysis-of-many-samples"><i class="fa fa-check"></i><b>3.2</b> Regression analysis of many samples</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="differential-abundance.html"><a href="differential-abundance.html#todo-rank-based-analyses"><i class="fa fa-check"></i><b>3.2.1</b> [TODO] Rank-based analyses</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="case-studies.html"><a href="case-studies.html"><i class="fa fa-check"></i><b>4</b> Case studies Â </a>
<ul>
<li class="chapter" data-level="4.1" data-path="case-studies.html"><a href="case-studies.html#foliar-fungi-experiment"><i class="fa fa-check"></i><b>4.1</b> Foliar fungi experiment</a></li>
<li class="chapter" data-level="4.2" data-path="case-studies.html"><a href="case-studies.html#vaginal-microbiomes-of-pregnant-women"><i class="fa fa-check"></i><b>4.2</b> Vaginal microbiomes of pregnant women</a></li>
<li class="chapter" data-level="4.3" data-path="case-studies.html"><a href="case-studies.html#human-gut-microbiomes"><i class="fa fa-check"></i><b>4.3</b> Human gut microbiomes</a></li>
<li class="chapter" data-level="4.4" data-path="case-studies.html"><a href="case-studies.html#microbial-growth-in-marine-sediments"><i class="fa fa-check"></i><b>4.4</b> Microbial growth in marine sediments</a></li>
<li class="chapter" data-level="4.5" data-path="case-studies.html"><a href="case-studies.html#summary-1"><i class="fa fa-check"></i><b>4.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="solutions.html"><a href="solutions.html"><i class="fa fa-check"></i><b>5</b> Potential solutions</a>
<ul>
<li class="chapter" data-level="5.1" data-path="solutions.html"><a href="solutions.html#use-ratio-based-relative-abundance-analyses"><i class="fa fa-check"></i><b>5.1</b> Use ratio-based relative-abundance analyses</a></li>
<li class="chapter" data-level="5.2" data-path="solutions.html"><a href="solutions.html#calibrate-compositions"><i class="fa fa-check"></i><b>5.2</b> Calibration using community controls</a></li>
<li class="chapter" data-level="5.3" data-path="solutions.html"><a href="solutions.html#calibration-using-reference-species"><i class="fa fa-check"></i><b>5.3</b> Calibration using reference species</a></li>
<li class="chapter" data-level="5.4" data-path="solutions.html"><a href="solutions.html#choosing-complementary-total-density-and-metagenomics-measurements"><i class="fa fa-check"></i><b>5.4</b> Choosing complementary total-density and metagenomics measurements</a></li>
<li class="chapter" data-level="5.5" data-path="solutions.html"><a href="solutions.html#bias-sensitivity-analysis"><i class="fa fa-check"></i><b>5.5</b> Bias-sensitivity analysis</a></li>
<li class="chapter" data-level="5.6" data-path="solutions.html"><a href="solutions.html#bias-aware-meta-analysis"><i class="fa fa-check"></i><b>5.6</b> Bias-aware meta-analysis</a></li>
<li class="chapter" data-level="5.7" data-path="solutions.html"><a href="solutions.html#todo-summary"><i class="fa fa-check"></i><b>5.7</b> [TODO] Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="model-details.html"><a href="model-details.html"><i class="fa fa-check"></i><b>A</b> Model details</a>
<ul>
<li class="chapter" data-level="A.1" data-path="model-details.html"><a href="model-details.html#mgs-measurement"><i class="fa fa-check"></i><b>A.1</b> MGS measurement</a></li>
<li class="chapter" data-level="A.2" data-path="model-details.html"><a href="model-details.html#total-density-bias"><i class="fa fa-check"></i><b>A.2</b> Taxonomic bias in total-density measurement</a></li>
<li class="chapter" data-level="A.3" data-path="model-details.html"><a href="model-details.html#total-density-ref"><i class="fa fa-check"></i><b>A.3</b> Using reference species for total-density normalization</a></li>
<li class="chapter" data-level="A.4" data-path="model-details.html"><a href="model-details.html#linearity-of-extraction"><i class="fa fa-check"></i><b>A.4</b> Linearity of extraction</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html"><i class="fa fa-check"></i><b>B</b> Review of experimental methods for obtaining absolute densities</a>
<ul>
<li class="chapter" data-level="B.1" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#measurement-of-total-cell-density"><i class="fa fa-check"></i><b>B.1</b> Measurement of total cell density</a></li>
<li class="chapter" data-level="B.2" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#measurement-of-total-dna-density-post-extraction"><i class="fa fa-check"></i><b>B.2</b> Measurement of total DNA density post extraction</a></li>
<li class="chapter" data-level="B.3" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#equivolumetric-protocol"><i class="fa fa-check"></i><b>B.3</b> Equivolumetric protocol</a></li>
<li class="chapter" data-level="B.4" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#housekeeping-species"><i class="fa fa-check"></i><b>B.4</b> Housekeeping species</a></li>
<li class="chapter" data-level="B.5" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#spike-ins"><i class="fa fa-check"></i><b>B.5</b> Spike-ins</a></li>
<li class="chapter" data-level="B.6" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#targeted-measurements"><i class="fa fa-check"></i><b>B.6</b> Targeted measurements</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="appendix-regression.html"><a href="appendix-regression.html"><i class="fa fa-check"></i><b>C</b> Taxonomic bias and linear regression</a>
<ul>
<li class="chapter" data-level="C.1" data-path="appendix-regression.html"><a href="appendix-regression.html#review-of-simple-linear-regression"><i class="fa fa-check"></i><b>C.1</b> Review of simple linear regression</a></li>
<li class="chapter" data-level="C.2" data-path="appendix-regression.html"><a href="appendix-regression.html#measurement-error-in-the-response"><i class="fa fa-check"></i><b>C.2</b> Measurement error in the response</a></li>
<li class="chapter" data-level="C.3" data-path="appendix-regression.html"><a href="appendix-regression.html#specific-application-to-taxonomic-bias"><i class="fa fa-check"></i><b>C.3</b> Specific application to taxonomic bias</a></li>
</ul></li>
<li class="chapter" data-level="D" data-path="theory-notes.html"><a href="theory-notes.html"><i class="fa fa-check"></i><b>D</b> Theory notes</a>
<ul>
<li class="chapter" data-level="D.1" data-path="theory-notes.html"><a href="theory-notes.html#effective-experimental-effort-intuition"><i class="fa fa-check"></i><b>D.1</b> Effective experimental effort: Intuition</a></li>
<li class="chapter" data-level="D.2" data-path="theory-notes.html"><a href="theory-notes.html#absolute-abundance-measurement"><i class="fa fa-check"></i><b>D.2</b> Absolute-abundance measurement</a></li>
</ul></li>
<li class="chapter" data-level="E" data-path="supplemental-figures.html"><a href="supplemental-figures.html"><i class="fa fa-check"></i><b>E</b> Supplemental figures</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Differential abundance analysis of taxonomically-biased microbiome measurements</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="theory-notes" class="section level1" number="10">
<h1><span class="header-section-number">D</span> Theory notes</h1>
<div id="effective-experimental-effort-intuition" class="section level2" number="10.1">
<h2><span class="header-section-number">D.1</span> Effective experimental effort: Intuition</h2>
<p>The effective experimental effort per cell for sample <span class="math inline">\(a\)</span> is
<span class="math display" id="eq:effort">\[\begin{align}
  \tag{D.1}
  \text{effort}_S(a)
    &amp;= \frac{\text{reads}_S(a)}{\text{abun}_S(a) \cdot \text{efficiency}_S(a)} \\
    &amp;= \frac{\sum_{i \in S}\text{reads}_i(a)}{\sum_{i \in S} \text{abun}_i(a) \cdot \text{efficiency}_i(a)};
\end{align}\]</span>
the second form follows from the definition of the mean efficiency.
The effort increases if more sequencing performed on the sample, leading to greater number of reads for all species; and decreases if a larger sample input volume or concentration is used, leading to fewer reads per cell.
Less intuitively, it also decreases with the mean efficiency of the sample.
To help interpret the effort light of this last behavior, we consider how the effort varies across a few illustrative scenarios.</p>
<p><strong>Scenario 1:</strong>
Consider Sample 1 in Figure <a href="abundance-measurement.html#fig:error-proportions">2.1</a>.
Imagine that sequencing reads can be perfectly assigned to the three species; the taxonomic bias is entirely due to steps prior to read assignment, such as DNA extraction and PCR.
Supposing that the sample was sequenced to a read depth of 100, the three species have read counts of <span class="math inline">\((4, 72, 24)\)</span>.
In addition, suppose the total cell concentration in the sample equals 1 in some particular units.
Since the mean efficiency is <span class="math inline">\((1 + 18 + 6) / 3 = 8.\bar 3\)</span>, the effort is <span class="math inline">\(100 / (1 \cdot 8.\bar 3) = 12\)</span>.</p>
<p><strong>Scenario 2:</strong>
Now consider a situation identical to Scenario 1, except that the bioinformatics pipeline is unable to assign reads from Species 3.
In this case, the <em>assigned</em> read counts will be <span class="math inline">\((4, 72, 0)\)</span>, for a total of <span class="math inline">\(76\)</span>, though the total number of sequenced reads remains 100.
Meanwhile, the efficiency of species 3 drops to 0 and the mean efficiency of the sample drops to <span class="math inline">\((1 + 18 + 0) / 3 = 6.\bar 3\)</span>.
The effort is therefore <span class="math inline">\(76 / (1 \cdot 6.\bar 3) = 12\)</span>, the same as in Scenario 1.
<!-- The decline in total reads is exactly offset by the decline in the mean efficiency. -->
<!-- This example illustrates a general pattern: The effort is invariant to removing from or adding to species' assigned reads. TODO: Make precise, and prove --></p>
<p><strong>Scenario 3:</strong>
Now consider a situation identical to Scenario 1, except that Species 3 fails to amplify during the PCR step in a marker-gene sequencing experiment.
But thanks to careful experimental normalization during library preparation, an equal amount of DNA is prepared in the library as in Scenario 1, leading to 100 sequenced reads from the sample.
The measured relative abundances are identical to in Scenario 2, but the read counts are larger, <span class="math inline">\((\approx 5.3, \approx 94.7)\)</span>.
(As in Section <a href="abundance-measurement.html#abundance-measurement">2</a>, we ignore the fact that read counts are discrete.)
The mean efficiency is also the same (<span class="math inline">\(6.\bar 3\)</span>) as in Scenario 2.
The effort is therefore <span class="math inline">\(100 / (1 \cdot 6.\bar 3) \approx 15.8\)</span>, larger than in Scenario 1 and Scenario 2.</p>
<p><strong>Scenario 4:</strong>
Now consider a situation identical to Scenario 1, but supposing Species 3 to be a reference species, such as the host or a spike-in.
Does the effort depend on whether we include the reference species in the species set <span class="math inline">\(S\)</span>?
As we saw in Scenario 1, the effort associated with all three species is 12.
For the two non-reference species alone, the total abundance is <span class="math inline">\(2/3\)</span>, the total read count is <span class="math inline">\(76\)</span>, and the mean efficiency is <span class="math inline">\((1 + 18) / 2 = 9.5\)</span>.
The effort is therefore <span class="math inline">\(76 /(\tfrac{2}{3} \cdot 9.5) = 12\)</span>, the same as in Scenario 1 and Scenario 2.
<!-- Compared to Scenario 2, the change in abundance and mean efficiency offset --></p>
<p>TODO:</p>
<ul>
<li>Make sure this result is general, and not just happening when the three species are all equally abundant.</li>
<li>consider moving secenario 4 to 2 or 3, leaving the scenario where effort changes to last.</li>
<li>note that for proving the general results, it may be best to go back to main model equation, where we can see that if the read count, abundance, and efficiency for at least one species is unchanged, than the effort is unchanged.
<ul>
<li>in scenario 3, the effort must go up since taxa 1 and 2 are unchanged but get more reads.</li>
<li>if we add or remove species, without changing the read count of other species, then the effort must stay the same.</li>
<li>this fact that effort is independent of the taxonomic scope S seems useful and convenient and intuitive.</li>
</ul></li>
</ul>
<p><strong>Summary:</strong>
The behavior of the effective experimental effort across these scenarios makes intuitive sense if we think of it as specifically describing the âwet labâ components of the MGS measurement, up to and including sequencing but excluding bioinformatic analysis of the sequencing data.
The taxonomic scope <span class="math inline">\(S\)</span> is an arbitrary decision made implicitly or explicitly by the researcher, which does not change the wet portion of the experiment; thus <span class="math inline">\(\text{effort}(a)\)</span> is independent of <span class="math inline">\(S\)</span>.
In addition, sequencing reads are routinely excluded during bioinformatic analysis, and such filtering does not change <span class="math inline">\(\text{effort}(a)\)</span>, instead reducing the read counts by reducing the speciesâ efficiencies.
On the other hand, if species drop out prior to sequencing due to low extraction efficiency and/or low PCR efficiency, then achieving the same total read count requires that a greater sample volume or concentration be passed on to subsequent steps.
These adjustments are reflected by a commensurate increase in <span class="math inline">\(\text{effort}(a)\)</span>.</p>
</div>
<div id="absolute-abundance-measurement" class="section level2" number="10.2">
<h2><span class="header-section-number">D.2</span> Absolute-abundance measurement</h2>
<p>Q: How to motivate this âgeneralâ form?
It comes from ignoring bias.</p>
<p>Make clear that R doesnât need to be in S. Perhaps want different sets - a superset of all species, and an set of species that each measurement method considers?</p>
<hr />
<p>Consider this general form for measuring of the absolute abundance of a taxon <span class="math inline">\(I\)</span> from a reference species <span class="math inline">\(R\)</span>,
<span class="math display">\[\begin{align}
  \widehat{\text{abun}}_{I}(a) 
  &amp;= \text{reads}_{I}(a) \cdot \frac{\widehat{\text{abun}}_{R}(a)}{\text{reads}_{R}(a)}.
\end{align}\]</span>
When <span class="math inline">\(R = S\)</span>, this measurement amounts to total-community normalization; when <span class="math inline">\(R=r\)</span>, it amounts to reference-species normalization.
(This form does not encompass using the geometric average of multiple species.)
Under our model, we can rewrite the right-hand-side to obtain
<span class="math display">\[\begin{align}
  \widehat{\text{abun}}_{I}(a) 
  &amp;= \text{abun}_{I}(a) \cdot \frac{\text{efficiency}_{I}(a)}{\text{efficiency}_{R}(a)}
  \cdot \text{FE}\left[\widehat{\text{abun}_R}(a)\right],
\end{align}\]</span>
where <span class="math inline">\(\text{FE}\left[\widehat{\text{abun}_R}(a)\right] = \widehat{\text{abun}_R}(a) / \text{abun}_R(a)\)</span> is the fold error in the reference abundance measurement.</p>
<p>From here, we can derive the other results.</p>
<p>Methods differ based on the choice of <span class="math inline">\(R\)</span> and in how the abundance of <span class="math inline">\(R\)</span> is measured.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="appendix-regression.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="supplemental-figures.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/mikemc/differential-abundance-theory/edit/main/appendix-notes.Rmd",
"text": "Edit"
},
"history": {
"link": "https://github.com/mikemc/differential-abundance-theory/commits/main/appendix-notes.Rmd",
"text": null
},
"view": {
"link": "https://github.com/mikemc/differential-abundance-theory/blob/main/appendix-notes.Rmd",
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
