
# HMP2 IBD study

```{r}
here('notebook/_data/gut') %>% fs::dir_ls()
ps <- here('notebook/_data/gut',
  '2021-03-31.HMP_2019_ibdmdb.relative_abundance.phyloseq.rds'
) %>% readRDS
sam <- ps %>% sample_data %>% as_tibble
tax <- ps %>% tax_table %>% as_tibble %>%
  add_column(sum = taxa_sums(ps))
```

```{r}
sam %>% glimpse
sam %>% dplyr::count(body_site)
sam %>% dplyr::count(dna_extraction_kit)
sam %>% dplyr::count(location)
sam %>% dplyr::count(subject_id) %>% dplyr::count(n)
sam %>% dplyr::count(study_condition, disease, disease_subtype)
```

Most subjects have many samples.

In the HMP2 paper, they classified some samples as dysbiotic, but I dont' think we have that info here.

Let's check out the tax table, and consider filtering some taxa.

```{r}
ps %>%
  tax_glom('phylum') %>%
  as_tibble %>%
  with_groups(c(kingdom, phylum), summarize,
    across(.abundance, list(median = median, mean = mean, max = max, sum = sum),
      .names = '{.fn}'
    ))
```

```{r}
tax %>% pull(sum) %>% sort %>% head
```

## Most abundant species

What is the proportion of the most abundant species in the sample?

```{r}
x <- ps %>% as_tibble
```

```{r}
most_abundant <- x %>%
  with_groups(.sample, slice_max, .abundance, n = 1)
```

```{r}
most_abundant %>%
  ggplot(aes(.abundance)) +
  scale_x_log10() +
  geom_histogram() +
  labs(x = 'Proportion')
```

```{r}
most_abundant %>%
  pull(.abundance) %>%
  summary
```

The most abundant species often forms a large fraction of the community.

Which species are most often most abundant?

```{r}
most_abundant %>%
  count(phylum, species) %>%
  arrange(-n) %>%
  print(n=20)
```

A relatively small set of species constitute the most abundant in most samples, and these are often from the same genera (Bacteroides, Prevotella, Roseburia) or phylum.
These observations suggest there is room for substantial variation in the mean efficiency across samples.


Let's compare to alpha diversity (effective number of species).

```{r}
div <- ps %>% as_tibble %>%
  with_groups(.sample, summarize,
    diversity_q1 = vegan::diversity(.abundance, index = 'shannon') %>% exp,
    diversity_q2 = vegan::diversity(.abundance, index = 'invsimpson')
    #> diversity_q2 = 1/sum(.abundance^2)
  ) %>%
  left_join(most_abundant)
```

```{r}
```

```{r}
div %>%
  ggplot(aes(diversity_q2)) +
  scale_x_log10() +
  geom_histogram() +
  labs(x = 'Diversity (q=2)')
```

```{r}
div %>%
  ggplot(aes(x = .abundance, y = diversity_q2)) +
  scale_x_log10() +
  scale_y_log10(limits = c(1, 30)) +
  geom_line(aes(y = 1/.abundance^2), color = 'grey') +
  geom_point(alpha = 0.5) +
  labs(x = 'Largest proportion', y = 'Diversity (q=2)')
```

Can see that there isn't a crazy amount of diversity - still typically less than 10.

## Variation in phylum proportions

If phyla vary substantially across samples, then any phylum-specific bias would drive variation in the mean efficiency.

I'll restrict plots to just the four main phyla.

```{r}
focal_phyla <- c('Actinobacteria', 'Bacteroidetes', 'Firmicutes',
  'Proteobacteria')
ps_phy <- ps %>% 
  filter_tax_table(kingdom %in% 'Bacteria') %>%
  tax_glom('phylum') %>%
  mutate_tax_table(.otu = phylum) %>%
  transform_sample_counts(close_elts) %>%
  mutate_tax_table(
    gram = case_when(
      phylum %in% c('Actinobacteria', 'Firmicutes') ~ 'Positive',
      TRUE ~ 'Negative'
    )
  )
x_phy <- ps_phy %>% as_tibble
```

```{r}
x_phy %>%
  filter(phylum %in% focal_phyla) %>%
  ggplot(aes(.abundance, fill = disease)) +
  facet_wrap(~phylum, scales = 'fixed', ncol = 1) +
  scale_x_log10() +
  geom_histogram()
```

There is substantial log-scale variation in all phyla, but the Bacteroidetes and Firmicutes are generally much more abundant than the other two.

```{r}
x_phy %>%
  filter(phylum %in% c('Bacteroidetes', 'Firmicutes')) %>%
  ggplot(aes(.abundance + 1e-4, fill = disease_subtype)) +
  facet_wrap(~phylum, ncol = 1) +
  scale_x_log10() +
  geom_histogram()
```

What about the B:F ratio?

```{r}
x_phy %>%
  filter(phylum %in% c('Bacteroidetes', 'Firmicutes')) %>%
  mutate(
    across(.abundance, ~ . + 1e-4),
    ratio_f = .abundance / .abundance[phylum == 'Firmicutes']
  ) %>%
  filter(phylum == 'Bacteroidetes') %>%
  ggplot(aes(ratio_f, fill = age_category)) +
  scale_x_log10(breaks = 10^(-3:4)) +
  facet_wrap(~disease, ncol = 1) +
  geom_histogram() +
  theme_minimal_vgrid()
```

There is a lot of variation, but most if it is generally not associated with disease.
This suggests that in this case, the main impact of variation in mean efficinecy from B:F ratio may be to create noise.

(note, I am ignoring structure in the data such as replicate sampling)

how correlated are species within a phylum?
what is the typical within-phylum diversity?

### Variation in mean efficiency due to phyla 

Suppose that these profiles are biased to favor Gram negative phyla.
We might expect this because of the high Bacteroidetes / Firmicutes ratio and previous findings that the HMP1 protocol had this issue.
(Question: is the HMP2 gut protocol the same as HMP1?)

Suppose a relatively modest bias of 10X.
What variation in the mean efficiency would we see?


```{r}
bias <- ps_phy %>% tax_table %>% as_tibble %>%
  mutate(
    efficiency = case_when(
      gram == 'Positive' ~ 1/10,
      TRUE ~ 1
    )
  ) %>%
  select(.otu, efficiency) %>%
  deframe
ps_phy_cal <- ps_phy %>% calibrate(bias)
```

```{r}
ps_phy_cal %>% 
  sample_data %>%
  as_tibble %>%
  ggplot(aes(.mean_efficiency)) +
  scale_x_log10() +
  geom_histogram()
```

We see sizable variation in the mean efficiency.

```{r}
bias %>% gm_sd
ps_phy_cal %>% sample_data %>% as_tibble %>% pull(.mean_efficiency) %>% gm_sd
```

Though the geometric standard deviation is modest.
Interestingly, if we increase the efficiency difference up from 10X to 100X, the geometric sd only increases from 1.7 to 2.3.
I think this is because the bias being sufficiently strong implies that the (calibrated) communities are dominated by Gram positive and the mean efficiency stabilises at whatever the G+ efficiency is.

## Diversity within phyla

What is the effective number of speices within phyla?
For q=2 (Inverse Simpson index), 

```{r}
div <- x %>% 
  filter(phylum %in% focal_phyla) %>%
  with_groups(c(.sample, phylum), mutate,
    across(.abundance, close_elts)
  ) %>%
  with_groups(c(.sample, phylum), summarize,
    diversity = 1/sum(.abundance^2)
  )
```

```{r}
div %>%
  ggplot(aes(diversity)) +
  scale_x_log10() +
  facet_wrap(~phylum, ncol = 1) +
  geom_histogram() +
  theme_minimal_vgrid()
```

```{r}
div %>%
  with_groups(phylum, summarize,
    qs = quantile(diversity, c(0.25, 0.5, 0.75), na.rm = TRUE), 
    prob = c(0.25, 0.5, 0.75)
  ) %>%
  pivot_wider(names_from = prob, values_from = qs)
```

We can see that there is typically diversity within phyla within an individual sample.

## Phylogenetic conservation

Question: How does the variation in mean efficiency across samples vary, if we increase phylogenetic variation in bias?

More generally, what does the variation in mean efficiency look like if we allow it to vary across species?
It may vary more than in the simple G+/G- case done above.


idea:

- fix the variation among species efficiencies, but vary a parameter that determines the phylogenetic conservation
    - Simpler: Do a sim where efficiencies are iid, and a sim where they come from the vcv matrix
- for each value of that parameter, generate 100 bias vectors, and compute the stnadard deviation in log mean efficiency.

Question: How does the variation in mean efficiency scale with the variation in simulated efficiencies? I would guess it would directly scale.

We can also consider a model where there is a correlation at the Gram-type level.


### Setup

To reduce the number of samples, let's grab just a couple samples per person.

Also subset to just bacteria - make sure to remove the sample that doesn't have bacterial taxa.

And drop some low prev taxa.

```{r}
```

```{r}
smpls <- sam %>%
  with_groups(subject_id, slice_min, visit_number, n = 3) %>%
  pull(.sample)

ps_focal <- ps %>%
  filter_tax_table(kingdom == 'Bacteria') %>%
  filter_taxa2(~sum(.>0) >= 10) %>%
  filter_sample_data(.sample %in% smpls) %>%
  transform_sample_counts(close_elts) %>%
  prune_samples(!is.na(sample_sums(.)), .) %>% # to drop samples w/o bacteria
  mutate_tax_table(.otu = species) %>%
  mutate_tax_table(
    gram = case_when(
      phylum %in% c('Actinobacteria', 'Firmicutes') ~ 'Positive',
      TRUE ~ 'Negative'
    )
  )

tree <- ps_focal %>% phy_tree 
phy_vcv <- ape::vcv(tree)
```

```{r}
diff_type <- ps_focal %>%
  tax_table %>%
  as_tibble %>%
  pull(gram) %>%
  {. == 'Positive'} %>%
  dist() %>%
  as('matrix')
same_type <- diff_type %>%
  {1 - .}
gram_vcv <- (same_type * 0.8) + (diff_type * -0.8)
diag(gram_vcv) <- 1
#> gram_vcv <- matrix(0, nrow = ntaxa(ps_focal), ncol = ntaxa(ps_focal))
rownames(gram_vcv) <- colnames(gram_vcv) <- taxa_names(ps_focal)
```

Idea: Fix the variance in species efficiencies, and simulate either as IID or with the given covariance.
The simplest way to do so is to sample the bias vector, then scale to the desired (geometric) standard deviation.

```{r}
# Returns a matrix with taxa as columns
sample_bias <- function(n, gsd, vcv, taxa_as = 'cols') {
  stopifnot(isSymmetric(vcv))
  stopifnot(n > 1)
  n_taxa <- nrow(vcv)
  gsd_log <- log(gsd)
  bias <- MASS::mvrnorm(n, mu = rep(0, n_taxa), Sigma = vcv) %>%
    t %>%
    scale %>%
    {. * gsd_log} %>%
    exp
  if (taxa_as == 'cols')
    bias <- t(bias)
  bias
}
#> x <- sample_bias(3, gsd = 3, phy_vcv)
#> x %>% apply(1, gm_sd) %>% head
#> x[1,] %>% qplot + scale_x_log10()
#> sample_bias(3, gsd = 3, phy_vcv, taxa_as = 'rows') %>% head
```

Now, generate R bias vectors of each type, and calibrate the data by them.

```{r}
set.seed(42)
R <- 300
gsd <- 3
reps <- list(
  iid = diag(ntaxa(ps_focal)),
  vcv = ps_focal %>% phy_tree %>% ape::vcv(),
  gram = gram_vcv
) %>%
  map_dfr(.id = 'sim_type',
    ~sample_bias(n = R, gsd = gsd, vcv = .x) %>%
      split(., row(.)) %>%
      enframe(".idx", "bias")
  ) %>%
  mutate(
    ps = map(bias, ~calibrate(ps_focal, .x))
  )
  # TODO: cache this, since it is a bit slow (due to large number of samples)
```

now, for each phyloseq object, compute the variation in the log mean efficiency, and its variation with disease state.

```{r}
reps_me <- reps %>%
  mutate(
    ps_sam = map(ps, ~sample_data(.) %>% as_tibble)
  ) %>%
  select(-bias, -ps) %>%
  unnest(ps_sam)
```

```{r}
reps_me_summ <- reps_me %>%
  with_groups(c(sim_type, .idx), summarize,
    across(.mean_efficiency, gm_sd)
  )
reps_me_summ %>%
  ggplot(aes(y = sim_type, x = .mean_efficiency)) +
  #> theme_minimal_hgrid() +
  scale_x_log10() +
  #> expand_limits(x = c(1, 3)) +
  geom_vline(xintercept = c(1,3), color = 'grey') +
  geom_boxplot(outlier.shape = NA) +
  geom_quasirandom(groupOnX = FALSE, alpha = 0.5) +
  labs(x = 'gm. std. of mean efficiency')
  # stat_slab(slab_type = 'histogram', breaks = 20)
```

As expected, the GSD's are larger under the VCV model; however, the differences is fairly small.
In particular, the median GSD isn't much larger, though the increase in the upper tail is a bit larger.
Perhaps the explanation has to do with our observation that the dominant species already often forms a large fraction of the community, and the bulk of the mean efficiency variation comes from the dominant few species swapping around, rather than of higher-level taxa.

Or perhaps it has something to do with my enforcing a constant gm-std across all taxa, and the structure of the phylogeny.


## Enterotypes? TODO

Work of others suggests we might be able to identify just a handul of species/genera/groups that tend to be dominant.

```{r}

```


what taxa tend to reach dominance/very high abundance?

```{r}
ps.gen <- ps %>% tax_glom('genus')
```

```{r}
ps.gen %>%
  transform_sample_counts(close_elts) %>%
  as_tibble %>%
  filter(.abundance > 0.3) %>%
  dplyr::count(phylum, family, genus) %>%
  arrange(-n) %>%
  print(n=20)
```

```{r}
ps %>%
  tax_glom('family') %>%
  transform_sample_counts(close_elts) %>%
  as_tibble %>%
  filter(.abundance > 0.3) %>%
  dplyr::count(phylum, family) %>%
  arrange(-n) %>%
  print(n=20)
```


This suggests that when we are considering phylogenetic conservation for the purposes of bias effect, then conservation within firmicutes may not be as important as that within specific sub-groups like Lachnospiraceae and Ruminococcaceae.

Note, the original HMP protocol was found to be biased in favor of Bacteroidetes.


# MOVE Thomas  et al

> Public metagenomic cohorts of patients with CRC, adenomas and controls. We downloaded five public fecal shotgun CRC datasets covering samples from six different countries, totaling 313patients with CRC, 143 adenomas and 308 controls (Table 1), and now available in curatedMetagenomicData27. We manually curated metadata tables for the public cohorts according to the curatedMetagenomicData27 Rpackage grammatical rules. The metadata table includes ten fields (sampleID, subjectID, body_site, country, sequencing_platform, PMID, number_reads, number_bases, minimum_read_length, median_read_length) that are mandatory for all datasets, in addition to other fields that are dataset-specific.


Check to see if I can get these, and use them for the analysis.


### What are the drivers of mean efficiency variation?

Hypothesis: The most abundant species in each sample can mostly explain the variation.


```{r, eval = F}
# zero out each sample to just the top n taxa, then compute the mean efficiency
# (without closing out? or closing out?)
mean_efficiency_top <- function(x, bias, type, n = 1) {


}
```


```{r, eval = F}
reps_me1 <- reps %>%
  mutate(
    mean_efficiency = map(bias, ~mean_efficiency(ps_gv, .x, type = 'actual')),
    across(mean_efficiency, map, enframe, '.sample', 'mean_efficiency')
    mean_efficiency_top = map(bias, ~mean_efficiency_top(ps_gv, .x, type = 'actual')),
    across(mean_efficiency_top, map, enframe, '.sample', 'mean_efficiency_top')
  ) %>%
  select(-bias) %>%
  unnest(c(mean_efficiency, mean_efficiency_top)) %>%
  left_join(ps_gv %>% sample_data %>% as_tibble, by = '.sample')
```


# Variation in species

We also need to choose what species to consider as candidates for a DA analysis in the given ecosystem.

Let's get a lightly filtered ps object for each type, and check out the prevalence patterns.

```{r}
ps_list <- list(
  stool = ps_gv %>% filter_sample_data(body_site == 'stool'),
  vagina = ps_gv %>% filter_sample_data(body_site == 'vagina')
) %>%
  map(filter_taxa2, ~ sum(. > 0) >= 2) %>%
  map(orient_taxa, as = 'cols')
ps_list %>% map_int(nsamples)
ps_list %>% map_int(ntaxa)
```

```{r, eval = T}
x <- ps_list %>%
  map_dfr(as_tibble)

min_vals <- c(0, 1e-5, 1e-4, 1e-3, 1e-2, 1e-1)

prev <- x %>%
  with_groups(c(.otu, body_site), summarize,
    min = min_vals,
    prevalence = map_dbl(min_vals, ~mean(.abundance > .x))
  )
```

```{r}
prev %>%
  ggplot(aes(x = prevalence, y = as.factor(min), 
      fill = body_site, color = body_site)) +
  scale_x_log10() +
  geom_density_ridges(stat = 'binline', alpha = 0.3, scale = 0.95)
```

It's a bit hard to know what threshold to use to pick a relevant set of taxa.

```{r}
prev %>%
  filter(min < 1e-3) %>%
  ggplot(aes(x = prevalence, y = body_site,
      fill = body_site, color = body_site)) +
  facet_wrap(~min) +
  geom_density_ridges(stat = 'binline', alpha = 0.3, scale = 0.95)
```
