---
title: "Compare impact of bias in HMP stool and vagina samples"
description: This report compares gut and vaginal microbiome profiles from the Human Microbiome Project to consider how the different ecological dynamics in each sample type might modulate the potential for bias to distort FCs in species proportions.
author:
  - name: Michael R. McLaren
    url: {}
date: 2022-01-30
draft: false
bibliography: ../../../main.bib
output:
  distill::distill_article:
    self_contained: false
    dev: svg
    toc: true
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  cache = TRUE,
  autodep = TRUE,
  cache.comments = FALSE,
  dpi = 300
)
```

# Setup

```{r}
library(here)
library(tidyverse)
library(speedyseq)
library(metacal)

library(cowplot)
library(patchwork)
theme_set(theme_cowplot())
library(ggbeeswarm)
library(ggdist)
library(ggridges)

library(fs)
dir_create('_output')
```

First, I load a phyloseq object with the Metaphlan3 profiles from the curatedMetagenomicData R package.
This phyloseq object was created in `notebook/_code/import-gut-datasets.Rmd`.
Filtering: I filter to just Bacteria, drop samples with no bacterial abundance, and drop species that don't appear in at least two samples.
I then renormalize to proportions.
I also simplify the OTU names to just the species name.

Then, I compute the most abundant species and its proportion, and add these to the sample data.
In addition, I add the plug-in estimates of three alpha-diversity metrics to the sample data.
These are the hill numbers of order 0, 1, and 2, equivalent to richness, the exponential of Shannon entropy, and Inverse Simpson index, respectively.

```{r}
#> here('notebook/_data/gut') %>% fs::dir_ls()
ps <- here('notebook/_data/gut',
  '2021-03-31.HMP_2012.relative_abundance.phyloseq.rds'
) %>% readRDS %>%
  filter_tax_table(kingdom == 'Bacteria') %>%
  prune_samples(sample_sums(.) > 0, .) %>% # to drop samples w/o bacteria %>%
  filter_taxa2(~sum(.>0) >= 2) %>%
  transform_sample_counts(close_elts) %>%
  mutate_tax_table(.otu = species)
# most abundant species and diversity
x <- ps %>% as_tibble
most_abundant <- x %>%
  with_groups(.sample, slice_max, .abundance, n = 1) %>%
  select(.sample, top_species = species, top_species_prop = .abundance)
fns <- list(
  #> richness = ~sum(.x > 0),
  diversity_q0 = ~sum(.x > 0),
  diversity_q1 = ~vegan::diversity(.x, index = 'shannon') %>% exp,
  diversity_q2 = ~vegan::diversity(.x, index = 'invsimpson')
) %>%
  map(as_mapper)
mat <- ps %>% otu_table %>% orient_taxa(as = 'cols') %>% as('matrix')
div <- fns %>%
  map_dfc(~apply(mat, 1, .x))
div1 <- bind_cols(most_abundant, div)
#  join w/ ps object
ps <- ps %>%
  left_join_sample_data(div1, by = '.sample')
rm(most_abundant, div, div1, x)
```

Let's check the breakdown of samples by body site and subsite,

```{r}
sam <- ps %>% sample_data %>% as_tibble
sam %>% count(body_site) %>% knitr::kable()
sam %>% count(body_site, body_subsite) %>% knitr::kable()
```

Since only the posterior fornix of the vaginal samples has more than a few samples, we'll restrict to just that subtype in what follows, along with the stool type/subtype.

```{r}
ps_gv <- ps %>%
  filter_sample_data(body_subsite %in% c('stool', 'posterior_fornix')) %>%
  filter_taxa2(~ sum(.) > 0)
sam <- ps_gv %>% sample_data %>% as_tibble
```

# Exploratory plots and stats

Distribution of alpha diversity across samples

```{r diversity, fig.dim = c(6,3.5)}
sam %>%
  pivot_longer(starts_with('diversity')) %>%
  mutate(
    q = str_extract(name, '[0-2]'),
    index = str_glue('q = {q}')
  ) %>%
  ggplot(aes(y = body_site, x = value)) +
  scale_x_log10() +
  facet_grid(.~index, scales = 'free', space = 'free') +
  theme_ridges(grid = FALSE, center_axis_labels = TRUE) +
  scale_y_discrete(expand = expansion(mult = c(0.01, .7))) +
  geom_density_ridges(stat = 'binline', scale = 0.90) +
  labs(y = 'Body site', x = 'Diversity')
```

Distribution of the proportion of the most abundant species in each sample,

```{r most-abundant, fig.dim = c(5, 3.5)}
sam %>%
  ggplot(aes(y = body_site, top_species_prop)) +
  scale_x_log10() +
  theme_ridges(grid = FALSE, center_axis_labels = TRUE) +
  expand_limits(x = 0.1) +
  scale_y_discrete(expand = expansion(mult = c(0.01, .7))) +
  geom_density_ridges(stat = 'binline', scale = 1.90) +
  labs(x = 'Proportion of most abundant species', y = 'Body site')
```

Which species tend to be most abundant?

```{r}
make_table <- function(x) {
  x %>%
    mutate(
      across(top_species, fct_infreq),
      across(top_species, fct_lump_n, n = 5),
    ) %>%
    count(top_species) %>%
    mutate(frac = close_elts(n)) %>%
    knitr::kable(digits = 2)
}

# Vagina
sam %>% filter(body_site == 'vagina') %>% make_table

# Stool
sam %>% filter(body_site == 'stool') %>% make_table
```

In both cases, a small number of closely related species dominant most samples.
Notably, we do not see Gardnerella or Lactobacillus BVAB1 in the vaginal samples, or any Firmicutes or other non-Bacteroidetes in the gut samples.
It is unclear whether this is due to the cohorts or bias in the protocol.

## Breakdown of total diversity (gamma) into alpha and beta components

This is more experimental.
We can use the hillR package to compute alpha, beta, and gamma diversity as Hill numbers (effective number of species) for q = 0 (richness), q = 1 (exponential of Shannon entropy), and q = 2 (Inverse Simpson index).
Note: Gamma diversity for q=0 (maybe also for other q) will increase with the number of samples, and so we'd expect it to be larger for stool simply due to the larger number of samples unless we subsample to a fixed number of samples for each body site.
q=0 is also sensitive to read depth, which we don't really have a way to account for here.
Since we have a phylogeny, we can compute phylogenetic diversity as well as species diversity.

First, get equal numbers of stool and vaginal (posterior fornix) samples.

```{r}
sam %>% count(body_site, body_subsite)
ps_list <- list(
  stool = ps_gv %>% filter_sample_data(body_site == 'stool'),
  vagina = ps_gv %>% filter_sample_data(body_site == 'vagina')
) %>%
  # subsample to equal number of samples
  map(filter_sample_data, row_number() <= 62) %>%
  map(filter_taxa2, ~sum(.) > 0) %>%
  map(orient_taxa, as = 'cols')
```

Then compute the diversity breakdown for each body site,

```{r}
# Function that, given a phyloseq object, computes phylogenetic and species
# diversity measures for the requested q values
hill <- function(ps, q) {
  ps <- ps %>% filter_taxa2(~sum(.) > 0) %>% orient_taxa(as = 'cols')
  suppressMessages(
    tibble(q = !!q) %>%
      mutate(
        div_taxa = map(q, ~hillR::hill_taxa_parti(otu_table(ps), .x)),
        div_phylo = map(q, ~hillR::hill_phylo_parti(otu_table(ps), phy_tree(ps), .x)),
      ) %>%
      unnest(c(div_taxa, div_phylo), names_repair = 'universal') %>%
      select(q = q...1, starts_with(c('TD', 'PD')))
  )
  }
div <- ps_list %>%
  map_dfr(hill, q = c(0, 1, 2), .id = 'body_site')
```


```{r}
div %>% knitr::kable(digits = 1)
```

```{r}
div %>%
  pivot_longer(starts_with(c('TD', 'PD'))) %>%
  pivot_wider(names_from = body_site) %>% 
  knitr::kable(digits = 1)
```

## Order-2 diversity

What is the GM, GSD, and GSE of order-2 diversity in each body site?

```{r}
div_q2_stats <- sam %>%
  with_groups(body_site, summarize, 
    gm = gm_mean(diversity_q2),
    gsd = gm_sd(diversity_q2),
    gse = gsd ^ (1 / sqrt(n()))
  )
div_q2_stats %>%
  knitr::kable(digits = 2)
```

We'll save this for use in the manuscript,

```{r}
write_csv(div_q2_stats, path('_output', 'div_q2_stats.csv'))
```

# Bias simulation

To assess the relative importance of bias for proportion-based DA analyses in the two ecosystems, we will consider the variation in the mean efficiency across samples for a large number of possible taxonomic biases, under the assumption that the measured profiles reflected the truth.

The following function samples random bias vectors from a multivariate lognormal distribution with a given covariance structure and geometric standard deviation across the efficiencies.

```{r}
# Returns a matrix with taxa as columns
sample_bias <- function(n, gsd, vcv, taxa_as = 'cols') {
  stopifnot(isSymmetric(vcv))
  stopifnot(n > 1)
  n_taxa <- nrow(vcv)
  bias <- MASS::mvrnorm(n, mu = rep(0, n_taxa), Sigma = vcv) %>%
    t %>%
    scale %>%
    {. * log(gsd)} %>%
    exp
  if (taxa_as == 'cols')
    bias <- t(bias)
  bias
}
#> x <- sample_bias(3, gsd = 3, phy_vcv)
#> x %>% apply(1, gm_sd) %>% head
#> x[1,] %>% qplot + scale_x_log10()
#> sample_bias(3, gsd = 3, phy_vcv, taxa_as = 'rows') %>% head
```

To choose a GSD for the simulations, let's examine the bias estimated by @mclaren2019cons for the @costea2017towa Phase 2 experimental protocols,

```{r}
bias_costea <- '_data/mclaren2019cons-table2-mod.tsv' %>% read_tsv
bias_costea_gsd <- bias_costea %>%
  summarize(across(-Taxon, gm_sd))
bias_costea_gsd %>%
  knitr::kable(digits = 1)
```

The strength of bias as quantified by the GSD in species efficiencies is much larger for protocol H than the other two protocols.
The GM of the GSD of the three protocols is `r bias_costea_gsd %>% select(H, Q, W) %>% unlist %>% gm_mean %>% round(2)`.
Let's round this to a simple whole number 4 and take this as the GSD for our simulations.

For the IID and phylogenetic-covariance models, simulate 1000 replicate bias vectors.

```{r, eval = T}
set.seed(42)
R <- 1000
gsd_sim <- 4
reps <- list(
  iid = diag(ntaxa(ps_gv)),
  vcv = ps_gv %>% phy_tree %>% ape::vcv()
  #> gram = gram_vcv
) %>%
  map_dfr(.id = 'sim_type',
    ~sample_bias(n = R, gsd = gsd_sim, vcv = .x) %>%
      split(., row(.)) %>%
      enframe(".idx", "bias")
  )
```

For each replicate, compute the mean efficiency of each sample, assuming that the observed profiles are the truth and the simulated vector is the bias.
Summarize the variation in the mean efficiency for the replicate by the GSD.

```{r}
reps_me <- reps %>%
  mutate(
    mean_efficiency = map(bias, ~mean_efficiency(ps_gv, .x, type = 'actual')),
    across(mean_efficiency, map, enframe, '.sample', 'mean_efficiency')
  ) %>%
  select(-bias) %>%
  unnest(mean_efficiency) %>%
  left_join(ps_gv %>% sample_data %>% as_tibble, by = '.sample')
reps_me_summ <- reps_me %>%
  with_groups(c(sim_type, body_site, .idx), summarize,
    across(mean_efficiency, gm_sd)
  )
```

```{r, include = F, eval = F}
reps_me_summ %>%
  ggplot(aes(y = sim_type, x = mean_efficiency)) +
  facet_wrap(~body_site, ncol = 1) +
  scale_x_log10() +
  geom_vline(xintercept = c(1, gsd_sim), color = 'grey') +
  geom_violin(draw_quantiles = c(0.5)) +
  # geom_boxplot(outlier.shape = NA) +
  # geom_quasirandom(groupOnX = FALSE, alpha = 0.2) +
  labs(x = 'GSD of mean efficiency', y = 'Simulation type')
```

```{r}
reps_me_summ %>%
  ggplot(aes(y = sim_type, x = mean_efficiency)) +
  facet_wrap(~body_site, ncol = 1) +
  scale_x_log10() +
  geom_vline(xintercept = c(1,gsd_sim), color = 'grey') +
  theme_ridges(grid = FALSE, center_axis_labels = TRUE) +
  scale_y_discrete(expand = expansion(mult = c(0.01, .7))) +
  stat_density_ridges(quantile_lines = TRUE, quantiles = c(0.5), scale = 0.95) +
  labs(x = 'GSD of mean efficiency', y = 'Simulation type')
```

Note, despite always forcing a GSD of 3 on the efficiencies, we sometimes see GSDs in the mean efficiency that are larger than 3, particularly in the vaginal communities.
This makes some sense: We expect the variation in the efficiency of vaginal communities to be driven by a just a few species, and the there is a non-negligible chance that they end up with a much higher/lower mean efficiency than typical.

Question: How does the mean efficiency GSD scale with the efficiency GSD?
When we compare this variation to that in the species proportions, the scale/value of GSD will matter.
However, it might not matter for comparing to the vaginal MB.

How much greater is the GM of the GSD in the vaginal comms versus stool?

```{r}
reps_me_summ %>%
  with_groups(c(body_site, sim_type), summarize, 
    across(mean_efficiency, gm_mean)) %>%
  pivot_wider(names_from = body_site, values_from = mean_efficiency) %>%
  mutate(vagina/stool) %>%
  knitr::kable(digits = 2)
```

Using the median instead of the GM should give a similar answer,

```{r}
reps_me_summ %>%
  with_groups(c(body_site, sim_type), summarize, 
    across(mean_efficiency, median)) %>%
  pivot_wider(names_from = body_site, values_from = mean_efficiency) %>%
  mutate(vagina/stool) %>%
  knitr::kable(digits = 2)
```

Note that the type of simulation has a negligible impact on these numbers.

Let's compute the GM, GSD, and GSE of the GM for the difference (ratio) between gut and vaginal samples.

```{r}
reps_me_summ %>%
  with_groups(c(body_site, sim_type), summarize, 
    GM = gm_mean(mean_efficiency),
    GSD = gm_sd(mean_efficiency),
    # GSE = exp(sd(log(mean_efficiency)) / sqrt(n())),
    GSE = GSD ^ (1 / sqrt(n()))
  ) %>%
  knitr::kable(digits = 4)
```

And if the difference,

```{r}
reps_me_summ_diff <- reps_me_summ %>%
  pivot_wider(names_from = body_site, values_from = mean_efficiency) %>%
  mutate(ratio = vagina / stool) %>%
  with_groups(c(sim_type), summarize, 
    gm = gm_mean(ratio),
    gsd = gm_sd(ratio),
    gse = gsd ^ (1 / sqrt(n()))
  )
reps_me_summ_diff %>%
  knitr::kable(digits = 3)
```

```{r}
write_csv(reps_me_summ_diff , path('_output', 'reps_me_summ_diff.csv'))
```

Let's also check the distribution of the ratio (geometric difference),

```{r}
reps_me_summ %>%
  pivot_wider(names_from = body_site, values_from = mean_efficiency) %>%
  mutate(ratio = vagina / stool) %>%
  ggplot(aes(y = sim_type, x = ratio)) +
  scale_x_log10() +
  theme_ridges(grid = FALSE, center_axis_labels = TRUE) +
  scale_y_discrete(expand = expansion(mult = c(0.01, .7))) +
  stat_density_ridges(quantile_lines = TRUE, quantiles = c(0.5), scale = 0.95) +
  labs(y = 'Simulation type', x = 'GSD in vagina / GSD in stool')
```

Note that there is a decent fraction of simulations in which the variation in the mean efficiency is lower in the vaginal samples, despite it typically being higher.

The large number of simulations (1000) makes the GSE very small; however, this number does not account for the limited number of gut and vaginal samples.
In other words, we have precisely estimated the difference for these particular samples.

# Variation in species 

For analysis of LFCs in proportions, what matters is the variation in the species proportions relative to the mean efficiency.
Therefore we should also consider whether the species proportions vary less in the more diverse gut.
To do so, I will measure the GSD in the (zero-replaced) proportion of each species within the given sample type that passes a prevalence filter within that sample type.

## Preliminary checks

To figure out how to replace zeros, let's check what the min positive abundances look like for various species.

```{r}
x <- ps_gv %>%
  as_tibble
```

```{r}
x %>%
  filter(.abundance > 0) %>%
  with_groups(c(.otu, body_site), summarize, 
    min_pos = min(.abundance)) %>%
  ggplot(aes(x = min_pos, y = body_site)) +
  scale_x_log10() +
  labs(x = 'Min. positive proportion of the species', y = 'Body site') +
  theme_ridges(grid = FALSE, center_axis_labels = TRUE) +
  scale_y_discrete(expand = expansion(mult = c(0.01, .7))) +
  geom_density_ridges(stat = 'binline', alpha = 0.3, scale = 0.95)
```

What about the min positive proportion in the sample?

```{r}
x %>%
  filter(.abundance > 0) %>%
  with_groups(c(.sample, body_site), summarize, 
    min_pos = min(.abundance)) %>%
  ggplot(aes(x = min_pos, y = body_site)) +
  scale_x_log10() +
  labs(x = 'Min. positive proportion of the sample', y = 'Body site') +
  theme_ridges(grid = FALSE, center_axis_labels = TRUE) +
  scale_y_discrete(expand = expansion(mult = c(0.01, .7))) +
  geom_density_ridges(stat = 'binline', alpha = 0.3, scale = 0.95)
```

We also need to choose what species to consider as candidates for a DA analysis in the given ecosystem.
For simplicity, let's use the filtering criterion we used in the MOMSPI diversity regression.
We'll do this filtering and the subsequent analysis separately within each sample type.

```{r}
ps_list <- list(
  stool = ps_gv %>% filter_sample_data(body_site == 'stool'),
  vagina = ps_gv %>% filter_sample_data(body_site == 'vagina')
) %>%
  map(filter_taxa2, ~ mean(. >= 1e-2) >= 0.05 | mean(. >= 1e-3) >= 0.15) %>%
  map(orient_taxa, as = 'cols')
ps_list %>% map_int(nsamples)
ps_list %>% map_int(ntaxa)
```

Note that there are many more species passing this criterion in stool.

I will replace zeros by adding a small pseudo-value to all proportions.
I expect the choice of pseudo-value will have a large impact on the GSDs of a species, so I will try a wide range of values.
Note, I am not renormalizing the proportions after adding the psuedo-value, since we've already filtered out many species.
However, the renormalization would involve dividing all proprotions by the same factor and so would not affect the resulting GSDs.
(This factor is equal to $1 + k\episilon$, where $k$ is the number of pre-filtered species and $\epsilon$ is the pseudo-value. Since $\episilon \ll 1/k$ except perhaps for the highest $\epsilon = 10^{-3}$, this factor is generally close to 1 anyways.) 

```{r}
min_vals <- c(1e-6, 1e-5, 1e-4, 1e-3)

gsd <- ps_list %>%
  map_dfr(as_tibble) %>%
  with_groups(c(.otu, body_site), summarize,
    min = min_vals,
    gsd = map_dbl(min_vals, ~gm_sd(.abundance + .x))
  )
```

```{r}
gsd %>%
  ggplot(aes(x = gsd, y = as.factor(min), 
      fill = body_site, color = body_site)) +
  # facet_wrap(~body_site) +
  scale_x_log10() +
  expand_limits(x = 1) +
  theme_ridges(grid = FALSE, center_axis_labels = TRUE) +
  scale_y_discrete(expand = expansion(mult = c(0.01, .7))) +
  geom_density_ridges(stat = 'binline', scale = 0.95, alpha = 0.3)
```

The GSDs tend to be larger in the vaginal microbiome, regardless of which min value is used.

What is the difference between gut and vaginal GSDs?
Determine by taking the geometric mean (GM) of the GSDs of various species.

```{r}
gsd %>%
  ggplot(aes(x = gsd, y = as.factor(min), 
      fill = body_site, color = body_site)) +
  scale_x_log10() +
  stat_summary() +
  expand_limits(x = 1)
```

```{r}
species_gsd_stats <- gsd %>%
  with_groups(c(body_site, min), summarize, across(gsd, gm_mean)) %>%
  pivot_wider(names_from = body_site, values_from = gsd) %>%
  mutate(vagina/stool)
species_gsd_stats 
```

Note, the difference between body sites has hardly any dependence on the pseudo-value.

```{r}
species_gsd_stats %>%
  janitor::clean_names() %>%
  write_csv(path('_output', 'species_gsd_stats.csv'))
```

# Final figures and numbers

```{r main, fig.dim = c(6,8)*1.1}
# Alpha diversity
p_div <- sam %>%
  ggplot(aes(y = body_site, x = diversity_q2)) +
  scale_x_log10() +
  geom_boxplot(outlier.shape = NA) +
  geom_quasirandom(groupOnX = FALSE, alpha = 0.75) +
  # theme_ridges(grid = FALSE, center_axis_labels = TRUE) +
  # scale_y_discrete(expand = expansion(mult = c(0.01, .7))) +
  # geom_density_ridges(stat = 'binline', scale = 0.90) +
  labs(y = 'Body site', x = 'Order-2 diversity')
# p_div 

# Variation in species proportions
p_species_gsd <- gsd %>%
  filter(min == 1e-4) %>%
  ggplot(aes(x = gsd, y = body_site)) +
  labs(y = 'Body site', x = "GSD[proportion]") +
  scale_x_log10() +
  expand_limits(x = 1) +
  # geom_vline(xintercept = c(1,gsd_sim), color = 'grey') +
  geom_boxplot(outlier.shape = NA) +
  geom_quasirandom(groupOnX = FALSE, alpha = 0.75)
  # theme_ridges(grid = FALSE, center_axis_labels = TRUE) +
  # scale_y_discrete(expand = expansion(mult = c(0.01, .7))) +
  # geom_density_ridges(stat = 'binline', scale = 0.90)

# GSD in mean eff in simulations
p_me_gsd <- reps_me_summ %>%
  filter(sim_type == 'iid') %>%
  ggplot(aes(y = body_site, x = mean_efficiency)) +
  scale_x_log10() +
  # expand_limits(x = 0.85) +
  geom_vline(xintercept = c(gsd_sim), color = 'grey') +
  theme_ridges(grid = FALSE, center_axis_labels = TRUE) +
  scale_y_discrete(expand = expansion(mult = c(0.01, .7))) +
  stat_density_ridges(quantile_lines = TRUE, quantiles = c(0.5), scale = 0.95) +
  labs(x = 'GSD[mean efficiency]', y = 'Body site')
# p_me_gsd 

(p_div + ggtitle('Within-sample diversity')) /
  (p_me_gsd + ggtitle("Cross-sample variation in the mean efficency")) /
  (p_species_gsd + ggtitle("Cross-sample variation in species' proportions")) +
  plot_annotation(tag_levels = 'A') +
  plot_layout() &
  theme_ridges(grid = FALSE, center_axis_labels = TRUE) 

```

We can also try putting the GSD in proportions and in mean efficiency on a common scale,

```{r, fig.dim = c(6,4)}
# Variation in species proportions
p_species_gsd <- gsd %>%
  filter(min == 1e-4) %>%
  ggplot(aes(x = gsd, y = body_site)) +
  labs(y = 'Body site', x = "GSD[proportion]") +
  scale_x_log10(limits = c(0.9, 200)) +
  theme_ridges(grid = FALSE, center_axis_labels = TRUE) +
  geom_vline(xintercept = c(1,gsd_sim), color = 'grey') +
  scale_y_discrete(expand = expansion(mult = c(0.01, .7))) +
  # geom_density_ridges(stat = 'binline', scale = 0.90)
  geom_density_ridges(jittered_points = TRUE, scale = 0.95)

# GSD in mean eff in simulations
p_me_gsd <- reps_me_summ %>%
  filter(sim_type == 'iid') %>%
  ggplot(aes(y = body_site, x = mean_efficiency)) +
  scale_x_log10(limits = c(0.9, 200)) +
  geom_vline(xintercept = c(1,gsd_sim), color = 'grey') +
  theme_ridges(grid = FALSE, center_axis_labels = TRUE) +
  scale_y_discrete(expand = expansion(mult = c(0.01, .7))) +
  stat_density_ridges(quantile_lines = TRUE, quantiles = c(0.5), scale = 0.95) +
  labs(x = 'GSD[mean efficiency]', y = 'Body site')
# p_me_gsd 

p_species_gsd / p_me_gsd
```

This plot makes it clear that the GSD in mean efficiency is generally smaller than that in the species proportions, suggesting that for the strength of bias in these simulations (GSD=4), that bias is unlikely to have a large impact.
However, the GSD in species proportions is highly dependent on the pseudocount; also, our choice of GSD=4 is simply for illustration.
Thus we should not overinterpret this result.

# Session info {.appendix}

<details><summary>Click for session info</summary>
```{r, R.options = list(width = 83)}
sessioninfo::session_info()
```
</details>
