<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 How taxonomic bias affects abundance measurements | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements</title>
  <meta name="description" content="2 How taxonomic bias affects abundance measurements | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="2 How taxonomic bias affects abundance measurements | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 How taxonomic bias affects abundance measurements | Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements" />
  
  
  

<meta name="author" content="Michael R. McLaren" />
<meta name="author" content="Jacob T. Nearing" />
<meta name="author" content="Amy D. Willis" />
<meta name="author" content="Karen G. Lloyd" />
<meta name="author" content="Benjamin J. Callahan" />


<meta name="date" content="2022-01-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introduction.html"/>
<link rel="next" href="differential-abundance.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
TeX: {extensions: ["cancel.js"]}
});
</script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="abundance-measurement.html"><a href="abundance-measurement.html"><i class="fa fa-check"></i><b>2</b> How taxonomic bias affects abundance measurements</a>
<ul>
<li class="chapter" data-level="2.1" data-path="abundance-measurement.html"><a href="abundance-measurement.html#relative-abundance-proportions-and-ratios"><i class="fa fa-check"></i><b>2.1</b> Relative abundance (proportions and ratios)</a></li>
<li class="chapter" data-level="2.2" data-path="abundance-measurement.html"><a href="abundance-measurement.html#absolute-abundance"><i class="fa fa-check"></i><b>2.2</b> Absolute abundance</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-abundance.html"><a href="differential-abundance.html"><i class="fa fa-check"></i><b>3</b> Taxonomic bias can cause errors in differential-abundance results</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-abundance.html"><a href="differential-abundance.html#fold-changes-between-a-pair-of-samples"><i class="fa fa-check"></i><b>3.1</b> Fold changes between a pair of samples</a></li>
<li class="chapter" data-level="3.2" data-path="differential-abundance.html"><a href="differential-abundance.html#regression-analysis-of-many-samples"><i class="fa fa-check"></i><b>3.2</b> Regression analysis of many samples</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="case-studies.html"><a href="case-studies.html"><i class="fa fa-check"></i><b>4</b> Case studies Â </a>
<ul>
<li class="chapter" data-level="4.1" data-path="case-studies.html"><a href="case-studies.html#foliar-fungi-experiment"><i class="fa fa-check"></i><b>4.1</b> Foliar fungi experiment</a></li>
<li class="chapter" data-level="4.2" data-path="case-studies.html"><a href="case-studies.html#vaginal-microbiomes-of-pregnant-women"><i class="fa fa-check"></i><b>4.2</b> Vaginal microbiomes of pregnant women</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="case-studies.html"><a href="case-studies.html#notes"><i class="fa fa-check"></i><b>4.2.1</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="case-studies.html"><a href="case-studies.html#microbial-growth-in-marine-sediments"><i class="fa fa-check"></i><b>4.3</b> Microbial growth in marine sediments</a></li>
<li class="chapter" data-level="4.4" data-path="case-studies.html"><a href="case-studies.html#summary-and-conclusions"><i class="fa fa-check"></i><b>4.4</b> Summary and conclusions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="solutions.html"><a href="solutions.html"><i class="fa fa-check"></i><b>5</b> Potential solutions</a>
<ul>
<li class="chapter" data-level="5.1" data-path="solutions.html"><a href="solutions.html#use-ratio-based-relative-abundance-analyses"><i class="fa fa-check"></i><b>5.1</b> Use ratio-based relative-abundance analyses</a></li>
<li class="chapter" data-level="5.2" data-path="solutions.html"><a href="solutions.html#calibrate-compositions"><i class="fa fa-check"></i><b>5.2</b> Calibration using community controls</a></li>
<li class="chapter" data-level="5.3" data-path="solutions.html"><a href="solutions.html#calibration-using-reference-species"><i class="fa fa-check"></i><b>5.3</b> Calibration using reference species</a></li>
<li class="chapter" data-level="5.4" data-path="solutions.html"><a href="solutions.html#choosing-complementary-total-density-and-metagenomics-measurements"><i class="fa fa-check"></i><b>5.4</b> Choosing complementary total-density and metagenomics measurements</a></li>
<li class="chapter" data-level="5.5" data-path="solutions.html"><a href="solutions.html#bias-sensitivity-analysis"><i class="fa fa-check"></i><b>5.5</b> Bias-sensitivity analysis</a></li>
<li class="chapter" data-level="5.6" data-path="solutions.html"><a href="solutions.html#bias-aware-meta-analysis"><i class="fa fa-check"></i><b>5.6</b> Bias-aware meta-analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>6</b> Conclusion</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="model-details.html"><a href="model-details.html"><i class="fa fa-check"></i><b>A</b> Model details</a>
<ul>
<li class="chapter" data-level="A.1" data-path="model-details.html"><a href="model-details.html#mgs-measurement"><i class="fa fa-check"></i><b>A.1</b> MGS measurement</a></li>
<li class="chapter" data-level="A.2" data-path="model-details.html"><a href="model-details.html#total-density-bias"><i class="fa fa-check"></i><b>A.2</b> Taxonomic bias in total-density measurement</a></li>
<li class="chapter" data-level="A.3" data-path="model-details.html"><a href="model-details.html#total-density-ref"><i class="fa fa-check"></i><b>A.3</b> Using reference species for total-density normalization</a></li>
<li class="chapter" data-level="A.4" data-path="model-details.html"><a href="model-details.html#linearity-of-extraction"><i class="fa fa-check"></i><b>A.4</b> Linearity of extraction</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html"><i class="fa fa-check"></i><b>B</b> Review of experimental methods for obtaining absolute densities</a>
<ul>
<li class="chapter" data-level="B.1" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#measurement-of-total-cell-density"><i class="fa fa-check"></i><b>B.1</b> Measurement of total cell density</a></li>
<li class="chapter" data-level="B.2" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#measurement-of-total-dna-density-post-extraction"><i class="fa fa-check"></i><b>B.2</b> Measurement of total DNA density post extraction</a></li>
<li class="chapter" data-level="B.3" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#equivolumetric-protocol"><i class="fa fa-check"></i><b>B.3</b> Equivolumetric protocol</a></li>
<li class="chapter" data-level="B.4" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#housekeeping-species"><i class="fa fa-check"></i><b>B.4</b> Housekeeping species</a></li>
<li class="chapter" data-level="B.5" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#spike-ins"><i class="fa fa-check"></i><b>B.5</b> Spike-ins</a></li>
<li class="chapter" data-level="B.6" data-path="review-absolute-methods.html"><a href="review-absolute-methods.html#targeted-measurements"><i class="fa fa-check"></i><b>B.6</b> Targeted measurements</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="appendix-regression.html"><a href="appendix-regression.html"><i class="fa fa-check"></i><b>C</b> Error in estimated regression coefficients and standard errors</a></li>
<li class="chapter" data-level="D" data-path="supplemental-figures.html"><a href="supplemental-figures.html"><i class="fa fa-check"></i><b>D</b> Supplemental figures</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Inferring true changes in microbial abundance from taxonomically-biased microbiome measurements</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="abundance-measurement" class="section level1" number="2">
<h1><span class="header-section-number">2</span> How taxonomic bias affects abundance measurements</h1>
<p>We begin by considering how taxonomic bias affects the relative- and absolute-abundance measurements which serve as the input to microbiome DA analyses.</p>
<p>Our primary tool for understanding the impact of taxonomic bias on MGS measurement is the theoretical model of MGS measurement developed and empirically validated by <span class="citation">McLaren, Willis, and Callahan (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span>.
This model is the simplest that respects the multiplicative nature of taxonomic bias and the <em>compositional</em> nature of MGS measurements, in which the total read count for a sample is unrelated to its total cell number or density (<span class="citation">Gloor et al. (<a href="#ref-gloor2017micr" role="doc-biblioref">2017</a>)</span>).
We consider a set of microbiome samples measured by a specific MGS protocol that extracts, sequences, and bioinformatically analyzes to taxonomically assign reads to a set of microbial species <span class="math inline">\(S\)</span>.
Various forms of taxonomic assignment are possible; for simplicity, we suppose that reads are assigned to the species level, with reads that cannot be uniquely assigned being discarded.
We ignore the possibility that reads are incorrectly assigned to the wrong species or sample.
Unless otherwise stated, we treat the sequencing measurement as deterministic, ignoring the ârandomâ variation in read counts that arise from the sampling of sequencing reads and other aspects of the MGS process.</p>
<p>In our model, the assigned read count of a species <span class="math inline">\(i\)</span> in a sample <span class="math inline">\(a\)</span> equals its abundance multiplied by species-specific and sample-specific factors,
<span class="math display" id="eq:measurement-model">\[\begin{align}
  \tag{2.1}
  \text{reads}_{i}(a)
  = \text{abun}_{i}(a) \quad \cdot
    \underbrace{\text{efficiency}_{i}}_{\substack{\text{species specific,} \\  \text{sample independent}}}
    \cdot \quad
    \underbrace{\text{effort}(a)}_{\substack{\text{species independent,} \\  \text{sample specific}}}.
\end{align}\]</span>
The species-specific factor, <span class="math inline">\(\text{efficiency}_{i}\)</span>, equals the <em>relative measurement efficiency</em> (or simply <em>efficiency</em>) of the speciesâhow much more easily that species is measured (converted from cells to assigned reads) relative to a arbitrary fixed reference species (<span class="citation">McLaren, Willis, and Callahan (<a href="#ref-mclaren2019cons" role="doc-biblioref">2019</a>)</span>).
We assume the efficiencies of particular species are consistent across samples.
The variation in efficiency among species corresponds to the taxonomic bias of the MGS protocol.
The sample-specific factor, <span class="math inline">\(\text{effort}(a)\)</span>, we call the <em>sequencing effort</em> for that sample; it captures the variation in the total number of assigned reads due to experimental features such as library normalization and total sequencing-run output.
Equation <a href="abundance-measurement.html#eq:measurement-model">(2.1)</a> implies that the total number of assigned reads in sample <span class="math inline">\(a\)</span> equals
<span class="math display" id="eq:total-reads">\[\begin{align}
  \tag{2.2}
  \text{reads}_S(a)
    = \text{abun}_S(a) \cdot \text{efficiency}_S(a) \cdot \text{effort}(a),
\end{align}\]</span>
where <span class="math inline">\(\text{abun}_{S}(a) \equiv \sum_{j\in S}\text{abun}_j(a)\)</span> is the total abundance of species in <span class="math inline">\(S\)</span> and
<span class="math display" id="eq:mean-efficiency">\[\begin{align}
  \tag{2.3}
  \text{efficiency}_S(a) 
    \equiv \frac{\sum_{j\in S}\text{abun}_j(a) \cdot \text{efficiency}_j}{\text{abun}_S(a)}
\end{align}\]</span>
is the <em>mean efficiency</em> over all species in <span class="math inline">\(S\)</span>.
<!-- The subscript $S$ serves as a reminder that many reads may not be assigned---for example, reads that fail to pass quality control or that come from the host or other taxonomic domains than what is being studied. --></p>
<div id="relative-abundance-proportions-and-ratios" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Relative abundance (proportions and ratios)</h2>
<p>We distinguish between two types of species-level <em>relative abundances</em> within a sample.
The <em>proportion</em> of species <span class="math inline">\(i\)</span> in sample <span class="math inline">\(a\)</span> equals to its abundance relative to the total abundance of all species in <span class="math inline">\(S\)</span>,
<span class="math display" id="eq:prop">\[\begin{align}
  \tag{2.4}
  \text{prop}_{i}(a) 
    &amp;\equiv \frac{\text{abun}_i(a)}{\text{abun}_S(a)}.
%  \\&amp;= \frac{\text{abun}_i(a)}{\sum_{i \in S}\text{abun}_i(a)}.
\end{align}\]</span>
The <em>ratio</em> between two species <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> equals the abundance of <span class="math inline">\(i\)</span> relative to that of <span class="math inline">\(j\)</span>,
<span class="math display" id="eq:ratio">\[\begin{align}
  \tag{2.5}
  \text{ratio}_{i/j}(a) = \frac{\text{abun}_i(a)}{\text{abun}_j(a)}.
\end{align}\]</span>
Proportions and ratios each form the basis for popular DA methods;
ratio-based methods are commonly referred to as Compositional Data Analysis (CoDA) methods.</p>
<p>Taxonomic bias has a different effect on species proportions versus species ratios.
The proportion of a species is typically measured by its proportion of assigned reads,
<span class="math display" id="eq:prop-meas">\[\begin{align}
  \tag{2.6}
  \widehat{\text{prop}}_{i}(a) = \frac{\text{reads}_i(a)}{\text{reads}_S(a)}.
\end{align}\]</span>
We can rewrite the right-hand side (using Equations <a href="abundance-measurement.html#eq:measurement-model">(2.1)</a>, <a href="abundance-measurement.html#eq:total-reads">(2.2)</a>, and <a href="abundance-measurement.html#eq:prop-meas">(2.6)</a>) to find
<span class="math display" id="eq:prop-error">\[\begin{align}
  \tag{2.7}
  \widehat{\text{prop}}_{i}(a)
  &amp;= \text{prop}_{i}(a) \cdot \underbrace{\frac{\text{efficiency}_{i}}{\text{efficiency}_S(a)}}_{\substack{\text{variable} \\ \text{fold error}}}.
\end{align}\]</span>
Taxonomic bias creates a multiplicative error in the speciesâ proportion equal to its efficiency relative to the mean efficiency in the sample.
Consequently, the speciesâs proportion is measured as too high in samples that are dominated by species with lower efficiencies, and measured as too low in samples that are dominated by species with higher efficiencies.
This phenomenon is illustrated in two hypothetical communities in Figure <a href="abundance-measurement.html#fig:error-proportions">2.1</a>.
Species 3 has an efficiency of 6; it is under-measured in Sample 1, which has a mean efficiency of 8.33, but over-measured in Sample 2, which has a mean efficiency of 3.15.</p>
<p>The measured ratio between species <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> is given by the ratio of their read counts,
<span class="math display" id="eq:ratio-meas">\[\begin{align}
  \tag{2.8}
  \widehat{\text{ratio}}_{i/j}(a) = \frac{\text{reads}_i(a)}{\text{reads}_j(a)}.
\end{align}\]</span>
From Equations <a href="abundance-measurement.html#eq:measurement-model">(2.1)</a> and <a href="abundance-measurement.html#eq:ratio-meas">(2.8)</a>, it follows that
<span class="math display" id="eq:ratio-error">\[\begin{align}
  \tag{2.9}
  \widehat{\text{ratio}}_{i/j}(a)
%  &amp;= \frac{\text{abun}_{i}(a)}{\text{abun}_{j}(a)} 
  &amp;= \text{ratio}_{i/j}(a) \cdot \underbrace{\frac{\text{efficiency}_{i}}{\text{efficiency}_{j}}}_{\substack{\text{constant} \\ \text{fold error}}}.
\end{align}\]</span>
Taxonomic bias creates a multiplicative measurement error in the ratio that is equal to the ratio in their efficiencies; the error is therefore constant across samples.
For instance, in Figure <a href="abundance-measurement.html#fig:error-proportions">2.1</a>, the ratio of Species 3 (with an efficiency of 6) to Species 1 (with an efficiency of 1) is over-estimated by a factor of 6 in both communities despite their varying compositions.</p>
<!-- begin figure -->
<div class="figure"><span style="display:block;" id="fig:error-proportions"></span>
<img src="figures/illustrations/error-proportions.svg" alt="Taxonomic bias creates sample-dependent multiplicative errors in species proportions, which can lead to inaccurate fold changes between samples. Top row: Error in proportions measured by MGS in two hypothetical microbiome samples that contain different relative abundances of three species. Bottom row: Error in the measured fold-change in the third species that is derived from these measurements. Speciesâ proportions may be measured as too high or too low depending on sample composition. For instance, Species 3 has an efficiency of 6 and is under-measured in Sample 1 (which has a mean efficiency of 8.33) but over-measured in Sample 2 (which has a mean efficiency of 3.15)." width="90%" />
<p class="caption">
Figure 2.1: <strong>Taxonomic bias creates sample-dependent multiplicative errors in species proportions, which can lead to inaccurate fold changes between samples.</strong> Top row: Error in proportions measured by MGS in two hypothetical microbiome samples that contain different relative abundances of three species. Bottom row: Error in the measured fold-change in the third species that is derived from these measurements. Speciesâ proportions may be measured as too high or too low depending on sample composition. For instance, Species 3 has an efficiency of 6 and is under-measured in Sample 1 (which has a mean efficiency of 8.33) but over-measured in Sample 2 (which has a mean efficiency of 3.15).
</p>
</div>

<!-- end figure -->
<p>The fact that the multiplicative error in proportions varies while that in ratios is constant forms the basis for understanding the effect of bias on different methods for measuring absolute abundances from MGS and the extent to which bias cancels in different DA analyses.</p>
</div>
<div id="absolute-abundance" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Absolute abundance</h2>
<p>Researchers often would like to know how the <em>absolute abundance</em> of a species changes.
In this context, <em>absolute</em> means not relative to other species; typically, the abundance of interest is relative to non-microbial entities such the volume, mass, or amount of host DNA in the sample.
A wide range of experimental methods have been used to convert the relative abundances from MGS measurements into measurements of absolute abundances.
Yet there has so far been little consideration as to how the resulting measurements are affected by taxonomic bias in the MGS measurements or in the supplemental measurements, such as flow cytometry and qPCR, that they often involve.
Here and in Appendix <a href="review-absolute-methods.html#review-absolute-methods">B</a> we describe how various methods for measuring absolute abundance are predicted to be affected by taxonomic bias, with an eye towards determining whether the measurement error is variable or constant.</p>
<p>Our discussion supposes the âabundanceâ of interest is cell number, though it applies to other quantities, such as biomass and genome copy number, that may be more relevant in different biological contexts.</p>
<p>We consider two general classes of methods: Those in which absolute-abundance information is derived from measurement of the aggregate abundance of the total community, and those in which it drives from targeted measurement (or prior knowledge) of one or more particular species.</p>
<p><strong>Normalization by total abundance:</strong>
Measurements of species absolute abundances are often obtained by making a (non-MGS) measurement of total community abundance, equating it with the aggregate abundance of the species <span class="math inline">\(S\)</span> measured by MGS, and multiplying this total by the proportions from MGS,
<span class="math display" id="eq:density-prop-meas">\[\begin{align}
  \tag{2.10}
  \widehat{\text{abun}}_{i}(a) 
  &amp;= \widehat{\text{prop}}_{i}(a) \cdot \widehat{\text{abun}}_S(a).
\end{align}\]</span>
This measurement is affected by taxonomic bias in the MGS measurement as well as in the total-abundance measurement.
For example, measurement of total community abundance by 16S qPCR is affected by variation among species in extraction efficiency, 16S copy number, and PCR binding and amplification bias.
If the species-level efficiencies of the total-abundance measurement are constant across samples and we neglect other sources, then we can express the total-abundance measurement as
<span class="math display" id="eq:total-density-error">\[\begin{align}
  \tag{2.11}
  \widehat{\text{abun}}_S(a) 
  &amp;= \sum_{i\in S} \text{abun}_i(a) \cdot \text{efficiency}^{\text{tot}}_i
\\&amp;= \text{abun}_S(a) \cdot \text{efficiency}^{\text{tot}}_S(a),
\end{align}\]</span>
where <span class="math inline">\(\text{efficiency}_{i}^{\text{tot}}(a)\)</span> is the absolute measurement efficiency of species <span class="math inline">\(i\)</span> for the total-abundance measurement and <span class="math inline">\(\text{efficiency}^{\text{tot}}_S(a)\)</span> is the mean efficiency of the total-abundance measurement in the sample.
<!-- Note, we have assumed that the total-abundance and MGS measurements are sensitive to the same species -->
The equations <a href="abundance-measurement.html#eq:total-density-error">(2.11)</a> and <a href="abundance-measurement.html#eq:prop-error">(2.7)</a> for the error in total-abundance and proportion measurements imply that the species abundance measurement <a href="abundance-measurement.html#eq:density-prop-meas">(2.10)</a> has error given by
<span class="math display" id="eq:density-prop-error">\[\begin{align}
  \tag{2.12}
  \widehat{\text{abun}}_{i}(a) 
  = \text{abun}_{i}(a) \cdot \frac{\text{efficiency}_{i} \cdot \text{efficiency}^{\text{tot}}_S(a)}{\text{efficiency}_S(a)}.
\end{align}\]</span></p>
<p>Equation <a href="abundance-measurement.html#eq:density-prop-error">(2.12)</a> indicates that the multiplicative error in the measured absolute abundance of a species equals its MGS efficiency relative to the mean MGS efficiency in the sample, multiplied by the mean efficiency of the total measurement.
As in the case of proportions (Equation <a href="abundance-measurement.html#eq:prop-error">(2.7)</a>), the error depends on sample composition through the two mean efficiency terms and so may vary across samples.
On the other hand, if the mean efficiency of the total-abundance measurement mirrors that of the MGS measurement, the two can offset and lead to a relatively stable error.
We discuss how this possibility might apply to real experimental workflows below.</p>
<p><strong>Normalization by one or more reference species:</strong>
Suppose we had a measurement of the (absolute) abundance of a <em>reference species</em> <span class="math inline">\(r\)</span>.
In the absence of taxonomic bias, all species are expected to have the same ratio of reads to abundance in a sample (Equation <a href="abundance-measurement.html#eq:measurement-model">(2.1)</a>).
Thus the abundance-to-reads ratio for species <span class="math inline">\(r\)</span> can serve as a conversion factor allowing us to obtain the abundance of an arbitrary species <span class="math inline">\(i\)</span>,
<span class="math display" id="eq:density-ratio-meas">\[\begin{align}
  \tag{2.13}
  \widehat{\text{abun}}_{i}(a) = \text{reads}_{i}(a) \cdot \frac{\widehat{\text{abun}}_{r}(a)}{\text{reads}_{r}(a)}.
\end{align}\]</span>
The abundance of one or more reference species can be directly measured using targeted measurement methods like species-specific qPCR and used with Equation
<a href="abundance-measurement.html#eq:density-ratio-meas">(2.13)</a> to obtain absolute abundance for all species.
To our knowledge, this approach has not previously been suggested.
<!-- (Appendix \@ref(model-details) considers possible extensions to multiple reference species.) -->
Instead, reference-based measurement has been used in the context of spike-in experiments or normalization to a species (such as the host) which is treated as having a constant abundance.
In a spike-in experiment, the abundance of reference species are added in known (up to experimental error) abundances to each sample.
When normalizing to a species that is treated as having a constant but unknown abundance, we set <span class="math inline">\(\widehat{\text{abun}}_{r}(a)\)</span> in Equation <a href="abundance-measurement.html#eq:density-ratio-meas">(2.13)</a> equal to 1 and interpret the abundances as having fixed but unknown units, which is sufficient for multiplicative DA analysis.</p>
<p>The error in the abundance measurement <a href="abundance-measurement.html#eq:density-ratio-meas">(2.13)</a> due to taxonomic bias in the MGS measurement is given by
<span class="math display" id="eq:density-ratio-error">\[\begin{align}
  \tag{2.14}
  \widehat{\text{abun}}_{i}(a) 
  = \text{abun}_{i}(a) \cdot \frac{\text{efficiency}_{i}}{\text{efficiency}_{r}}
  \cdot \begin{array}{c} \text{fold error in} \\ \widehat{\text{abun}_r}(a) \end{array}.
\end{align}\]</span>
The constant error in the measured ratio of species <span class="math inline">\(i\)</span> to <span class="math inline">\(r\)</span> (see Equation <a href="abundance-measurement.html#eq:ratio-error">(2.9)</a>) propagates to the abundance measurement.
There will generally also be systematic error in the abundance of the reference species; however, if the systematic fold error is constant across samples, so will be that of the abundances of other species.</p>
<p>Spike-ins are instead sometimes used to measure absolute abundances using Equation <a href="abundance-measurement.html#eq:density-prop-meas">(2.10)</a>.
In this case, an intermediate step is taken in which the total community abundance is first measured by the ratio of non-spike-in to spike-in reads.
<!-- CHECK -->
If done correctly, this calculation yields results that are identical to directly applying Equation <a href="abundance-measurement.html#eq:density-ratio-meas">(2.13)</a> (Appendix <a href="model-details.html#total-density-ref">A.3</a>).</p>
<p><strong>Difference between the two approaches:</strong>
Reference-species normalization yields constant fold errors because it is based on species-level read counts and abundances and we assume that efficiencies are constant at the species level.
In contrast, total-abundance normalization is based on aggregates of species (for the calculation of proportion and the total-abundance measurement) and so depends on mean efficiencies, which can vary across samples.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-gloor2017micr" class="csl-entry">
Gloor, Gregory B., Jean M. Macklaim, Vera Pawlowsky-Glahn, and Juan J. Egozcue. 2017. <span>â<span>Microbiome Datasets Are Compositional: And This Is Not Optional</span>.â</span> <em>Front. Microbiol.</em> 8 (November): 2224. <a href="https://doi.org/10.3389/fmicb.2017.02224">https://doi.org/10.3389/fmicb.2017.02224</a>.
</div>
<div id="ref-mclaren2019cons" class="csl-entry">
McLaren, Michael R, Amy D Willis, and Benjamin J Callahan. 2019. <span>â<span class="nocase">Consistent and correctable bias in metagenomic sequencing experiments</span>.â</span> <em>Elife</em> 8 (September): 46923. <a href="https://doi.org/10.7554/eLife.46923">https://doi.org/10.7554/eLife.46923</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-abundance.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/mikemc/differential-abundance-theory/edit/main/abundance-measurement.Rmd",
"text": "Edit"
},
"history": {
"link": "https://github.com/mikemc/differential-abundance-theory/commits/main/abundance-measurement.Rmd",
"text": null
},
"view": {
"link": "https://github.com/mikemc/differential-abundance-theory/blob/main/abundance-measurement.Rmd",
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
