<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007ba5; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #007ba5; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #007ba5; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #5e5e5e; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>Exploring the effect of bias in Leopold and Busby (2020)
  </title>



  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2019-10-21"/>
  <meta property="article:created" itemprop="dateCreated" content="2019-10-21"/>
  <meta name="article:author" content="Michael McLaren"/>

  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="Exploring the effect of bias in Leopold and Busby (2020)&#10;"/>
  <meta property="og:type" content="article"/>
  <meta property="og:locale" content="en_US"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="Exploring the effect of bias in Leopold and Busby (2020)&#10;"/>

  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Host Genotype and Colonist Arrival Order Jointly Govern Plant Microbiome Composition and Function;citation_publication_date=2020;citation_publisher=Elsevier Ltd.;citation_volume=30;citation_doi=10.1016/j.cub.2020.06.011;citation_author=Devin R Leopold;citation_author=Posy E Busby"/>
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","author","date","categories","description","bibliography","draft","output"]}},"value":[{"type":"character","attributes":{},"value":["Exploring the effect of bias in Leopold and Busby (2020)\n"]},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name"]}},"value":[{"type":"character","attributes":{},"value":["Michael McLaren"]}]}]},{"type":"character","attributes":{},"value":["2019-10-21"]},{"type":"character","attributes":{},"value":["bias sensitivity","R","ref:leopold2020host"]},{"type":"NULL"},{"type":"character","attributes":{},"value":["_references.bib"]},{"type":"logical","attributes":{},"value":[false]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["self_contained","toc","toc_float","dev"]}},"value":[{"type":"logical","attributes":{},"value":[false]},{"type":"logical","attributes":{},"value":[true]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["svg"]}]}]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-resource-manifest">
  {"type":"character","attributes":{},"value":["index_files/anchor-4.2.2/anchor.min.js","index_files/bowser-1.9.3/bowser.min.js","index_files/distill-2.2.21/template.v2.js","index_files/figure-html5/unnamed-chunk-10-1.svg","index_files/figure-html5/unnamed-chunk-11-1.svg","index_files/figure-html5/unnamed-chunk-12-1.svg","index_files/figure-html5/unnamed-chunk-13-1.svg","index_files/figure-html5/unnamed-chunk-14-1.svg","index_files/figure-html5/unnamed-chunk-15-1.svg","index_files/figure-html5/unnamed-chunk-16-1.svg","index_files/figure-html5/unnamed-chunk-17-1.svg","index_files/figure-html5/unnamed-chunk-18-1.svg","index_files/figure-html5/unnamed-chunk-19-1.svg","index_files/figure-html5/unnamed-chunk-2-1.svg","index_files/figure-html5/unnamed-chunk-20-1.svg","index_files/figure-html5/unnamed-chunk-23-1.svg","index_files/figure-html5/unnamed-chunk-24-1.svg","index_files/figure-html5/unnamed-chunk-26-1.svg","index_files/figure-html5/unnamed-chunk-27-1.svg","index_files/figure-html5/unnamed-chunk-3-1.svg","index_files/figure-html5/unnamed-chunk-7-1.svg","index_files/figure-html5/unnamed-chunk-8-1.svg","index_files/header-attrs-2.7/header-attrs.js","index_files/jquery-1.11.3/jquery.min.js","index_files/popper-2.6.0/popper.min.js","index_files/tippy-6.2.7/tippy-bundle.umd.min.js","index_files/tippy-6.2.7/tippy-light-border.css","index_files/tippy-6.2.7/tippy.css","index_files/tippy-6.2.7/tippy.umd.min.js","index_files/webcomponents-2.0.0/webcomponents.js"]}
  </script>
  <!--radix_placeholder_navigation_in_header-->
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          return "<p>" + $('#ref-' + ref).html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="index_files/header-attrs-2.7/header-attrs.js"></script>
  <script src="index_files/jquery-1.11.3/jquery.min.js"></script>
  <script src="index_files/popper-2.6.0/popper.min.js"></script>
  <link href="index_files/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="index_files/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="index_files/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="index_files/anchor-4.2.2/anchor.min.js"></script>
  <script src="index_files/bowser-1.9.3/bowser.min.js"></script>
  <script src="index_files/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="index_files/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Exploring the effect of bias in Leopold and Busby (2020)\n","authors":[{"author":"Michael McLaren","authorURL":"#","affiliation":"&nbsp;","affiliationURL":"#","orcidID":""}],"publishedDate":"2019-10-21T00:00:00.000-04:00","citationText":"McLaren, 2019"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Exploring the effect of bias in Leopold and Busby (2020)</h1>
<!--radix_placeholder_categories-->
<div class="dt-tags">
<div class="dt=tag">bias sensitivity</div>
<div class="dt=tag">R</div>
<div class="dt=tag">ref:leopold2020host</div>
</div>
<!--/radix_placeholder_categories-->

</div>

<div class="d-byline">
  
  Michael McLaren
  
<br/>2019-10-21
</div>

<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#setup">Setup</a></li>
<li><a href="#sample-mean-efficiency-across-samples">Sample mean efficiency across samples</a>
<ul>
<li><a href="#alt-plot">Alt plot</a></li>
<li><a href="#check-errors-in-fcs">Check errors in FCs</a>
<ul>
<li><a href="#mock-samples">mock samples</a></li>
<li><a href="#experiment-samples">Experiment samples</a></li>
</ul></li>
</ul></li>
<li><a href="#session-info">Session info</a></li>
</ul>
</nav>
</div>
<p>Analysis of bias in <span class="citation" data-cites="leopold2020host"><a href="#ref-leopold2020host" role="doc-biblioref">Leopold and Busby</a> (<a href="#ref-leopold2020host" role="doc-biblioref">2020</a>)</span>.</p>
<h1 id="setup">Setup</h1>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># Tools for microbiome data</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://mikemc.github.io/speedyseq'>speedyseq</a></span><span class='op'>)</span>
<span class='co'># Tools for general purpose data manipulation and plotting</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='http://tidyverse.tidyverse.org'>tidyverse</a></span><span class='op'>)</span>
<span class='co'># ggplot helpers</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/eclarke/ggbeeswarm'>ggbeeswarm</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://wilkelab.org/cowplot/'>cowplot</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://patchwork.data-imaginist.com'>patchwork</a></span><span class='op'>)</span>
<span class='fu'>theme_set</span><span class='op'>(</span><span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># stats helpers</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://broom.tidymodels.org/'>broom</a></span><span class='op'>)</span>

<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://mikemc.github.io/metacal'>metacal</a></span><span class='op'>)</span>; <span class='fu'><a href='https://rdrr.io/r/utils/packageDescription.html'>packageVersion</a></span><span class='op'>(</span><span class='st'>"metacal"</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code>[1] &#39;0.2.0&#39;</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ps.mock.pseudo</span> <span class='op'>&lt;-</span> <span class='va'>ps.mock</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/transform_sample_counts.html'>transform_sample_counts</a></span><span class='op'>(</span><span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='va'>x</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span>
<span class='va'>mc_fit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://mikemc.github.io/metacal/reference/estimate_bias.html'>estimate_bias</a></span><span class='op'>(</span><span class='va'>ps.mock.pseudo</span>, <span class='va'>mock_actual</span>, boot <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='va'>print</span>
</code></pre>
</div>
<pre><code>A metacal bias fit.

Estimated relative efficiencies:
   Melampsora     Dioszegia     Epicoccum      Fusarium   Penicillium 
    8.7486819     0.2947290     1.0064648     3.0660596     0.2233792 
 Cladosporium   Trichoderma    Alternaria Aureobasidium 
    0.8988898     0.5776648     1.4679841     0.7380894 

Contains 1000 bootstrap replicates.</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>bias</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/coef.html'>coef</a></span><span class='op'>(</span><span class='va'>mc_fit</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='va'>print</span>
</code></pre>
</div>
<pre><code>   Melampsora     Dioszegia     Epicoccum      Fusarium   Penicillium 
    8.7486819     0.2947290     1.0064648     3.0660596     0.2233792 
 Cladosporium   Trichoderma    Alternaria Aureobasidium 
    0.8988898     0.5776648     1.4679841     0.7380894 </code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>mc_fit.summary</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>mc_fit</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='va'>mc_fit.summary</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code>Summary of a metacal bias fit.

Estimated relative efficiencies:
# A tibble: 9 x 4
  taxon         estimate gm_mean gm_se
  &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;
1 Melampsora       8.75    8.76   1.06
2 Dioszegia        0.295   0.295  1.10
3 Epicoccum        1.01    0.995  1.50
4 Fusarium         3.07    3.08   1.14
5 Penicillium      0.223   0.224  1.09
6 Cladosporium     0.899   0.903  1.13
7 Trichoderma      0.578   0.579  1.09
8 Alternaria       1.47    1.46   1.08
9 Aureobasidium    0.738   0.736  1.12

Geometric standard error estimated from 1000 bootstrap replicates.</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>coef_tb</span> <span class='op'>&lt;-</span> <span class='va'>mc_fit.summary</span><span class='op'>$</span><span class='va'>coefficients</span>
<span class='va'>coef_tb</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span>taxon <span class='op'>=</span> <span class='fu'>fct_reorder</span><span class='op'>(</span><span class='va'>taxon</span>, <span class='va'>estimate</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>ggplot</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span><span class='va'>taxon</span>, <span class='va'>estimate</span>, 
      ymin <span class='op'>=</span> <span class='va'>estimate</span> <span class='op'>/</span> <span class='va'>gm_se</span><span class='op'>^</span><span class='fl'>2</span>, ymax <span class='op'>=</span> <span class='va'>estimate</span> <span class='op'>*</span> <span class='va'>gm_se</span><span class='op'>^</span><span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_hline</span><span class='op'>(</span>yintercept <span class='op'>=</span> <span class='fl'>1</span>, color <span class='op'>=</span> <span class='st'>"grey"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_pointrange</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_y_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>coord_flip</span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="index_files/figure-html5/unnamed-chunk-3-1.svg" width="480" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ps.pseudo</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/transform_sample_counts.html'>transform_sample_counts</a></span><span class='op'>(</span><span class='va'>ps</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='va'>x</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span>
<span class='va'>ps.pseudo.cal</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://mikemc.github.io/metacal/reference/calibrate.html'>calibrate</a></span><span class='op'>(</span><span class='va'>ps.pseudo</span>, <span class='va'>bias</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='va'>print</span>
</code></pre>
</div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:          [ 9 taxa and 567 samples ]:
sample_data() Sample Data:        [ 567 samples by 16 sample variables ]:
tax_table()   Taxonomy Table:     [ 9 taxa by 7 taxonomic ranks ]:
refseq()      DNAStringSet:       [ 9 reference sequences ]
taxa are columns</code></pre>
</div>
<h1 id="sample-mean-efficiency-across-samples">Sample mean efficiency across samples</h1>
<p>To estimate the SMRE with formula <code>sum_i A_i E_i</code>, use the calibrated abundances.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ps0</span> <span class='op'>&lt;-</span> <span class='va'>ps.pseudo.cal</span> <span class='op'>%&gt;%</span>
  <span class='fu'>prune_taxa</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>bias</span><span class='op'>)</span>, <span class='va'>.</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>otu</span> <span class='op'>&lt;-</span> <span class='va'>ps0</span> <span class='op'>%&gt;%</span> <span class='va'>otu_table</span> <span class='op'>%&gt;%</span> <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/orient_taxa.html'>orient_taxa</a></span><span class='op'>(</span>as <span class='op'>=</span> <span class='st'>"cols"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/transform_sample_counts.html'>transform_sample_counts</a></span><span class='op'>(</span><span class='va'>close_elts</span><span class='op'>)</span>
<span class='va'>sam0</span> <span class='op'>&lt;-</span> <span class='fu'>sample_data</span><span class='op'>(</span><span class='va'>ps0</span><span class='op'>)</span> <span class='op'>%&gt;%</span> 
  <span class='fu'>as</span><span class='op'>(</span><span class='st'>"data.frame"</span><span class='op'>)</span> <span class='op'>%&gt;%</span> 
  <span class='fu'>as_tibble</span><span class='op'>(</span>rownames <span class='op'>=</span> <span class='st'>"Sample"</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sme</span> <span class='op'>&lt;-</span> <span class='va'>otu</span> <span class='op'>%&gt;%</span> <span class='va'>psmelt</span> <span class='op'>%&gt;%</span>
  <span class='fu'>left_join</span><span class='op'>(</span><span class='va'>bias</span> <span class='op'>%&gt;%</span> <span class='fu'>enframe</span><span class='op'>(</span><span class='st'>"OTU"</span>, <span class='st'>"Efficiency"</span><span class='op'>)</span>, by <span class='op'>=</span> <span class='st'>"OTU"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>group_by</span><span class='op'>(</span><span class='va'>Sample</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>summarize</span><span class='op'>(</span>SMRE <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>Efficiency</span> <span class='op'>*</span> <span class='va'>Abundance</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>left_join</span><span class='op'>(</span><span class='va'>sam0</span>, by <span class='op'>=</span> <span class='st'>"Sample"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span>
    <span class='fu'>across</span><span class='op'>(</span><span class='va'>Timepoint</span>, <span class='va'>as.factor</span><span class='op'>)</span>,
    <span class='fu'>across</span><span class='op'>(</span><span class='va'>Timepoint</span>, <span class='va'>fct_explicit_na</span>, <span class='st'>"Other"</span><span class='op'>)</span>,
  <span class='op'>)</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>sme</span><span class='op'>$</span><span class='va'>SMRE</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code>[1] 0.2236076 8.7034225</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>bias</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code>[1] 0.2233792 8.7486819</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='fu'>ggplot</span><span class='op'>(</span><span class='va'>sme</span>, <span class='fu'>aes</span><span class='op'>(</span><span class='va'>SMRE</span>, fill <span class='op'>=</span> <span class='va'>Samp_type</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_histogram</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>annotate</span><span class='op'>(</span><span class='st'>"point"</span>, y <span class='op'>=</span> <span class='fl'>200</span>, x <span class='op'>=</span> <span class='va'>bias</span>, color <span class='op'>=</span> <span class='st'>"black"</span>, shape <span class='op'>=</span> <span class='st'>"|"</span>, size <span class='op'>=</span> <span class='fl'>7</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_y_sqrt</span><span class='op'>(</span>breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>10</span>, <span class='fl'>50</span>, <span class='fl'>100</span>, <span class='fl'>200</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>expand_limits</span><span class='op'>(</span>x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>bias</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_fill_brewer</span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"qual"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>labs</span><span class='op'>(</span>
    <span class='co'># x = "Sample mean relative efficiency (SMRE)",</span>
    x <span class='op'>=</span> <span class='st'>"Mean efficiency"</span>, 
    y <span class='op'>=</span> <span class='st'>"Number of samples"</span>,
    fill <span class='op'>=</span> <span class='st'>"Sample type"</span>
  <span class='op'>)</span>
<span class='va'>p1</span> <span class='op'>+</span>
  <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_annotation.html'>plot_annotation</a></span><span class='op'>(</span>
    title <span class='op'>=</span> <span class='st'>"Distribution of sample mean relative efficiencies"</span>,
  <span class='op'>)</span>
</code></pre>
</div>
<p><img src="index_files/figure-html5/unnamed-chunk-8-1.svg" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>ggsave</span><span class='op'>(</span><span class='st'>"/tmp/leopold2020host-smre-distribution.svg"</span>, units <span class='op'>=</span> <span class='st'>"in"</span>,
  width <span class='op'>=</span> <span class='fl'>6</span>, height <span class='op'>=</span> <span class='fl'>3</span>, scale <span class='op'>=</span> <span class='fl'>1.5</span><span class='op'>)</span>
<span class='fu'>ggsave</span><span class='op'>(</span><span class='st'>"/tmp/leopold2020host-smre-distribution.png"</span>, units <span class='op'>=</span> <span class='st'>"in"</span>,
  width <span class='op'>=</span> <span class='fl'>6</span>, height <span class='op'>=</span> <span class='fl'>3</span>, scale <span class='op'>=</span> <span class='fl'>1.5</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>In this case, the bimodal distribution is likely driven by the high proportion of Melampsora (pathogen) in ~half the samples.</p>
<p>TODO: check this hypothesis. by making a histogram of melampspora %, and a scatterplot of the melamspora vs.SMRE. Can do with facet matrix from ggforce</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>mel</span> <span class='op'>&lt;-</span> <span class='va'>otu</span> <span class='op'>%&gt;%</span>
  <span class='fu'>prune_taxa</span><span class='op'>(</span><span class='st'>"Melampsora"</span>, <span class='va'>.</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='va'>psmelt</span> <span class='op'>%&gt;%</span>
  <span class='fu'>select</span><span class='op'>(</span><span class='op'>-</span><span class='va'>OTU</span>, Melampsora <span class='op'>=</span> <span class='va'>Abundance</span><span class='op'>)</span>
<span class='va'>sme0</span> <span class='op'>&lt;-</span> <span class='fu'>left_join</span><span class='op'>(</span><span class='va'>sme</span>, <span class='va'>mel</span>, by <span class='op'>=</span> <span class='st'>"Sample"</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># p2 &lt;- ggplot(sme0, aes(SMRE, fill = Melampsora)) +</span>
<span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='fu'>ggplot</span><span class='op'>(</span><span class='va'>sme0</span>, <span class='fu'>aes</span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>Melampsora</span>, x <span class='op'>=</span> <span class='va'>SMRE</span>, color <span class='op'>=</span> <span class='va'>Samp_type</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_point</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='co'># annotate("point", y = 1, x = bias, color = "black", shape = "|", size = 7) +</span>
  <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='co'># scale_y_sqrt() +</span>
  <span class='fu'>scale_y_continuous</span><span class='op'>(</span>
    trans <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/log_trans.html'>pseudo_log_trans</a></span><span class='op'>(</span>sigma <span class='op'>=</span> <span class='fl'>1e-2</span><span class='op'>)</span>,
    breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>0.01</span>, <span class='fl'>0.1</span>, <span class='fl'>0.5</span>, <span class='fl'>1</span><span class='op'>)</span>
    <span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>expand_limits</span><span class='op'>(</span>x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>bias</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_color_brewer</span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"qual"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>labs</span><span class='op'>(</span>
    y <span class='op'>=</span> <span class='st'>"Melampsora proportion"</span>,
    <span class='co'># x = "Sample mean relative efficiency (SMRE)",</span>
    x <span class='op'>=</span> <span class='st'>"Mean efficiency"</span>,
    color <span class='op'>=</span> <span class='st'>"Sample type"</span>
  <span class='op'>)</span>
<span class='va'>p2</span>
</code></pre>
</div>
<p><img src="index_files/figure-html5/unnamed-chunk-11-1.svg" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='op'>(</span><span class='va'>p1</span> <span class='op'>+</span> <span class='fu'>ggtitle</span><span class='op'>(</span><span class='st'>"Distribution of the mean efficiency across samples"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>/</span> 
  <span class='op'>(</span><span class='va'>p2</span> <span class='op'>+</span> <span class='fu'>ggtitle</span><span class='op'>(</span><span class='st'>"Melampsora proportion vs. mean efficiency"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_annotation.html'>plot_annotation</a></span><span class='op'>(</span>tag_levels <span class='op'>=</span> <span class='st'>"A"</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="index_files/figure-html5/unnamed-chunk-12-1.svg" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>ggsave</span><span class='op'>(</span><span class='st'>"/tmp/leopold2020host-smre-distribution.svg"</span>, units <span class='op'>=</span> <span class='st'>"in"</span>,
  width <span class='op'>=</span> <span class='fl'>6</span>, height <span class='op'>=</span> <span class='fl'>6</span>, scale <span class='op'>=</span> <span class='fl'>1.5</span><span class='op'>)</span>
<span class='fu'>ggsave</span><span class='op'>(</span><span class='st'>"/tmp/leopold2020host-smre-distribution.pdf"</span>, units <span class='op'>=</span> <span class='st'>"in"</span>,
  width <span class='op'>=</span> <span class='fl'>6</span>, height <span class='op'>=</span> <span class='fl'>6</span>, scale <span class='op'>=</span> <span class='fl'>1.5</span><span class='op'>)</span>
<span class='fu'>ggsave</span><span class='op'>(</span><span class='st'>"/tmp/leopold2020host-smre-distribution.png"</span>, units <span class='op'>=</span> <span class='st'>"in"</span>,
  width <span class='op'>=</span> <span class='fl'>6</span>, height <span class='op'>=</span> <span class='fl'>6</span>, scale <span class='op'>=</span> <span class='fl'>1.5</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>Note, the high-Mel samples are the timepoint 2 samples, and were not used for some (all?) of his analysis due to the dominance of Mel. Lets check this out,</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sme</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span>
    <span class='fu'>across</span><span class='op'>(</span><span class='va'>Timepoint</span>, <span class='va'>as.factor</span><span class='op'>)</span>,
    <span class='fu'>across</span><span class='op'>(</span><span class='va'>Timepoint</span>, <span class='va'>fct_explicit_na</span>, <span class='st'>"Other"</span><span class='op'>)</span>,
  <span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='co'># filter(Samp_type == "Experiment") %&gt;%</span>
  <span class='fu'>ggplot</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span><span class='va'>SMRE</span>, fill <span class='op'>=</span> <span class='va'>Timepoint</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_histogram</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>annotate</span><span class='op'>(</span><span class='st'>"point"</span>, y <span class='op'>=</span> <span class='fl'>200</span>, x <span class='op'>=</span> <span class='va'>bias</span>, color <span class='op'>=</span> <span class='st'>"black"</span>, shape <span class='op'>=</span> <span class='st'>"|"</span>, size <span class='op'>=</span> <span class='fl'>7</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_y_sqrt</span><span class='op'>(</span>breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>10</span>, <span class='fl'>50</span>, <span class='fl'>100</span>, <span class='fl'>200</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>expand_limits</span><span class='op'>(</span>x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>bias</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_fill_brewer</span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"qual"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>labs</span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='st'>"Mean efficiency"</span>, 
    y <span class='op'>=</span> <span class='st'>"Number of samples"</span>
  <span class='op'>)</span>
</code></pre>
</div>
<p><img src="index_files/figure-html5/unnamed-chunk-14-1.svg" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sme</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>Samp_type</span> <span class='op'>==</span> <span class='st'>"Experiment"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>ggplot</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span><span class='va'>SMRE</span>, fill <span class='op'>=</span> <span class='va'>Treatment</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_histogram</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_y_sqrt</span><span class='op'>(</span>breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>10</span>, <span class='fl'>50</span>, <span class='fl'>100</span>, <span class='fl'>200</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_fill_brewer</span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"qual"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>labs</span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='st'>"Mean efficiency"</span>, 
    y <span class='op'>=</span> <span class='st'>"Number of samples"</span>
  <span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>facet_wrap</span><span class='op'>(</span><span class='op'>~</span><span class='va'>Timepoint</span>, ncol <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="index_files/figure-html5/unnamed-chunk-15-1.svg" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sme</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>Samp_type</span> <span class='op'>==</span> <span class='st'>"Experiment"</span>, <span class='va'>Treatment</span> <span class='op'>!=</span> <span class='st'>"Negative"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>ggplot</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span><span class='va'>SMRE</span>, fill <span class='op'>=</span> <span class='va'>Region</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_histogram</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_y_sqrt</span><span class='op'>(</span>breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>10</span>, <span class='fl'>50</span>, <span class='fl'>100</span>, <span class='fl'>200</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_fill_brewer</span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"qual"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>labs</span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='st'>"Mean efficiency"</span>, 
    y <span class='op'>=</span> <span class='st'>"Number of samples"</span>
  <span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>facet_wrap</span><span class='op'>(</span><span class='op'>~</span><span class='va'>Timepoint</span>, ncol <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="index_files/figure-html5/unnamed-chunk-16-1.svg" width="624" /></p>
</div>
<p>Can see that Eastern genotypes more susceptible (more Mel and higher SME)</p>
<p>Are some western genotypes less susceptible?</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sme</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>Samp_type</span> <span class='op'>==</span> <span class='st'>"Experiment"</span>, <span class='va'>Region</span> <span class='op'>==</span> <span class='st'>"West"</span>,
    <span class='va'>Treatment</span> <span class='op'>!=</span> <span class='st'>"Negative"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>ggplot</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span><span class='va'>SMRE</span>, fill <span class='op'>=</span> <span class='va'>Genotype</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_histogram</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_y_sqrt</span><span class='op'>(</span>breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>10</span>, <span class='fl'>50</span>, <span class='fl'>100</span>, <span class='fl'>200</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_fill_brewer</span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"qual"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>labs</span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='st'>"Mean efficiency"</span>, 
    y <span class='op'>=</span> <span class='st'>"Number of samples"</span>
  <span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>facet_wrap</span><span class='op'>(</span><span class='op'>~</span><span class='va'>Timepoint</span>, ncol <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="index_files/figure-html5/unnamed-chunk-17-1.svg" width="624" /></p>
</div>
<p>Eastern?</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sme</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>Samp_type</span> <span class='op'>==</span> <span class='st'>"Experiment"</span>, <span class='va'>Region</span> <span class='op'>==</span> <span class='st'>"East"</span>,
    <span class='va'>Treatment</span> <span class='op'>!=</span> <span class='st'>"Negative"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>ggplot</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span><span class='va'>SMRE</span>, fill <span class='op'>=</span> <span class='va'>Genotype</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_histogram</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_y_sqrt</span><span class='op'>(</span>breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>10</span>, <span class='fl'>50</span>, <span class='fl'>100</span>, <span class='fl'>200</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_fill_brewer</span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"qual"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>labs</span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='st'>"Mean efficiency"</span>, 
    y <span class='op'>=</span> <span class='st'>"Number of samples"</span>
  <span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>facet_wrap</span><span class='op'>(</span><span class='op'>~</span><span class='va'>Timepoint</span>, ncol <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="index_files/figure-html5/unnamed-chunk-18-1.svg" width="624" /></p>
</div>
<h3 id="alt-plot">Alt plot</h3>
<p>Might be better to just focus on the treated experimental samples; can note that the bimodal distribution in SMRE is due to the presence of Mel in the timepoint 2, and thus wed mess up our AA inferences if we tried to infer the change in AA after Mel introduction. We might also (if were were analyzing timepoint 2 samples)</p>
<p>Actually, including the bias values serves the important purpose of showing how the SME varies less than the efficiencies. For that reason it could even be worth having the single-isolate samples, though seems unneeded.</p>
<p>Points</p>
<ul>
<li>SME varies less across samples than efficiencies vary across taxa</li>
<li>SME much greater after rusting (timepoint 2) -&gt; it will look like the growth of the pathogen suppressed the proportion and absolute abundance of the colonizers 5-6X more than it actually did</li>
<li>The Mel prop and hence the SME varies significantly in the timepoint 2 samples. This may lead to systematic errors in certain inferences. E.g., if host genotype affects Mel and another taxon, that other taxons value will be distorted (and, I think, the effect on Mel will be inflated).
<ul>
<li>As an example, think through the case of more resistant host genotypes</li>
<li>Might also consider the taxa found to suppress Mel; see paper</li>
</ul></li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>p1.1</span> <span class='op'>&lt;-</span> <span class='va'>sme</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>Samp_type</span> <span class='op'>==</span> <span class='st'>"Experiment"</span>, <span class='va'>Treatment</span> <span class='op'>!=</span> <span class='st'>"Negative"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>ggplot</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span><span class='va'>SMRE</span>, fill <span class='op'>=</span> <span class='va'>Timepoint</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_histogram</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>annotate</span><span class='op'>(</span><span class='st'>"point"</span>, y <span class='op'>=</span> <span class='fl'>200</span>, x <span class='op'>=</span> <span class='va'>bias</span>, color <span class='op'>=</span> <span class='st'>"black"</span>, shape <span class='op'>=</span> <span class='st'>"|"</span>, size <span class='op'>=</span> <span class='fl'>7</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>expand_limits</span><span class='op'>(</span>x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>bias</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_y_sqrt</span><span class='op'>(</span>breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>10</span>, <span class='fl'>50</span>, <span class='fl'>100</span>, <span class='fl'>200</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_fill_brewer</span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"qual"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>labs</span><span class='op'>(</span>
    <span class='co'># x = "Sample mean relative efficiency (SMRE)",</span>
    x <span class='op'>=</span> <span class='st'>"Mean efficiency"</span>, 
    y <span class='op'>=</span> <span class='st'>"Number of samples"</span>
  <span class='op'>)</span>
<span class='va'>p2.1</span> <span class='op'>&lt;-</span> <span class='va'>sme0</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>Samp_type</span> <span class='op'>==</span> <span class='st'>"Experiment"</span>, <span class='va'>Treatment</span> <span class='op'>!=</span> <span class='st'>"Negative"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>ggplot</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>Melampsora</span>, x <span class='op'>=</span> <span class='va'>SMRE</span>, color <span class='op'>=</span> <span class='va'>Timepoint</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_point</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>expand_limits</span><span class='op'>(</span>x <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>bias</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='co'># scale_y_sqrt() +</span>
  <span class='fu'>scale_y_continuous</span><span class='op'>(</span>
    trans <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/log_trans.html'>pseudo_log_trans</a></span><span class='op'>(</span>sigma <span class='op'>=</span> <span class='fl'>1e-2</span><span class='op'>)</span>,
    breaks <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>0.01</span>, <span class='fl'>0.1</span>, <span class='fl'>0.5</span>, <span class='fl'>1</span><span class='op'>)</span>
    <span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_color_brewer</span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"qual"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>labs</span><span class='op'>(</span>
    y <span class='op'>=</span> <span class='st'>"Melampsora proportion"</span>,
    x <span class='op'>=</span> <span class='st'>"Mean efficiency"</span>
  <span class='op'>)</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='op'>(</span><span class='va'>p1.1</span> <span class='op'>+</span> <span class='fu'>ggtitle</span><span class='op'>(</span><span class='st'>"Distribution of the mean efficiency across samples"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>/</span> 
  <span class='op'>(</span><span class='va'>p2.1</span> <span class='op'>+</span> <span class='fu'>ggtitle</span><span class='op'>(</span><span class='st'>"Melampsora proportion vs. mean efficiency"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_annotation.html'>plot_annotation</a></span><span class='op'>(</span>tag_levels <span class='op'>=</span> <span class='st'>"A"</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="index_files/figure-html5/unnamed-chunk-20-1.svg" width="624" /></p>
</div>
<p>TODO: Still need to determien if the y axis is accurate on histogram; I think its off when using pseudo-log. But perhaps can get away with linear?</p>
<h2 id="check-errors-in-fcs">Check errors in FCs</h2>
<p>We can use this distribution to pick examples of samples where bias approximately cancels in the FC estimates (because their SMREs are similar), and samples where bias does not approximately cancel (because their SMREs are very different). Could do for mock samples (using the ground truth) or for natural samples (using the calibrated composition). Might as well do it both ways; not sure which would be best for a paper. The mocks have the benefit of knowing the ground truth, while the natural samples demonstrate practical significance. Also, we can give nice intuitive description of what were doing in the natural samples by saying that were comparing samples w/ low-low, high-high, or low-high pathogen %.</p>
<p>TODO ^^</p>
<h3 id="mock-samples">mock samples</h3>
<p>To pick from the mock samples:</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sme0</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>Samp_type</span> <span class='op'>==</span> <span class='st'>"Mock"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>select</span><span class='op'>(</span><span class='va'>Sample</span>, <span class='va'>SMRE</span>, <span class='va'>Melampsora</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>arrange</span><span class='op'>(</span><span class='va'>SMRE</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span>
    log10_SMRE <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log10</a></span><span class='op'>(</span><span class='va'>SMRE</span><span class='op'>)</span>,
    diff <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='cn'>NA</span>, <span class='fu'><a href='https://rdrr.io/r/base/diff.html'>diff</a></span><span class='op'>(</span><span class='va'>log10_SMRE</span><span class='op'>)</span><span class='op'>)</span>
  <span class='op'>)</span>
</code></pre>
</div>
<pre><code># A tibble: 10 x 5
   Sample   SMRE Melampsora log10_SMRE     diff
   &lt;chr&gt;   &lt;dbl&gt;      &lt;dbl&gt;      &lt;dbl&gt;    &lt;dbl&gt;
 1 Mock.3  0.541    0.00782   -0.267   NA      
 2 Mock.7  0.970    0.00761   -0.0132   0.254  
 3 Mock.9  0.981    0.0116    -0.00824  0.00492
 4 Mock.2  1.16     0.0524     0.0628   0.0711 
 5 Mock.6  1.24     0.0435     0.0931   0.0303 
 6 Mock.10 1.82     0.0948     0.261    0.168  
 7 Mock.1  2.00     0.0540     0.301    0.0400 
 8 Mock.4  2.12     0.164      0.327    0.0256 
 9 Mock.8  2.20     0.134      0.343    0.0161 
10 Mock.5  2.96     0.179      0.472    0.129  </code></pre>
</div>
<ul>
<li>Mock.3 -&gt; Mock.5 have the biggest difference in SMRE. Notably, its still only a factor of 5.5 difference.</li>
<li>Mock.7 -&gt; Mock.9 have the smallest log difference.</li>
</ul>
<p>Next: Compute the actual and observed FC in proportions, for each of these pairs of communities. Perhaps define a function to spit out the relevant data frame, given any pair of samples, the phyloseq object (or calibrated phyloseq object?), and the bias vector? Actually, for the mocks, should use the actual table instead of calibrated.</p>
<p>Use these objects,</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>mock_actual</span>
</code></pre>
</div>
<pre><code>OTU Table:          [ 9 taxa and 10 samples ]:
Taxa are columns
        Melampsora Dioszegia Epicoccum Fusarium Penicillium
Mock.1      0.0538    0.0108    0.0108   0.269       0.0108
Mock.2      0.0538    0.269     0.269    0.0108      0.0108
Mock.3      0.0108    0.269     0.0538   0.0108      0.269 
Mock.4      0.269     0.0538    0.0108   0.0538      0.269 
Mock.5      0.269     0.0108    0.0108   0.269       0.0538
Mock.6      0.0538    0.0538    0.0538   0.0108      0.0108
Mock.7      0.0108    0.269     0.0538   0.269       0.269 
Mock.8      0.269     0.0108    0.269    0.0538      0.0538
Mock.9      0.0108    0.0538    0.269    0.0538      0.0538
Mock.10     0.111     0.111     0.111    0.111       0.111 
#  with 4 more taxa (columns)</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ps.mock.pseudo</span>
</code></pre>
</div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:          [ 9 taxa and 10 samples ]:
sample_data() Sample Data:        [ 10 samples by 16 sample variables ]:
tax_table()   Taxonomy Table:     [ 9 taxa by 7 taxonomic ranks ]:
refseq()      DNAStringSet:       [ 9 reference sequences ]
taxa are columns</code></pre>
</div>
<p>We can use <code>metacal::pairwise_ratios()</code> to compute the FCs in abundances between all samples + for all taxa, given a phyloseq object.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>fcs</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>Actual <span class='op'>=</span> <span class='va'>mock_actual</span>, Observed <span class='op'>=</span> <span class='va'>ps.mock.pseudo</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>map</span><span class='op'>(</span><span class='va'>transform_sample_counts</span>, <span class='va'>close_elts</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>map</span><span class='op'>(</span><span class='va'>pairwise_ratios</span>, margin <span class='op'>=</span> <span class='st'>"samples"</span>, filter <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>map_dfr</span><span class='op'>(</span><span class='va'>psmelt</span>, .id <span class='op'>=</span> <span class='st'>"Type"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='co'># redo sample.1 and sample.2 since missing for the mock</span>
  <span class='fu'>select</span><span class='op'>(</span><span class='op'>-</span><span class='va'>sample.1</span>, <span class='op'>-</span><span class='va'>sample.2</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>rename</span><span class='op'>(</span>Pair <span class='op'>=</span> <span class='va'>Sample</span>, Fold_change <span class='op'>=</span> <span class='va'>Abundance</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>separate</span><span class='op'>(</span><span class='va'>Pair</span>, <span class='fu'>str_c</span><span class='op'>(</span><span class='st'>"Sample."</span>, <span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span><span class='op'>)</span>, sep <span class='op'>=</span> <span class='st'>":"</span>, remove <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span><span class='fu'>across</span><span class='op'>(</span><span class='va'>OTU</span>, <span class='va'>factor</span>, levels <span class='op'>=</span> <span class='va'>bias</span> <span class='op'>%&gt;%</span> <span class='va'>sort</span> <span class='op'>%&gt;%</span> <span class='va'>names</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>left_join</span><span class='op'>(</span><span class='va'>sme0</span> <span class='op'>%&gt;%</span> <span class='fu'>select</span><span class='op'>(</span>Sample.1 <span class='op'>=</span> <span class='va'>Sample</span>, <span class='va'>SMRE</span>, <span class='va'>Melampsora</span><span class='op'>)</span>, 
    by <span class='op'>=</span> <span class='st'>"Sample.1"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>left_join</span><span class='op'>(</span><span class='va'>sme0</span> <span class='op'>%&gt;%</span> <span class='fu'>select</span><span class='op'>(</span>Sample.2 <span class='op'>=</span> <span class='va'>Sample</span>, <span class='va'>SMRE</span>, <span class='va'>Melampsora</span><span class='op'>)</span>, 
    by <span class='op'>=</span> <span class='st'>"Sample.2"</span>, suffix <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>".1"</span>, <span class='st'>".2"</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>fcs</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>Pair</span> <span class='op'>%in%</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Mock.5:Mock.3"</span>, <span class='st'>"Mock.9:Mock.7"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>ggplot</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>OTU</span>, x <span class='op'>=</span> <span class='va'>Fold_change</span>, shape <span class='op'>=</span> <span class='va'>Type</span>, color <span class='op'>=</span> <span class='va'>Type</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>facet_grid</span><span class='op'>(</span><span class='va'>.</span> <span class='op'>~</span> <span class='va'>Pair</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_shape_manual</span><span class='op'>(</span>values <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>Actual <span class='op'>=</span> <span class='fl'>3</span>, Observed <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_color_brewer</span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"qual"</span>, palette <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_vline</span><span class='op'>(</span>xintercept <span class='op'>=</span> <span class='fl'>1</span>, color <span class='op'>=</span> <span class='st'>"black"</span>, size <span class='op'>=</span> <span class='fl'>0.2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_point</span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_minimal_grid.html'>theme_minimal_hgrid</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_x_log10</span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>3e-3</span>, <span class='fl'>1e2</span><span class='op'>)</span>, oob <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='va'><a href='https://scales.r-lib.org/reference/oob.html'>squish</a></span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>labs</span><span class='op'>(</span>x <span class='op'>=</span> <span class='st'>"Fold change in proportion"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_annotation.html'>plot_annotation</a></span><span class='op'>(</span>
    title <span class='op'>=</span> <span class='st'>"Actual and observed fold changes between pairs of mocks"</span>
  <span class='op'>)</span>
</code></pre>
</div>
<p><img src="index_files/figure-html5/unnamed-chunk-24-1.svg" width="624" /></p>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>ggsave</span><span class='op'>(</span><span class='st'>"/tmp/leopold2020-fc-mock-examples.png"</span>, units <span class='op'>=</span> <span class='st'>"in"</span>,
  width <span class='op'>=</span> <span class='fl'>6</span>, height <span class='op'>=</span> <span class='fl'>3</span>, scale <span class='op'>=</span> <span class='fl'>1.5</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>Can see that on the left, the error is positive for all taxa, and of a similar magnitude (on the order of ~6X, as predicted) for all taxa except for Epicoccum, which dropped out during sequencing in Mock.5; and on the right, the error is smaller and not consistently positive or negative. Sign errors correspond to the observed and actual fold changes being on opposing sides of the vertical red line at 1.</p>
<p>todo: add the SMRE changes to the facet labels, and perhaps mark the expected</p>
<p>What other types of plots, which might use all pairs of samples?</p>
<ul>
<li><p>we could plot the observed vs.predicted error</p></li>
<li><p>we could check that the error really is taxon independent; one way to do this is to calc the correlation in LFC between observed and actual</p></li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>lvls</span> <span class='op'>&lt;-</span> <span class='va'>sme0</span> <span class='op'>%&gt;%</span> <span class='fu'>arrange</span><span class='op'>(</span><span class='va'>SMRE</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='fu'>pull</span><span class='op'>(</span><span class='va'>Sample</span><span class='op'>)</span>
<span class='va'>fcs0</span> <span class='op'>&lt;-</span> <span class='va'>fcs</span> <span class='op'>%&gt;%</span>
  <span class='co'># filter(Sample.1 &lt; Sample.2) %&gt;%</span>
  <span class='fu'>select</span><span class='op'>(</span><span class='va'>Type</span><span class='op'>:</span><span class='va'>Fold_change</span>, <span class='va'>SMRE.1</span><span class='op'>:</span><span class='va'>Melampsora.2</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>pivot_wider</span><span class='op'>(</span>names_from <span class='op'>=</span> <span class='va'>Type</span>, values_from <span class='op'>=</span> <span class='va'>Fold_change</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span><span class='fu'>across</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>Sample.1</span>, <span class='va'>Sample.2</span><span class='op'>)</span>, <span class='va'>factor</span>, levels <span class='op'>=</span> <span class='va'>lvls</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>fcs0</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/integer.html'>as.integer</a></span><span class='op'>(</span><span class='va'>Sample.1</span><span class='op'>)</span> <span class='op'>&lt;</span> <span class='fu'><a href='https://rdrr.io/r/base/integer.html'>as.integer</a></span><span class='op'>(</span><span class='va'>Sample.2</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>ggplot</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span><span class='va'>Actual</span>, <span class='va'>Observed</span>, color <span class='op'>=</span> <span class='va'>OTU</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_abline</span><span class='op'>(</span>color <span class='op'>=</span> <span class='st'>"grey"</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_point</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_x_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_y_log10</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>facet_grid</span><span class='op'>(</span><span class='va'>Sample.1</span> <span class='op'>~</span> <span class='va'>Sample.2</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>scale_color_brewer</span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>"qual"</span>, palette <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="index_files/figure-html5/unnamed-chunk-27-1.svg" width="624" /></p>
</div>
<p>Note how (especially) for Mock 1 and 6, the points follow a straight line with a greater slope than predicted. This seems likely related to the systematic deviation from the model that I observed.</p>
<p>Check that there is indeed very high correlations; remove the Epicoccum Mock-5 dropout first to avoid major distortions just from that.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>corrs</span> <span class='op'>&lt;-</span> <span class='va'>fcs0</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span>
    <span class='va'>Sample.1</span> <span class='op'>!=</span> <span class='va'>Sample.2</span>,
    <span class='op'>!</span><span class='op'>(</span><span class='op'>(</span><span class='va'>Sample.1</span> <span class='op'>==</span> <span class='st'>"Mock.5"</span> <span class='op'>|</span> <span class='va'>Sample.2</span> <span class='op'>==</span> <span class='st'>"Mock.5"</span><span class='op'>)</span> <span class='op'>&amp;</span> <span class='va'>OTU</span> <span class='op'>==</span> <span class='st'>"Epicoccum"</span><span class='op'>)</span>
  <span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span><span class='fu'>across</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>Actual</span>, <span class='va'>Observed</span><span class='op'>)</span>, <span class='va'>log10</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>group_by</span><span class='op'>(</span><span class='va'>Pair</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>summarize</span><span class='op'>(</span>correlation <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/cor.html'>cor</a></span><span class='op'>(</span><span class='va'>Actual</span>, <span class='va'>Observed</span>, method <span class='op'>=</span> <span class='st'>"pearson"</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>corrs</span> <span class='op'>%&gt;%</span> <span class='fu'>pull</span><span class='op'>(</span><span class='va'>correlation</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='va'>summary</span>
</code></pre>
</div>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.9637  0.9827  0.9894  0.9874  0.9929  0.9989 </code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># qplot(corrs$correlation)</span>
</code></pre>
</div>
</div>
<h3 id="experiment-samples">Experiment samples</h3>
<p>We could just try picking a high and low Melampsora sample - perhaps from the mode of the peaks, rather than the extreme.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sme0</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/r/stats/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>Samp_type</span> <span class='op'>==</span> <span class='st'>"Experiment"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>select</span><span class='op'>(</span><span class='va'>Sample</span>, <span class='va'>SMRE</span>, <span class='va'>Melampsora</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>arrange</span><span class='op'>(</span><span class='va'>SMRE</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span>
    log10_SMRE <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log10</a></span><span class='op'>(</span><span class='va'>SMRE</span><span class='op'>)</span>,
    diff <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='cn'>NA</span>, <span class='fu'><a href='https://rdrr.io/r/base/diff.html'>diff</a></span><span class='op'>(</span><span class='va'>log10_SMRE</span><span class='op'>)</span><span class='op'>)</span>
  <span class='op'>)</span>
</code></pre>
</div>
<pre><code># A tibble: 548 x 5
   Sample         SMRE Melampsora log10_SMRE       diff
   &lt;chr&gt;         &lt;dbl&gt;      &lt;dbl&gt;      &lt;dbl&gt;      &lt;dbl&gt;
 1 G12.T6.R4.TP1 0.996  0.00182     -0.00185 NA        
 2 G1.T5.R2.TP1  1.24   0.0000218    0.0939   0.0958   
 3 G11.T2.R4.TP1 1.24   0.0000836    0.0951   0.00114  
 4 G12.T1.R2.TP1 1.28   0.0000166    0.108    0.0128   
 5 G12.T2.R4.TP1 1.28   0.0000309    0.108    0.000619 
 6 G11.T3.R3.TP1 1.29   0.000253     0.110    0.00119  
 7 G2.T2.R4.TP1  1.29   0.0000220    0.110    0.000570 
 8 G8.T4.R1.TP1  1.29   0.0000132    0.110    0.0000354
 9 G9.T5.R3.TP1  1.29   0.0000714    0.110    0.000185 
10 G4.T6.R3.TP1  1.29   0.000197     0.112    0.00139  
#  with 538 more rows</code></pre>
</div>
<h1 class="appendix" id="session-info">Session info</h1>
<details>
<summary>
Click for session info
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>sessioninfo</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/sessioninfo/man/session_info.html'>session_info</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code> Session info 
 setting  value                       
 version  R version 4.0.5 (2021-03-31)
 os       Arch Linux                  
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       America/New_York            
 date     2021-04-21                  

 Packages 
 package      * version    date       lib source                           
 ade4           1.7-16     2020-10-28 [1] CRAN (R 4.0.3)                   
 ape            5.4-1      2020-08-13 [1] CRAN (R 4.0.2)                   
 assertthat     0.2.1      2019-03-21 [1] CRAN (R 4.0.0)                   
 backports      1.2.1      2020-12-09 [1] CRAN (R 4.0.3)                   
 beeswarm       0.3.1      2021-03-07 [1] CRAN (R 4.0.5)                   
 Biobase        2.50.0     2020-10-27 [1] Bioconductor                     
 BiocGenerics   0.36.0     2020-10-27 [1] Bioconductor                     
 biomformat     1.18.0     2020-10-27 [1] Bioconductor                     
 Biostrings     2.58.0     2020-10-27 [1] Bioconductor                     
 broom          0.7.6      2021-04-05 [1] CRAN (R 4.0.5)                   
 bslib          0.2.4      2021-01-25 [1] CRAN (R 4.0.4)                   
 cellranger     1.1.0      2016-07-27 [1] CRAN (R 4.0.0)                   
 cli            2.4.0      2021-04-05 [1] CRAN (R 4.0.5)                   
 cluster        2.1.1      2021-02-14 [2] CRAN (R 4.0.5)                   
 codetools      0.2-18     2020-11-04 [2] CRAN (R 4.0.5)                   
 colorspace     2.0-0      2020-11-11 [1] CRAN (R 4.0.3)                   
 cowplot      * 1.1.1      2020-12-30 [1] CRAN (R 4.0.4)                   
 crayon         1.4.1      2021-02-08 [1] CRAN (R 4.0.4)                   
 data.table     1.14.0     2021-02-21 [1] CRAN (R 4.0.4)                   
 DBI            1.1.1      2021-01-15 [1] CRAN (R 4.0.4)                   
 dbplyr         2.1.1      2021-04-06 [1] CRAN (R 4.0.5)                   
 digest         0.6.27     2020-10-24 [1] CRAN (R 4.0.3)                   
 distill        1.2.4      2021-04-05 [1] Github (rstudio/distill@3ec93f1) 
 downlit        0.2.1      2020-11-04 [1] CRAN (R 4.0.3)                   
 dplyr        * 1.0.5      2021-03-05 [1] CRAN (R 4.0.4)                   
 ellipsis       0.3.1      2020-05-15 [1] CRAN (R 4.0.0)                   
 evaluate       0.14       2019-05-28 [1] CRAN (R 4.0.0)                   
 fansi          0.4.2      2021-01-15 [1] CRAN (R 4.0.3)                   
 farver         2.1.0      2021-02-28 [1] CRAN (R 4.0.4)                   
 forcats      * 0.5.1      2021-01-27 [1] CRAN (R 4.0.4)                   
 foreach        1.5.1      2020-10-15 [1] CRAN (R 4.0.3)                   
 fs             1.5.0      2020-07-31 [1] CRAN (R 4.0.2)                   
 generics       0.1.0      2020-10-31 [1] CRAN (R 4.0.3)                   
 ggbeeswarm   * 0.6.0      2017-08-07 [1] CRAN (R 4.0.0)                   
 ggplot2      * 3.3.3      2020-12-30 [1] CRAN (R 4.0.4)                   
 glue           1.4.2      2020-08-27 [1] CRAN (R 4.0.2)                   
 gtable         0.3.0      2019-03-25 [1] CRAN (R 4.0.0)                   
 haven          2.3.1      2020-06-01 [1] CRAN (R 4.0.1)                   
 here           1.0.1      2020-12-13 [1] CRAN (R 4.0.3)                   
 highr          0.8        2019-03-20 [1] CRAN (R 4.0.0)                   
 hms            1.0.0      2021-01-13 [1] CRAN (R 4.0.4)                   
 htmltools      0.5.1.1    2021-01-22 [1] CRAN (R 4.0.3)                   
 httr           1.4.2      2020-07-20 [1] CRAN (R 4.0.2)                   
 igraph         1.2.6      2020-10-06 [1] CRAN (R 4.0.3)                   
 IRanges        2.24.1     2020-12-12 [1] Bioconductor                     
 iterators      1.0.13     2020-10-15 [1] CRAN (R 4.0.3)                   
 jquerylib      0.1.3      2020-12-17 [1] CRAN (R 4.0.4)                   
 jsonlite       1.7.2      2020-12-09 [1] CRAN (R 4.0.3)                   
 knitr          1.31       2021-01-27 [1] CRAN (R 4.0.3)                   
 lattice        0.20-41    2020-04-02 [2] CRAN (R 4.0.5)                   
 lifecycle      1.0.0      2021-02-15 [1] CRAN (R 4.0.4)                   
 lubridate      1.7.10     2021-02-26 [1] CRAN (R 4.0.4)                   
 magrittr       2.0.1      2020-11-17 [1] CRAN (R 4.0.3)                   
 MASS           7.3-53.1   2021-02-12 [2] CRAN (R 4.0.5)                   
 Matrix         1.3-2      2021-01-06 [2] CRAN (R 4.0.5)                   
 metacal      * 0.2.0      2021-04-21 [1] Github (mikemc/metacal@cf3a3c7)  
 mgcv           1.8-34     2021-02-16 [2] CRAN (R 4.0.5)                   
 modelr         0.1.8      2020-05-19 [1] CRAN (R 4.0.0)                   
 multtest       2.46.0     2020-10-27 [1] Bioconductor                     
 munsell        0.5.0      2018-06-12 [1] CRAN (R 4.0.0)                   
 nlme           3.1-152    2021-02-04 [2] CRAN (R 4.0.5)                   
 nvimcom      * 0.9-102    2021-04-05 [1] local                            
 patchwork    * 1.1.1      2020-12-17 [1] CRAN (R 4.0.3)                   
 permute        0.9-5      2019-03-12 [1] CRAN (R 4.0.0)                   
 phyloseq     * 1.34.0     2020-10-27 [1] Bioconductor                     
 pillar         1.6.0      2021-04-13 [1] CRAN (R 4.0.5)                   
 pkgconfig      2.0.3      2019-09-22 [1] CRAN (R 4.0.0)                   
 plyr           1.8.6      2020-03-03 [1] CRAN (R 4.0.0)                   
 prettyunits    1.1.1      2020-01-24 [1] CRAN (R 4.0.0)                   
 progress       1.2.2      2019-05-16 [1] CRAN (R 4.0.2)                   
 ps             1.6.0      2021-02-28 [1] CRAN (R 4.0.4)                   
 purrr        * 0.3.4      2020-04-17 [1] CRAN (R 4.0.0)                   
 R6             2.5.0      2020-10-28 [1] CRAN (R 4.0.3)                   
 RColorBrewer   1.1-2      2014-12-07 [1] CRAN (R 4.0.0)                   
 Rcpp           1.0.6      2021-01-15 [1] CRAN (R 4.0.3)                   
 readr        * 1.4.0      2020-10-05 [1] CRAN (R 4.0.3)                   
 readxl         1.3.1      2019-03-13 [1] CRAN (R 4.0.0)                   
 reprex         2.0.0      2021-04-02 [1] CRAN (R 4.0.5)                   
 reshape2       1.4.4      2020-04-09 [1] CRAN (R 4.0.0)                   
 rhdf5          2.34.0     2020-10-27 [1] Bioconductor                     
 rhdf5filters   1.2.0      2020-10-27 [1] Bioconductor                     
 Rhdf5lib       1.12.1     2021-01-26 [1] Bioconductor                     
 rlang          0.4.10     2020-12-30 [1] CRAN (R 4.0.3)                   
 rmarkdown    * 2.7        2021-02-19 [1] CRAN (R 4.0.4)                   
 rprojroot      2.0.2      2020-11-15 [1] CRAN (R 4.0.3)                   
 rstudioapi     0.13       2020-11-12 [1] CRAN (R 4.0.3)                   
 rvest          1.0.0      2021-03-09 [1] CRAN (R 4.0.5)                   
 S4Vectors      0.28.1     2020-12-09 [1] Bioconductor                     
 sass           0.3.1      2021-01-24 [1] CRAN (R 4.0.4)                   
 scales         1.1.1      2020-05-11 [1] CRAN (R 4.0.0)                   
 sessioninfo    1.1.1      2018-11-05 [1] CRAN (R 4.0.0)                   
 speedyseq    * 0.5.3.9001 2020-10-06 [1] Github (mikemc/speedyseq@8daed32)
 stringi        1.5.3      2020-09-09 [1] CRAN (R 4.0.3)                   
 stringr      * 1.4.0      2019-02-10 [1] CRAN (R 4.0.0)                   
 survival       3.2-10     2021-03-16 [2] CRAN (R 4.0.5)                   
 tibble       * 3.1.1      2021-04-18 [1] CRAN (R 4.0.5)                   
 tidyr        * 1.1.3      2021-03-03 [1] CRAN (R 4.0.4)                   
 tidyselect     1.1.0      2020-05-11 [1] CRAN (R 4.0.0)                   
 tidyverse    * 1.3.0      2019-11-21 [1] CRAN (R 4.0.0)                   
 useful         1.2.6      2018-10-08 [1] CRAN (R 4.0.0)                   
 utf8           1.2.1      2021-03-12 [1] CRAN (R 4.0.5)                   
 vctrs          0.3.7      2021-03-29 [1] CRAN (R 4.0.5)                   
 vegan          2.5-7      2020-11-28 [1] CRAN (R 4.0.3)                   
 vipor          0.4.5      2017-03-22 [1] CRAN (R 4.0.0)                   
 withr          2.4.2      2021-04-18 [1] CRAN (R 4.0.5)                   
 xfun           0.22       2021-03-11 [1] CRAN (R 4.0.5)                   
 xml2           1.3.2      2020-04-23 [1] CRAN (R 4.0.0)                   
 XVector        0.30.0     2020-10-27 [1] Bioconductor                     
 yaml           2.2.1      2020-02-01 [1] CRAN (R 4.0.0)                   
 zlibbioc       1.36.0     2020-10-27 [1] Bioconductor                     

[1] /home/michael/R/x86_64-pc-linux-gnu-library/4.0
[2] /usr/lib/R/library</code></pre>
</div>
</details>
<div class="sourceCode" id="cb14"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-leopold2020host" class="csl-entry" role="doc-biblioentry">
Leopold, Devin R, and Posy E Busby. 2020. <span><span class="nocase">Host Genotype and Colonist Arrival Order Jointly Govern Plant Microbiome Composition and Function</span>.</span> <em>Curr. Biol.</em> 30 (16): 32603266.e5. <a href="https://doi.org/10.1016/j.cub.2020.06.011">https://doi.org/10.1016/j.cub.2020.06.011</a>.
</div>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
