<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { color: #20794d; } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007ba5; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #007ba5; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #007ba5; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #20794d; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>

  <!--radix_placeholder_meta_tags-->
  <title>Evaluate robustness example</title>

  <meta property="description" itemprop="description" content="Evaluate the robustness of an estimate and inference to bias."/>


  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2021-10-18"/>
  <meta property="article:created" itemprop="dateCreated" content="2021-10-18"/>
  <meta name="article:author" content="Amy D. Willis"/>

  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="Evaluate robustness example"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="Evaluate the robustness of an estimate and inference to bias."/>
  <meta property="og:locale" content="en_US"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="Evaluate robustness example"/>
  <meta property="twitter:description" content="Evaluate the robustness of an estimate and inference to bias."/>

  <!--/radix_placeholder_meta_tags-->
  
  <meta name="citation_reference" content="citation_title=Replication and refinement of a vaginal microbial signature of preterm birth in two racially distinct cohorts of US women;citation_publication_date=2017;citation_volume=114;citation_doi=10.1073/pnas.1705899114;citation_author=Benjamin J Callahan;citation_author=Daniel B DiGiulio;citation_author=Daniela S Aliaga Goltsman;citation_author=Christine L Sun;citation_author=Elizabeth K Costello;citation_author=Pratheepa Jeganathan;citation_author=Joseph R Biggio;citation_author=Ronald J Wong;citation_author=Maurice L Druzin;citation_author=Gary M Shaw;citation_author=David K Stevenson;citation_author=Susan P Holmes;citation_author=David A Relman"/>
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","categories","date","bibliography","output"]}},"value":[{"type":"character","attributes":{},"value":["Evaluate robustness example"]},{"type":"character","attributes":{},"value":["Evaluate the robustness of an estimate and inference to bias.\n"]},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name"]}},"value":[{"type":"character","attributes":{},"value":["Amy D. Willis"]}]}]},{"type":"character","attributes":{},"value":["bias sensitivity","ref:callahan2017repl"]},{"type":"character","attributes":{},"value":["10-18-2021"]},{"type":"character","attributes":{},"value":["../../../main.bib"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["self_contained","toc","dev"]}},"value":[{"type":"logical","attributes":{},"value":[false]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["svg"]}]}]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-resource-manifest">
  {"type":"character","attributes":{},"value":["evaluate-robustness-example_files/figure-html5/summary_plot-1.svg","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/model.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/out/r1_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/out/r1_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/out/r2_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/out/r2_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/out/r3_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/out/r3_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/out/r4_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/out/r4_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/r1.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/r2.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/r3.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/r4.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/model.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/out/r1_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/out/r1_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/out/r2_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/out/r2_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/out/r3_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/out/r3_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/out/r4_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/out/r4_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/r1.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/r2.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/r3.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/r4.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/model.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/out/r1_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/out/r1_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/out/r2_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/out/r2_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/out/r3_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/out/r3_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/out/r4_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/out/r4_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/r1.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/r2.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/r3.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/r4.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/model.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/out/r1_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/out/r1_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/out/r2_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/out/r2_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/out/r3_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/out/r3_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/out/r4_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/out/r4_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/r1.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/r2.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/r3.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/r4.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/model.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/out/r1_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/out/r1_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/out/r2_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/out/r2_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/out/r3_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/out/r3_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/out/r4_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/out/r4_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/r1.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/r2.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/r3.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/r4.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/model.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/out/r1_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/out/r1_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/out/r2_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/out/r2_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/out/r3_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/out/r3_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/out/r4_corncob_evals.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/out/r4_corncob.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/r1.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/r2.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/r3.Rdata","files/eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/r4.Rdata","files/sim-UndoGardnerella.Rdata"]}
  </script>
  <!--radix_placeholder_navigation_in_header-->
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure img {
    width: 100%;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        var refHtml = $('#ref-' + $(this).attr('data-cites')).html();
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="evaluate-robustness-example_files/header-attrs-2.11/header-attrs.js"></script>
  <script src="evaluate-robustness-example_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <script src="evaluate-robustness-example_files/popper-2.6.0/popper.min.js"></script>
  <link href="evaluate-robustness-example_files/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="evaluate-robustness-example_files/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="evaluate-robustness-example_files/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="evaluate-robustness-example_files/anchor-4.2.2/anchor.min.js"></script>
  <script src="evaluate-robustness-example_files/bowser-1.9.3/bowser.min.js"></script>
  <script src="evaluate-robustness-example_files/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="evaluate-robustness-example_files/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Evaluate robustness example","description":"Evaluate the robustness of an estimate and inference to bias.","authors":[{"author":"Amy D. Willis","authorURL":"#","affiliation":"&nbsp;","affiliationURL":"#","orcidID":""}],"publishedDate":"2021-10-18T00:00:00.000-04:00","citationText":"Willis, 2021"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Evaluate robustness example</h1>
<!--radix_placeholder_categories-->
<div class="dt-tags">
<div class="dt=tag">bias sensitivity</div>
<div class="dt=tag">ref:callahan2017repl</div>
</div>
<!--/radix_placeholder_categories-->
<p><p>Evaluate the robustness of an estimate and inference to bias.</p></p>
</div>

<div class="d-byline">
  Amy D. Willis  
  
<br/>10-18-2021
</div>

<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#robustness-of-an-analysis-to-bias">Robustness of an analysis to bias</a>
<ul>
<li><a href="#example-relative-abundance-of-gardnerella">Example: relative abundance of <em>Gardnerella</em></a></li>
</ul></li>
<li><a href="#data-setup">Data setup</a></li>
<li><a href="#session-info">Session info</a></li>
</ul>
</nav>
</div>
<h1 id="robustness-of-an-analysis-to-bias">Robustness of an analysis to bias</h1>
<p>We can investigate the sensitivity of any analysis to taxonomic bias by simulating possible efficiency vectors, calibrating our observed MGS measurements by these efficiencies, and rerunning our analysis on the simulated and calibrated data. Specifically, we can take the observed proportion of each species in our dataset <span class="math inline">\(\hat{\text{prop}}_{i}(a)\)</span>, and “undo” the efficiencies to obtain the calibrated proportions <span class="math display">\[\begin{align}
  \text{calibrated prop}_{i}(a)
  &amp;= \left( \frac{\hat{\text{prop}}_{i}(a)}{\text{efficiency}_{i}} \right) \bigg / \sum_{i&#39;} \left( \frac{\hat{\text{prop}}_{i&#39;}(a)}{\text{efficiency}_{i&#39;}} \right),
\end{align}\]</span> and the calibrated read counts <span class="math display">\[\begin{align}
  \text{reads}_{i}(a)
  &amp;= \text{total reads}(a) \times \text{calibrated prop}_{i}(a).
\end{align}\]</span> We can then rerun our data analysis using the calibrated read counts in place of our original data. If we do this repeatedly, each time using different realizations of the efficiency vector, we can investigate how the results of our analysis change. If the change is minimal, this suggests that our results are robust to unequal detection efficiencies in the different species. The converse would suggest that we need to be careful interpreting our results.</p>
<h2 id="example-relative-abundance-of-gardnerella">Example: relative abundance of <em>Gardnerella</em></h2>
<p>We illustrate this method using vaginal microbiome data from pregnant women published in a study by <span class="citation" data-cites="callahan2017repl">Callahan et al. (<a href="#ref-callahan2017repl" role="doc-biblioref">2017</a>)</span> seeking to replicate associations of specific genera (<em>Lactobacillus</em>, <em>Gardnerella</em>, and <em>Ureaplasma</em>) with preterm birth in two cohorts. For example, suppose we are interested in investigating if the genus <em>Gardnerella</em> is more abundant in the vaginal microbiome of women who give birth preterm, and we investigate this model by running <code>corncob</code> using preterm birth as a covariate in the mean and dispersion model. We could investigate the robustness of our findings using the following structure:</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>our_sim</span> <span class='op'>&lt;-</span> <span class='fu'>new_simulation</span><span class='op'>(</span><span class='st'>"UndoGardnerella"</span>, <span class='st'>"Undo Gardnerella relative abundance"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>generate_model</span><span class='op'>(</span><span class='va'>undo_efficiencies</span>,
                 ps <span class='op'>=</span> <span class='va'>cal</span> <span class='op'>%&gt;%</span>
                   <span class='fu'>filter_sample_data</span><span class='op'>(</span><span class='va'>cohort</span> <span class='op'>==</span> <span class='st'>"Stanford"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
                   <span class='fu'>tax_glom</span><span class='op'>(</span><span class='st'>"genus"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
                   <span class='va'>as_tibble</span>,
                 sig_sq_e <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>as.list</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>5</span>, length.out <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span><span class='op'>)</span>,
                 vary_along <span class='op'>=</span> <span class='st'>"sig_sq_e"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>simulate_from_model</span><span class='op'>(</span>nsim <span class='op'>=</span> <span class='fl'>25</span>, index <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>4</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>run_method</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='va'>corncob_Gardnerella</span><span class='op'>)</span>,
             parallel <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>socket_names <span class='op'>=</span> <span class='fl'>4</span>, libraries <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"corncob"</span>, <span class='st'>"speedyseq"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>evaluate</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='va'>betahat_fit</span>, <span class='va'>pvalue_fit</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>In brief, this approach uses the library <code>simulator</code> to generate random efficiencies and “undo” them as described above, then runs <code>corncob</code> on these generated datasets before finally obtaining the estimated coefficients on preterm birth and the p-value for the hypothesis test that the coefficient on preterm birth is zero. Let’s take a look at the components of the code.</p>
<p>First, we load the relevant libraries and our data (see <a href="#data-setup">Data setup</a>),</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># library(here)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://tidyverse.tidyverse.org'>tidyverse</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://magrittr.tidyverse.org'>magrittr</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://mikemc.github.io/speedyseq'>speedyseq</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/bryandmartin/corncob'>corncob</a></span><span class='op'>)</span>
<span class='co'># devtools::install_github("jacobbien/simulator")</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='http://github.com/jacobbien/simulator'>simulator</a></span><span class='op'>)</span>
<span class='va'>cal</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>readRDS</a></span><span class='op'>(</span>
  <span class='fu'>here</span><span class='fu'>::</span><span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"notebook/_data"</span>, <span class='st'>"callahan2017repl"</span>, <span class='st'>"callahan2017repl_phyloseq_clean.rds"</span><span class='op'>)</span>
<span class='op'>)</span>
</code></pre>
</div>
</div>
<p>Then we can create the function that generates random efficiencies and accordingly alters the observed counts:</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>undo_efficiencies</span> <span class='op'>&lt;-</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>ps</span>, <span class='va'>sig_sq_e</span><span class='op'>)</span> <span class='op'>{</span>

  <span class='va'>ps_ms</span> <span class='op'>&lt;-</span> <span class='va'>ps</span> <span class='op'>%&gt;%</span> <span class='fu'>group_by</span><span class='op'>(</span><span class='va'>.sample</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='fu'>mutate</span><span class='op'>(</span>mm <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>.abundance</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='va'>ungroup</span>
  <span class='va'>ps_tax</span> <span class='op'>&lt;-</span> <span class='va'>ps</span> <span class='op'>%&gt;%</span> <span class='fu'>select</span><span class='op'>(</span><span class='va'>`.otu`</span>, <span class='va'>`kingdom`</span><span class='op'>:</span><span class='va'>`species`</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='va'>distinct</span> <span class='op'>%&gt;%</span> <span class='va'>tax_table</span>
  <span class='va'>ps_sample</span> <span class='op'>&lt;-</span> <span class='va'>ps</span> <span class='op'>%&gt;%</span> <span class='fu'>select</span><span class='op'>(</span><span class='va'>`.sample`</span>, <span class='va'>`subject_id`</span><span class='op'>:</span><span class='va'>`outcome`</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='va'>distinct</span> <span class='op'>%&gt;%</span> <span class='va'>sample_data</span>

  <span class='fu'><a href='https://rdrr.io/pkg/simulator/man/new_model.html'>new_model</a></span><span class='op'>(</span>name <span class='op'>=</span> <span class='st'>"eff_undo"</span>, label <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/sprintf.html'>sprintf</a></span><span class='op'>(</span><span class='st'>"sig_sq_e = %s"</span>, <span class='va'>sig_sq_e</span><span class='op'>)</span>,
            params <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>ps_ms <span class='op'>=</span> <span class='va'>ps_ms</span>, ps_tax <span class='op'>=</span> <span class='va'>ps_tax</span>, ps_sample <span class='op'>=</span> <span class='va'>ps_sample</span>, sig_sq_e <span class='op'>=</span> <span class='va'>sig_sq_e</span><span class='op'>)</span>,
            simulate <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>ps_ms</span>, <span class='va'>ps_tax</span>, <span class='va'>ps_sample</span>, <span class='va'>sig_sq_e</span>, <span class='va'>nsim</span><span class='op'>)</span> <span class='op'>{</span>
              <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>replicate</a></span><span class='op'>(</span><span class='va'>nsim</span>,
                        <span class='op'>{</span>
                          <span class='va'>effs_tib</span> <span class='op'>&lt;-</span> <span class='va'>ps_ms</span> <span class='op'>%&gt;%</span>
                            <span class='fu'>select</span><span class='op'>(</span><span class='va'>genus</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='va'>distinct</span> <span class='op'>%&gt;%</span>
                            <span class='fu'>mutate</span><span class='op'>(</span><span class='st'>"efficiency"</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/Normal.html'>rnorm</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>.</span><span class='op'>)</span>, mean <span class='op'>=</span> <span class='fl'>0</span>, sd <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>sqrt</a></span><span class='op'>(</span><span class='va'>sig_sq_e</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='va'>exp</span><span class='op'>)</span>

                          <span class='va'>ps_tib</span> <span class='op'>&lt;-</span> <span class='va'>ps_ms</span> <span class='op'>%&gt;%</span>
                            <span class='fu'>full_join</span><span class='op'>(</span><span class='va'>effs_tib</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
                            <span class='fu'>group_by</span><span class='op'>(</span><span class='va'>.sample</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
                            <span class='fu'>mutate</span><span class='op'>(</span>p_ijs <span class='op'>=</span> <span class='op'>(</span><span class='va'>.abundance</span> <span class='op'>/</span> <span class='va'>efficiency</span><span class='op'>)</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>.abundance</span> <span class='op'>/</span> <span class='va'>efficiency</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
                            <span class='fu'>mutate</span><span class='op'>(</span>w_ij <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Round.html'>round</a></span><span class='op'>(</span><span class='va'>mm</span> <span class='op'>*</span> <span class='va'>p_ijs</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
                            <span class='va'>ungroup</span>

                          <span class='fu'>phyloseq</span><span class='op'>(</span><span class='va'>ps_tib</span> <span class='op'>%&gt;%</span>
                                     <span class='fu'>select</span><span class='op'>(</span><span class='va'>`.otu`</span>, <span class='va'>`.sample`</span>, <span class='va'>`w_ij`</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
                                     <span class='fu'>pivot_wider</span><span class='op'>(</span>names_from<span class='op'>=</span><span class='va'>`.otu`</span>, values_from<span class='op'>=</span><span class='va'>`w_ij`</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
                                     <span class='fu'>otu_table</span><span class='op'>(</span>taxa_are_rows <span class='op'>=</span> <span class='cn'>F</span><span class='op'>)</span>,
                                   <span class='va'>ps_tax</span>,
                                   <span class='va'>ps_sample</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
                            <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/tax_glom.html'>tax_glom</a></span><span class='op'>(</span><span class='st'>"genus"</span><span class='op'>)</span>
                        <span class='op'>}</span>,
                        simplify<span class='op'>=</span><span class='cn'>F</span><span class='op'>)</span>
            <span class='op'>}</span><span class='op'>)</span>
<span class='op'>}</span>
</code></pre>
</div>
</div>
<p>Next, we specify how we use our data to construct our estimates. We are using <code>bbdml</code> from the library <code>corncob</code> to model the relative abundance of ASV 8 (<em>Gardnerella</em>) using preterm birth as a covariate to model the mean and dispersion, but this is only one type of analysis that we might be interested in.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>corncob_Gardnerella</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/simulator/man/new_method.html'>new_method</a></span><span class='op'>(</span><span class='st'>"corncob"</span>, <span class='st'>"Corncob"</span>,
                               method <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>model</span>, <span class='va'>draw</span><span class='op'>)</span> <span class='op'>{</span>

                                 <span class='va'>fit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/corncob/man/bbdml.html'>bbdml</a></span><span class='op'>(</span>formula <span class='op'>=</span> <span class='va'>ASV8</span> <span class='op'>~</span> <span class='va'>preterm</span>,
                                              formula_null<span class='op'>=</span> <span class='op'>~</span><span class='fl'>1</span>,
                                              phi.formula<span class='op'>=</span> <span class='op'>~</span> <span class='va'>preterm</span>,
                                              phi.formula_null<span class='op'>=</span> <span class='op'>~</span><span class='fl'>1</span>,
                                              data <span class='op'>=</span> <span class='va'>draw</span> <span class='op'>%&gt;%</span>
                                                <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/merge_samples2.html'>merge_samples2</a></span><span class='op'>(</span><span class='st'>"subject_id"</span>, fun_otu <span class='op'>=</span> <span class='va'>sum</span><span class='op'>)</span><span class='op'>)</span>

                                 <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>fit <span class='op'>=</span> <span class='va'>fit</span>,
                                      summary <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>fit</span><span class='op'>)</span><span class='op'>)</span>
                               <span class='op'>}</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>Finally, we construct methods to obtain the estimated coefficient on preterm birth in the mean relative abundance model, and its p-value.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>betahat_fit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/simulator/man/new_metric.html'>new_metric</a></span><span class='op'>(</span><span class='st'>"betahat"</span>, <span class='st'>"beta-hat"</span>,
 metric <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>model</span>, <span class='va'>out</span><span class='op'>)</span> <span class='op'>{</span>
   <span class='va'>out</span><span class='op'>$</span><span class='va'>summary</span><span class='op'>$</span><span class='va'>coefficients</span><span class='op'>[</span><span class='fl'>2</span>,<span class='fl'>1</span><span class='op'>]</span>
 <span class='op'>}</span><span class='op'>)</span>
<span class='va'>pvalue_fit</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/simulator/man/new_metric.html'>new_metric</a></span><span class='op'>(</span><span class='st'>"pvalue"</span>, <span class='st'>"p-value"</span>,
                         metric <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>model</span>, <span class='va'>out</span><span class='op'>)</span> <span class='op'>{</span>
                           <span class='va'>out</span><span class='op'>$</span><span class='va'>summary</span><span class='op'>$</span><span class='va'>coefficients</span><span class='op'>[</span><span class='fl'>2</span>,<span class='fl'>4</span><span class='op'>]</span>
                         <span class='op'>}</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>Now that we have all of the pieces, we can run them together as follows. The below code runs 100 simulations split across 4 cores; for testing, the commented code runs just 4 iterations.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>our_sim</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/simulator/man/new_simulation.html'>new_simulation</a></span><span class='op'>(</span><span class='st'>"UndoGardnerella"</span>, <span class='st'>"Undo Gardnerella relative abundance"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/simulator/man/generate_model.html'>generate_model</a></span><span class='op'>(</span><span class='va'>undo_efficiencies</span>,
                 ps <span class='op'>=</span> <span class='va'>cal</span> <span class='op'>%&gt;%</span>
                   <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/filter-phyloseq.html'>filter_sample_data</a></span><span class='op'>(</span><span class='va'>cohort</span> <span class='op'>==</span> <span class='st'>"Stanford"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
                   <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/tax_glom.html'>tax_glom</a></span><span class='op'>(</span><span class='st'>"genus"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
                   <span class='va'>as_tibble</span>,
                 sig_sq_e <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>as.list</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>5</span>, length.out <span class='op'>=</span> <span class='fl'>6</span><span class='op'>)</span><span class='op'>)</span>,
                 vary_along <span class='op'>=</span> <span class='st'>"sig_sq_e"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='co'># simulate_from_model(nsim = 4, index = 1) %&gt;%</span>
  <span class='co'># run_method(list(corncob_Gardnerella)) %&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/simulator/man/simulate_from_model.html'>simulate_from_model</a></span><span class='op'>(</span>nsim <span class='op'>=</span> <span class='fl'>25</span>, index <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>4</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/simulator/man/run_method.html'>run_method</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='va'>corncob_Gardnerella</span><span class='op'>)</span>,
             parallel <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>socket_names <span class='op'>=</span> <span class='fl'>4</span>, libraries <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"corncob"</span>, <span class='st'>"speedyseq"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://rdrr.io/pkg/simulator/man/evaluate.html'>evaluate</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='va'>betahat_fit</span>, <span class='va'>pvalue_fit</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code>#&gt; ..Created model and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/model.Rdata
#&gt; ..Created model and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/model.Rdata
#&gt; ..Created model and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/model.Rdata
#&gt; ..Created model and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/model.Rdata
#&gt; ..Created model and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/model.Rdata
#&gt; ..Created model and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/model.Rdata
#&gt; ..Simulated 25 draws in 6.59 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/r1.Rdata
#&gt; ..Simulated 25 draws in 5.37 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/r2.Rdata
#&gt; ..Simulated 25 draws in 5.73 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/r3.Rdata
#&gt; ..Simulated 25 draws in 5.73 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0/r4.Rdata
#&gt; ..Simulated 25 draws in 8.59 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/r1.Rdata
#&gt; ..Simulated 25 draws in 7.25 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/r2.Rdata
#&gt; ..Simulated 25 draws in 6.43 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/r3.Rdata
#&gt; ..Simulated 25 draws in 6.47 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1/r4.Rdata
#&gt; ..Simulated 25 draws in 7.25 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/r1.Rdata
#&gt; ..Simulated 25 draws in 6.77 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/r2.Rdata
#&gt; ..Simulated 25 draws in 5.97 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/r3.Rdata
#&gt; ..Simulated 25 draws in 6.62 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2/r4.Rdata
#&gt; ..Simulated 25 draws in 7.09 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/r1.Rdata
#&gt; ..Simulated 25 draws in 5.28 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/r2.Rdata
#&gt; ..Simulated 25 draws in 6.12 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/r3.Rdata
#&gt; ..Simulated 25 draws in 5.89 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3/r4.Rdata
#&gt; ..Simulated 25 draws in 5.66 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/r1.Rdata
#&gt; ..Simulated 25 draws in 5.49 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/r2.Rdata
#&gt; ..Simulated 25 draws in 5.31 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/r3.Rdata
#&gt; ..Simulated 25 draws in 5.28 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4/r4.Rdata
#&gt; ..Simulated 25 draws in 5.67 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/r1.Rdata
#&gt; ..Simulated 25 draws in 6.16 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/r2.Rdata
#&gt; ..Simulated 25 draws in 5.3 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/r3.Rdata
#&gt; ..Simulated 25 draws in 5.54 sec and saved in eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5/r4.Rdata
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Created 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0&quot;, index = 1, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0&quot;, index = 2, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0&quot;, index = 3, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_0&quot;, index = 4, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; in parallel.
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Created 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1&quot;, index = 1, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1&quot;, index = 2, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1&quot;, index = 3, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_1&quot;, index = 4, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; in parallel.
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Created 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2&quot;, index = 1, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2&quot;, index = 2, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2&quot;, index = 3, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_2&quot;, index = 4, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; in parallel.
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Created 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3&quot;, index = 1, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3&quot;, index = 2, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3&quot;, index = 3, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_3&quot;, index = 4, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; in parallel.
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Created 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4&quot;, index = 1, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4&quot;, index = 2, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4&quot;, index = 3, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_4&quot;, index = 4, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; in parallel.
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Performed Corncob in 0.15 seconds (on average over 25 sims)
#&gt; ..Created 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5&quot;, index = 1, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5&quot;, index = 2, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5&quot;, index = 3, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; new(&quot;OutputRef&quot;, dir = &quot;/home/michael/research/differential-abundance-theory/notebook/_posts/2021-10-18-evaluate-robustness-example&quot;, model_name = &quot;eff_undo/ps_63d55ce90e7c2b5932d2349d0790c865933f3e0d/sig_sq_e_5&quot;, index = 4, method_name = &quot;corncob&quot;, out_loc = &quot;out&quot;, simulator.files = &quot;files&quot;) 
#&gt; in parallel.
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)
#&gt; ..Evaluated Corncob in terms of 
#&gt; beta-hat, p-value, Computing time (sec)</code></pre>
</div>
<p>Here’s how the estimated coefficient on preterm birth changes under different draws from a distribution given by <span class="math inline">\(\text{efficiency}_{i} \overset{iid}{\sim} \text{LogNormal}(0, \sigma^2_e)\)</span>, and how the p-value for testing the null hypothesis that the coefficient on preterm birth is zero:</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ev_df</span> <span class='op'>&lt;-</span> <span class='va'>our_sim</span> <span class='op'>%&gt;%</span> <span class='va'>evals</span> <span class='op'>%&gt;%</span> <span class='va'>as.data.frame</span>
<span class='va'>model_df</span> <span class='op'>&lt;-</span> <span class='va'>our_sim</span> <span class='op'>%&gt;%</span> <span class='va'>model</span> <span class='op'>%&gt;%</span> <span class='va'>as.data.frame</span>
<span class='va'>ev_with_model_params</span> <span class='op'>&lt;-</span> <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate-joins.html'>right_join</a></span><span class='op'>(</span><span class='va'>model_df</span>, <span class='va'>ev_df</span>, by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"name"</span> <span class='op'>=</span> <span class='st'>"Model"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='va'>tibble</span>
<span class='va'>f_names</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='st'>'p-value'</span> <span class='op'>=</span> <span class='st'>"p-value"</span>,
                <span class='st'>'beta-hat'</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/grDevices/plotmath.html'>hat</a></span><span class='op'>(</span><span class='va'>beta</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>f_labeller</span> <span class='op'>&lt;-</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>variable</span>, <span class='va'>value</span><span class='op'>)</span><span class='op'>{</span><span class='kw'><a href='https://rdrr.io/r/base/function.html'>return</a></span><span class='op'>(</span><span class='va'>f_names</span><span class='op'>[</span><span class='va'>value</span><span class='op'>]</span><span class='op'>)</span><span class='op'>}</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ev_with_model_params</span> <span class='op'>%&gt;%</span>
  <span class='fu'>pivot_longer</span><span class='op'>(</span>cols <span class='op'>=</span> <span class='va'>betahat</span><span class='op'>:</span><span class='va'>pvalue</span>, names_to<span class='op'>=</span><span class='st'>"eval"</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>mutate</span><span class='op'>(</span>eval <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>eval</span> <span class='op'>==</span> <span class='st'>"pvalue"</span>, <span class='st'>"p-value"</span>, <span class='st'>"beta-hat"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'>ggplot</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>sig_sq_e</span>, group <span class='op'>=</span> <span class='va'>sig_sq_e</span>, y <span class='op'>=</span> <span class='va'>value</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_boxplot</span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>geom_abline</span><span class='op'>(</span><span class='fu'>aes</span><span class='op'>(</span>slope <span class='op'>=</span> <span class='va'>slope</span>, intercept <span class='op'>=</span> <span class='va'>intercept</span><span class='op'>)</span>, color <span class='op'>=</span> <span class='st'>"blue"</span>,
              <span class='fu'>tibble</span><span class='op'>(</span>eval <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"p-value"</span>, <span class='st'>"beta-hat"</span><span class='op'>)</span>, slope <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>,<span class='fl'>0</span><span class='op'>)</span>, intercept <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0.05</span>, <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>xlab</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>sigma</span><span class='op'>[</span><span class='va'>e</span><span class='op'>]</span><span class='op'>^</span><span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='co'># ggtitle("Corncob results for testing\nH0: Mean Gardnerella relative abundance is equal for preterm &amp; full-term") +</span>
  <span class='fu'>facet_wrap</span><span class='op'>(</span><span class='op'>~</span><span class='va'>eval</span>, scales<span class='op'>=</span><span class='st'>"free_y"</span>, nrow <span class='op'>=</span> <span class='fl'>2</span>, labeller <span class='op'>=</span> <span class='va'>f_labeller</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>ylab</span><span class='op'>(</span><span class='st'>""</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>cowplot</span><span class='fu'>::</span><span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="evaluate-robustness-example_files/figure-html5/summary_plot-1.svg" width="691.2" /></p>
</div>
<p>While the estimate of the coefficient does not change on average, we see more variation in its estimate as the variation in the efficiencies increases. For some realizations from a model with large variation in the efficiencies, we even see a change in the sign of <span class="math inline">\(\hat{\beta}\)</span> from positive to negative. Furthermore, we see a corresponding increase in the range of p-values, suggesting that more variation in the efficiencies can lead to either deattenuated estimates (larger p-values) or more precise non-zero estimates (smaller p-values).</p>
<p>A major advantage of this approach is that any distribution of efficiency vectors can be investigated. For example, we could specifically investigate how <em>Gardnerella</em> being low- or high-efficiency impacts our results. We could also investigate more sophisticated distributions for the efficiencies, such as correlated efficiencies across phylogeny.</p>
<h1 class="appendix" id="data-setup">Data setup</h1>
<p>The following code was used to download and prepare the data from <span class="citation" data-cites="callahan2017repl">Callahan et al. (<a href="#ref-callahan2017repl" role="doc-biblioref">2017</a>)</span> for analysis; it is not evaluated when this document is knit. The first code chunk downloads the ASV abundance matrix, sample metadata, and taxonomy assignments for the vaginal samples.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://tidyverse.tidyverse.org'>tidyverse</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://mikemc.github.io/speedyseq'>speedyseq</a></span><span class='op'>)</span>

<span class='co'># Download --------------------------------------------------------------------</span>
<span class='va'>dl_path</span> <span class='op'>&lt;-</span> <span class='fu'>here</span><span class='fu'>::</span><span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"notebook/_data"</span>, <span class='st'>"callahan2017repl"</span><span class='op'>)</span>
<span class='va'>link</span> <span class='op'>&lt;-</span> <span class='st'>"https://stacks.stanford.edu/file/druid:yb681vm1809/RepRefine_Scripts.tar.gz"</span>
<span class='kw'>if</span> <span class='op'>(</span><span class='op'>!</span><span class='fu'><a href='https://rdrr.io/r/base/files2.html'>dir.exists</a></span><span class='op'>(</span><span class='va'>dl_path</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='fu'><a href='https://rdrr.io/r/base/files2.html'>dir.create</a></span><span class='op'>(</span><span class='va'>dl_path</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/utils/download.file.html'>download.file</a></span><span class='op'>(</span><span class='va'>link</span>, <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>dl_path</span>, <span class='fu'><a href='https://rdrr.io/r/base/basename.html'>basename</a></span><span class='op'>(</span><span class='va'>link</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
  <span class='fu'><a href='https://rdrr.io/r/base/system2.html'>system2</a></span><span class='op'>(</span><span class='st'>"tar"</span>, args <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"-xvf"</span>, <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>dl_path</span>, <span class='fu'><a href='https://rdrr.io/r/base/basename.html'>basename</a></span><span class='op'>(</span><span class='va'>link</span><span class='op'>)</span><span class='op'>)</span>, <span class='st'>"-C"</span>, <span class='va'>dl_path</span><span class='op'>)</span><span class='op'>)</span>
<span class='op'>}</span>
<span class='fu'><a href='https://rdrr.io/r/base/load.html'>load</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>dl_path</span>, <span class='st'>"RepRefine_Scripts"</span>, <span class='st'>"input"</span>, <span class='st'>"processed.rda"</span><span class='op'>)</span><span class='op'>)</span>
<span class='co'># This loads:</span>
<span class='co'># df: the sample metadata</span>
<span class='co'># st: the sequence table (in counts)</span>
<span class='co'># ft: the sequence table normalized to frequencies / proportions</span>
<span class='co'># tax: taxonomy assignment matrix</span>
<span class='co'># </span>
<span class='co'># To confirm relationship between ft and st, run </span>
<span class='co'># all.equal(ft, st %&gt;% apply(1, function(x) x / sum(x)) %&gt;% t)</span>
<span class='co'># </span>
<span class='co'># The taxonomy matrix has an extra column with a duplicate name of "Species"</span>
<span class='co'># that is all NAs, so let's remove that</span>
<span class='va'>tax</span> <span class='op'>&lt;-</span> <span class='va'>tax</span><span class='op'>[</span>, <span class='op'>-</span><span class='fl'>8</span><span class='op'>]</span>
<span class='co'># Import into phyloseq --------------------------------------------------------</span>
<span class='va'>sam</span> <span class='op'>&lt;-</span> <span class='fu'>sample_data</span><span class='op'>(</span><span class='va'>df</span><span class='op'>)</span>
<span class='va'>st</span> <span class='op'>&lt;-</span> <span class='fu'>otu_table</span><span class='op'>(</span><span class='va'>st</span>, taxa_are_rows <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='va'>tax</span> <span class='op'>&lt;-</span> <span class='fu'>tax_table</span><span class='op'>(</span><span class='va'>tax</span><span class='op'>)</span>
<span class='va'>ps</span> <span class='op'>&lt;-</span> <span class='fu'>phyloseq</span><span class='op'>(</span><span class='va'>sam</span>, <span class='va'>st</span>, <span class='va'>tax</span><span class='op'>)</span>
<span class='co'># Check that all samples and taxa made it into the phyloseq object</span>
<span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/all.equal.html'>all.equal</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>st</span><span class='op'>)</span>, <span class='fu'>taxa_names</span><span class='op'>(</span><span class='va'>ps</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/all.equal.html'>all.equal</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sam</span><span class='op'>)</span>, <span class='fu'>sample_names</span><span class='op'>(</span><span class='va'>ps</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='co'># Add ASV sequences to the refseq() slot</span>
<span class='va'>seqs</span> <span class='op'>&lt;-</span> <span class='fu'>taxa_names</span><span class='op'>(</span><span class='va'>ps</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='fu'>rlang</span><span class='fu'>::</span><span class='fu'><a href='https://rlang.r-lib.org/reference/set_names.html'>set_names</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>%&gt;%</span> <span class='fu'>Biostrings</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html'>DNAStringSet</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='va'>ps</span> <span class='op'>&lt;-</span> <span class='fu'>merge_phyloseq</span><span class='op'>(</span><span class='va'>ps</span>, <span class='va'>seqs</span><span class='op'>)</span>
<span class='co'># Rename the ASVs to simpler names (ASV1, ASV2, ...). The full sequences remain</span>
<span class='co'># in `refseq(ps)`.</span>
<span class='fu'>taxa_names</span><span class='op'>(</span><span class='va'>ps</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"ASV"</span>, <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fu'>ntaxa</span><span class='op'>(</span><span class='va'>ps</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='co'># Clean up objects</span>
<span class='fu'><a href='https://rdrr.io/r/base/rm.html'>rm</a></span><span class='op'>(</span><span class='va'>tax</span>, <span class='va'>df</span>, <span class='va'>sam</span>, <span class='va'>st</span>, <span class='va'>ft</span>, <span class='va'>seqs</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>Next, we use munging functions from janitor and speedyseq to select the metadata of primary interest, standardize format of column names, and convert categorical variables to factors.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>ps1</span> <span class='op'>&lt;-</span> <span class='va'>ps</span> <span class='op'>%&gt;%</span> 
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/select-phyloseq.html'>select_sample_data</a></span><span class='op'>(</span><span class='va'>SubjectID</span>, <span class='va'>Cohort</span>, <span class='va'>Race</span>, <span class='va'>preterm</span>, <span class='va'>Outcome</span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/rename-phyloseq.html'>rename_with_sample_data</a></span><span class='op'>(</span><span class='fu'>janitor</span><span class='fu'>::</span><span class='va'><a href='https://rdrr.io/pkg/janitor/man/make_clean_names.html'>make_clean_names</a></span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/rename-phyloseq.html'>rename_with_tax_table</a></span><span class='op'>(</span><span class='fu'>janitor</span><span class='fu'>::</span><span class='va'><a href='https://rdrr.io/pkg/janitor/man/make_clean_names.html'>make_clean_names</a></span><span class='op'>)</span> <span class='op'>%&gt;%</span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/mutate-phyloseq.html'>mutate_sample_data</a></span><span class='op'>(</span>
    <span class='fu'>across</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>subject_id</span>, <span class='va'>cohort</span>, <span class='va'>race</span>, <span class='va'>outcome</span><span class='op'>)</span>, <span class='va'>factor</span><span class='op'>)</span>,
  <span class='op'>)</span>
</code></pre>
</div>
</div>
<p>Finally, we save the phyloseq object for use in the analysis.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>saveRDS</a></span><span class='op'>(</span><span class='va'>ps1</span>, <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>dl_path</span>, <span class='st'>"callahan2017repl_phyloseq_clean.rds"</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/rm.html'>rm</a></span><span class='op'>(</span><span class='va'>ps</span>, <span class='va'>ps1</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<h1 class="appendix" id="session-info">Session info</h1>
<details>
<summary>
Click for session info
</summary>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>sessioninfo</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/sessioninfo/man/session_info.html'>session_info</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code>#&gt; ─ Session info ──────────────────────────────────────────────────────────────────
#&gt;  setting  value                       
#&gt;  version  R version 4.1.1 (2021-08-10)
#&gt;  os       Arch Linux                  
#&gt;  system   x86_64, linux-gnu           
#&gt;  ui       X11                         
#&gt;  language (EN)                        
#&gt;  collate  en_US.UTF-8                 
#&gt;  ctype    en_US.UTF-8                 
#&gt;  tz       America/New_York            
#&gt;  date     2021-10-25                  
#&gt; 
#&gt; ─ Packages ──────────────────────────────────────────────────────────────────────
#&gt;  package          * version    date       lib source                              
#&gt;  ade4               1.7-17     2021-06-17 [1] CRAN (R 4.1.0)                      
#&gt;  ape                5.5        2021-04-25 [1] CRAN (R 4.1.0)                      
#&gt;  assertthat         0.2.1      2019-03-21 [1] CRAN (R 4.0.0)                      
#&gt;  backports          1.2.1      2020-12-09 [1] CRAN (R 4.0.3)                      
#&gt;  Biobase            2.52.0     2021-05-19 [1] Bioconductor                        
#&gt;  BiocGenerics       0.38.0     2021-05-19 [1] Bioconductor                        
#&gt;  biomformat         1.20.0     2021-05-19 [1] Bioconductor                        
#&gt;  Biostrings         2.60.1     2021-06-06 [1] Bioconductor                        
#&gt;  bitops             1.0-7      2021-04-24 [1] CRAN (R 4.1.0)                      
#&gt;  bookdown           0.22       2021-04-22 [1] CRAN (R 4.1.0)                      
#&gt;  broom              0.7.9      2021-07-27 [1] CRAN (R 4.1.0)                      
#&gt;  bslib              0.2.5.1    2021-05-18 [1] CRAN (R 4.1.0)                      
#&gt;  cellranger         1.1.0      2016-07-27 [1] CRAN (R 4.0.0)                      
#&gt;  cli                3.0.1      2021-07-17 [1] CRAN (R 4.1.0)                      
#&gt;  cluster            2.1.2      2021-04-17 [2] CRAN (R 4.1.1)                      
#&gt;  codetools          0.2-18     2020-11-04 [2] CRAN (R 4.1.1)                      
#&gt;  colorspace         2.0-2      2021-08-11 [1] R-Forge (R 4.1.1)                   
#&gt;  corncob          * 0.2.0      2021-03-11 [1] CRAN (R 4.0.5)                      
#&gt;  cowplot            1.1.1      2021-08-27 [1] Github (wilkelab/cowplot@555c9ae)   
#&gt;  crayon             1.4.1      2021-02-08 [1] CRAN (R 4.0.4)                      
#&gt;  data.table         1.14.2     2021-09-27 [1] CRAN (R 4.1.1)                      
#&gt;  DBI                1.1.1      2021-01-15 [1] CRAN (R 4.0.4)                      
#&gt;  dbplyr             2.1.1      2021-04-06 [1] CRAN (R 4.0.5)                      
#&gt;  digest             0.6.28     2021-09-23 [1] CRAN (R 4.1.1)                      
#&gt;  distill            1.2        2021-01-13 [1] CRAN (R 4.1.0)                      
#&gt;  downlit            0.2.1      2020-11-04 [1] CRAN (R 4.0.3)                      
#&gt;  dplyr            * 1.0.7      2021-06-18 [1] CRAN (R 4.1.0)                      
#&gt;  ellipsis           0.3.2      2021-04-29 [1] CRAN (R 4.1.0)                      
#&gt;  evaluate           0.14       2019-05-28 [1] CRAN (R 4.0.0)                      
#&gt;  fansi              0.5.0      2021-05-25 [1] CRAN (R 4.1.0)                      
#&gt;  farver             2.1.0      2021-02-28 [1] CRAN (R 4.0.4)                      
#&gt;  fastmap            1.1.0      2021-01-25 [1] CRAN (R 4.0.4)                      
#&gt;  forcats          * 0.5.1      2021-01-27 [1] CRAN (R 4.0.4)                      
#&gt;  foreach            1.5.1      2020-10-15 [1] CRAN (R 4.0.3)                      
#&gt;  fs                 1.5.0      2020-07-31 [1] CRAN (R 4.0.2)                      
#&gt;  generics           0.1.0      2020-10-31 [1] CRAN (R 4.0.3)                      
#&gt;  GenomeInfoDb       1.28.1     2021-07-01 [1] Bioconductor                        
#&gt;  GenomeInfoDbData   1.2.6      2021-05-31 [1] Bioconductor                        
#&gt;  ggplot2          * 3.3.5      2021-06-25 [1] CRAN (R 4.1.0)                      
#&gt;  glue               1.4.2      2020-08-27 [1] CRAN (R 4.0.2)                      
#&gt;  gtable             0.3.0      2019-03-25 [1] CRAN (R 4.0.0)                      
#&gt;  haven              2.4.3      2021-08-04 [1] CRAN (R 4.1.1)                      
#&gt;  here               1.0.1      2020-12-13 [1] CRAN (R 4.0.5)                      
#&gt;  highr              0.9        2021-04-16 [1] CRAN (R 4.1.0)                      
#&gt;  hms                1.1.1      2021-09-26 [1] CRAN (R 4.1.1)                      
#&gt;  htmltools          0.5.2      2021-08-25 [1] CRAN (R 4.1.1)                      
#&gt;  httr               1.4.2      2020-07-20 [1] CRAN (R 4.0.2)                      
#&gt;  igraph             1.2.6      2020-10-06 [1] CRAN (R 4.0.3)                      
#&gt;  IRanges            2.26.0     2021-05-19 [1] Bioconductor                        
#&gt;  iterators          1.0.13     2020-10-15 [1] CRAN (R 4.0.3)                      
#&gt;  jquerylib          0.1.4      2021-04-26 [1] CRAN (R 4.1.0)                      
#&gt;  jsonlite           1.7.2      2020-12-09 [1] CRAN (R 4.0.3)                      
#&gt;  knitr              1.36       2021-09-29 [1] CRAN (R 4.1.1)                      
#&gt;  labeling           0.4.2      2020-10-20 [1] CRAN (R 4.0.3)                      
#&gt;  lattice            0.20-44    2021-05-02 [2] CRAN (R 4.1.1)                      
#&gt;  lifecycle          1.0.1      2021-09-24 [1] CRAN (R 4.1.1)                      
#&gt;  lubridate          1.7.10     2021-02-26 [1] CRAN (R 4.0.4)                      
#&gt;  magrittr         * 2.0.1      2020-11-17 [1] CRAN (R 4.0.3)                      
#&gt;  MASS               7.3-54     2021-05-03 [2] CRAN (R 4.1.1)                      
#&gt;  Matrix             1.3-4      2021-06-01 [2] CRAN (R 4.1.1)                      
#&gt;  mgcv               1.8-36     2021-06-01 [2] CRAN (R 4.1.1)                      
#&gt;  modelr             0.1.8      2020-05-19 [1] CRAN (R 4.0.0)                      
#&gt;  multtest           2.48.0     2021-05-19 [1] Bioconductor                        
#&gt;  munsell            0.5.0      2018-06-12 [1] CRAN (R 4.0.0)                      
#&gt;  nlme               3.1-152    2021-02-04 [2] CRAN (R 4.1.1)                      
#&gt;  nvimcom          * 0.9-102    2021-10-25 [1] local                               
#&gt;  permute            0.9-5      2019-03-12 [1] CRAN (R 4.0.0)                      
#&gt;  phyloseq         * 1.36.0     2021-05-19 [1] Bioconductor                        
#&gt;  pillar             1.6.3      2021-09-26 [1] CRAN (R 4.1.1)                      
#&gt;  pkgconfig          2.0.3      2019-09-22 [1] CRAN (R 4.0.0)                      
#&gt;  plyr               1.8.6      2020-03-03 [1] CRAN (R 4.0.0)                      
#&gt;  purrr            * 0.3.4      2020-04-17 [1] CRAN (R 4.0.0)                      
#&gt;  R6                 2.5.1      2021-08-19 [1] CRAN (R 4.1.1)                      
#&gt;  Rcpp               1.0.7      2021-07-07 [1] CRAN (R 4.1.0)                      
#&gt;  RCurl              1.98-1.3   2021-03-16 [1] CRAN (R 4.0.5)                      
#&gt;  readr            * 2.0.2      2021-09-27 [1] CRAN (R 4.1.1)                      
#&gt;  readxl             1.3.1      2019-03-13 [1] CRAN (R 4.0.0)                      
#&gt;  reprex             2.0.1      2021-08-05 [1] CRAN (R 4.1.1)                      
#&gt;  reshape2           1.4.4      2020-04-09 [1] CRAN (R 4.0.0)                      
#&gt;  rhdf5              2.36.0     2021-05-19 [1] Bioconductor                        
#&gt;  rhdf5filters       1.4.0      2021-05-19 [1] Bioconductor                        
#&gt;  Rhdf5lib           1.14.2     2021-07-06 [1] Bioconductor                        
#&gt;  rlang              0.4.11     2021-04-30 [1] CRAN (R 4.1.0)                      
#&gt;  rmarkdown        * 2.11       2021-09-14 [1] CRAN (R 4.1.1)                      
#&gt;  rprojroot          2.0.2      2020-11-15 [1] CRAN (R 4.0.3)                      
#&gt;  rstudioapi         0.13       2020-11-12 [1] CRAN (R 4.0.3)                      
#&gt;  rvest              1.0.1      2021-07-26 [1] CRAN (R 4.1.0)                      
#&gt;  S4Vectors          0.30.0     2021-05-19 [1] Bioconductor                        
#&gt;  sass               0.4.0      2021-05-12 [1] CRAN (R 4.1.0)                      
#&gt;  scales             1.1.1      2020-05-11 [1] CRAN (R 4.0.0)                      
#&gt;  sessioninfo        1.1.1      2018-11-05 [1] CRAN (R 4.0.0)                      
#&gt;  simulator        * 0.2.3      2021-10-25 [1] Github (jacobbien/simulator@aef35ab)
#&gt;  speedyseq        * 0.5.3.9018 2021-06-29 [1] Github (mikemc/speedyseq@ceb941f)   
#&gt;  stringi            1.7.5      2021-10-04 [1] CRAN (R 4.1.1)                      
#&gt;  stringr          * 1.4.0      2019-02-10 [1] CRAN (R 4.0.0)                      
#&gt;  survival           3.2-11     2021-04-26 [2] CRAN (R 4.1.1)                      
#&gt;  tibble           * 3.1.5      2021-09-30 [1] CRAN (R 4.1.1)                      
#&gt;  tidyr            * 1.1.4      2021-09-27 [1] CRAN (R 4.1.1)                      
#&gt;  tidyselect         1.1.1      2021-04-30 [1] CRAN (R 4.1.0)                      
#&gt;  tidyverse        * 1.3.1      2021-04-15 [1] CRAN (R 4.1.0)                      
#&gt;  tzdb               0.1.2      2021-07-20 [1] CRAN (R 4.1.0)                      
#&gt;  utf8               1.2.2      2021-07-24 [1] CRAN (R 4.1.0)                      
#&gt;  vctrs              0.3.8      2021-04-29 [1] CRAN (R 4.1.0)                      
#&gt;  vegan              2.5-7      2020-11-28 [1] CRAN (R 4.0.3)                      
#&gt;  withr              2.4.2      2021-04-18 [1] CRAN (R 4.0.5)                      
#&gt;  xfun               0.26       2021-09-14 [1] CRAN (R 4.1.1)                      
#&gt;  xml2               1.3.2      2020-04-23 [1] CRAN (R 4.0.0)                      
#&gt;  XVector            0.32.0     2021-05-19 [1] Bioconductor                        
#&gt;  yaml               2.2.1      2020-02-01 [1] CRAN (R 4.0.0)                      
#&gt;  zlibbioc           1.38.0     2021-05-19 [1] Bioconductor                        
#&gt; 
#&gt; [1] /home/michael/.local/lib/R/library
#&gt; [2] /usr/lib/R/library</code></pre>
</div>
</details>
<div class="sourceCode" id="cb3"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-callahan2017repl" class="csl-entry" role="doc-biblioentry">
Callahan, Benjamin J, Daniel B DiGiulio, Daniela S Aliaga Goltsman, Christine L Sun, Elizabeth K Costello, Pratheepa Jeganathan, Joseph R Biggio, et al. 2017. <span>“<span class="nocase">Replication and refinement of a vaginal microbial signature of preterm birth in two racially distinct cohorts of US women</span>.”</span> <em>Proc. Natl. Acad. Sci. U. S. A.</em> 114 (37): 9966–71. <a href="https://doi.org/10.1073/pnas.1705899114">https://doi.org/10.1073/pnas.1705899114</a>.
</div>
</div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
<h3 id="references">References</h3>
<div id="references-listing"></div>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
