<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007ba5; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #007ba5; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #007ba5; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #5e5e5e; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>


  <!--radix_placeholder_meta_tags-->
<title>Research notebook for manuscript on taxonomic bias and differential-abundance methods: MOMS-PI: Regression on diversity</title>

<meta property="description" itemprop="description" content="Analyze the impact of bias on a DA analysis of relative abundances versus diversity."/>


<!--  https://schema.org/Article -->
<meta property="article:published" itemprop="datePublished" content="2021-12-15"/>
<meta property="article:created" itemprop="dateCreated" content="2021-12-15"/>
<meta name="article:author" content="Michael R. McLaren"/>

<!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
<meta property="og:title" content="Research notebook for manuscript on taxonomic bias and differential-abundance methods: MOMS-PI: Regression on diversity"/>
<meta property="og:type" content="article"/>
<meta property="og:description" content="Analyze the impact of bias on a DA analysis of relative abundances versus diversity."/>
<meta property="og:locale" content="en_US"/>
<meta property="og:site_name" content="Research notebook for manuscript on taxonomic bias and differential-abundance methods"/>

<!--  https://dev.twitter.com/cards/types/summary -->
<meta property="twitter:card" content="summary"/>
<meta property="twitter:title" content="Research notebook for manuscript on taxonomic bias and differential-abundance methods: MOMS-PI: Regression on diversity"/>
<meta property="twitter:description" content="Analyze the impact of bias on a DA analysis of relative abundances versus diversity."/>

<!--/radix_placeholder_meta_tags-->
  
  <!--radix_placeholder_rmarkdown_metadata-->

<script type="text/json" id="radix-rmarkdown-metadata">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","categories","date","draft","bibliography","output"]}},"value":[{"type":"character","attributes":{},"value":["MOMS-PI: Regression on diversity"]},{"type":"character","attributes":{},"value":["Analyze the impact of bias on a DA analysis of relative abundances versus diversity."]},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url"]}},"value":[{"type":"character","attributes":{},"value":["Michael R. McLaren"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":[]}},"value":[]}]}]},{"type":"character","attributes":{},"value":["ref:brooks2015thet","ref:fettweis2019thev"]},{"type":"character","attributes":{},"value":["2021-12-15"]},{"type":"logical","attributes":{},"value":[false]},{"type":"character","attributes":{},"value":["../../../main.bib"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["self_contained","dev","toc"]}},"value":[{"type":"logical","attributes":{},"value":[false]},{"type":"character","attributes":{},"value":["svg"]},{"type":"logical","attributes":{},"value":[true]}]}]}]}
</script>
<!--/radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-resource-manifest">
  {"type":"character","attributes":{},"value":["momspi-regression-diversity_files/anchor-4.2.2/anchor.min.js","momspi-regression-diversity_files/bowser-1.9.3/bowser.min.js","momspi-regression-diversity_files/distill-2.2.21/template.v2.js","momspi-regression-diversity_files/figure-html5/regression-lm-1.svg","momspi-regression-diversity_files/figure-html5/regression-lm-alt-1.svg","momspi-regression-diversity_files/figure-html5/unnamed-chunk-16-1.svg","momspi-regression-diversity_files/figure-html5/unnamed-chunk-20-1.png","momspi-regression-diversity_files/figure-html5/unnamed-chunk-20-1.svg","momspi-regression-diversity_files/figure-html5/unnamed-chunk-7-1.png","momspi-regression-diversity_files/figure-html5/unnamed-chunk-7-1.svg","momspi-regression-diversity_files/figure-html5/unnamed-chunk-8-1.svg","momspi-regression-diversity_files/header-attrs-2.11/header-attrs.js","momspi-regression-diversity_files/jquery-3.6.0/jquery-3.6.0.js","momspi-regression-diversity_files/jquery-3.6.0/jquery-3.6.0.min.js","momspi-regression-diversity_files/jquery-3.6.0/jquery-3.6.0.min.map","momspi-regression-diversity_files/popper-2.6.0/popper.min.js","momspi-regression-diversity_files/tippy-6.2.7/tippy-bundle.umd.min.js","momspi-regression-diversity_files/tippy-6.2.7/tippy-light-border.css","momspi-regression-diversity_files/tippy-6.2.7/tippy.css","momspi-regression-diversity_files/tippy-6.2.7/tippy.umd.min.js","momspi-regression-diversity_files/webcomponents-2.0.0/webcomponents.js"]}
  </script>
  <!--radix_placeholder_navigation_in_header-->
<meta name="distill:offset" content="../.."/>

<script type="application/javascript">

  window.headroom_prevent_pin = false;

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // initialize headroom for banner
    var header = $('header').get(0);
    var headerHeight = header.offsetHeight;
    var headroom = new Headroom(header, {
      tolerance: 5,
      onPin : function() {
        if (window.headroom_prevent_pin) {
          window.headroom_prevent_pin = false;
          headroom.unpin();
        }
      }
    });
    headroom.init();
    if(window.location.hash)
      headroom.unpin();
    $(header).addClass('headroom--transition');

    // offset scroll location for banner on hash change
    // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
    window.addEventListener("hashchange", function(event) {
      window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
    });

    // responsive menu
    $('.distill-site-header').each(function(i, val) {
      var topnav = $(this);
      var toggle = topnav.find('.nav-toggle');
      toggle.on('click', function() {
        topnav.toggleClass('responsive');
      });
    });

    // nav dropdowns
    $('.nav-dropbtn').click(function(e) {
      $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
      $(this).parent().siblings('.nav-dropdown')
         .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
    });
    $("body").click(function(e){
      $('.nav-dropdown-content').removeClass('nav-dropdown-active');
    });
    $(".nav-dropdown").click(function(e){
      e.stopPropagation();
    });
  });
</script>

<style type="text/css">

/* Theme (user-documented overrideables for nav appearance) */

.distill-site-nav {
  color: rgba(255, 255, 255, 0.8);
  background-color: #0F2E3D;
  font-size: 15px;
  font-weight: 300;
}

.distill-site-nav a {
  color: inherit;
  text-decoration: none;
}

.distill-site-nav a:hover {
  color: white;
}

@media print {
  .distill-site-nav {
    display: none;
  }
}

.distill-site-header {

}

.distill-site-footer {

}


/* Site Header */

.distill-site-header {
  width: 100%;
  box-sizing: border-box;
  z-index: 3;
}

.distill-site-header .nav-left {
  display: inline-block;
  margin-left: 8px;
}

@media screen and (max-width: 768px) {
  .distill-site-header .nav-left {
    margin-left: 0;
  }
}


.distill-site-header .nav-right {
  float: right;
  margin-right: 8px;
}

.distill-site-header a,
.distill-site-header .title {
  display: inline-block;
  text-align: center;
  padding: 14px 10px 14px 10px;
}

.distill-site-header .title {
  font-size: 18px;
  min-width: 150px;
}

.distill-site-header .logo {
  padding: 0;
}

.distill-site-header .logo img {
  display: none;
  max-height: 20px;
  width: auto;
  margin-bottom: -4px;
}

.distill-site-header .nav-image img {
  max-height: 18px;
  width: auto;
  display: inline-block;
  margin-bottom: -3px;
}



@media screen and (min-width: 1000px) {
  .distill-site-header .logo img {
    display: inline-block;
  }
  .distill-site-header .nav-left {
    margin-left: 20px;
  }
  .distill-site-header .nav-right {
    margin-right: 20px;
  }
  .distill-site-header .title {
    padding-left: 12px;
  }
}


.distill-site-header .nav-toggle {
  display: none;
}

.nav-dropdown {
  display: inline-block;
  position: relative;
}

.nav-dropdown .nav-dropbtn {
  border: none;
  outline: none;
  color: rgba(255, 255, 255, 0.8);
  padding: 16px 10px;
  background-color: transparent;
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
  margin: 0;
  margin-top: 1px;
  z-index: 2;
}

.nav-dropdown-content {
  display: none;
  position: absolute;
  background-color: white;
  min-width: 200px;
  border: 1px solid rgba(0,0,0,0.15);
  border-radius: 4px;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
  z-index: 1;
  margin-top: 2px;
  white-space: nowrap;
  padding-top: 4px;
  padding-bottom: 4px;
}

.nav-dropdown-content hr {
  margin-top: 4px;
  margin-bottom: 4px;
  border: none;
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
}

.nav-dropdown-active {
  display: block;
}

.nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
  color: black;
  padding: 6px 24px;
  text-decoration: none;
  display: block;
  text-align: left;
}

.nav-dropdown-content .nav-dropdown-header {
  display: block;
  padding: 5px 24px;
  padding-bottom: 0;
  text-transform: uppercase;
  font-size: 14px;
  color: #999999;
  white-space: nowrap;
}

.nav-dropdown:hover .nav-dropbtn {
  color: white;
}

.nav-dropdown-content a:hover {
  background-color: #ddd;
  color: black;
}

.nav-right .nav-dropdown-content {
  margin-left: -45%;
  right: 0;
}

@media screen and (max-width: 768px) {
  .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
  .distill-site-header a.nav-toggle {
    float: right;
    display: block;
  }
  .distill-site-header .title {
    margin-left: 0;
  }
  .distill-site-header .nav-right {
    margin-right: 0;
  }
  .distill-site-header {
    overflow: hidden;
  }
  .nav-right .nav-dropdown-content {
    margin-left: 0;
  }
}


@media screen and (max-width: 768px) {
  .distill-site-header.responsive {position: relative; min-height: 500px; }
  .distill-site-header.responsive a.nav-toggle {
    position: absolute;
    right: 0;
    top: 0;
  }
  .distill-site-header.responsive a,
  .distill-site-header.responsive .nav-dropdown {
    display: block;
    text-align: left;
  }
  .distill-site-header.responsive .nav-left,
  .distill-site-header.responsive .nav-right {
    width: 100%;
  }
  .distill-site-header.responsive .nav-dropdown {float: none;}
  .distill-site-header.responsive .nav-dropdown-content {position: relative;}
  .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
    display: block;
    width: 100%;
    text-align: left;
  }
}

/* Site Footer */

.distill-site-footer {
  width: 100%;
  overflow: hidden;
  box-sizing: border-box;
  z-index: 3;
  margin-top: 30px;
  padding-top: 30px;
  padding-bottom: 30px;
  text-align: center;
}

/* Headroom */

d-title {
  padding-top: 6rem;
}

@media print {
  d-title {
    padding-top: 4rem;
  }
}

.headroom {
  z-index: 1000;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
}

.headroom--transition {
  transition: all .4s ease-in-out;
}

.headroom--unpinned {
  top: -100px;
}

.headroom--pinned {
  top: 0;
}

/* adjust viewport for navbar height */
/* helps vertically center bootstrap (non-distill) content */
.min-vh-100 {
  min-height: calc(100vh - 100px) !important;
}

</style>

<script src="../../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="../../site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet"/>
<link href="../../site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet"/>
<script src="../../site_libs/headroom-0.9.4/headroom.min.js"></script>
<script src="../../site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
<script src="../../site_libs/fuse-6.4.1/fuse.min.js"></script>

<script type="application/javascript">

function getMeta(metaName) {
  var metas = document.getElementsByTagName('meta');
  for (let i = 0; i < metas.length; i++) {
    if (metas[i].getAttribute('name') === metaName) {
      return metas[i].getAttribute('content');
    }
  }
  return '';
}

function offsetURL(url) {
  var offset = getMeta('distill:offset');
  return offset ? offset + '/' + url : url;
}

function createFuseIndex() {

  // create fuse index
  var options = {
    keys: [
      { name: 'title', weight: 20 },
      { name: 'categories', weight: 15 },
      { name: 'description', weight: 10 },
      { name: 'contents', weight: 5 },
    ],
    ignoreLocation: true,
    threshold: 0
  };
  var fuse = new window.Fuse([], options);

  // fetch the main search.json
  return fetch(offsetURL('search.json'))
    .then(function(response) {
      if (response.status == 200) {
        return response.json().then(function(json) {
          // index main articles
          json.articles.forEach(function(article) {
            fuse.add(article);
          });
          // download collections and index their articles
          return Promise.all(json.collections.map(function(collection) {
            return fetch(offsetURL(collection)).then(function(response) {
              if (response.status === 200) {
                return response.json().then(function(articles) {
                  articles.forEach(function(article) {
                    fuse.add(article);
                  });
                })
              } else {
                return Promise.reject(
                  new Error('Unexpected status from search index request: ' +
                            response.status)
                );
              }
            });
          })).then(function() {
            return fuse;
          });
        });

      } else {
        return Promise.reject(
          new Error('Unexpected status from search index request: ' +
                      response.status)
        );
      }
    });
}

window.document.addEventListener("DOMContentLoaded", function (event) {

  // get search element (bail if we don't have one)
  var searchEl = window.document.getElementById('distill-search');
  if (!searchEl)
    return;

  createFuseIndex()
    .then(function(fuse) {

      // make search box visible
      searchEl.classList.remove('hidden');

      // initialize autocomplete
      var options = {
        autoselect: true,
        hint: false,
        minLength: 2,
      };
      window.autocomplete(searchEl, options, [{
        source: function(query, callback) {
          const searchOptions = {
            isCaseSensitive: false,
            shouldSort: true,
            minMatchCharLength: 2,
            limit: 10,
          };
          var results = fuse.search(query, searchOptions);
          callback(results
            .map(function(result) { return result.item; })
          );
        },
        templates: {
          suggestion: function(suggestion) {
            var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
              ? `<img src="${offsetURL(suggestion.preview)}"</img>`
              : '';
            var html = `
              <div class="search-item">
                <h3>${suggestion.title}</h3>
                <div class="search-item-description">
                  ${suggestion.description || ''}
                </div>
                <div class="search-item-preview">
                  ${img}
                </div>
              </div>
            `;
            return html;
          }
        }
      }]).on('autocomplete:selected', function(event, suggestion) {
        window.location.href = offsetURL(suggestion.path);
      });
      // remove inline display style on autocompleter (we want to
      // manage responsive display via css)
      $('.algolia-autocomplete').css("display", "");
    })
    .catch(function(error) {
      console.log(error);
    });

});

</script>

<style type="text/css">

.nav-search {
  font-size: x-small;
}

/* Algolioa Autocomplete */

.algolia-autocomplete {
  display: inline-block;
  margin-left: 10px;
  vertical-align: sub;
  background-color: white;
  color: black;
  padding: 6px;
  padding-top: 8px;
  padding-bottom: 0;
  border-radius: 6px;
  border: 1px #0F2E3D solid;
  width: 180px;
}


@media screen and (max-width: 768px) {
  .distill-site-nav .algolia-autocomplete {
    display: none;
    visibility: hidden;
  }
  .distill-site-nav.responsive .algolia-autocomplete {
    display: inline-block;
    visibility: visible;
  }
  .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
    margin-left: 0;
    width: 400px;
    max-height: 400px;
  }
}

.algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
  width: 90%;
  outline: none;
  border: none;
}

.algolia-autocomplete .aa-hint {
  color: #999;
}
.algolia-autocomplete .aa-dropdown-menu {
  width: 550px;
  max-height: 70vh;
  overflow-x: visible;
  overflow-y: scroll;
  padding: 5px;
  margin-top: 3px;
  margin-left: -150px;
  background-color: #fff;
  border-radius: 5px;
  border: 1px solid #999;
  border-top: none;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
  cursor: pointer;
  padding: 5px 4px;
  border-bottom: 1px solid #eee;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
  border-bottom: none;
  margin-bottom: 2px;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
  overflow: hidden;
  font-size: 0.8em;
  line-height: 1.4em;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
  font-size: 1rem;
  margin-block-start: 0;
  margin-block-end: 5px;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
  display: inline-block;
  overflow: hidden;
  height: 2.8em;
  width: 80%;
  margin-right: 4%;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
  display: inline-block;
  width: 15%;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
  height: 3em;
  width: auto;
  display: none;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
  display: initial;
}

.algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
  background-color: #eee;
}
.algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
  font-weight: bold;
  font-style: normal;
}

</style>


<!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

<style type="text/css">

body {
  background-color: white;
}

.pandoc-table {
  width: 100%;
}

.pandoc-table>caption {
  margin-bottom: 10px;
}

.pandoc-table th:not([align]) {
  text-align: left;
}

.pagedtable-footer {
  font-size: 15px;
}

d-byline .byline {
  grid-template-columns: 2fr 2fr;
}

d-byline .byline h3 {
  margin-block-start: 1.5em;
}

d-byline .byline .authors-affiliations h3 {
  margin-block-start: 0.5em;
}

.authors-affiliations .orcid-id {
  width: 16px;
  height:16px;
  margin-left: 4px;
  margin-right: 4px;
  vertical-align: middle;
  padding-bottom: 2px;
}

d-title .dt-tags {
  margin-top: 1em;
  grid-column: text;
}

.dt-tags .dt-tag {
  text-decoration: none;
  display: inline-block;
  color: rgba(0,0,0,0.6);
  padding: 0em 0.4em;
  margin-right: 0.5em;
  margin-bottom: 0.4em;
  font-size: 70%;
  border: 1px solid rgba(0,0,0,0.2);
  border-radius: 3px;
  text-transform: uppercase;
  font-weight: 500;
}

d-article table.gt_table td,
d-article table.gt_table th {
  border-bottom: none;
}

.html-widget {
  margin-bottom: 2.0em;
}

.l-screen-inset {
  padding-right: 16px;
}

.l-screen .caption {
  margin-left: 10px;
}

.shaded {
  background: rgb(247, 247, 247);
  padding-top: 20px;
  padding-bottom: 20px;
  border-top: 1px solid rgba(0, 0, 0, 0.1);
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
}

.shaded .html-widget {
  margin-bottom: 0;
  border: 1px solid rgba(0, 0, 0, 0.1);
}

.shaded .shaded-content {
  background: white;
}

.text-output {
  margin-top: 0;
  line-height: 1.5em;
}

.hidden {
  display: none !important;
}

d-article {
  padding-top: 2.5rem;
  padding-bottom: 30px;
}

d-appendix {
  padding-top: 30px;
}

d-article>p>img {
  width: 100%;
}

d-article h2 {
  margin: 1rem 0 1.5rem 0;
}

d-article h3 {
  margin-top: 1.5rem;
}

d-article iframe {
  border: 1px solid rgba(0, 0, 0, 0.1);
  margin-bottom: 2.0em;
  width: 100%;
}

/* Tweak code blocks */

d-article div.sourceCode code,
d-article pre code {
  font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
}

d-article pre,
d-article div.sourceCode,
d-article div.sourceCode pre {
  overflow: auto;
}

d-article div.sourceCode {
  background-color: white;
}

d-article div.sourceCode pre {
  padding-left: 10px;
  font-size: 12px;
  border-left: 2px solid rgba(0,0,0,0.1);
}

d-article pre {
  font-size: 12px;
  color: black;
  background: none;
  margin-top: 0;
  text-align: left;
  white-space: pre;
  word-spacing: normal;
  word-break: normal;
  word-wrap: normal;
  line-height: 1.5;

  -moz-tab-size: 4;
  -o-tab-size: 4;
  tab-size: 4;

  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  hyphens: none;
}

d-article pre a {
  border-bottom: none;
}

d-article pre a:hover {
  border-bottom: none;
  text-decoration: underline;
}

d-article details {
  grid-column: text;
  margin-bottom: 0.8em;
}

@media(min-width: 768px) {

d-article pre,
d-article div.sourceCode,
d-article div.sourceCode pre {
  overflow: visible !important;
}

d-article div.sourceCode pre {
  padding-left: 18px;
  font-size: 14px;
}

d-article pre {
  font-size: 14px;
}

}

figure img.external {
  background: white;
  border: 1px solid rgba(0, 0, 0, 0.1);
  box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
  padding: 18px;
  box-sizing: border-box;
}

/* CSS for d-contents */

.d-contents {
  grid-column: text;
  color: rgba(0,0,0,0.8);
  font-size: 0.9em;
  padding-bottom: 1em;
  margin-bottom: 1em;
  padding-bottom: 0.5em;
  margin-bottom: 1em;
  padding-left: 0.25em;
  justify-self: start;
}

@media(min-width: 1000px) {
  .d-contents.d-contents-float {
    height: 0;
    grid-column-start: 1;
    grid-column-end: 4;
    justify-self: center;
    padding-right: 3em;
    padding-left: 2em;
  }
}

.d-contents nav h3 {
  font-size: 18px;
  margin-top: 0;
  margin-bottom: 1em;
}

.d-contents li {
  list-style-type: none
}

.d-contents nav > ul {
  padding-left: 0;
}

.d-contents ul {
  padding-left: 1em
}

.d-contents nav ul li {
  margin-top: 0.6em;
  margin-bottom: 0.2em;
}

.d-contents nav a {
  font-size: 13px;
  border-bottom: none;
  text-decoration: none
  color: rgba(0, 0, 0, 0.8);
}

.d-contents nav a:hover {
  text-decoration: underline solid rgba(0, 0, 0, 0.6)
}

.d-contents nav > ul > li > a {
  font-weight: 600;
}

.d-contents nav > ul > li > ul {
  font-weight: inherit;
}

.d-contents nav > ul > li > ul > li {
  margin-top: 0.2em;
}


.d-contents nav ul {
  margin-top: 0;
  margin-bottom: 0.25em;
}

.d-article-with-toc h2:nth-child(2) {
  margin-top: 0;
}


/* Figure */

.figure {
  position: relative;
  margin-bottom: 2.5em;
  margin-top: 1.5em;
}

.figure img {
  width: 100%;
}

.figure .caption {
  color: rgba(0, 0, 0, 0.6);
  font-size: 12px;
  line-height: 1.5em;
}

.figure img.external {
  background: white;
  border: 1px solid rgba(0, 0, 0, 0.1);
  box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
  padding: 18px;
  box-sizing: border-box;
}

.figure .caption a {
  color: rgba(0, 0, 0, 0.6);
}

.figure .caption b,
.figure .caption strong, {
  font-weight: 600;
  color: rgba(0, 0, 0, 1.0);
}

/* Citations */

d-article .citation {
  color: inherit;
  cursor: inherit;
}

div.hanging-indent{
  margin-left: 1em; text-indent: -1em;
}

/* Citation hover box */

.tippy-box[data-theme~=light-border] {
  background-color: rgba(250, 250, 250, 0.95);
}

.tippy-content > p {
  margin-bottom: 0;
  padding: 2px;
}


/* Tweak 1000px media break to show more text */

@media(min-width: 1000px) {
  .base-grid,
  distill-header,
  d-title,
  d-abstract,
  d-article,
  d-appendix,
  distill-appendix,
  d-byline,
  d-footnote-list,
  d-citation-list,
  distill-footer {
    grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
    grid-column-gap: 16px;
  }

  .grid {
    grid-column-gap: 16px;
  }

  d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
  }
  figure .caption, .figure .caption, figure figcaption {
    font-size: 13px;
  }
}

@media(min-width: 1180px) {
  .base-grid,
  distill-header,
  d-title,
  d-abstract,
  d-article,
  d-appendix,
  distill-appendix,
  d-byline,
  d-footnote-list,
  d-citation-list,
  distill-footer {
    grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
    grid-column-gap: 32px;
  }

  .grid {
    grid-column-gap: 32px;
  }
}


/* Get the citation styles for the appendix (not auto-injected on render since
   we do our own rendering of the citation appendix) */

d-appendix .citation-appendix,
.d-appendix .citation-appendix {
  font-size: 11px;
  line-height: 15px;
  border-left: 1px solid rgba(0, 0, 0, 0.1);
  padding-left: 18px;
  border: 1px solid rgba(0,0,0,0.1);
  background: rgba(0, 0, 0, 0.02);
  padding: 10px 18px;
  border-radius: 3px;
  color: rgba(150, 150, 150, 1);
  overflow: hidden;
  margin-top: -12px;
  white-space: pre-wrap;
  word-wrap: break-word;
}

/* Include appendix styles here so they can be overridden */

d-appendix {
  contain: layout style;
  font-size: 0.8em;
  line-height: 1.7em;
  margin-top: 60px;
  margin-bottom: 0;
  border-top: 1px solid rgba(0, 0, 0, 0.1);
  color: rgba(0,0,0,0.5);
  padding-top: 60px;
  padding-bottom: 48px;
}

d-appendix h3 {
  grid-column: page-start / text-start;
  font-size: 15px;
  font-weight: 500;
  margin-top: 1em;
  margin-bottom: 0;
  color: rgba(0,0,0,0.65);
}

d-appendix h3 + * {
  margin-top: 1em;
}

d-appendix ol {
  padding: 0 0 0 15px;
}

@media (min-width: 768px) {
  d-appendix ol {
    padding: 0 0 0 30px;
    margin-left: -30px;
  }
}

d-appendix li {
  margin-bottom: 1em;
}

d-appendix a {
  color: rgba(0, 0, 0, 0.6);
}

d-appendix > * {
  grid-column: text;
}

d-appendix > d-footnote-list,
d-appendix > d-citation-list,
d-appendix > distill-appendix {
  grid-column: screen;
}

/* Include footnote styles here so they can be overridden */

d-footnote-list {
  contain: layout style;
}

d-footnote-list > * {
  grid-column: text;
}

d-footnote-list a.footnote-backlink {
  color: rgba(0,0,0,0.3);
  padding-left: 0.5em;
}



/* Anchor.js */

.anchorjs-link {
  /*transition: all .25s linear; */
  text-decoration: none;
  border-bottom: none;
}
*:hover > .anchorjs-link {
  margin-left: -1.125em !important;
  text-decoration: none;
  border-bottom: none;
}

/* Social footer */

.social_footer {
  margin-top: 30px;
  margin-bottom: 0;
  color: rgba(0,0,0,0.67);
}

.disqus-comments {
  margin-right: 30px;
}

.disqus-comment-count {
  border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  cursor: pointer;
}

#disqus_thread {
  margin-top: 30px;
}

.article-sharing a {
  border-bottom: none;
  margin-right: 8px;
}

.article-sharing a:hover {
  border-bottom: none;
}

.sidebar-section.subscribe {
  font-size: 12px;
  line-height: 1.6em;
}

.subscribe p {
  margin-bottom: 0.5em;
}


.article-footer .subscribe {
  font-size: 15px;
  margin-top: 45px;
}


.sidebar-section.custom {
  font-size: 12px;
  line-height: 1.6em;
}

.custom p {
  margin-bottom: 0.5em;
}

/* Styles for listing layout (hide title) */
.layout-listing d-title, .layout-listing .d-title {
  display: none;
}

/* Styles for posts lists (not auto-injected) */


.posts-with-sidebar {
  padding-left: 45px;
  padding-right: 45px;
}

.posts-list .description h2,
.posts-list .description p {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
}

.posts-list .description h2 {
  font-weight: 700;
  border-bottom: none;
  padding-bottom: 0;
}

.posts-list h2.post-tag {
  border-bottom: 1px solid rgba(0, 0, 0, 0.2);
  padding-bottom: 12px;
}
.posts-list {
  margin-top: 60px;
  margin-bottom: 24px;
}

.posts-list .post-preview {
  text-decoration: none;
  overflow: hidden;
  display: block;
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  padding: 24px 0;
}

.post-preview-last {
  border-bottom: none !important;
}

.posts-list .posts-list-caption {
  grid-column: screen;
  font-weight: 400;
}

.posts-list .post-preview h2 {
  margin: 0 0 6px 0;
  line-height: 1.2em;
  font-style: normal;
  font-size: 24px;
}

.posts-list .post-preview p {
  margin: 0 0 12px 0;
  line-height: 1.4em;
  font-size: 16px;
}

.posts-list .post-preview .thumbnail {
  box-sizing: border-box;
  margin-bottom: 24px;
  position: relative;
  max-width: 500px;
}
.posts-list .post-preview img {
  width: 100%;
  display: block;
}

.posts-list .metadata {
  font-size: 12px;
  line-height: 1.4em;
  margin-bottom: 18px;
}

.posts-list .metadata > * {
  display: inline-block;
}

.posts-list .metadata .publishedDate {
  margin-right: 2em;
}

.posts-list .metadata .dt-authors {
  display: block;
  margin-top: 0.3em;
  margin-right: 2em;
}

.posts-list .dt-tags {
  display: block;
  line-height: 1em;
}

.posts-list .dt-tags .dt-tag {
  display: inline-block;
  color: rgba(0,0,0,0.6);
  padding: 0.3em 0.4em;
  margin-right: 0.2em;
  margin-bottom: 0.4em;
  font-size: 60%;
  border: 1px solid rgba(0,0,0,0.2);
  border-radius: 3px;
  text-transform: uppercase;
  font-weight: 500;
}

.posts-list img {
  opacity: 1;
}

.posts-list img[data-src] {
  opacity: 0;
}

.posts-more {
  clear: both;
}


.posts-sidebar {
  font-size: 16px;
}

.posts-sidebar h3 {
  font-size: 16px;
  margin-top: 0;
  margin-bottom: 0.5em;
  font-weight: 400;
  text-transform: uppercase;
}

.sidebar-section {
  margin-bottom: 30px;
}

.categories ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.categories li {
  color: rgba(0, 0, 0, 0.8);
  margin-bottom: 0;
}

.categories li>a {
  border-bottom: none;
}

.categories li>a:hover {
  border-bottom: 1px solid rgba(0, 0, 0, 0.4);
}

.categories .active {
  font-weight: 600;
}

.categories .category-count {
  color: rgba(0, 0, 0, 0.4);
}


@media(min-width: 768px) {
  .posts-list .post-preview h2 {
    font-size: 26px;
  }
  .posts-list .post-preview .thumbnail {
    float: right;
    width: 30%;
    margin-bottom: 0;
  }
  .posts-list .post-preview .description {
    float: left;
    width: 45%;
  }
  .posts-list .post-preview .metadata {
    float: left;
    width: 20%;
    margin-top: 8px;
  }
  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.5em;
    font-size: 16px;
  }
  .posts-with-sidebar .posts-list {
    float: left;
    width: 75%;
  }
  .posts-with-sidebar .posts-sidebar {
    float: right;
    width: 20%;
    margin-top: 60px;
    padding-top: 24px;
    padding-bottom: 24px;
  }
}


/* Improve display for browsers without grid (IE/Edge <= 15) */

.downlevel {
  line-height: 1.6em;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  margin: 0;
}

.downlevel .d-title {
  padding-top: 6rem;
  padding-bottom: 1.5rem;
}

.downlevel .d-title h1 {
  font-size: 50px;
  font-weight: 700;
  line-height: 1.1em;
  margin: 0 0 0.5rem;
}

.downlevel .d-title p {
  font-weight: 300;
  font-size: 1.2rem;
  line-height: 1.55em;
  margin-top: 0;
}

.downlevel .d-byline {
  padding-top: 0.8em;
  padding-bottom: 0.8em;
  font-size: 0.8rem;
  line-height: 1.8em;
}

.downlevel .section-separator {
  border: none;
  border-top: 1px solid rgba(0, 0, 0, 0.1);
}

.downlevel .d-article {
  font-size: 1.06rem;
  line-height: 1.7em;
  padding-top: 1rem;
  padding-bottom: 2rem;
}


.downlevel .d-appendix {
  padding-left: 0;
  padding-right: 0;
  max-width: none;
  font-size: 0.8em;
  line-height: 1.7em;
  margin-bottom: 0;
  color: rgba(0,0,0,0.5);
  padding-top: 40px;
  padding-bottom: 48px;
}

.downlevel .footnotes ol {
  padding-left: 13px;
}

.downlevel .base-grid,
.downlevel .distill-header,
.downlevel .d-title,
.downlevel .d-abstract,
.downlevel .d-article,
.downlevel .d-appendix,
.downlevel .distill-appendix,
.downlevel .d-byline,
.downlevel .d-footnote-list,
.downlevel .d-citation-list,
.downlevel .distill-footer,
.downlevel .appendix-bottom,
.downlevel .posts-container {
  padding-left: 40px;
  padding-right: 40px;
}

@media(min-width: 768px) {
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
  padding-left: 150px;
  padding-right: 150px;
  max-width: 900px;
}
}

.downlevel pre code {
  display: block;
  border-left: 2px solid rgba(0, 0, 0, .1);
  padding: 0 0 0 20px;
  font-size: 14px;
}

.downlevel code, .downlevel pre {
  color: black;
  background: none;
  font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  text-align: left;
  white-space: pre;
  word-spacing: normal;
  word-break: normal;
  word-wrap: normal;
  line-height: 1.5;

  -moz-tab-size: 4;
  -o-tab-size: 4;
  tab-size: 4;

  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  hyphens: none;
}

.downlevel .posts-list .post-preview {
  color: inherit;
}



</style>

<script type="application/javascript">

function is_downlevel_browser() {
  if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                 window.navigator.userAgent)) {
    return true;
  } else {
    return window.load_distill_framework === undefined;
  }
}

// show body when load is complete
function on_load_complete() {

  // add anchors
  if (window.anchors) {
    window.anchors.options.placement = 'left';
    window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
  }


  // set body to visible
  document.body.style.visibility = 'visible';

  // force redraw for leaflet widgets
  if (window.HTMLWidgets) {
    var maps = window.HTMLWidgets.findAll(".leaflet");
    $.each(maps, function(i, el) {
      var map = this.getMap();
      map.invalidateSize();
      map.eachLayer(function(layer) {
        if (layer instanceof L.TileLayer)
          layer.redraw();
      });
    });
  }

  // trigger 'shown' so htmlwidgets resize
  $('d-article').trigger('shown');
}

function init_distill() {

  init_common();

  // create front matter
  var front_matter = $('<d-front-matter></d-front-matter>');
  $('#distill-front-matter').wrap(front_matter);

  // create d-title
  $('.d-title').changeElementType('d-title');

  // create d-byline
  var byline = $('<d-byline></d-byline>');
  $('.d-byline').replaceWith(byline);

  // create d-article
  var article = $('<d-article></d-article>');
  $('.d-article').wrap(article).children().unwrap();

  // move posts container into article
  $('.posts-container').appendTo($('d-article'));

  // create d-appendix
  $('.d-appendix').changeElementType('d-appendix');

  // flag indicating that we have appendix items
  var appendix = $('.appendix-bottom').children('h3').length > 0;

  // replace footnotes with <d-footnote>
  $('.footnote-ref').each(function(i, val) {
    appendix = true;
    var href = $(this).attr('href');
    var id = href.replace('#', '');
    var fn = $('#' + id);
    var fn_p = $('#' + id + '>p');
    fn_p.find('.footnote-back').remove();
    var text = fn_p.html();
    var dtfn = $('<d-footnote></d-footnote>');
    dtfn.html(text);
    $(this).replaceWith(dtfn);
  });
  // remove footnotes
  $('.footnotes').remove();

  // move refs into #references-listing
  $('#references-listing').replaceWith($('#refs'));

  $('h1.appendix, h2.appendix').each(function(i, val) {
    $(this).changeElementType('h3');
  });
  $('h3.appendix').each(function(i, val) {
    var id = $(this).attr('id');
    $('.d-contents a[href="#' + id + '"]').parent().remove();
    appendix = true;
    $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
  });

  // show d-appendix if we have appendix content
  $("d-appendix").css('display', appendix ? 'grid' : 'none');

  // localize layout chunks to just output
  $('.layout-chunk').each(function(i, val) {

    // capture layout
    var layout = $(this).attr('data-layout');

    // apply layout to markdown level block elements
    var elements = $(this).children().not('details, div.sourceCode, pre, script');
    elements.each(function(i, el) {
      var layout_div = $('<div class="' + layout + '"></div>');
      if (layout_div.hasClass('shaded')) {
        var shaded_content = $('<div class="shaded-content"></div>');
        $(this).wrap(shaded_content);
        $(this).parent().wrap(layout_div);
      } else {
        $(this).wrap(layout_div);
      }
    });


    // unwrap the layout-chunk div
    $(this).children().unwrap();
  });

  // remove code block used to force  highlighting css
  $('.distill-force-highlighting-css').parent().remove();

  // remove empty line numbers inserted by pandoc when using a
  // custom syntax highlighting theme
  $('code.sourceCode a:empty').remove();

  // load distill framework
  load_distill_framework();

  // wait for window.distillRunlevel == 4 to do post processing
  function distill_post_process() {

    if (!window.distillRunlevel || window.distillRunlevel < 4)
      return;

    // hide author/affiliations entirely if we have no authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
    if (!have_authors)
      $('d-byline').addClass('hidden');

    // article with toc class
    $('.d-contents').parent().addClass('d-article-with-toc');

    // strip links that point to #
    $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

    // add orcid ids
    $('.authors-affiliations').find('.author').each(function(i, el) {
      var orcid_id = front_matter.authors[i].orcidID;
      if (orcid_id) {
        var a = $('<a></a>');
        a.attr('href', 'https://orcid.org/' + orcid_id);
        var img = $('<img></img>');
        img.addClass('orcid-id');
        img.attr('alt', 'ORCID ID');
        img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
        a.append(img);
        $(this).append(a);
      }
    });

    // hide elements of author/affiliations grid that have no value
    function hide_byline_column(caption) {
      $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
    }

    // affiliations
    var have_affiliations = false;
    for (var i = 0; i<front_matter.authors.length; ++i) {
      var author = front_matter.authors[i];
      if (author.affiliation !== "&nbsp;") {
        have_affiliations = true;
        break;
      }
    }
    if (!have_affiliations)
      $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

    // published date
    if (!front_matter.publishedDate)
      hide_byline_column("Published");

    // document object identifier
    var doi = $('d-byline').find('h3:contains("DOI")');
    var doi_p = doi.next().empty();
    if (!front_matter.doi) {
      // if we have a citation and valid citationText then link to that
      if ($('#citation').length > 0 && front_matter.citationText) {
        doi.html('Citation');
        $('<a href="#citation"></a>')
          .text(front_matter.citationText)
          .appendTo(doi_p);
      } else {
        hide_byline_column("DOI");
      }
    } else {
      $('<a></a>')
         .attr('href', "https://doi.org/" + front_matter.doi)
         .html(front_matter.doi)
         .appendTo(doi_p);
    }

     // change plural form of authors/affiliations
    if (front_matter.authors.length === 1) {
      var grid = $('.authors-affiliations');
      grid.children('h3:contains("Authors")').text('Author');
      grid.children('h3:contains("Affiliations")').text('Affiliation');
    }

    // remove d-appendix and d-footnote-list local styles
    $('d-appendix > style:first-child').remove();
    $('d-footnote-list > style:first-child').remove();

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // hoverable references
    $('span.citation[data-cites]').each(function() {
      var refs = $(this).attr('data-cites').split(" ");
      var refHtml = refs.map(function(ref) {
        return "<p>" + $('#ref-' + ref).html() + "</p>";
      }).join("\n");
      window.tippy(this, {
        allowHTML: true,
        content: refHtml,
        maxWidth: 500,
        interactive: true,
        interactiveBorder: 10,
        theme: 'light-border',
        placement: 'bottom-start'
      });
    });

    // clear polling timer
    clearInterval(tid);

    // show body now that everything is ready
    on_load_complete();
  }

  var tid = setInterval(distill_post_process, 50);
  distill_post_process();

}

function init_downlevel() {

  init_common();

   // insert hr after d-title
  $('.d-title').after($('<hr class="section-separator"/>'));

  // check if we have authors
  var front_matter = JSON.parse($("#distill-front-matter").html());
  var have_authors = front_matter.authors && front_matter.authors.length > 0;

  // manage byline/border
  if (!have_authors)
    $('.d-byline').remove();
  $('.d-byline').after($('<hr class="section-separator"/>'));
  $('.d-byline a').remove();

  // remove toc
  $('.d-contents').remove();

  // move appendix elements
  $('h1.appendix, h2.appendix').each(function(i, val) {
    $(this).changeElementType('h3');
  });
  $('h3.appendix').each(function(i, val) {
    $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
  });


  // inject headers into references and footnotes
  var refs_header = $('<h3></h3>');
  refs_header.text('References');
  $('#refs').prepend(refs_header);

  var footnotes_header = $('<h3></h3');
  footnotes_header.text('Footnotes');
  $('.footnotes').children('hr').first().replaceWith(footnotes_header);

  // move appendix-bottom entries to the bottom
  $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
  $('.appendix-bottom').remove();

  // remove appendix if it's empty
  if ($('.d-appendix').children().length === 0)
    $('.d-appendix').remove();

  // prepend separator above appendix
  $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

  // trim code
  $('pre>code').each(function(i, val) {
    $(this).html($.trim($(this).html()));
  });

  // move posts-container right before article
  $('.posts-container').insertBefore($('.d-article'));

  $('body').addClass('downlevel');

  on_load_complete();
}


function init_common() {

  // jquery plugin to change element types
  (function($) {
    $.fn.changeElementType = function(newType) {
      var attrs = {};

      $.each(this[0].attributes, function(idx, attr) {
        attrs[attr.nodeName] = attr.nodeValue;
      });

      this.replaceWith(function() {
        return $("<" + newType + "/>", attrs).append($(this).contents());
      });
    };
  })(jQuery);

  // prevent underline for linked images
  $('a > img').parent().css({'border-bottom' : 'none'});

  // mark non-body figures created by knitr chunks as 100% width
  $('.layout-chunk').each(function(i, val) {
    var figures = $(this).find('img, .html-widget');
    if ($(this).attr('data-layout') !== "l-body") {
      figures.css('width', '100%');
    } else {
      figures.css('max-width', '100%');
      figures.filter("[width]").each(function(i, val) {
        var fig = $(this);
        fig.css('width', fig.attr('width') + 'px');
      });

    }
  });

  // auto-append index.html to post-preview links in file: protocol
  // and in rstudio ide preview
  $('.post-preview').each(function(i, val) {
    if (window.location.protocol === "file:")
      $(this).attr('href', $(this).attr('href') + "index.html");
  });

  // get rid of index.html references in header
  if (window.location.protocol !== "file:") {
    $('.distill-site-header a[href]').each(function(i,val) {
      $(this).attr('href', $(this).attr('href').replace("index.html", "./"));
    });
  }

  // add class to pandoc style tables
  $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
  $('.kable-table').children('table').addClass('pandoc-table');

  // add figcaption style to table captions
  $('caption').parent('table').addClass("figcaption");

  // initialize posts list
  if (window.init_posts_list)
    window.init_posts_list();

  // implmement disqus comment link
  $('.disqus-comment-count').click(function() {
    window.headroom_prevent_pin = true;
    $('#disqus_thread').toggleClass('hidden');
    if (!$('#disqus_thread').hasClass('hidden')) {
      var offset = $(this).offset();
      $(window).resize();
      $('html, body').animate({
        scrollTop: offset.top - 35
      });
    }
  });
}

document.addEventListener('DOMContentLoaded', function() {
  if (is_downlevel_browser())
    init_downlevel();
  else
    window.addEventListener('WebComponentsReady', init_distill);
});

</script>

<!--/radix_placeholder_distill-->
  <script src="../../site_libs/header-attrs-2.11/header-attrs.js"></script>
  <script src="../../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <script src="../../site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="../../site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="../../site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="../../site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="../../site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="../../site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="../../site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="../../site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
<!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"MOMS-PI: Regression on diversity","description":"Analyze the impact of bias on a DA analysis of relative abundances versus diversity.","authors":[{"author":"Michael R. McLaren","authorURL":{},"affiliation":"&nbsp;","affiliationURL":"#","orcidID":""}],"publishedDate":"2021-12-15T00:00:00.000-05:00","citationText":"McLaren, 2021"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="../../index.html" class="title">Research notebook for manuscript on taxonomic bias and differential-abundance methods</a>
<input id="distill-search" class="nav-search hidden" type="text" placeholder="Search..."/>
</div>
<div class="nav-right">
<a href="../../index.html">Home</a>
<a href="../../about.html">About</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>MOMS-PI: Regression on diversity</h1>
<!--radix_placeholder_categories-->
<div class="dt-tags">
  <a href="../../index.html#category:ref:brooks2015thet" class="dt-tag">ref:brooks2015thet</a>
  <a href="../../index.html#category:ref:fettweis2019thev" class="dt-tag">ref:fettweis2019thev</a>
</div>
<!--/radix_placeholder_categories-->
<p>Analyze the impact of bias on a DA analysis of relative abundances versus diversity.</p>
</div>

<div class="d-byline">
  Michael R. McLaren true 
  
<br/>2021-12-15
</div>

<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#setup">Setup</a>
<ul>
<li><a href="#r-setup">R setup</a></li>
<li><a href="#moms-pi-setup">MOMS-PI setup</a></li>
</ul></li>
<li><a href="#regression-analysis">Regression analysis</a>
<ul>
<li><a href="#classify-samples-into-high-and-low-diversity-groups">Classify samples into high and low diversity groups</a></li>
<li><a href="#regression-setup">Regression setup</a></li>
<li><a href="#linear-regression-on-log-proportion">Linear regression on log proportion</a>
<ul>
<li><a href="#alternate-zero-replacement-strategy">Alternate zero-replacement strategy</a></li>
</ul></li>
<li><a href="#interpretion">Interpretion</a></li>
</ul></li>
</ul>
</nav>
</div>
<h1 id="setup">Setup</h1>
<h2 id="r-setup">R setup</h2>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># Tools for microbiome data</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://mikemc.github.io/speedyseq'>speedyseq</a></span><span class='op'>)</span>
<span class='co'># Tools for general purpose data manipulation and plotting</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://tidyverse.tidyverse.org'>tidyverse</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://fs.r-lib.org'>fs</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://here.r-lib.org/'>here</a></span><span class='op'>)</span>
<span class='co'># ggplot helpers</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/eclarke/ggbeeswarm'>ggbeeswarm</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://wilkelab.org/cowplot/'>cowplot</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://patchwork.data-imaginist.com'>patchwork</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://scales.r-lib.org'>scales</a></span><span class='op'>)</span>
<span class='co'># stats helpers</span>
<span class='co'># library(broom)</span>

<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://mikemc.github.io/metacal'>metacal</a></span><span class='op'>)</span>; <span class='fu'><a href='https://rdrr.io/r/utils/packageDescription.html'>packageVersion</a></span><span class='op'>(</span><span class='st'>"metacal"</span><span class='op'>)</span>
</code></pre>
</div>
<pre><code>[1] &#39;0.2.0.9008&#39;</code></pre>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/theme_get.html'>theme_set</a></span><span class='op'>(</span><span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>scale_y_custom</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_continuous</a></span><span class='op'>(</span>
    trans <span class='op'>=</span> <span class='st'>'log10'</span>,
    breaks <span class='op'>=</span> <span class='fu'><a href='https://scales.r-lib.org/reference/trans_breaks.html'>trans_breaks</a></span><span class='op'>(</span><span class='st'>'log10'</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fl'>10</span><span class='op'>^</span><span class='va'>x</span><span class='op'>)</span>,
    labels <span class='op'>=</span> <span class='fu'><a href='https://scales.r-lib.org/reference/trans_format.html'>trans_format</a></span><span class='op'>(</span><span class='st'>'log10'</span>, <span class='fu'><a href='https://scales.r-lib.org/reference/label_parse.html'>math_format</a></span><span class='op'>(</span><span class='fl'>10</span><span class='op'>^</span><span class='va'>.x</span><span class='op'>)</span><span class='op'>)</span>
  <span class='op'>)</span>
</code></pre>
</div>
</div>
<h2 id="moms-pi-setup">MOMS-PI setup</h2>
<p>Load the MOMS-PI profiles and estimate bias,</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='kw'><a href='https://rdrr.io/r/base/source.html'>source</a></span><span class='op'>(</span><span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>'notebook/_code/momspi-setup.R'</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>To simplify interpretation of some of the regression models, lets rarefy samples to 10K reads (well be dropping some samples that are below this limit).</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>momspi_rare</span> <span class='op'>&lt;-</span> <span class='fu'>xfun</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/xfun/man/cache_rds.html'>cache_rds</a></span><span class='op'>(</span><span class='op'>{</span>
  <span class='va'>momspi</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> 
    <span class='fu'><a href='https://rdrr.io/pkg/phyloseq/man/rarefy_even_depth.html'>rarefy_even_depth</a></span><span class='op'>(</span>
      sample.size <span class='op'>=</span> <span class='fl'>1e4</span>, 
      rngseed <span class='op'>=</span> <span class='fl'>42</span>,
      replace <span class='op'>=</span> <span class='cn'>FALSE</span>,
      trimOTUs <span class='op'>=</span> <span class='cn'>FALSE</span>
    <span class='op'>)</span>
  <span class='op'>}</span>, hash <span class='op'>=</span> <span class='va'>momspi</span>
<span class='op'>)</span>
<span class='co'># Make sure no taxa have 0 counts, in case this causes issues later on</span>
<span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/all.html'>all</a></span><span class='op'>(</span><span class='va'>momspi_rare</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='va'>taxa_sums</span> <span class='op'>&gt;</span> <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>Lets also add the new sample sums and estimated mean efficiency and</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>mean_eff</span> <span class='op'>&lt;-</span> <span class='va'>momspi_rare</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
 <span class='fu'><a href='https://mikemc.github.io/metacal/reference/calibrate.html'>calibrate</a></span><span class='op'>(</span><span class='va'>bias_all_vec</span>, norm <span class='op'>=</span> <span class='st'>'close'</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
 <span class='fu'><a href='https://mikemc.github.io/metacal/reference/perturb.html'>perturb</a></span><span class='op'>(</span><span class='va'>bias_all_vec</span>, norm <span class='op'>=</span> <span class='st'>'none'</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> 
 <span class='va'>sample_sums</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
 <span class='fu'><a href='https://tibble.tidyverse.org/reference/enframe.html'>enframe</a></span><span class='op'>(</span><span class='st'>'.sample'</span>, <span class='st'>'mean_efficiency'</span><span class='op'>)</span>

<span class='va'>momspi_rare</span> <span class='op'>&lt;-</span> <span class='va'>momspi_rare</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> 
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/mutate-phyloseq.html'>mutate_sample_data</a></span><span class='op'>(</span><span class='va'>.</span>, 
    sample_sum_original <span class='op'>=</span> <span class='va'>sample_sum</span>,
    sample_sum <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/phyloseq/man/sample_sums.html'>sample_sums</a></span><span class='op'>(</span><span class='va'>.</span><span class='op'>)</span>,
  <span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/join-phyloseq.html'>left_join_sample_data</a></span><span class='op'>(</span><span class='va'>mean_eff</span>, by <span class='op'>=</span> <span class='st'>'.sample'</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<p>Lets also add the top species in each sample,</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>get_top_species</span> <span class='op'>&lt;-</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>ps</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>x</span> <span class='op'>&lt;-</span> <span class='va'>ps</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/transform_sample_counts.html'>transform_sample_counts</a></span><span class='op'>(</span><span class='va'>close_elts</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='va'>otu_table</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'>as_tibble</span><span class='op'>(</span>pivot <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://dplyr.tidyverse.org/reference/with_groups.html'>with_groups</a></span><span class='op'>(</span><span class='va'>.sample</span>, <span class='va'>slice_max</span>, <span class='va'>.abundance</span>, n <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>
  <span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/duplicated.html'>anyDuplicated</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>$</span><span class='va'>.sample</span><span class='op'>)</span><span class='op'>)</span>
    <span class='kw'><a href='https://rdrr.io/r/base/stop.html'>stop</a></span><span class='op'>(</span><span class='st'>'Ties present'</span><span class='op'>)</span>
  <span class='va'>x</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span><span class='va'>.sample</span>, top_species <span class='op'>=</span> <span class='va'>.otu</span><span class='op'>)</span>
<span class='op'>}</span>
<span class='va'>top_species</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>
  <span class='st'>'obs'</span> <span class='op'>=</span> <span class='va'>momspi_rare</span>,
  <span class='st'>'cal'</span> <span class='op'>=</span> <span class='va'>momspi_rare</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='fu'><a href='https://mikemc.github.io/metacal/reference/calibrate.html'>calibrate</a></span><span class='op'>(</span><span class='va'>bias_all_vec</span><span class='op'>)</span>
  <span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://purrr.tidyverse.org/reference/map.html'>map_dfr</a></span><span class='op'>(</span><span class='va'>get_top_species</span>, .id <span class='op'>=</span> <span class='st'>'type'</span><span class='op'>)</span>
<span class='va'>top_species_wide</span> <span class='op'>&lt;-</span> <span class='va'>top_species</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/pivot_wider.html'>pivot_wider</a></span><span class='op'>(</span>values_from <span class='op'>=</span> <span class='va'>top_species</span>, names_from <span class='op'>=</span> <span class='va'>type</span>, 
    names_prefix <span class='op'>=</span> <span class='st'>'top_species_'</span><span class='op'>)</span> 
<span class='va'>momspi_rare</span> <span class='op'>&lt;-</span> <span class='va'>momspi_rare</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/join-phyloseq.html'>left_join_sample_data</a></span><span class='op'>(</span><span class='va'>top_species_wide</span>, by <span class='op'>=</span> <span class='st'>'.sample'</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<h1 id="regression-analysis">Regression analysis</h1>
<p>NOTE: Consider saving output so can be used in other notebook files</p>
<h2 id="classify-samples-into-high-and-low-diversity-groups">Classify samples into high and low diversity groups</h2>
<p>Compute Shannon and Inverse-Simpson diversity indices and add to the sample metadata. <code>Phyloseq::estimate_richness()</code> will complain about the lack of singletons, but we can ignore this since we arent using species richness estimators. I will exponentiate the Shannon index to convert it to the q1 Hill number so that (like Inverse Simpson) it has the scale of number of species.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>div</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>
  <span class='st'>'obs'</span> <span class='op'>=</span> <span class='va'>momspi_rare</span>,
  <span class='st'>'cal'</span> <span class='op'>=</span> <span class='va'>momspi_rare</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='fu'><a href='https://mikemc.github.io/metacal/reference/calibrate.html'>calibrate</a></span><span class='op'>(</span><span class='va'>bias_all_vec</span><span class='op'>)</span>
<span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://purrr.tidyverse.org/reference/map.html'>map</a></span><span class='op'>(</span><span class='va'>estimate_richness</span>, measures <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'Shannon'</span>, <span class='st'>'InvSimpson'</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://purrr.tidyverse.org/reference/map.html'>map_dfr</a></span><span class='op'>(</span><span class='va'>as_tibble</span>, rownames <span class='op'>=</span> <span class='st'>'.sample'</span>, .id <span class='op'>=</span> <span class='st'>'type'</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>transmute</a></span><span class='op'>(</span>
    <span class='va'>.sample</span>, <span class='va'>type</span>,
    div_q1 <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>exp</a></span><span class='op'>(</span><span class='va'>Shannon</span><span class='op'>)</span>,
    div_q2 <span class='op'>=</span> <span class='va'>InvSimpson</span>,
  <span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/pivot_longer.html'>pivot_longer</a></span><span class='op'>(</span><span class='fu'><a href='https://tidyselect.r-lib.org/reference/starts_with.html'>starts_with</a></span><span class='op'>(</span><span class='st'>'div_'</span><span class='op'>)</span>, names_to <span class='op'>=</span> <span class='st'>'index'</span>, values_to <span class='op'>=</span> <span class='st'>'diversity'</span><span class='op'>)</span>
<span class='va'>div_wide</span> <span class='op'>&lt;-</span> <span class='va'>div</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/pivot_wider.html'>pivot_wider</a></span><span class='op'>(</span>values_from <span class='op'>=</span> <span class='va'>diversity</span>, names_from <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'index'</span>, <span class='st'>'type'</span><span class='op'>)</span><span class='op'>)</span> 
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>p</span> <span class='op'>&lt;-</span> <span class='va'>div_wide</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='fu'><a href='https://tidyselect.r-lib.org/reference/starts_with.html'>starts_with</a></span><span class='op'>(</span><span class='st'>'div'</span><span class='op'>)</span>, <span class='va'>log2</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>.panel_x</span>, y <span class='op'>=</span> <span class='va'>.panel_y</span><span class='op'>)</span>, alpha <span class='op'>=</span> <span class='fl'>0.5</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>ggforce</span><span class='fu'>::</span><span class='fu'><a href='https://ggforce.data-imaginist.com/reference/geom_autohistogram.html'>geom_autodensity</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_bin_2d.html'>geom_bin2d</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>.panel_x</span>, y <span class='op'>=</span> <span class='va'>.panel_y</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>ggforce</span><span class='fu'>::</span><span class='fu'><a href='https://ggforce.data-imaginist.com/reference/facet_matrix.html'>facet_matrix</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/vars.html'>vars</a></span><span class='op'>(</span><span class='fu'><a href='https://tidyselect.r-lib.org/reference/starts_with.html'>starts_with</a></span><span class='op'>(</span><span class='st'>'div'</span><span class='op'>)</span><span class='op'>)</span>, layer.diag <span class='op'>=</span> <span class='fl'>2</span>, layer.lower <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span>
<span class='fu'>ggrastr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ggrastr/man/rasterise.html'>rasterize</a></span><span class='op'>(</span><span class='va'>p</span>, dpi <span class='op'>=</span> <span class='fl'>300</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="momspi-regression-diversity_files/figure-html5/unnamed-chunk-7-1.svg" width="2100" /></p>
</div>
<p>The observed (Log) order-1 and order-2 diversity show a tight correlation with each other, as do the calibrated diversities, while the correlation is much lower when comparing observed and calibrated numbers. The reason is likely that calibration will tend to increase the diversity of samples that are observed to be dominated by a high-efficiency taxon (like L. iners) and decrease the diversity of samples dominated by a low efficiency tason (like G. vaginalis).</p>
<p>Shannon diversity of the observed profiles is what is most often considered in vaginal microbiome studies, so lets use this metric to partition the samples.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>div_wide</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>div_q1_obs</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_x_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/annotate.html'>annotate</a></span><span class='op'>(</span><span class='st'>'rect'</span>, xmin <span class='op'>=</span> <span class='fl'>0</span>, xmax <span class='op'>=</span> <span class='fl'>1.5</span>, ymin <span class='op'>=</span> <span class='op'>-</span><span class='cn'>Inf</span>, ymax <span class='op'>=</span> <span class='cn'>Inf</span>, 
    fill <span class='op'>=</span> <span class='st'>'blue'</span>, alpha <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/annotate.html'>annotate</a></span><span class='op'>(</span><span class='st'>'rect'</span>, xmin <span class='op'>=</span> <span class='fl'>1.5</span>, xmax <span class='op'>=</span> <span class='fl'>3</span>, ymin <span class='op'>=</span> <span class='op'>-</span><span class='cn'>Inf</span>, ymax <span class='op'>=</span> <span class='cn'>Inf</span>, 
    fill <span class='op'>=</span> <span class='st'>'grey'</span>, alpha <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/annotate.html'>annotate</a></span><span class='op'>(</span><span class='st'>'rect'</span>, xmin <span class='op'>=</span> <span class='fl'>3</span>, xmax <span class='op'>=</span> <span class='cn'>Inf</span>, ymin <span class='op'>=</span> <span class='op'>-</span><span class='cn'>Inf</span>, ymax <span class='op'>=</span> <span class='cn'>Inf</span>, 
    fill <span class='op'>=</span> <span class='st'>'green'</span>, alpha <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_histogram.html'>geom_histogram</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_vline</a></span><span class='op'>(</span>xintercept <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1.5</span>, <span class='fl'>3</span><span class='op'>)</span>, color <span class='op'>=</span> <span class='st'>'black'</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="momspi-regression-diversity_files/figure-html5/unnamed-chunk-8-1.svg" width="1950" /></p>
</div>
<p>Ill pick the first mode as the Low diversity group, treat the second mode as Medium, and the tail as High.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>div_wide</span> <span class='op'>&lt;-</span> <span class='va'>div_wide</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span>
    div_group <span class='op'>=</span> <span class='fu'><a href='https://dplyr.tidyverse.org/reference/case_when.html'>case_when</a></span><span class='op'>(</span>
      <span class='va'>div_q1_obs</span> <span class='op'>&lt;</span> <span class='fl'>1.5</span> <span class='op'>~</span> <span class='st'>'Low'</span>,
      <span class='va'>div_q1_obs</span> <span class='op'>&gt;</span> <span class='fl'>3</span> <span class='op'>~</span> <span class='st'>'High'</span>,
      <span class='cn'>TRUE</span> <span class='op'>~</span> <span class='st'>'Medium'</span>
    <span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span>levels <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'Low'</span>, <span class='st'>'Medium'</span>, <span class='st'>'High'</span><span class='op'>)</span><span class='op'>)</span>
  <span class='op'>)</span>
</code></pre>
</div>
</div>
<h2 id="regression-setup">Regression setup</h2>
<p>Add the diversity grouping information, subset to the Low and High samples, and compute the offsets for the Gamma-Poisson model</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>momspi_div</span> <span class='op'>&lt;-</span> <span class='va'>momspi_rare</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/join-phyloseq.html'>left_join_sample_data</a></span><span class='op'>(</span><span class='va'>div_wide</span>, by <span class='op'>=</span> <span class='st'>'.sample'</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/filter-phyloseq.html'>filter_sample_data</a></span><span class='op'>(</span><span class='va'>div_group</span> <span class='op'><a href='https://rdrr.io/r/base/match.html'>%in%</a></span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'Low'</span>, <span class='st'>'High'</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/mutate-phyloseq.html'>mutate_sample_data</a></span><span class='op'>(</span>
    gp_offset_obs <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log</a></span><span class='op'>(</span><span class='va'>sample_sum</span><span class='op'>)</span>,
    gp_offset_cal <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log</a></span><span class='op'>(</span><span class='va'>sample_sum</span><span class='op'>)</span> <span class='op'>-</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log</a></span><span class='op'>(</span><span class='va'>mean_efficiency</span><span class='op'>)</span>,
  <span class='op'>)</span>
</code></pre>
</div>
</div>
<p>Well use the criterion from Fettweis et al to pick the taxa to test. Note, we may not get the same exact taxa as them since were using a larger set of women and ignoring the medium-diversity samples. Note, the log proportions used in the regression analysis below are with respect to the full species set.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>taxa_to_test</span> <span class='op'>&lt;-</span> <span class='va'>momspi_div</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/transform_sample_counts.html'>transform_sample_counts</a></span><span class='op'>(</span><span class='va'>close_elts</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/filter_taxa.html'>filter_taxa2</a></span><span class='op'>(</span><span class='op'>~</span> <span class='fu'><a href='https://rdrr.io/r/base/mean.html'>mean</a></span><span class='op'>(</span><span class='va'>.</span> <span class='op'>&gt;=</span> <span class='fl'>1e-2</span><span class='op'>)</span> <span class='op'>&gt;=</span> <span class='fl'>0.05</span> <span class='op'>|</span> <span class='fu'><a href='https://rdrr.io/r/base/mean.html'>mean</a></span><span class='op'>(</span><span class='va'>.</span> <span class='op'>&gt;=</span> <span class='fl'>1e-3</span><span class='op'>)</span> <span class='op'>&gt;=</span> <span class='fl'>0.15</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='va'>taxa_names</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='va'>print</span>
</code></pre>
</div>
<pre><code> [1] &quot;Lachnospiraceae_BVAB1&quot;         &quot;Aerococcus_christensenii&quot;     
 [3] &quot;Atopobium_vaginae&quot;             &quot;Campylobacter_ureolyticus&quot;    
 [5] &quot;Clostridiales_BVAB2&quot;           &quot;Clostridiales_OTU22&quot;          
 [7] &quot;Coriobacteriaceae_OTU27&quot;       &quot;Corynebacterium_cluster45&quot;    
 [9] &quot;Dialister_cluster51&quot;           &quot;Dialister_micraerophilus&quot;     
[11] &quot;Dialister_propionicifaciens&quot;   &quot;Finegoldia_magna&quot;             
[13] &quot;Gardnerella_vaginalis&quot;         &quot;Lactobacillus_crispatus&quot;      
[15] &quot;Lactobacillus_gasseri_cluster&quot; &quot;Lactobacillus_iners&quot;          
[17] &quot;Lactobacillus_jensenii&quot;        &quot;Megasphaera_OTU70_type1&quot;      
[19] &quot;Mycoplasma_hominis&quot;            &quot;Parvimonas_OTU142&quot;            
[21] &quot;Peptoniphilus_indolicus&quot;       &quot;Prevotella_amnii&quot;             
[23] &quot;Prevotella_bivia&quot;              &quot;Prevotella_cluster2&quot;          
[25] &quot;Prevotella_cluster50&quot;          &quot;Prevotella_disiens&quot;           
[27] &quot;Sneathia_amnii&quot;                &quot;Sneathia_sanguinegens&quot;        
[29] &quot;TM7_OTU-H1&quot;                    &quot;Ureaplasma_cluster23&quot;         </code></pre>
</div>
<h2 id="linear-regression-on-log-proportion">Linear regression on log proportion</h2>
<p>The simple linear regression is the fastest and simplest approach, and the easiest to understand using results from the manuscript.</p>
<p>First, we need to ensure there are no zeros. Ill try two approaches; the first is to use a pseudocount of 0.5.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>stat_test</span> <span class='op'>&lt;-</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>ps</span>, <span class='va'>taxa</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>ps</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/filter-phyloseq.html'>filter_sample_data</a></span><span class='op'>(</span><span class='va'>div_group</span> <span class='op'><a href='https://rdrr.io/r/base/match.html'>%in%</a></span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'Low'</span>, <span class='st'>'High'</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/transform_sample_counts.html'>transform_sample_counts</a></span><span class='op'>(</span><span class='va'>close_elts</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://rdrr.io/pkg/phyloseq/man/prune_taxa-methods.html'>prune_taxa</a></span><span class='op'>(</span><span class='va'>taxa</span>, <span class='va'>.</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='va'>as_tibble</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://dplyr.tidyverse.org/reference/with_groups.html'>with_groups</a></span><span class='op'>(</span><span class='va'>.otu</span>, <span class='va'>nest</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span>
      fit <span class='op'>=</span> <span class='fu'><a href='https://purrr.tidyverse.org/reference/map.html'>map</a></span><span class='op'>(</span><span class='va'>data</span>,
        <span class='op'>~</span><span class='fu'><a href='https://rdrr.io/r/stats/lm.html'>lm</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log</a></span><span class='op'>(</span><span class='va'>.abundance</span><span class='op'>)</span> <span class='op'>~</span> <span class='va'>div_group</span>, data <span class='op'>=</span> <span class='va'>.x</span><span class='op'>)</span>
      <span class='op'>)</span>,
    <span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span><span class='op'>-</span><span class='va'>data</span><span class='op'>)</span>
<span class='op'>}</span>
<span class='va'>res_div_lm</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>
  <span class='st'>'Observed'</span> <span class='op'>=</span> <span class='va'>momspi_div</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/transform_sample_counts.html'>transform_sample_counts</a></span><span class='op'>(</span><span class='op'>~</span><span class='va'>.</span> <span class='op'>+</span> <span class='fl'>0.5</span><span class='op'>)</span>,
  <span class='st'>'Calibrated'</span> <span class='op'>=</span> <span class='va'>momspi_div</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='fu'><a href='https://mikemc.github.io/speedyseq/reference/transform_sample_counts.html'>transform_sample_counts</a></span><span class='op'>(</span><span class='op'>~</span><span class='va'>.</span> <span class='op'>+</span> <span class='fl'>0.5</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://mikemc.github.io/metacal/reference/calibrate.html'>calibrate</a></span><span class='op'>(</span><span class='va'>bias_all_vec</span><span class='op'>)</span>
<span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://purrr.tidyverse.org/reference/map.html'>map_dfr</a></span><span class='op'>(</span><span class='va'>stat_test</span>, taxa <span class='op'>=</span> <span class='va'>taxa_to_test</span>, .id <span class='op'>=</span> <span class='st'>'type'</span><span class='op'>)</span>
<span class='va'>res_div_lm_tidy</span> <span class='op'>&lt;-</span> <span class='va'>res_div_lm</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='va'>fit</span>, <span class='va'>map</span>, <span class='fu'>broom.mixed</span><span class='fu'>::</span><span class='va'><a href='https://generics.r-lib.org/reference/tidy.html'>tidy</a></span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/nest.html'>unnest</a></span><span class='op'>(</span><span class='va'>fit</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> 
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>term</span> <span class='op'>!=</span> <span class='st'>'(Intercept)'</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='co'># x should be the outout of a tidier</span>
<span class='va'>plot_estimates</span> <span class='op'>&lt;-</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='op'>{</span>
  <span class='va'>err</span> <span class='op'>&lt;-</span> <span class='va'>x</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span>name <span class='op'>=</span> <span class='va'>type</span>, <span class='va'>.otu</span>, value <span class='op'>=</span> <span class='va'>estimate</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://tidyr.tidyverse.org/reference/pivot_wider.html'>pivot_wider</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span>
      error <span class='op'>=</span> <span class='va'>Observed</span> <span class='op'>-</span> <span class='va'>Calibrated</span>,
      error_sign <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/sign.html'>sign</a></span><span class='op'>(</span><span class='va'>Observed</span><span class='op'>)</span> <span class='op'>==</span> <span class='fu'><a href='https://rdrr.io/r/base/sign.html'>sign</a></span><span class='op'>(</span><span class='va'>Calibrated</span><span class='op'>)</span>,
      sign <span class='op'>=</span> <span class='fu'><a href='https://dplyr.tidyverse.org/reference/case_when.html'>case_when</a></span><span class='op'>(</span>
        <span class='fu'><a href='https://rdrr.io/r/base/sign.html'>sign</a></span><span class='op'>(</span><span class='va'>Observed</span><span class='op'>)</span> <span class='op'>&gt;</span> <span class='fl'>0</span> <span class='op'>&amp;</span> <span class='fu'><a href='https://rdrr.io/r/base/sign.html'>sign</a></span><span class='op'>(</span><span class='va'>Calibrated</span><span class='op'>)</span> <span class='op'>&gt;</span> <span class='fl'>0</span> <span class='op'>~</span> <span class='st'>'Positive'</span>,
        <span class='fu'><a href='https://rdrr.io/r/base/sign.html'>sign</a></span><span class='op'>(</span><span class='va'>Observed</span><span class='op'>)</span> <span class='op'>&lt;</span> <span class='fl'>0</span> <span class='op'>&amp;</span> <span class='fu'><a href='https://rdrr.io/r/base/sign.html'>sign</a></span><span class='op'>(</span><span class='va'>Calibrated</span><span class='op'>)</span> <span class='op'>&lt;</span> <span class='fl'>0</span> <span class='op'>~</span> <span class='st'>'Negative'</span>,
        <span class='cn'>TRUE</span> <span class='op'>~</span> <span class='st'>'Depends'</span>,
      <span class='op'>)</span>
    <span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span><span class='va'>.otu</span>, <span class='va'>error_sign</span>, <span class='va'>sign</span><span class='op'>)</span>

  <span class='va'>p1</span> <span class='op'>&lt;-</span> <span class='va'>x</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate-joins.html'>left_join</a></span><span class='op'>(</span><span class='va'>err</span>, by <span class='op'>=</span> <span class='st'>'.otu'</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>type</span>, x <span class='op'>=</span> <span class='va'>estimate</span>, fill <span class='op'>=</span> <span class='va'>sign</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_brewer.html'>scale_fill_brewer</a></span><span class='op'>(</span>type <span class='op'>=</span> <span class='st'>'qual'</span>, palette <span class='op'>=</span> <span class='fl'>2</span>,
      guide <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guide_legend.html'>guide_legend</a></span><span class='op'>(</span>reverse <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_vline</a></span><span class='op'>(</span>xintercept <span class='op'>=</span> <span class='fl'>0</span>, color <span class='op'>=</span> <span class='st'>'grey'</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'>ggdist</span><span class='fu'>::</span><span class='fu'><a href='http://mjskay.github.io/ggdist/reference/geom_dotsinterval.html'>geom_dots</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>labs</a></span><span class='op'>(</span>fill <span class='op'>=</span> <span class='st'>'Sign of estimate'</span>, x <span class='op'>=</span> <span class='st'>'Estimated LFC'</span>, y <span class='op'>=</span> <span class='st'>'Meas. type'</span><span class='op'>)</span>
  <span class='va'>p2</span> <span class='op'>&lt;-</span> <span class='va'>x</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span>
      .otu_fct <span class='op'>=</span> <span class='fu'><a href='https://forcats.tidyverse.org/reference/fct_reorder.html'>fct_reorder</a></span><span class='op'>(</span><span class='va'>.otu</span>, <span class='va'>estimate</span>, .fun <span class='op'>=</span> <span class='va'>min</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
        <span class='fu'><a href='https://forcats.tidyverse.org/reference/fct_relabel.html'>fct_relabel</a></span><span class='op'>(</span><span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'><a href='https://stringr.tidyverse.org/reference/str_replace.html'>str_replace_all</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='st'>'_'</span>, <span class='st'>' '</span><span class='op'>)</span><span class='op'>)</span>
    <span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>.otu_fct</span>, x <span class='op'>=</span> <span class='va'>estimate</span>, color <span class='op'>=</span> <span class='va'>type</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='co'># scale_color_brewer(type = 'qual', palette = 7) +</span>
    <span class='fu'>colorblindr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/colorblindr/man/scale_OkabeIto.html'>scale_color_OkabeIto</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_vline</a></span><span class='op'>(</span>xintercept <span class='op'>=</span> <span class='fl'>0</span>, color <span class='op'>=</span> <span class='st'>'grey'</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_linerange.html'>geom_linerange</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>1</span>,
      <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>
        xmin <span class='op'>=</span> <span class='va'>estimate</span> <span class='op'>-</span> <span class='fl'>2</span><span class='op'>*</span><span class='va'>std.error</span>,
        xmax <span class='op'>=</span> <span class='va'>estimate</span> <span class='op'>+</span> <span class='fl'>2</span><span class='op'>*</span><span class='va'>std.error</span>
      <span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>1.5</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>labs</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='st'>'Species-level OTU'</span>, x <span class='op'>=</span> <span class='st'>'Estimated LFC'</span>, color <span class='op'>=</span> <span class='st'>'Meas. type'</span><span class='op'>)</span>
  <span class='va'>p1</span> <span class='op'>/</span> <span class='va'>p2</span> <span class='op'>+</span>
    <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_layout.html'>plot_layout</a></span><span class='op'>(</span>ncol <span class='op'>=</span> <span class='fl'>1</span>, heights <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0.3</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
    <span class='fu'><a href='https://patchwork.data-imaginist.com/reference/plot_annotation.html'>plot_annotation</a></span><span class='op'>(</span>tag_levels <span class='op'>=</span> <span class='st'>'A'</span><span class='op'>)</span> <span class='op'>&amp;</span>
    <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_minimal_grid.html'>theme_minimal_hgrid</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='op'>}</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='fu'>plot_estimates</span><span class='op'>(</span><span class='va'>res_div_lm_tidy</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="momspi-regression-diversity_files/figure-html5/regression-lm-1.svg" width="2310" /></p>
</div>
<p>The data is unlikely to meet the assumptions generally considered necessary for simple linear regression. In particular, the standard errors (and confidence intervals) are likely to be much too small. Bootstrapped intervals could be used to address this issue, if we are interested in understanding the impact of bias on confidence intervals in a more realistic scenario.</p>
<h3 id="alternate-zero-replacement-strategy">Alternate zero-replacement strategy</h3>
<p>Lets also try adjust the proportions to the Dirichlet posterior mean, with the prior set to the average proportions across all samples. This amounts to a pseudocount of <code>ntaxa</code> times the mean (observed) proportion of the species, and so adjusts for differences in mean proportion among species.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>momspi_div_alt</span> <span class='op'>&lt;-</span> <span class='va'>momspi_div</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> 
  <span class='va'>adjust_dirchlet</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>res_div_alt_lm</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>
  <span class='st'>'Observed'</span> <span class='op'>=</span> <span class='va'>momspi_div_alt</span>,
  <span class='st'>'Calibrated'</span> <span class='op'>=</span> <span class='va'>momspi_div_alt</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='fu'><a href='https://mikemc.github.io/metacal/reference/calibrate.html'>calibrate</a></span><span class='op'>(</span><span class='va'>bias_all_vec</span><span class='op'>)</span>
<span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://purrr.tidyverse.org/reference/map.html'>map_dfr</a></span><span class='op'>(</span><span class='va'>stat_test</span>, taxa <span class='op'>=</span> <span class='va'>taxa_to_test</span>, .id <span class='op'>=</span> <span class='st'>'type'</span><span class='op'>)</span>
<span class='va'>res_div_alt_lm_tidy</span> <span class='op'>&lt;-</span> <span class='va'>res_div_alt_lm</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/across.html'>across</a></span><span class='op'>(</span><span class='va'>fit</span>, <span class='va'>map</span>, <span class='fu'>broom.mixed</span><span class='fu'>::</span><span class='va'><a href='https://generics.r-lib.org/reference/tidy.html'>tidy</a></span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/nest.html'>unnest</a></span><span class='op'>(</span><span class='va'>fit</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> 
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>term</span> <span class='op'>!=</span> <span class='st'>'(Intercept)'</span><span class='op'>)</span>

<span class='fu'>plot_estimates</span><span class='op'>(</span><span class='va'>res_div_alt_lm_tidy</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="momspi-regression-diversity_files/figure-html5/regression-lm-alt-1.svg" width="2310" /></p>
</div>
<h2 id="interpretion">Interpretion</h2>
<p>Check the diversity-efficiency relationship</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>sam</span> <span class='op'>&lt;-</span> <span class='va'>momspi_div</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='va'>sample_data</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='va'>as_tibble</span>
<span class='va'>sam</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>y <span class='op'>=</span> <span class='va'>div_group</span>, x <span class='op'>=</span> <span class='va'>mean_efficiency</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_x_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>ggdist</span><span class='fu'>::</span><span class='fu'><a href='http://mjskay.github.io/ggdist/reference/stat_sample_slabinterval.html'>stat_slab</a></span><span class='op'>(</span><span class='op'>)</span>
</code></pre>
</div>
<p><img src="momspi-regression-diversity_files/figure-html5/unnamed-chunk-16-1.svg" width="1950" /></p>
</div>
<p>From this plot, we can see that the mean efficiency is 3-10X lower in the High diversity relative to the low diversity group.</p>
<p>The results of a linear regression analysis give us the systematic error in LFCs for the simple linear regression,</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>mean_eff_lm</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/lm.html'>lm</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log</a></span><span class='op'>(</span><span class='va'>mean_efficiency</span><span class='op'>)</span> <span class='op'>~</span> <span class='va'>div_group</span>,
  data <span class='op'>=</span> <span class='va'>sam</span>
<span class='op'>)</span>
<span class='va'>mean_eff_lm</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='va'>summary</span>
</code></pre>
</div>
<pre><code>
Call:
lm(formula = log(mean_efficiency) ~ div_group, data = sam)

Residuals:
     Min       1Q   Median       3Q      Max 
-3.02856 -0.24524 -0.01627  0.28631  1.61043 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   -0.34795    0.02371  -14.68   &lt;2e-16 ***
div_groupHigh -1.56081    0.03122  -50.00   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.5494 on 1267 degrees of freedom
Multiple R-squared:  0.6637,    Adjusted R-squared:  0.6634 
F-statistic:  2500 on 1 and 1267 DF,  p-value: &lt; 2.2e-16</code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>mean_eff_lfc</span> <span class='op'>&lt;-</span> <span class='va'>mean_eff_lm</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='fu'>broom</span><span class='fu'>::</span><span class='fu'><a href='https://generics.r-lib.org/reference/tidy.html'>tidy</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> 
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span><span class='va'>term</span> <span class='op'>==</span> <span class='st'>'div_groupHigh'</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='fu'><a href='https://dplyr.tidyverse.org/reference/pull.html'>pull</a></span><span class='op'>(</span><span class='va'>estimate</span><span class='op'>)</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>res_div_lm_tidy</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span>name <span class='op'>=</span> <span class='va'>type</span>, <span class='va'>.otu</span>, value <span class='op'>=</span> <span class='va'>estimate</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://tidyr.tidyverse.org/reference/pivot_wider.html'>pivot_wider</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span>
    error <span class='op'>=</span> <span class='va'>Observed</span> <span class='op'>-</span> <span class='va'>Calibrated</span>,
  <span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/pull.html'>pull</a></span><span class='op'>(</span><span class='va'>error</span><span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='va'>summary</span>
</code></pre>
</div>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  1.517   1.517   1.517   1.517   1.517   1.517 </code></pre>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>mean_eff_lfc</span> 
</code></pre>
</div>
<pre><code>[1] -1.560813</code></pre>
</div>
<p>Can see that the errors are the same for all species, and nearly equal to the negative LFC in mean efficiency. A small difference is expected from the addition of pseudocounts for the species regression.</p>
<p>To understand why the mean efficiency is lower, can try a scatterplot of mean efficiency against diversity with samples colored by most abundant species.</p>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>div1</span> <span class='op'>&lt;-</span> <span class='va'>div</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate-joins.html'>left_join</a></span><span class='op'>(</span>
    <span class='va'>momspi_rare</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='va'>sample_data</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span> <span class='va'>as_tibble</span>,
    by <span class='op'>=</span> <span class='st'>'.sample'</span>
  <span class='op'>)</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://dplyr.tidyverse.org/reference/mutate.html'>mutate</a></span><span class='op'>(</span>
    top_species_plot <span class='op'>=</span> <span class='fu'><a href='https://forcats.tidyverse.org/reference/fct_lump.html'>fct_lump_n</a></span><span class='op'>(</span><span class='va'>top_species_cal</span>, <span class='fl'>7</span><span class='op'>)</span>
  <span class='op'>)</span>
</code></pre>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span class='va'>colors_top</span> <span class='op'>&lt;-</span> <span class='fu'>RColorBrewer</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html'>brewer.pal</a></span><span class='op'>(</span><span class='fl'>8</span>, <span class='st'>'Accent'</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>,<span class='fl'>2</span>,<span class='fl'>3</span>,<span class='fl'>5</span>,<span class='fl'>4</span>,<span class='fl'>6</span>,<span class='fl'>7</span>,<span class='fl'>8</span><span class='op'>)</span><span class='op'>]</span> 
<span class='va'>div1</span> <span class='op'><a href='https://mikemc.github.io/speedyseq/reference/pipe.html'>%&gt;%</a></span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span><span class='va'>diversity</span>, <span class='va'>mean_efficiency</span>, color <span class='op'>=</span> <span class='va'>top_species_plot</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>type</span><span class='op'>~</span><span class='va'>index</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_x_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'>ggrastr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/ggrastr/man/rasterise.html'>rasterize</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>alpha <span class='op'>=</span> <span class='fl'>0.6</span><span class='op'>)</span>, dpi <span class='op'>=</span> <span class='fl'>300</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_smooth.html'>stat_smooth</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>group <span class='op'>=</span> <span class='st'>'none'</span><span class='op'>)</span>, color <span class='op'>=</span> <span class='st'>'black'</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>labs</a></span><span class='op'>(</span>color <span class='op'>=</span> <span class='st'>'Top species'</span>, x <span class='op'>=</span> <span class='st'>'Diversity'</span>, y <span class='op'>=</span> <span class='st'>'Mean efficiency'</span><span class='op'>)</span> <span class='op'>+</span>
  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_manual.html'>scale_color_manual</a></span><span class='op'>(</span>
    values <span class='op'>=</span> <span class='va'>colors_top</span>,
    labels <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'><a href='https://stringr.tidyverse.org/reference/str_replace.html'>str_replace_all</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='st'>'_'</span>, <span class='st'>' '</span><span class='op'>)</span>
  <span class='op'>)</span>
</code></pre>
</div>
<p><img src="momspi-regression-diversity_files/figure-html5/unnamed-chunk-20-1.svg" width="2400" /></p>
</div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom">
  <h3 id="references">References</h3>
  <div id="references-listing"></div>
</div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
